\documentclass[onecolumn, compsoc,10pt]{IEEEtran}
\let\labelindent\relax
\usepackage{enumitem}
\input{preamble.tex}  
\input{customdef} 
\usepackage{flushend}
\externaldocument[SI-]{SI}
% \externaldocument[EXT-]{exfig}
\newif\iftikzX
\tikzXtrue
\tikzXfalse
\def\Extended{}
\newif\ifFIGS
\FIGSfalse  
\FIGStrue 
 
% Zoonotic risk technology
% FluLeap~\cite{eng2014predicting} is a machine-learning algorithm
% that uses sequence data to identify host tropism: classify influenza
% viruses as either avian or human.
% PREDICT,  a US\$200-million project funded by the
% US Agency for International Development
% (USAID), spent around a decade looking for
% animal viruses. By the time it ended in 2020, it
% had identified 949 new viruses in samples from
% wildlife, livestock and people, in 34 countries

% Zoonotic Risk Technology Enters the Viral Emergence Toolkit
% Grange $\etal$~\cite{grange2021ranking} look for diverse factors,
% but are not strain specific assessments, and identify species
% level risk.

% ~\cite{wille2021accurately} How accurately can we assess zoonotic risk? underscore the
% inherent challanges in assessing  risk from viruses that have not been studied
% well enough.
% genome size may be a useful surrogate in screening for potential zoonotic viruses~\cite{grewelle2020larger}
%
% Among these traits-genomic material, genome segmentation, and replication without nuclear entry-the last is the strongest predictor of cross-species transmission. This finding highlights nuclear entry as a barrier to transmission and suggests that the ability to complete replication in the cytoplasm may prove to be a useful indicator of the threat of cross-species transmission. ~\cite{pulliam2009ability}

  
\tikzexternalenable   
% \pgfplotsset{compat=1.18}
\begin{document}  

\maketitle

{\bf \sffamily \fontsize{10}{12}\selectfont \noindent   
  {\normalfont \itshape Abstract:} Novel \infl  strains emerging into humans from animal reservoirs can cause large antigenic shifts in the circulating strain, and such events are suspected to have triggered numerous  pandemics in the past~\cite{shao2017evolution,mills2004transmissibility,reid2003origin,landolt2007up}, often exacting a devastating toll on human lives. Yet, our current ability to  scalably  evaluate the pandemic potential of   individual strains that do not yet circulate in humans remains limited. In this study, we introduce a computational approach, known as the \enet,  to learn how viable genotypic variations are shaped by  emergent evolutionary constraints using only genomic sequence data of collected strains.    Analyzing    Hemaglutinnin (HA) and Neuraminidase (NA) amino acid sequences from nearly 100,000 unique \infl  strains from public  databases, our proposed algorithm merges machine learning and large deviation theory to  estimate the likelihood of  specific future  mutations, ultimately  yielding the  numerical odds of one parent strain giving rise to a specific descendant via natural evolutionary processes. After validating our model on the problem of forecasting the dominant strain(s) of the upcoming flu season, with \enet-based forecasts significantly outperforming World Health Organization (WHO) recommended flu vaccine compositions almost consistently over the past two decades for  H1N1 and H3N2 subtypes, individually in the Northern and the Southern hemispheres (HA: $34.8\% $ improvement, NA: $12.5\%$ improvement), we assess  the pandemic potential  of novel animal  strains that do not yet circulate in humans. While the state-of-the-art  Influenza Risk Assessment Tool (IRAT) from the Center for Disease Control (CDC) comprises multiple time-consuming experimental assays, our proposed \erisk score can be evaluated in $6-8$ seconds for each new strain, while strongly correlating with the published IRAT scores (correlation=0.703, p-value $= 0.00026$). This approximately six orders of magnitude speedup (weeks vs seconds) in  identifying risky strains is a necessary step to fully exploit the  current surveillance capacity via scalably analyzing tens of thousands of strains collected every year. Thus, our results potentially  enable  meaningful preemptive pandemic avoidance and mitigation strategies, which are especially relevant in the light of the recent COVID-19 pandemic.
}
  
\vspace{10pt}
\section*{Introduction}
Influenza viruses constantly evolve~\cite{dos2016influenza},  rapidly producing  sequence alterations over a time scale of months  that  perturb surface protein structures sufficiently to evade the prevailing host immunity, and cause the  recurring seasonal flu epidemic. These periodic  infection peaks claim a quarter to half a million lives~\cite{huddleston2020integrating} globally,  and currently our response hinges on  successfully inoculating a substantial fraction of the  population with a  reformulated  vaccine annually~\cite{boni2008vaccination,dos2016influenza}.  Among numerous factors that hinder optimal design of the seasonal flu shot, failing to correctly predict the dominant strain in the upcoming season dramatically reduces vaccine effectiveness~\cite{tricco2013comparing}, and despite  recent advances~\cite{neher2014predicting,huddleston2020integrating} such predictions remain imperfect. In addition to  the seasonal  epidemic, novel \infl strains spilling over into humans from animal reservoirs have triggered  pandemics  at least four times (1918 Spanish flu/H1N1, 1957 Asian flu/H2N2, 1968 hongkong flu/H3N2, 2009 swine flu/H1N1) in the past 100 years~\cite{shao2017evolution}. With the memory of the  sudden emergence of \cov and the ensuing devastating pandemic fresh in our minds, a looming question  is whether we can  preempt and mitigate such events in the future.  \infl,  partly on account of its segmented genome and its wide prevalence is common animal hosts, can easily incorporate genes from multiple strains and (re)emerge as novel human pathogens~\cite{reid2003origin,vergara2014ns}, and   thus harbors  a high potential  of triggering the next  pandemic.

\input{figuresandtables.tex}

A possible approach to mitigating such risk is to identify  strains in animal hosts that do not yet circulate in humans but is likely to spill-over and quickly achieve human-to-human (HH) transmission capability. Despite global surveillance efforts to  collect wild specimens from diverse hosts and geo-locations, our  current ability to objectively, reliably and scalably  evaluate such risk posed to  humans by  individual animal  strains  is  limited~\cite{wille2021accurately}. Despite recent progress towards understanding emergence risk~\cite{pulliam2009ability,grewelle2020larger,grange2021ranking}, scalable ranking of individual strains remains out of reach.


CDC's current solution to preempting strain-specific emergence  is the Influenza Risk Assessment Tool (IRAT), which  uses  ten weighted and pre-determined risk elements to factor in   1) expert-selected properties of the strain, 2) attributes of the population, and 3) ecology and epidemiological characteristics~\cite{Influenz24:online}. Influenza subject matter experts CDC and other key organizations such as the Food and Drug Administration (FDA), Animal and Plant Health Inspection Service (APHIS), and Agricultural Research Service (ARS) are asked to evaluate are asked to evaluate strain lineages using the ten risk elements defined in the IRAT, namely the number of human infections, transmission in laboratory animals, receptor binding characteristics, population immunity, infections in animals, genomic analysis, antigenic relatedness, global distribution in animals, disease severity and  pathogenesis, and antiviral treatment options. Each SME score 1 to 3 elements based on their
particular areas of expertise, and the point estimate scores for each risk element are averaged, multiplied by predetermined weights, and summed to give an aggregate score for each of the two IRAT risk questions related to potential risk for emergence in humans and potential public health impact if the virus gains HH capability. Generally  the scores obtained are subjective to some degree, with some variability observed among SME point estimates. Additionally, evaluating the score  involves multiple experimental assays, possibly taking  weeks to compile the final  evaluation for a single strain or lineage. This results in  a scalability bottleneck: while  current global surveillance efforts annually  collect  thousands of sequences, most  will never be analyzed in time. IRAT assessment protocols are  not fast enough to leverage the full capacity of current surveillance output.%, and thus have low odds of successfully  preempting a \infl pandemic. 



% Variability was seen among SME point estimate scores in the risk elements of Genomic Variation, Global
% Distribution in Animals, and Antigenic Relatedness. The greatest variation was seen in the scores for
% Genomic Variation where the range of scores spanned from a low to high risk range indicating some
% differences in interpretation of the available data. The different categories include: 
% Human Infections, Transmission in Lab Animals, Receptor Binding, Population Immunity, Infections in Animals, Genomic Analysis, Antigenic Relatedness, Global Distribution in Animals, Disease Severity and  Pathogenesis, and Antiviral/Treatment Options. 



% CDC's current solution to preempting strain-specific emergence  is the Influenza Risk Assessment Tool (IRAT), which  uses  ten weighted risk elements to factor in   1) expert-selected properties of the strain, 2) attributes of the population, and 3) ecology and epidemiological characteristics~\cite{Influenz24:online}. The IRAT score assigns  a grade between 1 and 10 individually for emergence risk and public health impact to individual strains, which typically are not circulating in humans. Evaluating the score  involves multiple experimental assays, possibly taking  weeks to compile the final  evaluation for a single strain. This results in  a scalability bottleneck: while  current global surveillance efforts annually  collect  thousands of sequences, most  will never be analyzed in time. IRAT assessment protocols are  not fast enough to leverage the full capacity of current surveillance output, and thus have low odds of successfully  preempting a \infl pandemic.

In this study, we introduce a pattern recognition algorithm to automatically parse out emergent evolutionary constraints operating on \infl viruses in the wild, to numerically estimate the probability $Pr(x \rightarrow y)$ of a strain $x$ spontaneously giving rise to  $y$. We show that this capability is key to preempting  strains which would be in circulation in future, and thus 1) reliably forecast dominant strains of seasonal epidemics, and 2) approximate IRAT scores of non-human strains without direct experimental assays, and without the involvement of SMEs.

To uncover relevant evolutionary constraints, we analyzed observed variations (point substitutions and indels) of the  amino acid (AA)  sequences  of the two key proteins implicated  in cellular entry and exit~\cite{gamblin2010influenza,shao2017evolution}, namely HA and NA respectively. By representing these constraints within a predictive framework -- the \enet -- we estimated the  odds of a specific mutation to arise in a given strain, and consequently the probability of a specific strain spontaneously  evolving into another.  Such explicit calculations are difficult  without first inferring the emergent constraints at play,  as well as  the variation of mutational probabilities and the potential residue replacements from one positional index to the next along the AA sequence of a protein. The many well-known classical  DNA  substitution models~\cite{posada1998modeltest} or standard approaches to phylogenetic tree inference which assume a constant mutation rate or some variation thereof, do not address these issues, and are not applicable to the problem at hand. Similalrly, recently reported algorithms such  as FluLeap~\cite{eng2014predicting}  which identfies host tropism from sequence data, or estimating risk posed by different viral species~\cite{grange2021ranking} do not allow train-specific risk assessment.

The genomic dependencies we uncover are shaped by  a  functional necessity of conserving/augmenting  fitness in the wild. A  strain must be sufficiently common  to be recorded, implying that the sequences from public databases that we train  with have  high replicative fitness. Lacking kinetic proofreading  in  RNA-polymerase,  \infl integrates  faulty nucleotides   at a relatively high rate ($10^{-3}$ to $10^{-4}$) during  replication~\cite{ahlquist2002rna,chen2006avian}. However, few   of these variations are actually viable, with only specific patterns  maintaining/gaining fitness;  leading to emergent dependencies between such changes. Furthermore, these fitness constraints are not time-invariant. The background distribution of strains, and selection pressure from the evolution of cytotoxic T lymphocyte  epitopes~\cite{woolthuis2016long,fan2012role,van2016differential,berkhoff2007assessment,van2012evasion} in humans can change quickly. With a sufficient number of unique samples to train on for each flu season, the \enet (recomputed for each time-period) is expected to  track these  constraints, automatically reflecting the effect of  evolving host immunity, and the current background strain distribution.  


Structurally, an \enet comprises an interdependent collection of  local predictors: each aiming to model the observed amino acid ``outcome'' at a specific positional index of the proteins using as features (input variables) the residues appearing  at other  locations  (Fig.~\ref{figscheme}b). The algorithm automatically identifies the set of features (AA positions) that influence the outcome at a particular index, implying that an \enet comprises atmost as many such  position-specific predictors as the length of the AA sequence. These individual predictors are implemented as conditional inference trees~\cite{Hothorn06unbiasedrecursive}, in which  nodal splits  have  a minimum pre-specified significance in differentiating the  child nodes. Inferring residue predictors at each  index as  functions of the rest of the AA sequence essentially yields an estimated conditional residue distribution  at each index. The set of indices acting as features in each tree varies; for example, as shown in the fragment of the  H1N1 HA \enet (2020-2021) in Fig~\ref{figscheme}b, the predictor for index 63 is dependent on the AA residue at index 155, and the predictor for index 155 is dependent on index 223, the predictor for index 223 is dependent on index 14, and the residue at index 14 is dependent on index 63, revealing a cyclic dependency. The complete \enet harbors a vast number of such  relationships, wherein each internal node of a tree may be recursively ``expanded'' to its own tree. Owing to this recursive expansion,  a complete \enet is hard to visualize; nevertheless it captures the complexity of the rules guiding evolutionary change substantially better than earlier attempts, as evidenced by our validation results.

In this study we used  HA and NA protein seqwuences from  all unique \infl strains in the NCBI and GISAID databases between the year 2000 to present time (2022 April,  98,299 strains in total). We  construct \enet{s} separately for H1N1 and H3N2 subtypes, and for each flu season, constructing in total 85 models for predicting seasonal dominance. % # of models: 16 that i computed + 69 precomputed in quasinet package %Dominant strain forecasts are made for each  season and  each hemisphere separately.
We  use AA sequences with no additional phenotypic annotation, other than identifying the host animal, the subtype, and the time and place of collection. This is advantageous, since antigenic characterization of  \infl strains tend to be substantially  laborious and low-throughput compared to genome sequencing~\cite{wood2012reproducibility}; however,  incorporation of detailed phenotypic information, $e.g.$ from deep mutational scanning (DMS) assays, has been shown to improve prediction of seasonal strains~\cite{huddleston2020integrating}. Despite limiting ourselves to only genotypic and subtype information, our approach is able to distill deep emergent  fitness-preserving constraints   outperforming reported DMS  augmented strategies for \infl.

 
Inference of the \enet components is the first step in our approach, which then induces  an intrinsic distance metric between genomic sequences. The \qdist  (Eq.~\eqref{q-distance} in Methods) is defined as the square-root of the Jensen-Shannon (JS) divergence~\cite{cover} of the position-specific conditional distributions, averaged over the entire sequence length. Unlike the classical edit distance measuring the number of mutations by which the sequences differ, the \qdist is informed by the   dependencies that the \enet infers, and adapts to the specific subtype, allele frequencies, and variations in the background population. Central to our approach is the theoretical result (Theorem~\ref{SI-thmbnd} in Supplementary text) that   $\log Pr(x \rightarrow y )$   may be approximated by the  \qdist $\theta(x,y)$.
%
%Thus, the \qdist provides a measure of genomic similarity that is substantially more biologically meaningful compared to the common edit distance.

The mathematical intuition  relating  \qdist  to the loglikelihood of spontaneous change  is similar to the prediction of  a biased outcome when we  toss a fair coin. With an overwhelming probability, a sequence of such tosses should result in roughly equal number of heads and tails. However, large deviations, $i.e$, substantial deviations from the expected fraction,  do happen, and the probability of such rare events is explicitly quantifiable~\cite{varadhan2010large} with elementary results from large deviation theory. Generalizing to non-uniform conditional probabilities inferred by the \enet, we show  that the likelihood of a spontaneous transition  by random chance may also be similarly bounded. As an important distinction from the edit distance, the \qdist  between two fixed sequences may change even if only the background population changes (SI-Table~\ref{SI-tabex}, example where  the distance between two fixed  sequences vary when we vary their collection years, and hence the environment or the background strain distribution). Thus, we can not only estimate the risk of emergence of a particular animal strain, but can track how that risk evolves over time [Figure ref XXX].

%Sequences may have a large \qdist and a small edit distance, and vice versa (although on average the two distances tend to be positively correlated, see SI-Table~\ref{SI-tabcor}).



The ability of the \enet framework to determine the numerical odds of spontaneous jump $ Pr(x \rightarrow y)$ (Fig.~\ref{figscheme}) suggests that we are able to frame the problem of forecasting  dominant strain(s), and that of estimating the  pandemic potential of an animal strain as  mathematical propositions (albeit with some simplifying assumptions), with  approximate solutions (Fig.~\ref{figscheme}c-d). Thus,  a dominant strain for an upcoming  season may be identified as one which maximizes the joint probability of simultaneosly arising from each (or most)  of the currently circulating strains (Fig.~\ref{figscheme}c).  This does not deterministically specify the dominant strain, but a strain satisfying this criterion  has  high odds of emerging as the dominant one. And, a pandemic risk score of a novel strain may be estimated by the probability of it giving rise to a well-adapted human strain.  We validate  our proposed solutions for these problems in out-of-sample data.
%Thus, the \qdist enables us to estimate the numerical value of the probability $Pr(x \rightarrow y)$, and hence solve the mathematical propositions described above.
%

In the context of the first problem namely the  forecast of dominant strain(s) for the next flu season,  we derive the following search criteria (See Methods) from the proposition stated above, to identify a historical strain that is expected to be close to the upcoming season's dominant strain:
%
\calign{
\label{dompred}&\dst = \argmin_{y \in \cup_{\tau \leqq t} H^\tau}  \left ( \sum_{x\in H^t}  \theta^{[t]}(x,y) - \abs{H^t}A \ln \mem{y}  \right )
}%
where $\dst$ is the predicted dominant strain  at time $t+\delta$, $H^t$ is the set of currently circulating human strains at time $t$  observed over the past year, $\theta^{[t]}$ is the \qdist informed by the inferred \enet using sequences in $H^t$, $\mem{y}$ is the estimated probability of strain $y$ being generated by the \enet, and $A$ is a constant dependent on the sequence length and significance threshold used (See Methods). Note that the first term gets the solution close to the centroid of the current strain distribution (in the \qdist metric, which is different from the centroid if the standard edit distance is used), and the second term relates to how common the genomic patterns are amongst recent human strains. 

Prediction of the future dominant strain as  a close historical strain  allows direct validation of the approach with past WHO recommendations. Notably, the flu shot  is annually prepared at least six months in advance, and is based on a cocktail of historical strains determined by the WHO via global surveillance~\cite{agor2018models}, hoping to match the circulating strain(s) in the upcoming  season (recommendations for the northern hemisphere are given in February, while that for the southern hemisphere are given at the end of December the previous year). % A variety of hard-to-model effects, despite observed cross-reactive effects~\cite{tricco2013comparing}, have limited vaccine effectiveness in recent years~\cite{cdceff}.
For each year of the past two decades, we calculated strain forecasts using  Eq.~\eqref{dompred} with data available 6 months before the target season. % is essentially an approximation to the corrected  ``q''-centroid of the strain distribution from the previous season (See  Methods).
We  measured forecast performance by the number of mutations by which the predicted HA/NA AA sequence deviated from the realized dominant strain, which we approximated as the one closest to the centroid of the observed strains in the target season in the sense of the edit distance $i.e.$ number of mutations. Our \enet-informed forecasts outperform  WHO/CDC recommended flu vaccine compositions almost consistently over the past two decades, for both H1N1 and H3N2 subtypes, individually in the northern and the southern hemispheres. %The \enet  predictions consistently outperform the WHO recommendations. % For H1N1 HA, the \enet induced recommendation outperforms the WHO suggestion by $>33\%$ on average over the last two decades, and $>71\%$ on average in the last decade. The gains for H1N1 NA over the same time periods are $>15\%$ and $>52\%$, respectively. For H3N2 HA, the \enet induced recommendation outperforms the WHO suggestion by $>37\%$ on average over the last two decades, and $>40\%$ in the last decade. The gains for H3N2 NA over the same time periods are $>14\%$ and $>15\%$, respectively. Finding multi-cluster predictions has the potential to yield even more improved results, as seen in Fig.~\ref{figseasonal} and SI-Table~\ref{SI-tabrec8} through SI-Table~\ref{SI-tabrec11}. \DQS{what is multicluster}
%
%\DQS{concrete numbers}
The  results  broken down by hemisphere/protein/subtypes is given in Table~\ref{tabperf}. Fig.~\ref{figseasonal} illustrates the relative gains computed for both subtypes and the two hemispheres. % (the flu season occupy distinct time periods and may have different dominant strains in the two hemispheres~\cite{boni2008vaccination}).
Additional improvement is possible if we recommend multiple strains every season for the vaccine cocktail (Fig.~\ref{figseasonal}e,f,k,l). The details of the specific strain  recommendations made by the \enet approach for two subtypes (H1N1, H3N2), for two genes (HA, NA) and for the northern and the southern hemispheres over the previous two decades are enumerated in the Supplementary Text in Tables SI-Table~\ref{SI-tabrec0} through SI-Table~\ref{SI-tabrec11}. While it is recognized that even well-matched strains can fail to induce a strong immune response due to previous infection history of vaccine recipients~\cite{cobey2018poor}, strain-matching is a crucial component to realizing high vaccine effectiveness~\cite{gouma2020antigenic}. Thus, our results may improve the effectiveness of the flu shot via sophisticated  pattern-recognition, outperforming current practice (WHO/CDC) as well as  recently reported prediction strategies  using more standard computational and/or experimental frameworks~\cite{huddleston2020integrating,neher2014predicting},  without using  detailed phenotypic information such as DMS annotations~\cite{huddleston2020integrating,neher2014predicting}.

%(put concrete results on season forecasts)

Our primary claim in this study, however,  is the ability to estimate the pandemic potential of novel animal strains, via a  time-varying \erisk score $\rho_t(x)$ for a strain $x$ not yet found to circulate in human hosts. We show that:%
\cgather{\label{eqrho}
\rho_t(x) \triangleq -\frac{1}{\abs{H^t}} \sum_{y \in H^t} \theta^{[t]}(x,y)
}%
scales as the average log-likelihood of $Pr(x \rightarrow y)$ where $y$ is any human strain of a similar subtype to $x$, and  $\theta^{[t]}$ is the \qdist informed by the \enet computed from recent human strains $H_t$ at time $t$ of the same subtype as $x$, observed over the past year. As before, the \enet inference makes it possible to estimate $\rho_t(x)$ explicitly. 

To validate our score against CDC-estimated IRAT emergence scores, we construct \enet models for HA and NA segments using all human strains of the same  subtypes collected within the past year of the IRAT assessment date. Since our measure is time-varying, making sure that we estimate the risk at the time of the IRAT assessment is important. For example, the A/swine/Shandong/1207/2016 strain was assessed by IRAT in July 2020, so we  use human H1N1 strains collected  between July 1, 2019 - June 30, 2020. For sub-types with few recorded human strains (H1N2, H5N1, H5N6, H7N7, H9N2), we consider all human strains of the corresponding subtypes collected upto the IRAT evaluation date. We then compute the average \qdist between a given animal  strain  and the circulating human strains for both HA and HA segments (using Eq.~\eqref{eqrho}), with finally reporting the geometric mean of the estimates as the estimated risk. Considering IRAT scores of 22 strains published by the CDC, we find strong support  (correlation of $-0.7032$, pvalue $< 0.005$, Fig.~\ref{figirat}) for this claim. Importantly, each \erisk score  is  computable in approximately $8-12$ seconds as opposed to potentially weeks taken by IRAT experimental assays, and this dramatic reduction in time and cost  opens the door to fully exploiting the  current surveillance capacity.

Importantly,  \erisk is a function of time; the background distribution of strains matters, and
we show that recomputing the risks for a different time changes (worsens) the correlation with the IRAT estimate. We also show how the risks evolve over time according to our model (See Table~\ref{tabwwXXXX}  and SI-Fig.~\ref{SI-irat})

To map the \enet distances to more recognizable IRAT scores, we  train a linear model to estimate the IRAT emergence score from the \erisk values. Also, since the IRAT impact score is strongly correlated with the emergence score (correlation of $0.8015$), we  trained a linear model to estimate the impact score as well from the  \erisk. We show these results in Table~\ref{tabXXXX}. With our linear models trained, we compute a simulated IRAT analysis of all \infl strains collected over past two years (6,254 sequences in total). (Table~\ref{tabeeeXXXX} shows the top 5 strains of each available subtype). The geolocation of the strains, along with risk information is shown in Fig.~\ref{figXXXXXeee}.


\DQS{discuss world prediction results}

In conclusion, while numerous tools exist for ad hoc quantification of genomic similarity~\cite{posada1998modeltest,goldberger2005genomic,huelsenbeck1997phylogeny,neher2014predicting,VanderMeer2010,Smith2009}, a smaller  distance $i.e.$ a higher similarity of two strains in  these frameworks is not sufficient to imply a high likelihood of a jump. % These measures tend to be  variations of the edit distance between  sequences, and their connection to selection pressures is at best qualitative, with the  missing piece being effectively learning which changes are likely in the wild, conditioned on possibly the entire sequence of the current strain.
To the best of our knowledge, the \enet algorithm is  the first of its kind to learn an appropriate comparison metric from data, without assuming any model of DNA or amino acid substitution, or a genealogical tree a priori. While the effect of the environment and selection cannot be inferred from a single sequence, an entire database of observed strains, processed through the right lens, can parse out useful predictive models of these complex interactions. Our results are  aligned with recent studies demonstrating effective  predictability of  future mutations  for different organisms~\cite{mollentze2021identifying,maher2021predicting}.
   
The \qdist calculation   is currently limited to analogous sequences (such as point variations of the same protein from different viral subtypes), and the \enet inference requires a  sufficient diversity of observed strains. A multi-variate regression analysis indicates  that the most important factor for our approach to succeed is  the diversity of the sequence dataset (see Supplementary Text, SI Table~\ref{SI-tabreg}), which would exclude applicability to completely novel pathogens with no related human variants, and ones that evolve very slowly. Nevertheless, the tools reported here can improve effectiveness of the annual flu shot, and perhaps allow for the development of premeptive vaccines to  target risky animal strains  before the first human infection in the next pandemic.

% The available IRAT scores are well approximated within our framework, albeit with  four outliers. For three of these, the approximation would be better if IRAT had actually  underestimated their risk, which include one H3N2 CIV and two H7N9. While the zoonotic potential of H3 CIV is unknown~\cite{martinez2020characterizing}, susceptibility to infection with at least three different \infl subtypes (H3N8, H3N2 and H1N1)~\cite{chang1976influenza,houser1980evidence,chen2018emergence} and diverse human influenza viruses implies the potential to produce novel human-adapted reassortment viruses~\cite{lowen2017constraints}.% Compared to strains in which the full genome has evolved in a non-human host, such
% % reassortant viruses may more easily overcome the species barrier to cause an outbreak in humans~\cite{lowen2017constraints}.
% %Thus, the risk for the 2015 H3N2 IL CIV might have been underestimated by IRAT.
% Risk from the H7N9 strains could also have been underestimated.
% To date, a total of 1,568 laboratory-confirmed human infections with avian influenza A(H7N9) virus including 616 fatal cases have been reported to WHO since early 2013, and HA mutations have been shown to cause high pathogenicity in poultry, and  WHO recognizes the  need for close monitoring to identify 
% changes in the virus and transmission behaviour\url{https://www.who.int/docs/default-source/wpro---documents/emergency/surveillance/avian-influenza/ai_20220930.pdf?sfvrsn=22ea0816_18}.% The last case of human infection with avian influenza A(H7N9) reported to WHO in the Western Pacific Region was in 2019.
% % Of the 1,568 human infections with avian influenza A(H7N9), 33 have reported mutations in the hemagglutinin gene indicating a change to high pathogenicity in poultry, and while the current virus does not  transmit easily among humans,
% %the WHO recognizes the  need for close monitoring to identify 
% %changes in the virus and transmission behaviour\url{https://www.who.int/docs/default-source/wpro---documents/emergency/surveillance/avian-influenza/ai_20220930.pdf?sfvrsn=22ea0816_18}.
%  Importantly, %T% his might be corrobortated by the fact that
% the \erisk  increase when recomputed with current background for these strains, reflecting the possibility of underestimation in the IRAT scores.  Similarly, for  A/swine/Shandong/1207/2016  that was given the highest IRAT risk score, appears to be an outlier and a possible case of risk overestimation, and the \erisk  decreases  when recomputed with current background.



% It is important to restrict the comparison to strains encountered in relatively recent past, expanding the search eliminated  the ability to track the IRAT scores (SI-Fig.~\ref{SI-irat}).
%\DQS{we can have a linera estimation of both emergenec score and impact score}

% For further analysis, we also performed q-analysis on IRAT H1- and H3- sub-types by taking average q-distance between the target strain and all human-circulating strains available, with no upper or lower collection date bound. We expected the correlation to be worse than with bounded strains, since a strain being ``close" to humans at some point in the past does not necessarily mean being close now. Indeed, our results showed almost no correlation to the IRAT emergence risk scores. Bounded results for H1- and H3- sub-types yielded a correlation of $-0.6916$, while unbounded results yielded a correlation of $0.0545$; see SI-Fig.~\ref{SI-irat}.

%Given the efficiency of the q-distance computations, we can track how risk of emergence changes over time by continually updating the estimated risk score of the surveyed strains as the human strain population updates over time.
%For exact average q-distance and \enet sample size statistics, please see SI-Table~\ref{SI-irattab} in the Supplementary Text.









% The zoonotic potential of H3 CIV is unknown~\cite{martinez2020characterizing}. Serological studies suggested that human H1N1 and H3N2 viruses could infect dogs, albeit inefficiently, and A/Hong Kong/68-like and influenza B viruses were isolated from dogs in Taiwan in 1971~\cite{chang1976influenza,houser1980evidence}. Swine-origin H1N1 CIV have also been isolated in dogs in China~\cite{chen2018emergence}. Since dogs are susceptible to infection with at least three different influenza A subtypes (H3N8, H3N2 and H1N1) and can also be infected with diverse human influenza viruses, these animals have the potential to act as mixing vessels in which novel viruses are generated by reassortment. Genetic exchange through reassortment allows the formation of chimeric genotypes in which some gene segments are well adapted to human hosts. Compared to strains in which the full genome has evolved in a non-human host, such reassortant viruses may more easily overcome the species barrier to cause an outbreak in humans~\cite{lowen2017constraints}. Thus, the risk for the 2015 H3N2 IL CIV might have been underestimated by IRAT.

%Risk from the H7N9 strains could also have been underestimated
%To date, a total of 1,568 laboratory-confirmed human infections with avian influenza A(H7N9) virus including 616 fatal cases (CFR:39\%) have been reported to WHO since early 2013. The last case of human infection with avian influenza A(H7N9) reported to WHO in the Western Pacific Region was in 2019.
%Of the 1,568 human infections with avian influenza A(H7N9), 33 have reported mutations in the hemagglutinin gene indicating a change to high pathogenicity in poultry. % No increased transmissibility or virulence of the virus within human
% cases has been detected related to the HPAI A(H7N9) virus.
%WHO is continuing to assess the epidemiological situation and will conduct further risk assessments as new information becomes available.
%The number and geographical distribution of human infections with
%avian influenza A(H7N9) viruses in the fifth epidemic wave (1 October 2016 to 30 September 2017) were
%greater than previous waves and the subsequent waves.
%Further sporadic human cases of avian influenza A(H7N9) virus infection are expected in affected and
%possibly neighbouring areas. Should human cases from affected areas travel internationally, their infection
%may be detected in another country during or after arrival. However, if this were to occur,
% The current virus does not have the ability to transmit easily among humans, however the WHO recognizes the  need for close monitoring to identify 
% changes in the virus and transmission behaviour to humans as this may have serious public health impacts. This might be corrobortated by teh fact that the risk from our analysis increase when recomputed with current information, rather than when the IRAT scores were generated  (\qdist has decreases).
% \url{https://www.who.int/docs/default-source/wpro---documents/emergency/surveillance/avian-influenza/ai_20220930.pdf?sfvrsn=22ea0816_18}

% Similar arguments apply for the H1N1 swine strain (A/swine/Shandong/1207/2016) that was given the highest risk score. Our analysis identifies this as an outlier, and suggest the risk was overestimated, which might again be corroborated by the decrease in the \erisk when recomputed with current data on circulating strains (\qdist has increased).



% {\color{magenta}
%   Result requirements:
%   0) Check if changes happen on incorporating the additional term from eqn 9
  
%   1) percent improvement data table augnment with comparison against literature data (in number of average mutational deviation)
  
%   2) table for fig 2
  
%   3) table/figure for risk scores of all h3n2/h1n1 strains in swine collected in 2021/2022
  
%   4) How many sequences were used from which databases. Where is this info described (need this to appear in teh abstract as well)
  
%   5) all qnets should be submitted to zenodo


%   It seems the old results were generated using a different version of quasinet package. Maybe we should indeed regenerate the results to make sure we can answer reviewer questions.

%   Target results:

%   1. HA NA recommendations, northern outhern hemispheres (earlier we used 2 years of data, and chose the time over which we consider last season sequences as different for north vs the south, we did not use separate qnets over north/south)

%   2. check if recommendations match or errors match at least
%   3. check similar approach for latest years
%   4. must complete IRAT prediction table (using geometric mean between ha and na estimates seems to do better. try)


% }


% \section*{Results}

% Our \enet models are inferred using $>98,000XXXX$ Influenza A HA/NA nucleotide sequences from two public databases (NCBI:  \href{https://www.ncbi.nlm.nih.gov/genome/viruses/}{https: //www.ncbi.nlm.nih.gov/genome/viruses/}
% and GISAID:  \href{https://www.gisaid.org/}{https://www.gisaid.org/}
% see SI-Table~\ref{SI-tabseq}).




% We begin by collecting  $>98,000XXXX$ Influenza A HA/NA nucleotide sequences from two public databases (NCBI:  \href{https://www.ncbi.nlm.nih.gov/genome/viruses/}{https: //www.ncbi.nlm.nih.gov/genome/viruses/}
% and GISAID:  \href{https://www.gisaid.org/}{https://www.gisaid.org/}
% see SI-Table~\ref{SI-tabseq}), uncovering a network of  dependencies between individual mutations revealed through subtle variations of the aligned sequences. The representation of these dependencies as a recursive forest of conditional inference trees is referred to as the \enet (see Fig.~\ref{figscheme}). 
% 
% Finally, we define the q-distance (see Eq.~\eqref{q-distance} in Methods) as the square-root of the Jensen-Shannon (JS) divergence~\cite{cover} of these conditional distributions from one sequence to another, averaged over the entire sequence. The q-distance, informed by the dependencies modeled by the inferred {\qnet}s, adapts to the specific organism, allele frequencies, and variations in the background population. Invoking Sanov's theorem on large deviations~\cite{cover}, we show  that the  likelihood of spontaneous change is bounded above and below by a simple exponential function of the  q-distance. 

% The mathematical intuition behind relating the \qdist distance to change-probability  is the same as in the prediction of  a biased outcome when we sequentially toss a fair coin. With an overwhelming probability, such an experiment with a fair coin should result in roughly equal number of heads and tails. However, large deviations, $i.e$, substantial differences from the expected fraction of heads and tails can happen, and the probability of such rare events is quantifiable~\cite{varadhan2010large} with existing theory. We show here that the likelihood of a spontaneous transition of a genomic sequence to a  different variant by random chance may also be similarly bounded, given we have the \qnet as  an estimated model of the evolutionary constraints. As an important distinction from the edit distance, the \qdist  between two sequences may change even if only the background population changes (see SI-Table~\ref{SI-tabex}, where  the distance between two  fixed sequences vary when we vary their collection years). Sequences may have a large \qdist and a small edit distance, and vice versa (although on average the two distances tend to be positively correlated, see SI-Table~\ref{SI-tabcor}).  %Hence for tracking drift in Influenza A, we construct a seasonal \qnet for each sub-type and protein that we consider.


% For predicting future strains, we hypothesized that since the probability of a drift exponentially decreases with an increasing q-distance, the centroid of the strain distribution in our metric will change slowly. If true, the strain selected closest to the ``q''-centroid will be a good approximation of next season's dominant strain.

% For each year of the last two decades, we computed our prediction for the current season's dominant strain as the strain selected from previous seasons closet to the ``q''-centroid of the strain distribution from the previous season (See Eq.~\eqref{dompred} in Methods). We then computed the dominant strain in each season as the centroid of the strain distribution observed in a given season in the classical sense (no. of mutations), since the edit distance metric is widely used and offers a point of comparison between WHO predictions and \qnet predictions. Note that the recommendations for the northern hemisphere are given in February, while that for the southern hemisphere are given at the end of December the previous year, keeping in mind that the flu season in the south begins a few months early, as described in Fig.~\ref{figseasonal}. Finally, we computed the edit distance (no. of mutations) between the dominant strain and the WHO and \qnet predictions.

% Our second application aims to compare emergence risk grades given by the CDC through its Influenza Risk Assessment Tool (IRAT) with results using our q-distance metric. While IRAT uses a combination of 10 weighted risk elements evaluated slowly over the course of several months per strain, we attempt to quantify emergence risk quickly with the q-distance metric. We looked at the same strains that were analyzed by IRAT. For each strain previously analyzed by IRAT, we construct \qnet models for HA and NA segments using all human strains of the same variety circulating in the year prior to risk assessment. For example, the "A/swine/Shandong/1207/2016" strain was assessed by IRAT in July 2020, so we will use human H1N1 strains circulating between July 1, 2019 through June 30, 2020. For sub-types with few human strains (H1N2, H5N1, H5N6, H7N7, H9N2), we only use the upper bound of the date. We then compute the average q-distance between the strain in question and the circulating human strains for both HA and HA segments. Seven of the 23 strains are not included in our comparison due to having zero or too few human strains in the sample space to construct a \qnet; see Supplementary Text, SI-Table~\ref{SI-irattab}. We hypothesize that a lower average q-distance between the strain in question and circulating human strains should correspond to a higher emergence risk. Hence, we expect to see a high negative correlation between q-distance and IRAT grade, which assigns 1 to be the lowest risk and 10 to be the highest risk.

% \subsection*{Concrete Results} 

% We tested the hypothesis of our first application, computing the strain closest to the ``q''-centroid for each flu season and selecting that strain as the prediction for the next season's dominant strain. We performed this analysis on past two decades of sequence data for Influenza A (H1N1 and H3N2) with promising results: the q-distance based prediction demonstrably outperforms WHO recommendations by reducing  the  distance between the predicted and  the dominant  strain (Fig.~\ref{figseasonal}). Recall that we identify the dominant strain to be the one that occurs most frequently, computed as the centroid of the strain distribution observed in a given season in the classical sense (no. of mutations).

% The \qnet single-cluster predictions consistently outperform the WHO recommendations. For H1N1 HA, the \qnet induced recommendation outperforms the WHO suggestion by $>33\%$ on average over the last two decades, and $>71\%$ on average in the last decade. The gains for H1N1 NA over the same time periods are $>15\%$ and $>52\%$, respectively. For H3N2 HA, the \qnet induced recommendation outperforms the WHO suggestion by $>37\%$ on average over the last two decades, and $>40\%$ in the last decade. The gains for H3N2 NA over the same time periods are $>14\%$ and $>15\%$, respectively. Finding multi-cluster predictions has the potential to yield even more improved results, as seen in Fig.~\ref{figseasonal} and SI-Table~\ref{SI-tabrec8} through SI-Table~\ref{SI-tabrec11}.

% The full table of single-cluster results with improvement broken down by hemisphere is given in Table~\ref{tabperf}. Fig.~\ref{figseasonal} illustrates the relative gains computed for both subtypes and the two hemispheres (since the flu season occupy distinct time periods and may have different dominant strains in the northern and southern hemispheres~\cite{boni2008vaccination}). Additional improvement is possible if we recommend multiple strains every season for the vaccine cocktail (Fig.~\ref{figseasonal}e,f,k,l). The details of the specific strain  recommendations made by the \qnet approach for two subtypes (H1N1, H3N2), for two genes (HA, NA) and for the northern and the southern hemispheres over the previous two decades are enumerated in the Supplementary Text in Tables SI-Table~\ref{SI-tabrec0} through SI-Table~\ref{SI-tabrec11}.


%\section*{Discussion}

%\subsubsection*{Summary \& Conclusion}

%In conclusion, we introduce a data-driven distance metric to track subtle deviations in sequences. We show that we can use the q-distance metric to make recommendations for the flu-shot composition, outperforming the WHO's recommendations in relation to the dominant strain. We also show that we can roughly replicate the CDC's IRAT grades for emergence risk of strains not currently circulating among humans in an efficient manner that can be scaled to rank many more strains than is currently done. The ability to predict future flu strains via subtle variations in a limited set of immunologically important residues suggest that the tools developed here could lead to more effective escape-resistant vaccines, which could be essential in preempting and mitigating the next pandemic.

% Thus, in a strict mathematical sense, the distance we propose is not a distance between two strains $x,y$, but a distance between strain $x^A$ ($x$ in a background environment $A$) and a strain  $y^B$ ($y$ in a background environment $B$). Indeed we can show that the distance between the same pair of  strains of \infl HA is different based on if they are collected in 2008 vs in 2009, reflecting that the background environment and circulating diversity changed over the two years.  Thus, our  distance metric  is fundamentally different from measures that exist in the literature. In particular, our mathematical framework leads to the key result that the q-distance is a scaled representation of the log-likelihood of spontaneous jump  between strains. This interpretation is missing in existing tools, and makes way for leveraging the q-distance to model emergence of new strains. Thus, we can predict entirely new sequences -- which differ by a non-trivial number of edits from any observed strain -- that still lead to functional proteins, as demonstrated in our preliminary studies.

%Very recently, two articles have explored the possibility of predicting  pathogenicity from genomic sequences (Mollentze~\cite{mollentze2021identifying}) and forecasting which amongst observed mutations will dominate the circulating population (Maher et al~\cite{maher2021predicting}). These studies provide strong pieces of evidence that forecasting future variants of virus strains is indeed impossible. While their questions overlap with our framework, our approach is distinct and more ambitious. For example,  Mollentze  uses  classical  sequence similarity; extended to include similarity to  human housekeeping genes hoping to identify viruses  evading the human immune system more easily.  The demonstrated  performance is poor (incorrectly tagging all SARS-related coronaviruses as potentially pathogenic), implying un-actionable specificity. On the other hand,  Maher outright assumes mutations to be independent. Features are found manually,  are specific to \hcov, and the authors take  a meta-analysis-esque route, compiling together a ``kitchen-sink'' of features via standard machine learning. Importantly, these approaches  only aim to predict point mutations, and their ability to track a strain through a high-dimensional sequence space is limited.
%Thus, even the question if whether a yet-to-be-seen strain is indeed a valid biological encoding of a virus (which is simpler to determining risk posed by such future variants) cannot answered by our peers, limiting such approaches to analyzing mutations already seen, or strains  already collected.  Additionally, generalizability and actionability is  suspect, given that Maher's features are \hcov specific, and Mollentze's similarity to house-keeping genes might not be universal. Finally, both these  approaches apply to a mutation or a combination of mutations that already exist, and cannot predict new mutations, or new strains.

%\subsubsection*{Limitations \& Conclusion}






\section*{Methods}
\input{method}





\section*{Data Sharing} 

Working software is publicly available at \href{https://pypi.org/project/quasinet/}{https://pypi.org/project/emergenet/}.
Accession numbers of all sequences used, and acknowledgement documentation for GISAID sequences is available as supplementary information.





\subsection*{Data Source}

In this study, we use sequences for the Hemaglutinnin (HA)  and Neuraminidase (NA) for Influenza A (for subtypes H1N1 and H3N2), which are key enablers of cellular entry and exit mechanisms respectively~\cite{mcauley2019influenza}. We use two sequences databases: 1) National Center for Biotechnology Information (NCBI) virus~\cite{hatcher2017virus} and 2) GISAID~\cite{bogner2006global} databases. The former is a community portal for viral sequence data, aiming to increase the usability of data archived in various NCBI repositories. GISAID has a somewhat more restricted user agreement, and use of GISAID data in an analysis requires acknowledgment of the contributions of both the submitting and the originating laboratories (Corresponding acknowledgment tables are included as supplementary information). We collected a total of 98,299 sequences in our analysis, although not all were used due to some being duplicates (see SI-Table~\ref{SI-tabseq}).








% #############################################
% #############################################

% Bibliography
% \bibliography{qnet,BibLib1,bioshock_refs,bioshock,keck}
% \bibliographystyle{vancouver}
\bibliographystyle{naturemag}
\bibliography{allbib}


\clearpage                                                                      
\setcounter{figure}{0}                                    \renewcommand{\figurename}{Extended Data Figure}                               
\setcounter{table}{0}                                     
\renewcommand{\tablename}{Extended Data Table}                                 
\input{extfiguresandtables.tex}


\clearpage
empty


\clearpage



{\color{magenta}

  

{\color{Red1}
  \section*{Discussion}
  In the aftermath of the COVID-19 pandemic that caused one of the most devastating disasters of the past century, a looming question is whether we can prepare for, preempt and mitigate such events in the future. Evolving viruses, whether currently circulating in the human population, or in animal reservoirs that might spillover and attain human-to-human transmission capability, pose an ever-present epidemic risk.  Current surveillance paradigms, while crucial for mapping disease ecosystems, are limited in their ability to address this challenge. Habitat encroachment, climate change, and other ecological factors~\cite{rulli2017nexus,chua2002anthropogenic,childs2004zoonotic} unquestionably drive up the odds of zoonotic spill-overs. Nevertheless, current efforts at tracking these effects have not improved our ability to quantify future risk of emergence~\cite{fair2019viral}. Tracking viral diversity in animal hosts, while important, often does not transparently map to emergence risk.  This is particularly true for \infl, which partly on account of its segmented genome, can easily incorporate genes from multiple strains and emerge as novel human pathogens, and thus harbor a high pandemic potential. While large antigenic shifts in \infl are relatively rare, even the smaller seasonal sequence alterations in cause sufficient variation in the surface proteins to evade existing immunity, and require yearly reformulation of the flu vaccine.

  However, for the  vaccine to be effective, we need to  predict the dominant circulating strain of the upcoming season with sufficient accuracy.
  Currently, the composition of the flu shot is decided at least six months in advance of the seasonal infection peak, and targets three to four  historical strains as recommended by the CDC/WHO, who identify these specific strains by  sampling the current circulation~\cite{agor2018models}, hoping to match the  dominant strain(s) in the upcoming season. A variety of hard-to-model effects hinder this prediction, which, despite observed cross-reactive effects~\cite{tricco2013comparing}, have had  limited vaccine effectiveness in recent years~\cite{cdceff}. Rank-ordering strains which do not yet circulate in humans according to either their spillover risk or their pandemic potential, has proven to be even more difficult [REF]. CDC's current, somewhat subjective, solution to this problem is the Influenza Risk Assessment Tool (IRAT), which  uses a combination of ten weighted risk elements, including  1) properties of the virus, 2) attributes of the population, and 3) ecology and epidemiological characteristics of the virus~\cite{Influenz24:online} that are expert-selected. Evaluating these factors involve several experimental assay for each strain, taking possibly weeks to return the final IRAT  score for a single strain. Thus, we have a scalability problem: with  the current global biosurveillance efforts  collecting tens of thousands of sequences every year, IRAT assessment is simply not fast enough to  preempt a pandemic. }



\section*{Discussion \& Sequence Comparisons}

For further discussion, we looked at our \qnet predictions more closely. Comparing the \qnet inferred strain (QNT) against the one recommended by the WHO, we find: 1) the residues that only the QNT matches correctly with DOM (while the WHO fails) are largely localized within the receptor binding domain (RBD), with $>57\%$ occurring within  the RBD on average (see Fig.~\ref{figseq}a for a specific example), and 2) when the WHO strain deviates from  the QNT/DOM  matched residue, the ``correct'' residue is often replaced in the WHO recommendation with one that has very different side chain, hydropathy  and/or chemical properties (see Fig.~\ref{figseq}b-f), suggesting deviations in recognition characteristics. Combined with the fact that we find circulating strains are almost always within a few edits of the DOM (see SI-Fig.~\ref{SI-figdom}), these observations suggest that hosts vaccinated with the QNT recommendation is more likely to have season-specific antibodies that are more likely to recognize a larger cross-section of the circulating strains.

High season-to-season genomic variation in the key  Influenza capsidic proteins is driven by two opposing influences: 1) the need to conserve function  limiting random mutations, and 2) hyper-variability to escape recognition by neutralizing antibodies. Even a  single residue change in the surface proteins might dramatically alter recognition characteristics, brought about by unpredictable~\cite{carugo2001normalized,righetto2014comparative} changes in local or regional properties such as charge, hydropathy, side chain solvent accessibility~\cite{lee1971interpretation,shrake1973environment,momen2008impact,adamczak2005combining}.

Focusing on the average localization of the QNT to WHO deviations in the HA molecular  structure, the changes are observed to primarily occur in the HA1 sub-unit (see Fig.~\ref{figseq}g-i, HA0 numbering used, other numbering conversions are given in SI-Table~\ref{SI-tabnum}), with the most frequent deviations  occurring around the $\approx 200$ loop, the $\approx 220$ loop, the $\approx 180$ helix, and the $\approx 100$ helix, in addition to some residues in the HA2 sub-unit ($\approx 49$ \& $\approx 124$). Unsurprisingly, the residues we find to be most impacted in the HA1 sub-unit (the globular top of the fusion protein) have been repeatedly implicated in receptor binding interactions~\cite{tzarum2015structure,lazniewski2018structural,garcia2015dynamic}. Thus, we are able to fine tune the future recommendation over the state of the art, largely by modifying residue recommendations around the RBD and  structures affecting recognition dynamics.



}

{\color{Green1}


  It is well known that the influenza viral RNA-polymerase represents the lack of proofreading function. Thus, the integration of faulty nucleotides often occurs during the viral replication process with a rate of $10^{-3}$ to $10^{-4}$, which results in high mutation rates [39,40].



  
  Due to its crucial role in receptor recognition and attachment, IAV HA is considered to be a principal determinant of the host-range. The specificity of the HA of avian influenza viruses is for $\alpha-2,3$ SA receptors found in the intestinal tract of the bird, whereas $\alpha-2,6$ SA receptors are predominantly found in the upper respiratory tract of humans. Recently, it has been shown that mutations in the HA protein alter its receptor-binding preference that allows the highly pathogenic avian H5N1 IAV to transmit between mammals [41]. Therefore, it is not surprising that multiple changes in gene segments of the avian influenza virus could result in its adaptation to humans [1]. On the other hand, owing to having both $\alpha-2,3$ and $\alpha-2,6$  linkages, pigs and several avian species (pheasants, turkeys, quails) may act as mixing vessels and can generate re-assortment viruses [42,43].


  Influenza proteins must evade immune recognition while maintaining their ability to function and interact with host cellular factors [44]. The three mechanisms by which influenza viruses undergo evolutionary change include mutation (antigenic drift), re-assortment (antigenic shift), and, in rare instances, recombination. The different virus lineages are predominantly host specific, but there are periodic exchanges of influenza virus gene segments between species, giving rise to pandemics of disease in humans, lower animals, and birds [45]. Influenza virus evolution proceeds via re-assortment and mutation, and such evolution can influence the host specificity and pathogenicity of these viruses [46]. Genetic variations of influenza A virus lead to possible changes in upcoming epidemiological behavior and may result in human pandemics.


  Significant mutations in antigenic sites resulting from constant point mutations in the influenza virus contribute to the gradual evolution of the virus, leading to antigen migration to produce new influenza virus subtypes to escape the immune pressure of the population [47]. All subtypes of influenza A virus antigenic drift can occur, but such antigenic drift often occurs in the general human influenza. Immune escape can be achieved by mutation in IAV proteins such as HA and/or NA. The minimal structural changes can occur in these surface proteins and so the immune protection of the host (acquired through previous infections or immunization) will no longer be effective against the invading virus. As a consequence, the immune system is unable to identify the newly changed virus variants and the recognition pattern of the antigen-antibody-interaction is not fully functional anymore. In addition, amino acid substitutions in HA protein can change the receptor preference of influenza virus. Some studies have shown that the G186V mutation in HA protein was noted as a potential adaptation of avian H7 to human-type receptors [48,49]. In A/Vietnam/1203/2004 (H5N1) virus, K58I substitution in HA protein is associated with increased viral replication of upper respiratory tracts in mice and ferrets [50]. Remarkably, the K58I substitution combined with a G219S mutation in HA protein increased the overall affinities of binding to  $\alpha-2,3$ and  $\alpha-2,6$ SA of the A/Anhui/1/13 (H7N9) virus [51]. Furthermore, there is a R292K mutation in NA protein in H7N9 virus strains which had been isolated from a patient after drug treatment. This substitution was found to promote drug resistance; in particular, it gave a high resistance to oseltamivir which is the most commonly used anti-influenza drug [52]. Antigenic drifts are the main reason for new variants and cause annual influenza outbreaks. Although these changes may not lead to pandemics, antigenic drift over a period of time can make a strain considerably different from the original pandemic virus.

  It has been confirmed that the long-term evolution of cytotoxic T lymphocyte (CTL) epitopes is associated with CTL-mediated clearance of infection and it is thought that the selection pressures imposed by CTL immunity shape the long-term evolution of IAV [53,54]. Viruses mutate amino acid residues within CTL epitopes to evade CTL recognition [55]. Under certain circumstances, amino acid substitutions occur at the anchoring residues, while in other cases they occur at the T cell receptor contact residues [56]. For instance, mutations at the anchored residues of the CTL epitope have been described in the human leukocyte antigen (HLA)-B* 2705 restricted NP383–391 epitope, which has the R-to-G substitution at position 384 (R384G) [57,58]. This replacement significantly reduced the in vitro virus-specific CTL response in HLA-B* 2705-positive individuals.

  4.2. Re-Assortment
  It has been well recognized that the segmented genome of the influenza virus allows the exchange of RNA segments between genotypically different influenza viruses, resulting in the production of new strains and/or subtypes [67], which is referred to as re-assortment. A pandemic IAV can be produced by transmission from animals to humans or by reconfiguration between avian influenza viruses and human influenza viruses [68]. As the influenza virus has a segmented genome, re-assortment is an important mechanism for generation of the "novel" virus [69]. Thus, re-assortment of the virus achieves a new antigenic pattern known as "antigenic shift". Pandemic influenza emerges as a result of such major genetic changes of IAV. These modifications occur due to mechanistic errors during the replication of viral RNA polymerase, evolutionary pressure, the novel environment of the host, immune pressure, or antiviral drug pressure [70]. Two of the three major human influenza pandemics in the twentieth century (1957 and 1968) and this century (2009) were due to the re-assortment between the human IAV and other host species.


  There is evidence indicating that the HA, NA, and PB1 genes of the H2N2 1957 pandemic strain in addition to the HA and PB1 fragments of the H3N2 1968 pandemic strain are both avian, and the remaining fragments may come directly from 1918 [67]. The first influenza pandemic in this century, the influenza A H1N1 virus, is a re-assortant caused by a multiple mixed recombination between the European H1N1 swine influenza virus, North American H1N2 swine influenza virus, North American avian influenza virus, and H3N2 influenza virus [71].

  In addition to mutation and re-assortment, IAVs still have another relatively rare means of evolution called recombination. Genetic recombination is one of the primary processes that produce the genetic diversity upon which natural selection acts. Recombination in IAVs can occur through two main mechanisms: one is the non-homologous recombination that occurs between two different RNA fragments [81,82]; the other is the controversial homologous recombination, often considered to be absent or very rare, which is thought to participate in template switching while the polymerase is copying the RNA.



  Wild waterfowl and shorebirds belong to the main natural host species of IAV [88]. IAV has been able to establish the successful infection of a variety of animals, including avian and mammalian species, and its evolution has led to the emergence of IAV in human beings for a long time [89]. Since the pandemic outbreak of influenza virus in 1918, the re-assortment of influenza virus has occurred among bird and human viruses. As described above, the re-assortment of influenza viruses has resulted in the pandemic of H2N2 in 1957 and of H3N2 in 1968 [90]. During the year 2009, there was an outbreak of H1N1 in humans that caused the first pandemic of influenza through human transmission in the 21st century [91].

  Usually, an avian influenza subtype does not infect humans and a human influenza subtype is unable to infect the birds. However, swine acts as a virus mixer vessel, leading to the generation of new influenza viruses, which can infect both humans and poultry. The mutation and re-assortment of the IAV genome are susceptible to forming new subtypes of influenza virus that may result in widely propagated and destructive pandemics due to the lack of immunity to the emerging pathogen [67]. For example, the outbreak of H5N1 avian influenza in 1997 and the outbreak of H1N1 swine influenza in 2009 caused great panic and brought serious economic losses to the breeding industry.



}


\section*{Brief Methods}
\DQS{review more craefully phenotypic info used in teh literature and why it is claimed to be necessary in those papers. Why dont we need it}
A key barrier to making progress  on both the problems cited above, namely predicting the dominant strain(s) in seasonal flu, as well as estimating the numerical odds of an animal strain  to spillover and attain HH capability,  is our limited understanding of the emergent dependencies across individual mutations that  constrain evolutionary trajectories. Thus, to the best of our knowledge, the state of the art has no tools to estimate the numerical likelihood of specific mutations in the future, and in general  the likelihood of a wild strain spontaneously giving rise to another by random chance. Currently, this likelihood is often qualitatively equated to sequence similarity, which is measured by the number of mutations it takes to change one strain to another. However, the odds of one sequence mutating to another is not just a function of how many mutations separate them, but also of how specific mutations incrementally affect fitness. Ignoring the constraints arising from the need to conserve function makes any assessment of the mutation likelihood open to subjective bias. Here, we show that a precise calculation is possible when sequence similarity is evaluated via a new biologically-aware metric, which we call the \textit{q-distance}.

Some recent efforts have recognized this gap, and have attempted to predict future dominant strain by incorporating other phenotypic details. 

As an applications of the q-distance, we show that we can improve seasonal forecasts for the future dominant circulating strain by learning from the mutational patterns of key surface proteins: Hemaglutinnin (HA) and Neuraminidase (NA) for Influenza A. We outperform the WHO's recommendations for the flu-shot composition consistently over past two decades, measured as the number of mutations that separate the predicted from the dominant circulating strain in each season. Our recommendations repeatedly end up closer to the dominant circulating strain, illustrating the potential of our approach to correctly predict evolutionary trajectories. 

We also show that this new metric allows us to  assess the risk posed by novel strains  effectively and quickly. We compare q-distance results to the CDC's Influenza Risk Assessment Tool (IRAT)~\cite{Influenz24:online}, which gives a grade between 1-10 for emergence risk and public health impact to Influenza A viruses not currently circulating among humans. Our results show strong negative correlations between IRAT emergence risk grades and q-distances to the nearest human strains to the strains in question. However, while IRAT may take weeks to analyze a single strain -- hence the small number of analyzed strains -- q-analysis can be done within milliseconds for each new strain. Moreover, q-analysis only requires sequence data, while IRAT requires information for 10 risk elements, grouped into three categories: 1) properties of the virus, 2) attributes of the population, and 3) ecology and epidemiology of the virus~\cite{Influenz24:online}. Thus, our method could potentially be a low-cost, efficient substitute to IRAT, which could used at scale to rank the risk of emergence of non-circulating strains.

{\color{Red1} Discussion? 
  Thus, the tool proposed in this study  can  profoundly impact  bio-surveillance strategies. The ability to rank newly collected strains by risk at scale, allows actionable estimates of  pandemic risks via  quantifying the odds of a particular strain spilling into to the human population. Additionally,  for strains already circulating in humans, our tools can estimate the odds  of specific  new mutan varaints emerging, and their ability to  escape current vaccines. %This study potentially represents an important step forward in modeling emerging pathogens, with uncharted impact on science and health, particularly as we prepare for the aftermath of \cov.
}

\end{document}
% LocalWords:  Neuraminidase subtype
