{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "import copy\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import Levenshtein\n",
    "from tqdm import tqdm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 39,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "1773"
      ]
     },
     "execution_count": 39,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmpf=pd.read_csv('./combined_results.csv')\n",
    "tmpf[tmpf.predicted_emergence_score>6].index.size"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 40,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>subtype</th>\n",
       "      <th>ha_accession</th>\n",
       "      <th>na_accession</th>\n",
       "      <th>ha</th>\n",
       "      <th>na</th>\n",
       "      <th>HA_Avg_Qdist</th>\n",
       "      <th>ha_variance</th>\n",
       "      <th>NA_Avg_Qdist</th>\n",
       "      <th>na_variance</th>\n",
       "      <th>Geometric_Mean</th>\n",
       "      <th>predicted_impact_score</th>\n",
       "      <th>predicted_emergence_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A/swine/Missouri/A02524459/2020</td>\n",
       "      <td>H1N1</td>\n",
       "      <td>EPI1765551</td>\n",
       "      <td>EPI1765552</td>\n",
       "      <td>MKAILVVLLYTFTTANADTLCIGYHANNSTDTVDTVLEKNVTVTHS...</td>\n",
       "      <td>MNANQRIIAIGTVCMIVGIISLLLQIGNIVSLWISHSIQTGWENHT...</td>\n",
       "      <td>0.014602</td>\n",
       "      <td>0.000009</td>\n",
       "      <td>0.029655</td>\n",
       "      <td>0.000013</td>\n",
       "      <td>0.020809</td>\n",
       "      <td>6.379932</td>\n",
       "      <td>6.391580</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A/swine/Ohio/A02524448/2020</td>\n",
       "      <td>H1N1</td>\n",
       "      <td>EPI1765553</td>\n",
       "      <td>EPI1765554</td>\n",
       "      <td>MKAVLVVLLYTFTTANADTLCIGYHANNSTDTVDTVLEKNVTVTHS...</td>\n",
       "      <td>MNTNQRIITIGTVCLIVGIISLLLQIGNIVSLWVSHSIQTRWENHT...</td>\n",
       "      <td>0.019032</td>\n",
       "      <td>0.000012</td>\n",
       "      <td>0.030905</td>\n",
       "      <td>0.000014</td>\n",
       "      <td>0.024253</td>\n",
       "      <td>6.330661</td>\n",
       "      <td>6.329795</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A/swine/South_Dakota/A02524453/2020</td>\n",
       "      <td>H1N1</td>\n",
       "      <td>EPI1765555</td>\n",
       "      <td>EPI1765556</td>\n",
       "      <td>MKAILVVLLYTFTTANADTLCIGYHANNSTDTVDTVLEKNVTVTHS...</td>\n",
       "      <td>MNTNQRIITIGTVCMIVGIISLLLQIGNIVSLWISHSIQIEWENHT...</td>\n",
       "      <td>0.015378</td>\n",
       "      <td>0.000012</td>\n",
       "      <td>0.030887</td>\n",
       "      <td>0.000013</td>\n",
       "      <td>0.021794</td>\n",
       "      <td>6.365780</td>\n",
       "      <td>6.373840</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A/swine/Indiana/A02524458/2020</td>\n",
       "      <td>H1N1</td>\n",
       "      <td>EPI1765559</td>\n",
       "      <td>EPI1765560</td>\n",
       "      <td>MKAVLVVLLYTFTTANADTLCIGYHANNSTDTVDTVLEKNVTVTHS...</td>\n",
       "      <td>MNTNQRIITIGTVCLIVGIISLLLQIGNIVSLWVSHSIQTRWENHT...</td>\n",
       "      <td>0.019020</td>\n",
       "      <td>0.000012</td>\n",
       "      <td>0.030905</td>\n",
       "      <td>0.000014</td>\n",
       "      <td>0.024245</td>\n",
       "      <td>6.330775</td>\n",
       "      <td>6.329937</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A/swine/Iowa/A02479424/2020</td>\n",
       "      <td>H1N1</td>\n",
       "      <td>EPI1766606</td>\n",
       "      <td>EPI1766607</td>\n",
       "      <td>MKAVLVVMLYTFTTANADTLCIGYHANNSTDTVDTVLEKNVTVTHS...</td>\n",
       "      <td>MNTNQRIITIGTVCLIVGIISLLLQIGNIVSLWVSHSIQTRWENHT...</td>\n",
       "      <td>0.016578</td>\n",
       "      <td>0.000012</td>\n",
       "      <td>0.034251</td>\n",
       "      <td>0.000016</td>\n",
       "      <td>0.023829</td>\n",
       "      <td>6.336689</td>\n",
       "      <td>6.337357</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6061</th>\n",
       "      <td>A/chicken/Guangdong/F94/2022</td>\n",
       "      <td>H9N2</td>\n",
       "      <td>EPI2167482</td>\n",
       "      <td>EPI2167484</td>\n",
       "      <td>MKTVSLITILLAATVSNADKICIGYQSTNSTETVDTLTENNVPVTH...</td>\n",
       "      <td>MNPNQKITAIGSVSLIIAIICLLMQIAILTTTMTLHFGQKECSNPL...</td>\n",
       "      <td>0.021926</td>\n",
       "      <td>0.000166</td>\n",
       "      <td>0.073419</td>\n",
       "      <td>0.011472</td>\n",
       "      <td>0.040122</td>\n",
       "      <td>6.111827</td>\n",
       "      <td>6.054706</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6062</th>\n",
       "      <td>A/chicken/Guangdong/F99/2022</td>\n",
       "      <td>H9N2</td>\n",
       "      <td>EPI2167503</td>\n",
       "      <td>EPI2167506</td>\n",
       "      <td>MEAVSLITILLAATVSNADKICIGYQSTNSTETVDTLTENNVPVTH...</td>\n",
       "      <td>MNPNQKITAIGSVSLIIAIICLLMQIAILTTTMTLHFRQKECSNPS...</td>\n",
       "      <td>0.018988</td>\n",
       "      <td>0.000092</td>\n",
       "      <td>0.052028</td>\n",
       "      <td>0.014803</td>\n",
       "      <td>0.031431</td>\n",
       "      <td>6.230001</td>\n",
       "      <td>6.203400</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6063</th>\n",
       "      <td>A/chicken/Egypt/Menoufia/2021</td>\n",
       "      <td>H9N2</td>\n",
       "      <td>EPI2175832</td>\n",
       "      <td>EPI2175833</td>\n",
       "      <td>MEVIPLMIMLLLVTTNNADKICIGHQSTNSTETVDTLTETGVPVTH...</td>\n",
       "      <td>MNPNQRIIALGSASLTVATICLLIQIAILATTMTLHFKQNEYANTS...</td>\n",
       "      <td>0.055346</td>\n",
       "      <td>0.000136</td>\n",
       "      <td>0.461097</td>\n",
       "      <td>0.015276</td>\n",
       "      <td>0.159750</td>\n",
       "      <td>4.898218</td>\n",
       "      <td>4.492293</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6064</th>\n",
       "      <td>A/chicken/South_Korea/SL20/2020</td>\n",
       "      <td>H9N2</td>\n",
       "      <td>EPI2176583</td>\n",
       "      <td>EPI2176585</td>\n",
       "      <td>METVSLITILVVATVSSADKICIGYQSTNSTETVDTLTENNVPVTH...</td>\n",
       "      <td>MNPNQKITAIGSVSLIIAIICLLMQIAILTTTMTFHFGQKECSNPS...</td>\n",
       "      <td>0.029999</td>\n",
       "      <td>0.000109</td>\n",
       "      <td>0.051293</td>\n",
       "      <td>0.014672</td>\n",
       "      <td>0.039227</td>\n",
       "      <td>6.123818</td>\n",
       "      <td>6.069810</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6065</th>\n",
       "      <td>A/Greylag_Goose/South_Korea/SW21/2021</td>\n",
       "      <td>H9N2</td>\n",
       "      <td>EPI2188134</td>\n",
       "      <td>EPI2188136</td>\n",
       "      <td>METIALVAILLMATASNADKICIGYQSTNSTETVDTLIENNVPVTH...</td>\n",
       "      <td>MNPNQKIITIGSVSLTIAIVCFLMQIAILATTITLHFKQNECSIPS...</td>\n",
       "      <td>0.046408</td>\n",
       "      <td>0.000060</td>\n",
       "      <td>0.461309</td>\n",
       "      <td>0.015407</td>\n",
       "      <td>0.146316</td>\n",
       "      <td>4.996130</td>\n",
       "      <td>4.622751</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6066 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                         id subtype ha_accession na_accession  \\\n",
       "0           A/swine/Missouri/A02524459/2020    H1N1   EPI1765551   EPI1765552   \n",
       "1               A/swine/Ohio/A02524448/2020    H1N1   EPI1765553   EPI1765554   \n",
       "2       A/swine/South_Dakota/A02524453/2020    H1N1   EPI1765555   EPI1765556   \n",
       "3            A/swine/Indiana/A02524458/2020    H1N1   EPI1765559   EPI1765560   \n",
       "4               A/swine/Iowa/A02479424/2020    H1N1   EPI1766606   EPI1766607   \n",
       "...                                     ...     ...          ...          ...   \n",
       "6061           A/chicken/Guangdong/F94/2022    H9N2   EPI2167482   EPI2167484   \n",
       "6062           A/chicken/Guangdong/F99/2022    H9N2   EPI2167503   EPI2167506   \n",
       "6063          A/chicken/Egypt/Menoufia/2021    H9N2   EPI2175832   EPI2175833   \n",
       "6064        A/chicken/South_Korea/SL20/2020    H9N2   EPI2176583   EPI2176585   \n",
       "6065  A/Greylag_Goose/South_Korea/SW21/2021    H9N2   EPI2188134   EPI2188136   \n",
       "\n",
       "                                                     ha  \\\n",
       "0     MKAILVVLLYTFTTANADTLCIGYHANNSTDTVDTVLEKNVTVTHS...   \n",
       "1     MKAVLVVLLYTFTTANADTLCIGYHANNSTDTVDTVLEKNVTVTHS...   \n",
       "2     MKAILVVLLYTFTTANADTLCIGYHANNSTDTVDTVLEKNVTVTHS...   \n",
       "3     MKAVLVVLLYTFTTANADTLCIGYHANNSTDTVDTVLEKNVTVTHS...   \n",
       "4     MKAVLVVMLYTFTTANADTLCIGYHANNSTDTVDTVLEKNVTVTHS...   \n",
       "...                                                 ...   \n",
       "6061  MKTVSLITILLAATVSNADKICIGYQSTNSTETVDTLTENNVPVTH...   \n",
       "6062  MEAVSLITILLAATVSNADKICIGYQSTNSTETVDTLTENNVPVTH...   \n",
       "6063  MEVIPLMIMLLLVTTNNADKICIGHQSTNSTETVDTLTETGVPVTH...   \n",
       "6064  METVSLITILVVATVSSADKICIGYQSTNSTETVDTLTENNVPVTH...   \n",
       "6065  METIALVAILLMATASNADKICIGYQSTNSTETVDTLIENNVPVTH...   \n",
       "\n",
       "                                                     na  HA_Avg_Qdist  \\\n",
       "0     MNANQRIIAIGTVCMIVGIISLLLQIGNIVSLWISHSIQTGWENHT...      0.014602   \n",
       "1     MNTNQRIITIGTVCLIVGIISLLLQIGNIVSLWVSHSIQTRWENHT...      0.019032   \n",
       "2     MNTNQRIITIGTVCMIVGIISLLLQIGNIVSLWISHSIQIEWENHT...      0.015378   \n",
       "3     MNTNQRIITIGTVCLIVGIISLLLQIGNIVSLWVSHSIQTRWENHT...      0.019020   \n",
       "4     MNTNQRIITIGTVCLIVGIISLLLQIGNIVSLWVSHSIQTRWENHT...      0.016578   \n",
       "...                                                 ...           ...   \n",
       "6061  MNPNQKITAIGSVSLIIAIICLLMQIAILTTTMTLHFGQKECSNPL...      0.021926   \n",
       "6062  MNPNQKITAIGSVSLIIAIICLLMQIAILTTTMTLHFRQKECSNPS...      0.018988   \n",
       "6063  MNPNQRIIALGSASLTVATICLLIQIAILATTMTLHFKQNEYANTS...      0.055346   \n",
       "6064  MNPNQKITAIGSVSLIIAIICLLMQIAILTTTMTFHFGQKECSNPS...      0.029999   \n",
       "6065  MNPNQKIITIGSVSLTIAIVCFLMQIAILATTITLHFKQNECSIPS...      0.046408   \n",
       "\n",
       "      ha_variance  NA_Avg_Qdist  na_variance  Geometric_Mean  \\\n",
       "0        0.000009      0.029655     0.000013        0.020809   \n",
       "1        0.000012      0.030905     0.000014        0.024253   \n",
       "2        0.000012      0.030887     0.000013        0.021794   \n",
       "3        0.000012      0.030905     0.000014        0.024245   \n",
       "4        0.000012      0.034251     0.000016        0.023829   \n",
       "...           ...           ...          ...             ...   \n",
       "6061     0.000166      0.073419     0.011472        0.040122   \n",
       "6062     0.000092      0.052028     0.014803        0.031431   \n",
       "6063     0.000136      0.461097     0.015276        0.159750   \n",
       "6064     0.000109      0.051293     0.014672        0.039227   \n",
       "6065     0.000060      0.461309     0.015407        0.146316   \n",
       "\n",
       "      predicted_impact_score  predicted_emergence_score  \n",
       "0                   6.379932                   6.391580  \n",
       "1                   6.330661                   6.329795  \n",
       "2                   6.365780                   6.373840  \n",
       "3                   6.330775                   6.329937  \n",
       "4                   6.336689                   6.337357  \n",
       "...                      ...                        ...  \n",
       "6061                6.111827                   6.054706  \n",
       "6062                6.230001                   6.203400  \n",
       "6063                4.898218                   4.492293  \n",
       "6064                6.123818                   6.069810  \n",
       "6065                4.996130                   4.622751  \n",
       "\n",
       "[6066 rows x 13 columns]"
      ]
     },
     "execution_count": 40,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmpf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# parameters\n",
    "threshold=6  # draw tree above this emergenec risk threshold\n",
    "CONSTRUCT_PHYLO=False  # use constructed tree (tree construction takes hours)\n",
    "OUTPUT_DIR='./'\n",
    "PHYLO_TREE_DIR='./'\n",
    "PHYLO_DIR='./'\n",
    "COMBINED_RESULTS='./combined_results.csv'\n",
    "#CONSTRUCT_PHYLO=False\n",
    "num_collapsed=19  # number of mutations within which leaves are collapsed\n",
    "VERBOSE=False"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# class definition to hold multi sequence data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "class SeqInfo(object):\n",
    "    \"\"\"Holds information regarding the sequence.\n",
    "    \n",
    "    \"\"\"\n",
    "    def __init__(self, seq, \n",
    "                 protein,\n",
    "                 accession,\n",
    "                 subtype=None,\n",
    "                 id=None,\n",
    "                 name=None,\n",
    "                 host=None, \n",
    "                 date=None, \n",
    "                 erisk=None,\n",
    "                 irisk=None,\n",
    "                 risk_flag=None,\n",
    "                 country=None):\n",
    "        self.name = name\n",
    "        self.id = id\n",
    "        self.protein=protein\n",
    "        self.subtype=subtype        \n",
    "        self.seq = seq\n",
    "        self.accession = accession \n",
    "        self.host = host\n",
    "        self.date = date\n",
    "        self.erisk = erisk\n",
    "        self.irisk = irisk\n",
    "        self.risk_flag = risk_flag\n",
    "        self.country = country\n",
    "        \n",
    "class MultipleSeqInfo(object):\n",
    "    \"\"\"Holds information regarding multiple sequences.\n",
    "    \n",
    "    Args:\n",
    "        dataframe (pandas.DataFrame): list of records parsed from NCBI\n",
    "        accessionname (str): column name for accession id\n",
    "        proteinname (str): protein name \n",
    "        risk_threshold (float): emergence risk threshold to compute distance matrix\n",
    "    \"\"\"\n",
    "    def __init__(self,\n",
    "                 dataframe,\n",
    "                 accessionname,\n",
    "                 proteinname,\n",
    "                 risk_threshold=6.2):\n",
    "        \n",
    "        self.seq_infos = {}\n",
    "        self.risk_threshold = risk_threshold\n",
    "        for i in np.arange(dataframe.index.size):\n",
    "            record=dataframe.iloc[i,:]\n",
    "            seqinfo = SeqInfo(\n",
    "                name=record.id,\n",
    "                seq=record[proteinname], \n",
    "                protein=proteinname,\n",
    "                accession=record[accessionname],\n",
    "                subtype=record.subtype,\n",
    "                erisk=record.predicted_emergence_score,\n",
    "                irisk=record.predicted_impact_score,\n",
    "                risk_flag = record.predicted_emergence_score > self.risk_threshold,\n",
    "                host=None,\n",
    "                date=None,\n",
    "                country=None)\n",
    "            #print(record.predicted_emergence_score > self.risk_threshold)\n",
    "            self.seq_infos[seqinfo.accession] = seqinfo\n",
    "            \n",
    "    \n",
    "    def compute_L_distance_matrix(self):\n",
    "        highriskseq = pd.DataFrame.from_dict({key:val.seq \n",
    "                                              for (key,val) in self.seq_infos.items() \n",
    "                                              if val.risk_flag},orient='index',columns=['seq'])\n",
    "        num=highriskseq.index.size\n",
    "        d=np.zeros([num,num])\n",
    "        for x in tqdm(np.arange(num*num)):\n",
    "            j=x//num\n",
    "            i=x-num*j\n",
    "            if i > j:\n",
    "                d[i,j] = Levenshtein.distance(highriskseq.seq.values[i],\n",
    "                                                  highriskseq.seq.values[j])\n",
    "        ds=pd.DataFrame(d)        \n",
    "        ds=(ds+ds.transpose())\n",
    "        ds.columns=highriskseq.index.values\n",
    "        self.highriskdistancematrix=ds.copy()\n",
    "        \n",
    "        self.highriskdistancematrix.to_csv('dm'+str(self.risk_threshold)+'.csv',index=None)\n",
    "        return \n",
    "    \n",
    "    \n",
    "    def accessions_to_subtype(self, accessions):\n",
    "        \"\"\"Create a dictionary mapping the accession to the host.\n",
    "        \"\"\"\n",
    "        \n",
    "        subtypes = []\n",
    "        for accession in accessions:\n",
    "            seqinfo = self.seq_infos[accession]\n",
    "            subtypes.append(seqinfo.subtype)\n",
    "            \n",
    "        return subtypes\n",
    "\n",
    "    def accessions_to_host(self, accessions):\n",
    "        \"\"\"Create a dictionary mapping the accession to the host.\n",
    "        \"\"\"\n",
    "        \n",
    "        hosts = []\n",
    "        for accession in accessions:\n",
    "            seqinfo = self.seq_infos[accession]\n",
    "            hosts.append(seqinfo.host)\n",
    "        return hosts\n",
    "           \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "df=pd.read_csv(COMBINED_RESULTS,index_col=0).reset_index()\n",
    "#ALLinfoHA=MultipleSeqInfo(df.reset_index(),'ha_accession','ha',risk_threshold=6.054)\n",
    "ALLinfoHA=MultipleSeqInfo(df.reset_index(),'ha_accession','ha',risk_threshold=threshold)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "H1N1    906\n",
       "H3N2    821\n",
       "H9N2     36\n",
       "H7N9     10\n",
       "Name: subtype, dtype: int64"
      ]
     },
     "execution_count": 6,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[df.predicted_emergence_score>threshold].subtype.value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "scrolled": true
   },
   "source": [
    "# generate distance matrix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████| 3143529/3143529 [01:16<00:00, 40963.36it/s]\n"
     ]
    }
   ],
   "source": [
    "ALLinfoHA.compute_L_distance_matrix()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# tree construction from distance matrix using biopython"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio.Phylo import TreeConstruction\n",
    "from Bio import Phylo\n",
    "from Bio.SeqRecord import SeqRecord\n",
    "from Bio.Seq import Seq\n",
    "from Bio.Align import MultipleSeqAlignment\n",
    "from Bio import Entrez\n",
    "from Bio import SeqIO\n",
    "\n",
    "\n",
    "def load_dm(file_, upper_diag=True):\n",
    "    \"\"\"Load the distance matrix. \n",
    "    \n",
    "    Also, do some preprocessing. \n",
    "    \"\"\"\n",
    "    \n",
    "    df = pd.read_csv(file_)\n",
    "    #df.set_index('Unnamed: 0', inplace=True)\n",
    "    #assert np.all(df.columns == df.index)\n",
    "    \n",
    "    # drop duplicate columns after reading csv\n",
    "    #df = df.loc[:, ~df.columns.str.replace(\"(\\.\\d+)$\", \"\").duplicated()]\n",
    "    \n",
    "    if upper_diag:\n",
    "        df = df + df.T\n",
    "    return df\n",
    "\n",
    "def save_tree(tree, file_name, save_type='xml'):\n",
    "    \"\"\"Saved the created phylogenetic tree.\"\"\"\n",
    "    \n",
    "    if save_type == 'pickle':\n",
    "        graph = Phylo.to_networkx(tree)\n",
    "        save_pickled(graph, file_name)\n",
    "    elif save_type == 'xml':\n",
    "        Phylo.write(tree, file_name, 'phyloxml')\n",
    "    else:\n",
    "        raise ValueError('Not a correct save type.')\n",
    "    \n",
    "def pandas_dm_to_biopython_dm(dm):\n",
    "    \"\"\"Convert the pandas distance matrix to the biopython distance matrix.\n",
    "    \n",
    "    Returns:\n",
    "        biopython distance matrix\n",
    "    \"\"\"\n",
    "    \n",
    "    accessions = dm.columns\n",
    "    bio_dm = []\n",
    "    for i, accession in enumerate(accessions):\n",
    "        bio_dm.append(list(dm.iloc[i, :i+1].values))\n",
    "        \n",
    "    bio_dm = TreeConstruction._DistanceMatrix(\n",
    "        list(dm.columns), \n",
    "        bio_dm)\n",
    "    \n",
    "    return bio_dm\n",
    "\n",
    "def distance_matrix_to_phylo_tree(dm, outfile=None):\n",
    "    \"\"\"Create a phylogenetic tree from the distance matrix.\"\"\"\n",
    "    \n",
    "    dm = pandas_dm_to_biopython_dm(dm)\n",
    "    \n",
    "    treeConstructor = TreeConstruction.DistanceTreeConstructor()\n",
    "    tree = treeConstructor.nj(dm)\n",
    "    \n",
    "    if outfile is not None:\n",
    "        save_tree(tree, outfile)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# ete3 function, not all of these are used here"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "from ete3 import Tree, TreeStyle\n",
    "from ete3 import Phyloxml\n",
    "from ete3 import AttrFace, faces, Tree, NodeStyle, TreeStyle\n",
    "\n",
    "def load_pickled(file_name):\n",
    "    with open(file_name, 'rb') as f:\n",
    "        return pickle.load(f, encoding='latin')\n",
    "\n",
    "\n",
    "def get_farthest_node(tree, sequence):\n",
    "    return (tree&sequence).get_farthest_node()\n",
    "\n",
    "def get_all_accessions_from_tree(tree):\n",
    "    return [leaf_node.name for leaf_node in tree.get_leaves()]\n",
    "\n",
    "def remove_certain_hosts_from_tree(tree, hosts):\n",
    "    \"\"\"Remove leaf nodes if the host of that leaf is in `hosts`\"\"\"\n",
    "    \n",
    "    tree = copy.deepcopy(tree)\n",
    "    \n",
    "    removed_accessions = []\n",
    "    for leaf_node in tree.get_leaves():\n",
    "        if leaf_node.host in hosts:\n",
    "            leaf_node.detach()\n",
    "            \n",
    "    return tree\n",
    "\n",
    "def set_midpoint_outgroup(tree):\n",
    "    tree.set_outgroup(tree.get_midpoint_outgroup())\n",
    "\n",
    "\n",
    "def load_tree(filename, type_='phyloxml'):\n",
    "    \"\"\"Load saved phylogenetic tree.\n",
    "    \"\"\"\n",
    "    \n",
    "    if type_ == 'phyloxml':\n",
    "        project = Phyloxml()\n",
    "        project.build_from_file(filename)\n",
    "\n",
    "        for tree in project.get_phylogeny():\n",
    "            break\n",
    "\n",
    "        t=tree\n",
    "        \n",
    "    elif type_ == 'newick':\n",
    "        t = Tree(filename, format=1)\n",
    "    else:\n",
    "        raise ValueError('Not a correct type.')\n",
    "    \n",
    "    return t"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# construct tree from distance matrix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "#CONSTRUCT_PHYLO=True\n",
    "if CONSTRUCT_PHYLO:\n",
    "    ALL_dm_ldistance = load_dm(\n",
    "        OUTPUT_DIR + 'dm'+str(threshold)+'.csv', \n",
    "        upper_diag=False)\n",
    "    \n",
    "    distance_matrix_to_phylo_tree(\n",
    "        ALL_dm_ldistance, PHYLO_TREE_DIR + 'ldistance'+str(threshold)+'.xml')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# convert phyloxml tree to newick tree to manipulate trees"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "Phylo.convert(\n",
    "    PHYLO_DIR + 'ldistance'+str(threshold)+'.xml','phyloxml',\n",
    "    PHYLO_DIR + 'ldistance'+str(threshold)+'.nhx','newick')\n",
    "\n",
    "ltree = load_tree(\n",
    "    PHYLO_DIR + 'ldistance'+str(threshold)+'.nhx',\n",
    "    type_='newick')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# label nodes in tree to add other attributes like subtype risk etc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "def bandify(val,min=6.05,max=6.71):\n",
    "    maptoten=int(np.ceil(((val-min)/(max-min))*10))\n",
    "    return ' '+u'\\u2580'*maptoten\n",
    "\n",
    "def label_nodes(\n",
    "        tree, \n",
    "        recordinfo):\n",
    "    \"\"\"Label the nodes of the tree.\n",
    "    \n",
    "    We label nodes on whether:\n",
    "        it is covid19\n",
    "    \"\"\"\n",
    "    \n",
    "    tree = copy.deepcopy(tree)\n",
    "    \n",
    "    for node in tree:\n",
    "        name = node.name      \n",
    "        node.subtype = recordinfo.seq_infos[name].subtype\n",
    "        node.erisk =recordinfo.seq_infos[name].erisk\n",
    "        node.id = recordinfo.seq_infos[name].name + bandify(recordinfo.seq_infos[name].erisk,min=threshold)\n",
    "        if VERBOSE:\n",
    "            print(node.name,node.subtype,node.id,node.erisk)\n",
    "    return tree"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# construct labelled tree"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "labelled_tree=label_nodes(\n",
    "    ltree, ALLinfoHA)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# functions to collapse simialr leaves"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "def prune_nodes(t):\n",
    "    # collapsed nodes are labeled, so you locate them and prune them\n",
    "    for n in t.search_nodes(collapsed=True):\n",
    "        for ch in n.get_children():\n",
    "            ch.detach()\n",
    "            \n",
    "            \n",
    "def mean(array):\n",
    "    return sum(array)/float(len(array))\n",
    "\n",
    "def cache_distances(tree):\n",
    "    ''' precalculate distances of all nodes to the root''' \n",
    "    node2rootdist = {tree:0}\n",
    "    for node in tree.iter_descendants('preorder'):\n",
    "        node2rootdist[node] = node.dist + node2rootdist[node.up]\n",
    "    return node2rootdist\n",
    "\n",
    "def closest_node(node, node2tips, root_distance):\n",
    "    \"\"\"Find the closest node.\"\"\"\n",
    "    \n",
    "    tips = []\n",
    "    distances = []\n",
    "    for tip in node2tips[node]:\n",
    "        distances.append(root_distance[tip]-root_distance[node])\n",
    "        tips.append(tip)\n",
    "        #     index = np.argmin([root_distance[tip]-root_distance[node] for tip in node2tips[node]])\n",
    "    index = np.argmin(distances)\n",
    "    return tips[index]\n",
    "\n",
    "def riskiest_node(node, node2tips):\n",
    "    \"\"\"Find the closest node.\"\"\"\n",
    "    \n",
    "    tips = []\n",
    "    risks = []\n",
    "    for tip in node2tips[node]:\n",
    "        risks.append(tip.erisk)\n",
    "        tips.append(tip)\n",
    "        #     index = np.argmin([root_distance[tip]-root_distance[node] for tip in node2tips[node]])\n",
    "    index = np.argmax(risks)\n",
    "    return tips[index]\n",
    "\n",
    "def collapse(tree, min_dist,AllrecordInfo):\n",
    "    # cache the tip content of each node to reduce the number of times the tree is traversed\n",
    "    \n",
    "    tree = copy.deepcopy(tree)\n",
    "    \n",
    "    node2tips = tree.get_cached_content()\n",
    "    root_distance = cache_distances(tree)\n",
    "\n",
    "    for node in tree.get_descendants('preorder'):\n",
    "        if not node.is_leaf():\n",
    "            avg_distance_to_tips = mean([root_distance[tip]-root_distance[node]\n",
    "                                         for tip in node2tips[node]])\n",
    "            if VERBOSE:\n",
    "                print(avg_distance_to_tips)\n",
    "            if avg_distance_to_tips < min_dist:\n",
    "                # do whatever, ete support node annotation, deletion, labeling, etc.\n",
    "            \n",
    "                #closest_name = closest_node(node, node2tips, root_distance).name\n",
    "                closest_name = riskiest_node(node, node2tips).name\n",
    "                node.subtype = AllrecordInfo.seq_infos[closest_name].subtype\n",
    "                node.id = AllrecordInfo.seq_infos[closest_name].name + bandify(AllrecordInfo.seq_infos[closest_name].erisk,min=threshold)\n",
    "                node.name = '%s (%g)' %(closest_name,avg_distance_to_tips)\n",
    "                \n",
    "            \n",
    "                node.add_features(collapsed=True)\n",
    "\n",
    "                # set drawing attribute so they look collapsed when displayed with tree.show()\n",
    "                node.img_style['draw_descendants'] = False\n",
    "\n",
    "    return tree"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "# collapse leaved"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "num_collapsed=26\n",
    "ltree_collapsed = collapse(\n",
    "    labelled_tree, \n",
    "    min_dist=num_collapsed, \n",
    "    AllrecordInfo=ALLinfoHA)\n",
    "\n",
    "prune_nodes(ltree_collapsed)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# code for actual rendering"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {},
   "outputs": [],
   "source": [
    "# COLBAT='DarkRed'\n",
    "# COLRAT='SteelBlue'\n",
    "COLH3N2='Blue'\n",
    "COLH1N1='DarkRed'\n",
    "COLH7N9='DarkGreen'\n",
    "COLH9N2='#BD890F'\n",
    "COLDEF='BLACK'\n",
    "\n",
    "FS=50\n",
    "PW=10\n",
    "\n",
    "\n",
    "def nodeAttribConstruct(color, node):\n",
    "    N = AttrFace(\n",
    "        \"id\", fsize=FS, \n",
    "        text_prefix=\" \",penwidth=PW,ftype='Arial',\n",
    "        fgcolor=color,fstyle='bold')\n",
    "    faces.add_face_to_node(N, node, 1, position=\"branch-right\")\n",
    "    return N\n",
    "\n",
    "def layout(node):\n",
    "    if node.is_leaf():\n",
    "        if  node.subtype == 'H1N1':\n",
    "            N = nodeAttribConstruct(COLH1N1,node)\n",
    "        elif node.subtype == 'H3N2':\n",
    "            N = nodeAttribConstruct(COLH3N2,node)\n",
    "        elif node.subtype == 'H7N9':\n",
    "            N = nodeAttribConstruct(COLH7N9,node)\n",
    "        elif node.subtype == 'H9N2':\n",
    "            N = nodeAttribConstruct(COLH9N2,node)\n",
    "        else:\n",
    "            N = nodeAttribConstruct(COLDEF,node)\n",
    "            \n",
    "\n",
    "            \n",
    "def render_tree(tree, outfile):# all_seq_data, display_type='nearest_host'):\n",
    "    \"\"\"Render the tree inside the file to a circular \n",
    "    phylogenetic tree.\n",
    "    \n",
    "    NOTE: outfile should be in .pdf for best visuals\n",
    "    Returns:\n",
    "    \"\"\"\n",
    "    #tree = Tree(nwfile,format=1)\n",
    "\n",
    "    ts = TreeStyle()\n",
    "    ns = NodeStyle()\n",
    "    ts.show_leaf_name = False\n",
    "    #ts.rotation = 90\n",
    "    ts.mode = \"r\"\n",
    "    #ts.arc_start = -360 # 0 degrees = 3 o'clock\n",
    "    #ts.arc_span = 360\n",
    "    ts.scale=5\n",
    "    ts.show_scale=False\n",
    "    ts.branch_vertical_margin = .5 # 10 pixels between adjacent branches\n",
    "    # ts.show_branch_length=True\n",
    "    #ts.min_leaf_separation=10\n",
    "    #ts.optimal_scale_level='full'\n",
    "    #ts.branch_vertical_margin=0\n",
    "    \n",
    "    ns.hz_line_width=2\n",
    "    ns.vt_line_width=1\n",
    "    #ts.layout_fn = layout\n",
    "    ns[\"vt_line_width\"] = 16\n",
    "    ns[\"hz_line_width\"] = 16\n",
    "    #     ns['fsize'] = 20\n",
    "    for n in tree.traverse():\n",
    "        n.set_style(ns)\n",
    "        \n",
    "    #all_accessions = all_seq_data['accessions'].values\n",
    "    for n in tree:\n",
    "        ts.layout_fn = layout\n",
    "\n",
    "        \n",
    "    tree.set_style(ns)\n",
    "    tree.set_style(ts)\n",
    "    \n",
    "    ax=tree.render(\n",
    "        outfile, \n",
    "        dpi=300, \n",
    "        w=500,\n",
    "        tree_style=ts)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>subtype</th>\n",
       "      <th>ha_accession</th>\n",
       "      <th>na_accession</th>\n",
       "      <th>predicted_impact_score</th>\n",
       "      <th>predicted_emergence_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [id, subtype, ha_accession, na_accession, predicted_impact_score, predicted_emergence_score]\n",
       "Index: []"
      ]
     },
     "execution_count": 18,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "r=5\n",
    "medrisknames=[]\n",
    "for node in ltree_collapsed:\n",
    "    if ALLinfoHA.seq_infos[node.name.split()[0]].erisk > r:\n",
    "        medrisknames=np.append(medrisknames,\n",
    "                                ALLinfoHA.seq_infos[node.name.split()[0]].name)\n",
    "medrisknames=df[df.id.isin(medrisknames)][['id','subtype',\n",
    "                                               'ha_accession',\n",
    "                                               'na_accession',\n",
    "                                               'predicted_impact_score',\n",
    "                                               'predicted_emergence_score']].sort_values('predicted_emergence_score',\n",
    "                                                                                         ascending=False)\n",
    "medrisknames.to_csv('allriskystrains_collapsed.csv',index=None)\n",
    "medrisknames[medrisknames.id.str.contains('Karaganda')]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Render phylogenetic trees"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [],
   "source": [
    "ax=render_tree(\n",
    "    labelled_tree,\n",
    "    './riskyphylo'+str(threshold)+'.pdf')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "ax=render_tree(\n",
    "    ltree_collapsed,\n",
    "    'riskyphylo'+str(threshold)+'_collapsed_'+str(num_collapsed)+'.pdf')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Tree node 'Inner1771' (0x7f07f5c55dd)"
      ]
     },
     "execution_count": 21,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ltree_collapsed"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Find high risk strains which are on distinct branches on phylogenetic tree"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>subtype</th>\n",
       "      <th>HA accession</th>\n",
       "      <th>NA accession</th>\n",
       "      <th>predicted IRAT impact</th>\n",
       "      <th>predicted IRAT emergence</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>strain</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A/swine/Missouri/A02524711/2020</th>\n",
       "      <td>H1N1</td>\n",
       "      <td>EPI1818121</td>\n",
       "      <td>EPI1818122</td>\n",
       "      <td>6.631798</td>\n",
       "      <td>6.706662</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A/Camel/Inner_Mongolia/XL/2020</th>\n",
       "      <td>H7N9</td>\n",
       "      <td>EPI2026200</td>\n",
       "      <td>EPI2026202</td>\n",
       "      <td>6.628552</td>\n",
       "      <td>6.702610</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A/swine/Indiana/A02524710/2020</th>\n",
       "      <td>H3N2</td>\n",
       "      <td>EPI1818137</td>\n",
       "      <td>EPI1818138</td>\n",
       "      <td>6.606970</td>\n",
       "      <td>6.675656</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A/swine/North_Carolina/A02479173/2020</th>\n",
       "      <td>H1N1</td>\n",
       "      <td>EPI1780425</td>\n",
       "      <td>EPI1780426</td>\n",
       "      <td>6.587768</td>\n",
       "      <td>6.651667</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A/swine/Tennessee/A02524414/2022</th>\n",
       "      <td>H1N1</td>\n",
       "      <td>EPI2149257</td>\n",
       "      <td>EPI2149258</td>\n",
       "      <td>6.537854</td>\n",
       "      <td>6.589280</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A/swine/Iowa/A02479343/2020</th>\n",
       "      <td>H1N1</td>\n",
       "      <td>EPI1769114</td>\n",
       "      <td>EPI1769115</td>\n",
       "      <td>6.338728</td>\n",
       "      <td>6.339915</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A/canine/Texas/21-011409-001/2021</th>\n",
       "      <td>H3N2</td>\n",
       "      <td>EPI1896555</td>\n",
       "      <td>EPI1896557</td>\n",
       "      <td>6.309324</td>\n",
       "      <td>6.303022</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A/swine/Kansas/A02245381/2020(H1N1)</th>\n",
       "      <td>H1N1</td>\n",
       "      <td>EPI1777723</td>\n",
       "      <td>EPI1777724</td>\n",
       "      <td>6.295775</td>\n",
       "      <td>6.286015</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A/swine/Iowa/A02246996/2021</th>\n",
       "      <td>H1N1</td>\n",
       "      <td>EPI2146133</td>\n",
       "      <td>EPI2146134</td>\n",
       "      <td>6.281749</td>\n",
       "      <td>6.268407</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A/mink/China/chick_embryo/2020</th>\n",
       "      <td>H9N2</td>\n",
       "      <td>EPI2161544</td>\n",
       "      <td>EPI2161548</td>\n",
       "      <td>6.278700</td>\n",
       "      <td>6.264578</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>68 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                      subtype HA accession NA accession  \\\n",
       "strain                                                                    \n",
       "A/swine/Missouri/A02524711/2020          H1N1   EPI1818121   EPI1818122   \n",
       "A/Camel/Inner_Mongolia/XL/2020           H7N9   EPI2026200   EPI2026202   \n",
       "A/swine/Indiana/A02524710/2020           H3N2   EPI1818137   EPI1818138   \n",
       "A/swine/North_Carolina/A02479173/2020    H1N1   EPI1780425   EPI1780426   \n",
       "A/swine/Tennessee/A02524414/2022         H1N1   EPI2149257   EPI2149258   \n",
       "...                                       ...          ...          ...   \n",
       "A/swine/Iowa/A02479343/2020              H1N1   EPI1769114   EPI1769115   \n",
       "A/canine/Texas/21-011409-001/2021        H3N2   EPI1896555   EPI1896557   \n",
       "A/swine/Kansas/A02245381/2020(H1N1)      H1N1   EPI1777723   EPI1777724   \n",
       "A/swine/Iowa/A02246996/2021              H1N1   EPI2146133   EPI2146134   \n",
       "A/mink/China/chick_embryo/2020           H9N2   EPI2161544   EPI2161548   \n",
       "\n",
       "                                       predicted IRAT impact  \\\n",
       "strain                                                         \n",
       "A/swine/Missouri/A02524711/2020                     6.631798   \n",
       "A/Camel/Inner_Mongolia/XL/2020                      6.628552   \n",
       "A/swine/Indiana/A02524710/2020                      6.606970   \n",
       "A/swine/North_Carolina/A02479173/2020               6.587768   \n",
       "A/swine/Tennessee/A02524414/2022                    6.537854   \n",
       "...                                                      ...   \n",
       "A/swine/Iowa/A02479343/2020                         6.338728   \n",
       "A/canine/Texas/21-011409-001/2021                   6.309324   \n",
       "A/swine/Kansas/A02245381/2020(H1N1)                 6.295775   \n",
       "A/swine/Iowa/A02246996/2021                         6.281749   \n",
       "A/mink/China/chick_embryo/2020                      6.278700   \n",
       "\n",
       "                                       predicted IRAT emergence  \n",
       "strain                                                           \n",
       "A/swine/Missouri/A02524711/2020                        6.706662  \n",
       "A/Camel/Inner_Mongolia/XL/2020                         6.702610  \n",
       "A/swine/Indiana/A02524710/2020                         6.675656  \n",
       "A/swine/North_Carolina/A02479173/2020                  6.651667  \n",
       "A/swine/Tennessee/A02524414/2022                       6.589280  \n",
       "...                                                         ...  \n",
       "A/swine/Iowa/A02479343/2020                            6.339915  \n",
       "A/canine/Texas/21-011409-001/2021                      6.303022  \n",
       "A/swine/Kansas/A02245381/2020(H1N1)                    6.286015  \n",
       "A/swine/Iowa/A02246996/2021                            6.268407  \n",
       "A/mink/China/chick_embryo/2020                         6.264578  \n",
       "\n",
       "[68 rows x 5 columns]"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "r=6.057\n",
    "r=6.25\n",
    "highrisknames=[]\n",
    "for node in ltree_collapsed:\n",
    "    if ALLinfoHA.seq_infos[node.name.split()[0]].erisk > r:\n",
    "        highrisknames=np.append(highrisknames,\n",
    "                                ALLinfoHA.seq_infos[node.name.split()[0]].name)\n",
    "highrisknamesdf=df[df.id.isin(highrisknames)][['id','subtype',\n",
    "                                               'ha_accession',\n",
    "                                               'na_accession',\n",
    "                                               'predicted_impact_score',\n",
    "                                               'predicted_emergence_score']].sort_values('predicted_emergence_score',\n",
    "                                                                                         ascending=False)\n",
    "highrisknamesdf = highrisknamesdf.rename(columns={'id':'strain',\n",
    "                                                  'ha_accession':'HA accession',\n",
    "                                                  'na_accession':'NA accession',\n",
    "                                                  'predicted_impact_score':'predicted IRAT impact',\n",
    "                                                  'predicted_emergence_score':'predicted IRAT emergence'}).set_index('strain')\n",
    "highrisknamesdf#.drop_duplicates()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>subtype</th>\n",
       "      <th>HA accession</th>\n",
       "      <th>NA accession</th>\n",
       "      <th>predicted IRAT impact</th>\n",
       "      <th>predicted IRAT emergence</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>strain</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>\\rowcolor{Red3!20}A/swine/Missouri/A02524711/2020</th>\n",
       "      <td>H1N1</td>\n",
       "      <td>EPI1818121</td>\n",
       "      <td>EPI1818122</td>\n",
       "      <td>6.631798</td>\n",
       "      <td>6.706662</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>\\rowcolor{Green3!50}A/Camel/Inner_Mongolia/XL/2020</th>\n",
       "      <td>H7N9</td>\n",
       "      <td>EPI2026200</td>\n",
       "      <td>EPI2026202</td>\n",
       "      <td>6.628552</td>\n",
       "      <td>6.702610</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>\\rowcolor{Blue1!30}A/swine/Indiana/A02524710/2020</th>\n",
       "      <td>H3N2</td>\n",
       "      <td>EPI1818137</td>\n",
       "      <td>EPI1818138</td>\n",
       "      <td>6.606970</td>\n",
       "      <td>6.675656</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>\\rowcolor{Red3!20}A/swine/North_Carolina/A02479173/2020</th>\n",
       "      <td>H1N1</td>\n",
       "      <td>EPI1780425</td>\n",
       "      <td>EPI1780426</td>\n",
       "      <td>6.587768</td>\n",
       "      <td>6.651667</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>\\rowcolor{Red3!20}A/swine/Tennessee/A02524414/2022</th>\n",
       "      <td>H1N1</td>\n",
       "      <td>EPI2149257</td>\n",
       "      <td>EPI2149258</td>\n",
       "      <td>6.537854</td>\n",
       "      <td>6.589280</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>\\rowcolor{Red3!20}A/swine/Iowa/A02479343/2020</th>\n",
       "      <td>H1N1</td>\n",
       "      <td>EPI1769114</td>\n",
       "      <td>EPI1769115</td>\n",
       "      <td>6.338728</td>\n",
       "      <td>6.339915</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>\\rowcolor{Blue1!30}A/canine/Texas/21-011409-001/2021</th>\n",
       "      <td>H3N2</td>\n",
       "      <td>EPI1896555</td>\n",
       "      <td>EPI1896557</td>\n",
       "      <td>6.309324</td>\n",
       "      <td>6.303022</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>\\rowcolor{Red3!20}A/swine/Kansas/A02245381/2020(H1N1)</th>\n",
       "      <td>H1N1</td>\n",
       "      <td>EPI1777723</td>\n",
       "      <td>EPI1777724</td>\n",
       "      <td>6.295775</td>\n",
       "      <td>6.286015</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>\\rowcolor{Red3!20}A/swine/Iowa/A02246996/2021</th>\n",
       "      <td>H1N1</td>\n",
       "      <td>EPI2146133</td>\n",
       "      <td>EPI2146134</td>\n",
       "      <td>6.281749</td>\n",
       "      <td>6.268407</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>\\rowcolor{DarkOrange!40}A/mink/China/chick_embryo/2020</th>\n",
       "      <td>H9N2</td>\n",
       "      <td>EPI2161544</td>\n",
       "      <td>EPI2161548</td>\n",
       "      <td>6.278700</td>\n",
       "      <td>6.264578</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>68 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   subtype HA accession  \\\n",
       "strain                                                                    \n",
       "\\rowcolor{Red3!20}A/swine/Missouri/A02524711/2020     H1N1   EPI1818121   \n",
       "\\rowcolor{Green3!50}A/Camel/Inner_Mongolia/XL/2020    H7N9   EPI2026200   \n",
       "\\rowcolor{Blue1!30}A/swine/Indiana/A02524710/2020     H3N2   EPI1818137   \n",
       "\\rowcolor{Red3!20}A/swine/North_Carolina/A02479...    H1N1   EPI1780425   \n",
       "\\rowcolor{Red3!20}A/swine/Tennessee/A02524414/2022    H1N1   EPI2149257   \n",
       "...                                                    ...          ...   \n",
       "\\rowcolor{Red3!20}A/swine/Iowa/A02479343/2020         H1N1   EPI1769114   \n",
       "\\rowcolor{Blue1!30}A/canine/Texas/21-011409-001...    H3N2   EPI1896555   \n",
       "\\rowcolor{Red3!20}A/swine/Kansas/A02245381/2020...    H1N1   EPI1777723   \n",
       "\\rowcolor{Red3!20}A/swine/Iowa/A02246996/2021         H1N1   EPI2146133   \n",
       "\\rowcolor{DarkOrange!40}A/mink/China/chick_embr...    H9N2   EPI2161544   \n",
       "\n",
       "                                                   NA accession  \\\n",
       "strain                                                            \n",
       "\\rowcolor{Red3!20}A/swine/Missouri/A02524711/2020    EPI1818122   \n",
       "\\rowcolor{Green3!50}A/Camel/Inner_Mongolia/XL/2020   EPI2026202   \n",
       "\\rowcolor{Blue1!30}A/swine/Indiana/A02524710/2020    EPI1818138   \n",
       "\\rowcolor{Red3!20}A/swine/North_Carolina/A02479...   EPI1780426   \n",
       "\\rowcolor{Red3!20}A/swine/Tennessee/A02524414/2022   EPI2149258   \n",
       "...                                                         ...   \n",
       "\\rowcolor{Red3!20}A/swine/Iowa/A02479343/2020        EPI1769115   \n",
       "\\rowcolor{Blue1!30}A/canine/Texas/21-011409-001...   EPI1896557   \n",
       "\\rowcolor{Red3!20}A/swine/Kansas/A02245381/2020...   EPI1777724   \n",
       "\\rowcolor{Red3!20}A/swine/Iowa/A02246996/2021        EPI2146134   \n",
       "\\rowcolor{DarkOrange!40}A/mink/China/chick_embr...   EPI2161548   \n",
       "\n",
       "                                                    predicted IRAT impact  \\\n",
       "strain                                                                      \n",
       "\\rowcolor{Red3!20}A/swine/Missouri/A02524711/2020                6.631798   \n",
       "\\rowcolor{Green3!50}A/Camel/Inner_Mongolia/XL/2020               6.628552   \n",
       "\\rowcolor{Blue1!30}A/swine/Indiana/A02524710/2020                6.606970   \n",
       "\\rowcolor{Red3!20}A/swine/North_Carolina/A02479...               6.587768   \n",
       "\\rowcolor{Red3!20}A/swine/Tennessee/A02524414/2022               6.537854   \n",
       "...                                                                   ...   \n",
       "\\rowcolor{Red3!20}A/swine/Iowa/A02479343/2020                    6.338728   \n",
       "\\rowcolor{Blue1!30}A/canine/Texas/21-011409-001...               6.309324   \n",
       "\\rowcolor{Red3!20}A/swine/Kansas/A02245381/2020...               6.295775   \n",
       "\\rowcolor{Red3!20}A/swine/Iowa/A02246996/2021                    6.281749   \n",
       "\\rowcolor{DarkOrange!40}A/mink/China/chick_embr...               6.278700   \n",
       "\n",
       "                                                    predicted IRAT emergence  \n",
       "strain                                                                        \n",
       "\\rowcolor{Red3!20}A/swine/Missouri/A02524711/2020                   6.706662  \n",
       "\\rowcolor{Green3!50}A/Camel/Inner_Mongolia/XL/2020                  6.702610  \n",
       "\\rowcolor{Blue1!30}A/swine/Indiana/A02524710/2020                   6.675656  \n",
       "\\rowcolor{Red3!20}A/swine/North_Carolina/A02479...                  6.651667  \n",
       "\\rowcolor{Red3!20}A/swine/Tennessee/A02524414/2022                  6.589280  \n",
       "...                                                                      ...  \n",
       "\\rowcolor{Red3!20}A/swine/Iowa/A02479343/2020                       6.339915  \n",
       "\\rowcolor{Blue1!30}A/canine/Texas/21-011409-001...                  6.303022  \n",
       "\\rowcolor{Red3!20}A/swine/Kansas/A02245381/2020...                  6.286015  \n",
       "\\rowcolor{Red3!20}A/swine/Iowa/A02246996/2021                       6.268407  \n",
       "\\rowcolor{DarkOrange!40}A/mink/China/chick_embr...                  6.264578  \n",
       "\n",
       "[68 rows x 5 columns]"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "COLDICT={'H1N1':'Red3!20','H3N2':'Blue1!30','H7N9':'Green3!50','H9N2':'DarkOrange!40'}\n",
    "def rowcolor(row):\n",
    "    return '\\\\rowcolor{' + COLDICT[row.subtype]+'}' + row['strain']\n",
    "\n",
    "highrisknamesdf1 = highrisknamesdf.reset_index(drop=False)\n",
    "highrisknamesdf1['strain']=highrisknamesdf1.apply(rowcolor,axis=1)\n",
    "highrisknamesdf1=highrisknamesdf1.set_index('strain')\n",
    "highrisknamesdf1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 25,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\\begin{tabular}{L{1.95in}|L{.25in}|L{.60in}|L{.6in}|C{1in}|C{1in}}\\hline\r\n",
      "strain&subtype& HA  accession & NA  accession & predicted  IRAT  impact & predicted  IRAT  emergence \\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Missouri/A02524711/2020 &H1N1&EPI1818121&EPI1818122&6.6318&6.7067\\\\\r\n",
      " \\rowcolor{Green3!50}A/Camel/Inner\\_Mongolia/XL/2020 &H7N9&EPI2026200&EPI2026202&6.6286&6.7026\\\\\r\n",
      " \\rowcolor{Blue1!30}A/swine/Indiana/A02524710/2020 &H3N2&EPI1818137&EPI1818138&6.6070&6.6757\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/North\\_Carolina/A02479173/2020 &H1N1&EPI1780425&EPI1780426&6.5878&6.6517\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Tennessee/A02524414/2022 &H1N1&EPI2149257&EPI2149258&6.5379&6.5893\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Minnesota/A02635976/2021 &H1N1&EPI1912208&EPI1912209&6.4841&6.5220\\\\\r\n",
      " \\rowcolor{Blue1!30}A/swine/Chile/VN1401-5054/2020 &H3N2&EPI1974975&EPI1974978&6.4578&6.4891\\\\\r\n",
      " \\rowcolor{Blue1!30}A/swine/Italy/56910/2020 &H3N2&EPI2142217&EPI2142173&6.4537&6.4840\\\\\r\n",
      " \\rowcolor{Blue1!30}A/swine/Minnesota/A02245643/2020 &H3N2&EPI1769178&EPI1769179&6.4370&6.4631\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Iowa/A02479005/2020 &H1N1&EPI1777621&EPI1777622&6.4103&6.4297\\\\\r\n",
      " \\rowcolor{Blue1!30}A/swine/Iowa/A02524874/2020 &H3N2&EPI1907838&EPI1907839&6.3836&6.3962\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Indiana/A02636526/2022 &H1N1&EPI2148186&EPI2148187&6.3830&6.3954\\\\\r\n",
      " \\rowcolor{Blue1!30}A/swine/Illinois/A02479007/2020 &H3N2&EPI1777629&EPI1777630&6.3769&6.3877\\\\\r\n",
      " \\rowcolor{Blue1!30}A/swine/Spain/44579-1/2020 &H3N2&EPI1930744&EPI1930748&6.3737&6.3838\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Minnesota/A02248037/2021 &H1N1&EPI1912188&EPI1912189&6.3694&6.3784\\\\\r\n",
      " \\rowcolor{Blue1!30}A/swine/Minnesota/A02245699/2020 &H3N2&EPI1833007&EPI1833008&6.3692&6.3781\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Iowa/A02635917/2021 &H1N1&EPI1911753&EPI1911754&6.3687&6.3774\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Minnesota/A02711801/2022 &H1N1&EPI2153420&EPI2153421&6.3674&6.3759\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Illinois/A02635936/2021 &H1N1&EPI1911791&EPI1911792&6.3671&6.3754\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/South\\_Dakota/A02524453/2020 &H1N1&EPI1765555&EPI1765556&6.3658&6.3738\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Minnesota/A02248061/2021 &H1N1&EPI1912494&EPI1912495&6.3624&6.3696\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Iowa/A02636439/2022 &H1N1&EPI2147475&EPI2147476&6.3616&6.3686\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Iowa/A02524875/2020 &H1N1&EPI1907858&EPI1907859&6.3610&6.3678\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Minnesota/A02248060/2021 &H1N1&EPI1912500&EPI1912501&6.3592&6.3656\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Nebraska/A02636117/2021 &H1N1&EPI1932937&EPI1932938&6.3582&6.3644\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Iowa/A02524513/2020 &H1N1&EPI1832647&EPI1832648&6.3580&6.3641\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Nebraska/A02479337/2020 &H1N1&EPI1769116&EPI1769117&6.3579&6.3640\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Iowa/A02479383/2020 &H1N1&EPI1771027&EPI1771028&6.3579&6.3640\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Nebraska/A02479212/2020 &H1N1&EPI1775884&EPI1775885&6.3579&6.3640\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Minnesota/A02479051/2020 &H1N1&EPI1778572&EPI1778573&6.3579&6.3640\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Iowa/A02635719/2021 &H1N1&EPI1910907&EPI1910908&6.3579&6.3640\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Minnesota/A02245424/2020 &H1N1&EPI1780207&EPI1780208&6.3579&6.3640\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Iowa/A02525313/2021 &H1N1&EPI1910761&EPI1910762&6.3579&6.3640\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Iowa/A02524646/2020 &H1N1&EPI1817164&EPI1817165&6.3579&6.3640\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Missouri/A02525065/2021 &H1N1&EPI1908581&EPI1908582&6.3579&6.3640\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Missouri/A02524951/2020 &H1N1&EPI1908429&EPI1908430&6.3579&6.3640\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Iowa/A02524994/2020 &H1N1&EPI1908427&EPI1908428&6.3579&6.3640\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Nebraska/A02524954/2020 &H1N1&EPI1908393&EPI1908394&6.3579&6.3640\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Iowa/A02524724/2020 &H1N1&EPI1818387&EPI1818388&6.3579&6.3640\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Iowa/A02524892/2020 &H1N1&EPI1907881&EPI1907882&6.3579&6.3640\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Nebraska/A02479186/2020 &H1N1&EPI1774141&EPI1774142&6.3578&6.3638\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Iowa/A02479156/2020 &H1N1&EPI1780249&EPI1780250&6.3574&6.3634\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Iowa/A02479229/2020 &H1N1&EPI1775914&EPI1775915&6.3574&6.3633\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Iowa/A02479303/2020 &H1N1&EPI1768639&EPI1768640&6.3567&6.3625\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Minnesota/A02710691/2021 &H1N1&EPI2146090&EPI2146091&6.3561&6.3618\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Iowa/A02635881/2021 &H1N1&EPI1911668&EPI1911669&6.3560&6.3616\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Iowa/A02525354/2021 &H1N1&EPI1910789&EPI1910790&6.3560&6.3616\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Iowa/A02524968/2020 &H1N1&EPI1908419&EPI1908420&6.3543&6.3595\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Iowa/A02635823/2021 &H1N1&EPI1911263&EPI1911264&6.3543&6.3594\\\\\r\n",
      " \\rowcolor{Blue1!30}A/swine/Minnesota/A02711797/2022 &H3N2&EPI2153382&EPI2153383&6.3537&6.3587\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Iowa/A02479141/2020 &H1N1&EPI1780241&EPI1780242&6.3531&6.3579\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Nebraska/A02525367/2021 &H1N1&EPI1910817&EPI1910818&6.3516&6.3561\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Iowa/A02750621/2022 &H1N1&EPI2161576&EPI2161577&6.3509&6.3552\\\\\r\n",
      " \\rowcolor{Blue1!30}A/swine/Illinois/A02525253/2021 &H3N2&EPI1910375&EPI1910376&6.3506&6.3548\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Iowa/A02245587/2020 &H1N1&EPI1775817&EPI1775818&6.3497&6.3537\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Iowa/A02525217/2021 &H1N1&EPI1909087&EPI1909088&6.3487&6.3525\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Iowa/A02636114/2021 &H1N1&EPI1931853&EPI1931854&6.3487&6.3525\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Iowa/A02636145/2021 &H1N1&EPI1932055&EPI1932930&6.3487&6.3525\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Iowa/A02636496/2022 &H1N1&EPI2148086&EPI2148087&6.3485&6.3522\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Iowa/A02635871/2021 &H1N1&EPI1911656&EPI1911657&6.3473&6.3507\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Iowa/A02479067/2020 &H1N1&EPI1778734&EPI1778735&6.3469&6.3501\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Minnesota/A02246459/2021 &H1N1&EPI1912518&EPI1912519&6.3465&6.3496\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Minnesota/A02525348/2021 &H1N1&EPI1910795&EPI1910796&6.3400&6.3415\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Iowa/A02479343/2020 &H1N1&EPI1769114&EPI1769115&6.3387&6.3399\\\\\r\n",
      " \\rowcolor{Blue1!30}A/canine/Texas/21-011409-001/2021 &H3N2&EPI1896555&EPI1896557&6.3093&6.3030\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Kansas/A02245381/2020(H1N1) &H1N1&EPI1777723&EPI1777724&6.2958&6.2860\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Iowa/A02246996/2021 &H1N1&EPI2146133&EPI2146134&6.2817&6.2684\\\\\r\n",
      " \\rowcolor{DarkOrange!40}A/mink/China/chick\\_embryo/2020 &H9N2&EPI2161544&EPI2161548&6.2787&6.2646\\\\\r\n",
      "\\hline\\end{tabular}\r\n"
     ]
    }
   ],
   "source": [
    "from zedstat.textable import textable\n",
    "textable(highrisknamesdf1,#pd.concat([highrisknamesdf1.head(30),highrisknamesdf1.tail(10)]),\n",
    "         tabname='highrisk35.tex',\n",
    "         FORMAT='%1.4f',INDEX=True,\n",
    "         TABFORMAT='L{1.95in}|L{.25in}|L{.60in}|L{.6in}|C{1in}|C{1in}',LNTERM='\\\\\\\\\\n')\n",
    "! cat highrisk35.tex"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>H1N1</th>\n",
       "      <th>H3N2</th>\n",
       "      <th>H7N9</th>\n",
       "      <th>H9N2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A/swine/Missouri/A02524711/2020</td>\n",
       "      <td>A/swine/Indiana/A02524710/2020</td>\n",
       "      <td>A/Camel/Inner_Mongolia/XL/2020</td>\n",
       "      <td>A/mink/China/chick_embryo/2020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>6.706662</td>\n",
       "      <td>6.675656</td>\n",
       "      <td>6.70261</td>\n",
       "      <td>6.264578</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                              H1N1                            H3N2  \\\n",
       "0  A/swine/Missouri/A02524711/2020  A/swine/Indiana/A02524710/2020   \n",
       "1                         6.706662                        6.675656   \n",
       "\n",
       "                             H7N9                            H9N2  \n",
       "0  A/Camel/Inner_Mongolia/XL/2020  A/mink/China/chick_embryo/2020  \n",
       "1                         6.70261                        6.264578  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>6.0</th>\n",
       "      <th>6.2</th>\n",
       "      <th>6.3</th>\n",
       "      <th>6.4</th>\n",
       "      <th>6.5</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>H1N1</th>\n",
       "      <td>62</td>\n",
       "      <td>57</td>\n",
       "      <td>53</td>\n",
       "      <td>5</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>H3N2</th>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>H7N9</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>H9N2</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      6.0  6.2  6.3  6.4  6.5\n",
       "H1N1   62   57   53    5    4\n",
       "H3N2   11   11   11    4    1\n",
       "H7N9    1    1    1    1    1\n",
       "H9N2    1    1    0    0    0"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "75\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>subtype</th>\n",
       "      <th>ha_accession</th>\n",
       "      <th>na_accession</th>\n",
       "      <th>ha</th>\n",
       "      <th>na</th>\n",
       "      <th>HA_Avg_Qdist</th>\n",
       "      <th>ha_variance</th>\n",
       "      <th>NA_Avg_Qdist</th>\n",
       "      <th>na_variance</th>\n",
       "      <th>Geometric_Mean</th>\n",
       "      <th>predicted_impact_score</th>\n",
       "      <th>predicted_emergence_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>218</th>\n",
       "      <td>A/swine/Missouri/A02524711/2020</td>\n",
       "      <td>H1N1</td>\n",
       "      <td>EPI1818121</td>\n",
       "      <td>EPI1818122</td>\n",
       "      <td>MKAILVVMLYTFTTANADTLCIGYHANNSTDTVDTVLEKNVTVTHS...</td>\n",
       "      <td>MNPNQKIITIGSICMTIGTANLILQIGNIISIWVSHSIQIGNQSQI...</td>\n",
       "      <td>0.007931</td>\n",
       "      <td>9.678707e-06</td>\n",
       "      <td>0.001876</td>\n",
       "      <td>2.438430e-06</td>\n",
       "      <td>0.003857</td>\n",
       "      <td>6.631798</td>\n",
       "      <td>6.706662</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1902</th>\n",
       "      <td>A/swine/Indiana/A02524710/2020</td>\n",
       "      <td>H3N2</td>\n",
       "      <td>EPI1818137</td>\n",
       "      <td>EPI1818138</td>\n",
       "      <td>MKTIIALSYILCLVFAQKIPGNDNGTATLCLGHHAVPNGTIVKTIT...</td>\n",
       "      <td>MNPNQKIITIGSVSLTISTICFFMQIAILITTIALHFKQYEFNSPP...</td>\n",
       "      <td>0.004542</td>\n",
       "      <td>4.497950e-07</td>\n",
       "      <td>0.006619</td>\n",
       "      <td>1.048722e-06</td>\n",
       "      <td>0.005483</td>\n",
       "      <td>6.606970</td>\n",
       "      <td>6.675656</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5814</th>\n",
       "      <td>A/Camel/Inner_Mongolia/XL/2020</td>\n",
       "      <td>H7N9</td>\n",
       "      <td>EPI2026200</td>\n",
       "      <td>EPI2026202</td>\n",
       "      <td>MNTQILVFALIAIIPTNADKICLGHHAVSNGTKVNTLTEREVEVVN...</td>\n",
       "      <td>MNPNQKILCTSATAITIGAIAVLIGIANLGLNIGLHLKPGCNCSHS...</td>\n",
       "      <td>0.014006</td>\n",
       "      <td>2.854999e-03</td>\n",
       "      <td>0.001182</td>\n",
       "      <td>2.218687e-07</td>\n",
       "      <td>0.004069</td>\n",
       "      <td>6.628552</td>\n",
       "      <td>6.702610</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6049</th>\n",
       "      <td>A/mink/China/chick_embryo/2020</td>\n",
       "      <td>H9N2</td>\n",
       "      <td>EPI2161544</td>\n",
       "      <td>EPI2161548</td>\n",
       "      <td>METVSLITILLAATVSNADKICIGYQSSNSTETVDTLTENNVPVTH...</td>\n",
       "      <td>MNPNQKITAIGSVSLIIAIICLLMQIAILTMTMTLHFGQKECSNPS...</td>\n",
       "      <td>0.016908</td>\n",
       "      <td>1.293100e-04</td>\n",
       "      <td>0.046149</td>\n",
       "      <td>1.553612e-02</td>\n",
       "      <td>0.027933</td>\n",
       "      <td>6.278700</td>\n",
       "      <td>6.264578</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   id subtype ha_accession na_accession  \\\n",
       "218   A/swine/Missouri/A02524711/2020    H1N1   EPI1818121   EPI1818122   \n",
       "1902   A/swine/Indiana/A02524710/2020    H3N2   EPI1818137   EPI1818138   \n",
       "5814   A/Camel/Inner_Mongolia/XL/2020    H7N9   EPI2026200   EPI2026202   \n",
       "6049   A/mink/China/chick_embryo/2020    H9N2   EPI2161544   EPI2161548   \n",
       "\n",
       "                                                     ha  \\\n",
       "218   MKAILVVMLYTFTTANADTLCIGYHANNSTDTVDTVLEKNVTVTHS...   \n",
       "1902  MKTIIALSYILCLVFAQKIPGNDNGTATLCLGHHAVPNGTIVKTIT...   \n",
       "5814  MNTQILVFALIAIIPTNADKICLGHHAVSNGTKVNTLTEREVEVVN...   \n",
       "6049  METVSLITILLAATVSNADKICIGYQSSNSTETVDTLTENNVPVTH...   \n",
       "\n",
       "                                                     na  HA_Avg_Qdist  \\\n",
       "218   MNPNQKIITIGSICMTIGTANLILQIGNIISIWVSHSIQIGNQSQI...      0.007931   \n",
       "1902  MNPNQKIITIGSVSLTISTICFFMQIAILITTIALHFKQYEFNSPP...      0.004542   \n",
       "5814  MNPNQKILCTSATAITIGAIAVLIGIANLGLNIGLHLKPGCNCSHS...      0.014006   \n",
       "6049  MNPNQKITAIGSVSLIIAIICLLMQIAILTMTMTLHFGQKECSNPS...      0.016908   \n",
       "\n",
       "       ha_variance  NA_Avg_Qdist   na_variance  Geometric_Mean  \\\n",
       "218   9.678707e-06      0.001876  2.438430e-06        0.003857   \n",
       "1902  4.497950e-07      0.006619  1.048722e-06        0.005483   \n",
       "5814  2.854999e-03      0.001182  2.218687e-07        0.004069   \n",
       "6049  1.293100e-04      0.046149  1.553612e-02        0.027933   \n",
       "\n",
       "      predicted_impact_score  predicted_emergence_score  \n",
       "218                 6.631798                   6.706662  \n",
       "1902                6.606970                   6.675656  \n",
       "5814                6.628552                   6.702610  \n",
       "6049                6.278700                   6.264578  "
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "count=0\n",
    "Subtype={'H1N1':0,'H3N2':0,'H7N9':0,'H9N2':0}\n",
    "MaxriskStrain={'H1N1':None,'H3N2':None,'H7N9':None,'H9N2':None}\n",
    "Subtype_strat={ 6.0:{'H1N1':0,'H3N2':0,'H7N9':0,'H9N2':0},  6.2:{'H1N1':0,'H3N2':0,'H7N9':0,'H9N2':0},              6.3:{'H1N1':0,'H3N2':0,'H7N9':0,'H9N2':0},\n",
    "               6.4:{'H1N1':0,'H3N2':0,'H7N9':0,'H9N2':0},6.5:{'H1N1':0,'H3N2':0,'H7N9':0,'H9N2':0}}\n",
    "for node in ltree_collapsed:\n",
    "    Subtype[node.subtype]=Subtype[node.subtype]+1\n",
    "    if MaxriskStrain[node.subtype] is None:\n",
    "        MaxriskStrain[node.subtype]=(ALLinfoHA.seq_infos[node.name.split()[0]].name,\n",
    "                                     ALLinfoHA.seq_infos[node.name.split()[0]].erisk)\n",
    "    else:\n",
    "        if MaxriskStrain[node.subtype][1]<ALLinfoHA.seq_infos[node.name.split()[0]].erisk:\n",
    "            MaxriskStrain[node.subtype]=(ALLinfoHA.seq_infos[node.name.split()[0]].name,\n",
    "                                         ALLinfoHA.seq_infos[node.name.split()[0]].erisk)\n",
    "    for r in [6.0, 6.2,6.3,6.4,6.5]:\n",
    "        if r<ALLinfoHA.seq_infos[node.name.split()[0]].erisk:\n",
    "            #print(node.subtype,ALLinfoHA.seq_infos[node.name.split()[0]].erisk)\n",
    "            Subtype_strat[r][node.subtype]=Subtype_strat[r][node.subtype]+1\n",
    "        \n",
    "    count=count+1\n",
    "maxriskdf = pd.DataFrame(MaxriskStrain)\n",
    "Subtype_strat_df = pd.DataFrame(Subtype_strat)\n",
    "display(maxriskdf)\n",
    "display(Subtype_strat_df)\n",
    "print(count)\n",
    "df[df.id.isin(maxriskdf.iloc[0,:].values)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\\begin{tabular}{L{.75in}|L{.25in}|L{.60in}|L{.6in}|L{.6in}|L{.6in}}\\hline\r\n",
      "subtype&6.0&6.2&6.3&6.4&6.5\\\\\r\n",
      "H1N1&62&57&53&5&4\\\\\r\n",
      "H3N2&11&11&11&4&1\\\\\r\n",
      "H7N9&1&1&1&1&1\\\\\r\n",
      "H9N2&1&1&0&0&0\\\\\r\n",
      "\\hline\\end{tabular}\r\n"
     ]
    }
   ],
   "source": [
    "from zedstat.textable import textable\n",
    "Subtype_strat_df.columns=[str(x) for x in Subtype_strat_df.columns]\n",
    "Subtype_strat_df.index.name='subtype'\n",
    "Subtype_strat_df=Subtype_strat_df.reset_index()\n",
    "textable(Subtype_strat_df,#pd.concat([highrisknamesdf1.head(30),highrisknamesdf1.tail(10)]),\n",
    "         tabname='riskycount.tex',\n",
    "         FORMAT='%1.4f',INDEX=False,\n",
    "         TABFORMAT='L{.75in}|L{.25in}|L{.60in}|L{.6in}|L{.6in}|L{.6in}',LNTERM='\\\\\\\\\\n')\n",
    "! cat riskycount.tex"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [],
   "source": [
    "def write_fasta(seqs, fasta_file, wrap=80):\n",
    "    \"\"\"Write sequences to a fasta file.\n",
    "\n",
    "    Parameters\n",
    "    ----------\n",
    "    seqs : dict[seq_id] -> seq\n",
    "        Sequences indexed by sequence id.\n",
    "    fasta_file : str\n",
    "        Path to write the sequences to.\n",
    "    wrap: int\n",
    "        Number of AA/NT before the line is wrapped.\n",
    "    \"\"\"\n",
    "    with open(fasta_file, 'w') as f:\n",
    "        for gid, gseq in seqs.items():\n",
    "            f.write('>{}\\n'.format(gid))\n",
    "            for i in range(0, len(gseq), wrap):\n",
    "                f.write('{}\\n'.format(gseq[i:i + wrap])) \n",
    "\n",
    "#for i,name in zip(df[df.id.isin(maxriskdf.iloc[0,:].values)].ha.values,\n",
    "#                 df[df.id.isin(maxriskdf.iloc[0,:].values)].ha_accession.values):\n",
    "#    write_fasta({name:i},name+'.fasta')\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "id_list=['A/swine/North_Carolina/A02479173/2020','A/swine/Tennessee/A02524414/2022']\n",
    "RFILES={name+'('+subtype+')':i for i,name,subtype in zip(df[df.id.isin(id_list)].ha.values,\n",
    "                 df[df.id.isin(id_list)].ha_accession.values,\n",
    "                 df[df.id.isin(id_list)].subtype.values)}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "FASTANAME='../../../tex/Figures/plotdata/seqanal/risky4.fasta'\n",
    "FASTAFILES={name+'('+subtype+')':i for i,name,subtype in zip(df[df.id.isin(maxriskdf.iloc[0,:].values)].ha.values,\n",
    "                 df[df.id.isin(maxriskdf.iloc[0,:].values)].ha_accession.values,\n",
    "                 df[df.id.isin(maxriskdf.iloc[0,:].values)].subtype.values)}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "human_dom_seq='../../../qnet_predictions/results/dominant_sequences_2021_2022.csv'\n",
    "domf=pd.read_csv(human_dom_seq,index_col=0)\n",
    "domf=domf.assign(subtype=[x.split('_')[1].upper() for x in domf.index.values],protein=[x.split('_')[2].upper() for x in domf.index.values])\n",
    "domf=domf[domf.protein=='HA'].head(2)\n",
    "DOMFILES={'DOM_HUMAN'+'('+subtype+')':i for i,name,subtype in zip(domf.sequence.values,\n",
    "                 domf.name.values,\n",
    "                 domf.subtype.values)}\n",
    "DOMFILES.update(FASTAFILES)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [],
   "source": [
    "DOMFILES.update(RFILES)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "ename": "FileNotFoundError",
     "evalue": "[Errno 2] No such file or directory: '../../../tex/Figures/plotdata/seqanal/risky4.fasta'",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mFileNotFoundError\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m/tmp/ipykernel_17618/1584862752.py\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[0;32m----> 1\u001b[0;31m \u001b[0mwrite_fasta\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mDOMFILES\u001b[0m\u001b[0;34m,\u001b[0m\u001b[0mFASTANAME\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m",
      "\u001b[0;32m/tmp/ipykernel_17618/1946063178.py\u001b[0m in \u001b[0;36mwrite_fasta\u001b[0;34m(seqs, fasta_file, wrap)\u001b[0m\n\u001b[1;32m     11\u001b[0m         \u001b[0mNumber\u001b[0m \u001b[0mof\u001b[0m \u001b[0mAA\u001b[0m\u001b[0;34m/\u001b[0m\u001b[0mNT\u001b[0m \u001b[0mbefore\u001b[0m \u001b[0mthe\u001b[0m \u001b[0mline\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0mwrapped\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     12\u001b[0m     \"\"\"\n\u001b[0;32m---> 13\u001b[0;31m     \u001b[0;32mwith\u001b[0m \u001b[0mopen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mfasta_file\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'w'\u001b[0m\u001b[0;34m)\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0mf\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m     14\u001b[0m         \u001b[0;32mfor\u001b[0m \u001b[0mgid\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mgseq\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mseqs\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mitems\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m     15\u001b[0m             \u001b[0mf\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mwrite\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'>{}\\n'\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mformat\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mgid\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mFileNotFoundError\u001b[0m: [Errno 2] No such file or directory: '../../../tex/Figures/plotdata/seqanal/risky4.fasta'"
     ]
    }
   ],
   "source": [
    "write_fasta(DOMFILES,FASTANAME)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "ename": "KeyboardInterrupt",
     "evalue": "",
     "output_type": "error",
     "traceback": [
      "\u001b[0;31m---------------------------------------------------------------------------\u001b[0m",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m                         Traceback (most recent call last)",
      "\u001b[0;32m/tmp/ipykernel_2348532/2222584185.py\u001b[0m in \u001b[0;36m<module>\u001b[0;34m\u001b[0m\n\u001b[1;32m      2\u001b[0m \u001b[0;32mfrom\u001b[0m \u001b[0mBio\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mBlast\u001b[0m \u001b[0;32mimport\u001b[0m \u001b[0mNCBIWWW\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      3\u001b[0m \u001b[0mmy_query\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mSeqIO\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"EPI1818121.fasta\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mformat\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0;34m\"fasta\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m----> 4\u001b[0;31m \u001b[0mresult_handle\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mNCBIWWW\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mqblast\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"blastp\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m\"nr\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmy_query\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mseq\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m      5\u001b[0m \u001b[0mblast_result\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mopen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"my_blast.xml\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m\"w\"\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m      6\u001b[0m \u001b[0mblast_result\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mwrite\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mresult_handle\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m~/.local/lib/python3.10/site-packages/Bio/Blast/NCBIWWW.py\u001b[0m in \u001b[0;36mqblast\u001b[0;34m(program, database, sequence, url_base, auto_format, composition_based_statistics, db_genetic_code, endpoints, entrez_query, expect, filter, gapcosts, genetic_code, hitlist_size, i_thresh, layout, lcase_mask, matrix_name, nucl_penalty, nucl_reward, other_advanced, perc_ident, phi_pattern, query_file, query_believe_defline, query_from, query_to, searchsp_eff, service, threshold, ungapped_alignment, word_size, short_query, alignments, alignment_view, descriptions, entrez_links_new_window, expect_low, expect_high, format_entrez_query, format_object, format_type, ncbi_gi, results_file, show_overview, megablast, template_type, template_length)\u001b[0m\n\u001b[1;32m    254\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    255\u001b[0m         \u001b[0mrequest\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mRequest\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0murl_base\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmessage\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m{\u001b[0m\u001b[0;34m\"User-Agent\"\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0;34m\"BiopythonClient\"\u001b[0m\u001b[0;34m}\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 256\u001b[0;31m         \u001b[0mhandle\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0murlopen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mrequest\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    257\u001b[0m         \u001b[0mresults\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mhandle\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mread\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdecode\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    258\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/lib64/python3.10/urllib/request.py\u001b[0m in \u001b[0;36murlopen\u001b[0;34m(url, data, timeout, cafile, capath, cadefault, context)\u001b[0m\n\u001b[1;32m    214\u001b[0m     \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    215\u001b[0m         \u001b[0mopener\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0m_opener\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 216\u001b[0;31m     \u001b[0;32mreturn\u001b[0m \u001b[0mopener\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mopen\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0murl\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdata\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mtimeout\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    217\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    218\u001b[0m \u001b[0;32mdef\u001b[0m \u001b[0minstall_opener\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mopener\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/lib64/python3.10/urllib/request.py\u001b[0m in \u001b[0;36mopen\u001b[0;34m(self, fullurl, data, timeout)\u001b[0m\n\u001b[1;32m    517\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    518\u001b[0m         \u001b[0msys\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0maudit\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m'urllib.Request'\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mreq\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mfull_url\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mreq\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mdata\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mreq\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mheaders\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mreq\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mget_method\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 519\u001b[0;31m         \u001b[0mresponse\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_open\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mreq\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mdata\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    520\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    521\u001b[0m         \u001b[0;31m# post-process response\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/lib64/python3.10/urllib/request.py\u001b[0m in \u001b[0;36m_open\u001b[0;34m(self, req, data)\u001b[0m\n\u001b[1;32m    534\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    535\u001b[0m         \u001b[0mprotocol\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mreq\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mtype\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 536\u001b[0;31m         result = self._call_chain(self.handle_open, protocol, protocol +\n\u001b[0m\u001b[1;32m    537\u001b[0m                                   '_open', req)\n\u001b[1;32m    538\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mresult\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/lib64/python3.10/urllib/request.py\u001b[0m in \u001b[0;36m_call_chain\u001b[0;34m(self, chain, kind, meth_name, *args)\u001b[0m\n\u001b[1;32m    494\u001b[0m         \u001b[0;32mfor\u001b[0m \u001b[0mhandler\u001b[0m \u001b[0;32min\u001b[0m \u001b[0mhandlers\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    495\u001b[0m             \u001b[0mfunc\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mgetattr\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mhandler\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmeth_name\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 496\u001b[0;31m             \u001b[0mresult\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0mfunc\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m*\u001b[0m\u001b[0margs\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    497\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mresult\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    498\u001b[0m                 \u001b[0;32mreturn\u001b[0m \u001b[0mresult\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/lib64/python3.10/urllib/request.py\u001b[0m in \u001b[0;36mhttps_open\u001b[0;34m(self, req)\u001b[0m\n\u001b[1;32m   1389\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1390\u001b[0m         \u001b[0;32mdef\u001b[0m \u001b[0mhttps_open\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mreq\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1391\u001b[0;31m             return self.do_open(http.client.HTTPSConnection, req,\n\u001b[0m\u001b[1;32m   1392\u001b[0m                 context=self._context, check_hostname=self._check_hostname)\n\u001b[1;32m   1393\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/lib64/python3.10/urllib/request.py\u001b[0m in \u001b[0;36mdo_open\u001b[0;34m(self, http_class, req, **http_conn_args)\u001b[0m\n\u001b[1;32m   1346\u001b[0m         \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1347\u001b[0m             \u001b[0;32mtry\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1348\u001b[0;31m                 h.request(req.get_method(), req.selector, req.data, headers,\n\u001b[0m\u001b[1;32m   1349\u001b[0m                           encode_chunked=req.has_header('Transfer-encoding'))\n\u001b[1;32m   1350\u001b[0m             \u001b[0;32mexcept\u001b[0m \u001b[0mOSError\u001b[0m \u001b[0;32mas\u001b[0m \u001b[0merr\u001b[0m\u001b[0;34m:\u001b[0m \u001b[0;31m# timeout error\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/lib64/python3.10/http/client.py\u001b[0m in \u001b[0;36mrequest\u001b[0;34m(self, method, url, body, headers, encode_chunked)\u001b[0m\n\u001b[1;32m   1280\u001b[0m                 encode_chunked=False):\n\u001b[1;32m   1281\u001b[0m         \u001b[0;34m\"\"\"Send a complete request to the server.\"\"\"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1282\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_send_request\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmethod\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0murl\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mbody\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mheaders\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mencode_chunked\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1283\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1284\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0m_send_request\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mmethod\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0murl\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mbody\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mheaders\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mencode_chunked\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/lib64/python3.10/http/client.py\u001b[0m in \u001b[0;36m_send_request\u001b[0;34m(self, method, url, body, headers, encode_chunked)\u001b[0m\n\u001b[1;32m   1326\u001b[0m             \u001b[0;31m# default charset of iso-8859-1.\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1327\u001b[0m             \u001b[0mbody\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0m_encode\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mbody\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0;34m'body'\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1328\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mendheaders\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mbody\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mencode_chunked\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mencode_chunked\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1329\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1330\u001b[0m     \u001b[0;32mdef\u001b[0m \u001b[0mgetresponse\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/lib64/python3.10/http/client.py\u001b[0m in \u001b[0;36mendheaders\u001b[0;34m(self, message_body, encode_chunked)\u001b[0m\n\u001b[1;32m   1275\u001b[0m         \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1276\u001b[0m             \u001b[0;32mraise\u001b[0m \u001b[0mCannotSendHeader\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1277\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_send_output\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmessage_body\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mencode_chunked\u001b[0m\u001b[0;34m=\u001b[0m\u001b[0mencode_chunked\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1278\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1279\u001b[0m     def request(self, method, url, body=None, headers={}, *,\n",
      "\u001b[0;32m/usr/lib64/python3.10/http/client.py\u001b[0m in \u001b[0;36m_send_output\u001b[0;34m(self, message_body, encode_chunked)\u001b[0m\n\u001b[1;32m   1035\u001b[0m         \u001b[0mmsg\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;34mb\"\\r\\n\"\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mjoin\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_buffer\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1036\u001b[0m         \u001b[0;32mdel\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_buffer\u001b[0m\u001b[0;34m[\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m]\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1037\u001b[0;31m         \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msend\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0mmsg\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1038\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1039\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mmessage_body\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mnot\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/lib64/python3.10/http/client.py\u001b[0m in \u001b[0;36msend\u001b[0;34m(self, data)\u001b[0m\n\u001b[1;32m    973\u001b[0m         \u001b[0;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0msock\u001b[0m \u001b[0;32mis\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    974\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mauto_open\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 975\u001b[0;31m                 \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mconnect\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    976\u001b[0m             \u001b[0;32melse\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    977\u001b[0m                 \u001b[0;32mraise\u001b[0m \u001b[0mNotConnected\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/lib64/python3.10/http/client.py\u001b[0m in \u001b[0;36mconnect\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m   1445\u001b[0m             \u001b[0;34m\"Connect to a host on a given (SSL) port.\"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1446\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m-> 1447\u001b[0;31m             \u001b[0msuper\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mconnect\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m   1448\u001b[0m \u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m   1449\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0m_tunnel_host\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/lib64/python3.10/http/client.py\u001b[0m in \u001b[0;36mconnect\u001b[0;34m(self)\u001b[0m\n\u001b[1;32m    939\u001b[0m         \u001b[0;34m\"\"\"Connect to the host and port specified in __init__.\"\"\"\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    940\u001b[0m         \u001b[0msys\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0maudit\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0;34m\"http.client.connect\"\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mhost\u001b[0m\u001b[0;34m,\u001b[0m \u001b[0mself\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mport\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 941\u001b[0;31m         self.sock = self._create_connection(\n\u001b[0m\u001b[1;32m    942\u001b[0m             (self.host,self.port), self.timeout, self.source_address)\n\u001b[1;32m    943\u001b[0m         \u001b[0;31m# Might fail in OSs that don't implement TCP_NODELAY\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;32m/usr/lib64/python3.10/socket.py\u001b[0m in \u001b[0;36mcreate_connection\u001b[0;34m(address, timeout, source_address)\u001b[0m\n\u001b[1;32m    831\u001b[0m             \u001b[0;32mif\u001b[0m \u001b[0msource_address\u001b[0m\u001b[0;34m:\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    832\u001b[0m                 \u001b[0msock\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mbind\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msource_address\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0;32m--> 833\u001b[0;31m             \u001b[0msock\u001b[0m\u001b[0;34m.\u001b[0m\u001b[0mconnect\u001b[0m\u001b[0;34m(\u001b[0m\u001b[0msa\u001b[0m\u001b[0;34m)\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[0m\u001b[1;32m    834\u001b[0m             \u001b[0;31m# Break explicitly a reference cycle\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n\u001b[1;32m    835\u001b[0m             \u001b[0merr\u001b[0m \u001b[0;34m=\u001b[0m \u001b[0;32mNone\u001b[0m\u001b[0;34m\u001b[0m\u001b[0;34m\u001b[0m\u001b[0m\n",
      "\u001b[0;31mKeyboardInterrupt\u001b[0m: "
     ]
    }
   ],
   "source": [
    "from Bio import SeqIO\n",
    "from Bio.Blast import NCBIWWW\n",
    "my_query = SeqIO.read(\"EPI1818121.fasta\", format=\"fasta\")\n",
    "result_handle = NCBIWWW.qblast(\"blastp\", \"nr\", my_query.seq)\n",
    "blast_result = open(\"my_blast.xml\", \"w\")\n",
    "blast_result.write(result_handle.read())\n",
    "blast_result.close()\n",
    "result_handle.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def parse_blast(resultfile): #takes in the BLAST result, outputs list that can be made into csv\n",
    "    from Bio.Blast import NCBIXML\n",
    "    result_handle = open(resultfile)\n",
    "    blast_records = NCBIXML.parse(result_handle)\n",
    "    csv_list = []\n",
    "    \n",
    "    header = [  'accession','Query',\n",
    "                'Name', 'Length', 'Score', 'Expect',\n",
    "                'QueryStart', 'QueryEnd',\n",
    "                'SubjectStart', 'SubjectEnd','pct'\n",
    "            ]\n",
    "    \n",
    "    #csv_list.append(header)\n",
    "    count = 0\n",
    "    for blast_record in blast_records:\n",
    "        '''help(blast_record.alignments[0].hsps[0])''' # these give help info for the parts \n",
    "        '''help(blast_record.alignments[0])        '''\n",
    "        count +=1\n",
    "        \n",
    "        query = blast_record.query\n",
    "        for alignment in blast_record.alignments:\n",
    "\n",
    "            name = alignment.title\n",
    "            length = alignment.length\n",
    "    \n",
    "            hsp = alignment.hsps[0] # I don't know if we will ever have more than one, so might as well take the first one.\n",
    "            score = hsp.score\n",
    "            expect = hsp.expect\n",
    "            querystart = hsp.query_start\n",
    "            queryend = hsp.query_end\n",
    "            subjectstart = hsp.sbjct_start\n",
    "            subjectend = hsp.sbjct_end\n",
    "            pct=hsp.positives/hsp.align_length\n",
    "            accession=alignment.accession\n",
    "            row = [accession,query,name,length,score,expect,querystart,queryend,subjectstart,subjectend,pct]\n",
    "            csv_list.append(row)\n",
    "            \n",
    "    result_handle.close()\n",
    "    return pd.DataFrame(csv_list,columns=header)\n",
    " "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "csv_list=parse_blast('my_blast.xml')\n",
    "csv_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import Entrez\n",
    "Entrez.email = \"ishanu@uchicago.edu\"     # Always tell NCBI who you are\n",
    "handle = Entrez.esearch(db=\"protein\", term=\"QJT24340\")\n",
    "record = Entrez.read(handle)\n",
    "id=record[\"IdList\"]\n",
    "id"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "handle = Entrez.efetch(db=\"protein\", rettype=\"gb\", retmode=\"text\", id=\"1834373412\")\n",
    "#record = SeqIO.read(handle, \"genbank\")\n",
    "handle.read()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "H9N2: minks are probable mixing vessels\n",
    "https://www.ncbi.nlm.nih.gov/nuccore/ON870694.1\n",
    " https://www.tandfonline.com/doi/full/10.1080/22221751.2021.1899058\n",
    " Pandemic influenza, typically caused by the reassortment of human and avian influenza viruses, can result in severe or fatal infections in humans. Timely identification of potential pandemic viruses must be a priority in influenza virus surveillance. However, the range of host species responsible for the generation of novel pandemic influenza viruses remains unclear. In this study, we conducted serological surveys for avian and human influenza virus infections in farmed mink and determined the susceptibility of mink to prevailing avian and human virus subtypes. The results showed that farmed mink were commonly infected with human (H3N2 and H1N1/pdm) and avian (H7N9, H5N6, and H9N2) influenza A viruses. Correlational analysis indicated that transmission of human influenza viruses occurred from humans to mink, and that feed source was a probable route of avian influenza virus transmission to farmed mink. Animal experiments showed that mink were susceptible and permissive to circulating avian and human influenza viruses, and that human influenza viruses (H3N2 and H1N1/pdm), but not avian viruses, were capable of aerosol transmission among mink. These results indicate that farmed mink could be highly permissive “mixing vessels” for the reassortment of circulating human and avian influenza viruses. \n",
    " \n",
    " H7N9\n",
    " https://www.bmj.com/content/347/bmj.f4752?tab=responses\n",
    " HH transmission has been suspected\n",
    " Asian lineage H7N9 virus is rated by the Influenza Risk Assessment Tool as having the greatest potential to cause a pandemic, as well as potentially posing the greatest risk to severely impact public health if it were to achieve sustained human-to-human transmission.\n",
    " \n",
    " H3N2 swine\n",
    " https://www.cdc.gov/flu/swineflu/spotlights/first-human-infection-2022.htm\n",
    " Human infection detected in US this Aug\n",
    " \n",
    " H1N1\n",
    " Submitted (03-NOV-2020) USDA Swine Surveillance, USDA Swine\n",
    "            Surveillance, 1920 Dayton, Ames, IA 50010, USA\n",
    "COMMENT     Method: conceptual translation.\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
