{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 117,
   "metadata": {},
   "outputs": [],
   "source": [
    "import copy\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import Levenshtein\n",
    "from tqdm import tqdm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 118,
   "metadata": {},
   "outputs": [],
   "source": [
    "tmpf=pd.read_csv('../../../../paper_data/irat_enet/results/animal_predictions/combined_results_augmented_with_IRAT.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 119,
   "metadata": {},
   "outputs": [],
   "source": [
    "tmpfirat=tmpf[tmpf.is_irat==1]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 120,
   "metadata": {},
   "outputs": [],
   "source": [
    "ES=tmpfirat.ha_accession.values.astype(float)\n",
    "IS=tmpfirat.na_accession.values.astype(float)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 121,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "array([7.5, 6.6, 6.5, 6.4, 6.2, 6. , 5.8, 5.6, 5.3, 5.2, 5. , 4.6, 4.6,\n",
       "       4.4, 4.3, 4.2, 3.8, 3.7, 3.6, 3.4, 3.1, 2.8])"
      ]
     },
     "execution_count": 121,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ES"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 122,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/home/ishanu/.local/lib/python3.7/site-packages/pandas/core/generic.py:5516: SettingWithCopyWarning: \n",
      "A value is trying to be set on a copy of a slice from a DataFrame.\n",
      "Try using .loc[row_indexer,col_indexer] = value instead\n",
      "\n",
      "See the caveats in the documentation: https://pandas.pydata.org/pandas-docs/stable/user_guide/indexing.html#returning-a-view-versus-a-copy\n",
      "  self[name] = value\n"
     ]
    }
   ],
   "source": [
    "tmpfirat.emergence_risk=ES\n",
    "tmpfirat.impact_risk=IS\n",
    "\n",
    "accha=['IRAT'+x for x in tmpfirat.ha_accession.values]\n",
    "accna=['IRAT'+x for x in tmpfirat.na_accession.values]\n",
    "tmpfirat.ha_accession=accha\n",
    "tmpfirat.na_accession=accna"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 123,
   "metadata": {},
   "outputs": [],
   "source": [
    "mf=pd.concat([tmpfirat,tmpf[tmpf.is_irat==0]])\n",
    "mf.to_csv('combined_with_irat.csv')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 124,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>subtype</th>\n",
       "      <th>ha_accession</th>\n",
       "      <th>na_accession</th>\n",
       "      <th>ha</th>\n",
       "      <th>na</th>\n",
       "      <th>ha_risk</th>\n",
       "      <th>na_risk</th>\n",
       "      <th>geometric_mean_risk</th>\n",
       "      <th>emergence_risk</th>\n",
       "      <th>emergence_risk_var</th>\n",
       "      <th>impact_risk</th>\n",
       "      <th>impact_risk_var</th>\n",
       "      <th>is_irat</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A/swine/Shandong/1207/2016</td>\n",
       "      <td>H1N1</td>\n",
       "      <td>IRAT7.5</td>\n",
       "      <td>IRAT6.9</td>\n",
       "      <td>MEARLFVLFCAFTTLKADTICVGYHANNSTDTVDTILEKNVTVTHS...</td>\n",
       "      <td>MNPNQKIITIGSICMTIGIASLILQIGNIISIWISHSIQIENQNQS...</td>\n",
       "      <td>[0.059268942695459764, 0.059207549599012387, 0...</td>\n",
       "      <td>[0.04153677158036683, 0.04121188550974603, 0.0...</td>\n",
       "      <td>[0.04961693798040306, 0.04939691038300986, 0.0...</td>\n",
       "      <td>7.500000</td>\n",
       "      <td>1.860546e-03</td>\n",
       "      <td>6.900000</td>\n",
       "      <td>2.573638e-06</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A/Ohio/13/2017</td>\n",
       "      <td>H3N2</td>\n",
       "      <td>IRAT6.6</td>\n",
       "      <td>IRAT5.8</td>\n",
       "      <td>MKTIIALSHILCLVFAQKLPGNDNNMATLCLGHHAVPNGTIVKTIT...</td>\n",
       "      <td>MNPNQKIITIGSVSLIIATICFLMQIAILVTTITLHFKQHNCDSSP...</td>\n",
       "      <td>[0.007443436845452373, 0.007528108280118501, 0...</td>\n",
       "      <td>[0.0684551151913223, 0.06848866622907772, 0.06...</td>\n",
       "      <td>[0.02257302209884124, 0.0227066090672604, 0.02...</td>\n",
       "      <td>6.600000</td>\n",
       "      <td>7.193062e-02</td>\n",
       "      <td>5.800000</td>\n",
       "      <td>3.940741e-03</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A/Hong Kong/125/2017</td>\n",
       "      <td>H7N9</td>\n",
       "      <td>IRAT6.5</td>\n",
       "      <td>IRAT7.5</td>\n",
       "      <td>MNTQILVFALIAIIPTNADKICLGHHAVSNGTKVNTLTERGVEVVN...</td>\n",
       "      <td>MNPNQKILCTSATAITIGAIAVLIGIANLGLNIGLHLKPGCNCSHS...</td>\n",
       "      <td>[0.008053012400112526, 0.011702039255706398, 0...</td>\n",
       "      <td>[0.001702640937331002, 0.0015986998079791897, ...</td>\n",
       "      <td>[0.0037028892207661, 0.004325280096255358, 0.0...</td>\n",
       "      <td>6.500000</td>\n",
       "      <td>5.773995e-02</td>\n",
       "      <td>7.500000</td>\n",
       "      <td>2.592511e-03</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A/Shanghai/02/2013</td>\n",
       "      <td>H7N9</td>\n",
       "      <td>IRAT6.4</td>\n",
       "      <td>IRAT7.2</td>\n",
       "      <td>MNTQILVFALIAIIPTNADKICLGHHAVSNGTKVNTLTERGVEVVN...</td>\n",
       "      <td>MNPNQKILCTSATAIIIGAIAVLIGMANLGLNIGLHLKPGCNCSHS...</td>\n",
       "      <td>[0.0029756019607934496, 0.003893470493957003, ...</td>\n",
       "      <td>[0.0014946170772583095, 0.0014623661892494138,...</td>\n",
       "      <td>[0.00210888252534967, 0.002386143249933443, 0....</td>\n",
       "      <td>6.400000</td>\n",
       "      <td>2.576900e-02</td>\n",
       "      <td>7.200000</td>\n",
       "      <td>5.179005e-04</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A/Anhui-Lujiang/39/2018</td>\n",
       "      <td>H9N2</td>\n",
       "      <td>IRAT6.2</td>\n",
       "      <td>IRAT5.9</td>\n",
       "      <td>METVSLITILLVATASNADKICIGYQSTNSTETVDTLTENNVPVTH...</td>\n",
       "      <td>MNPNQKITAIGSVSLIIAIICLLMQIAILTTTMTLHFGQKECSNPS...</td>\n",
       "      <td>[0.013913606506215747, 0.014682700295856159, 0...</td>\n",
       "      <td>[0.025868069685588593, 0.04350993746295142, 0....</td>\n",
       "      <td>[0.018971508708604274, 0.025275351068975476, 0...</td>\n",
       "      <td>6.200000</td>\n",
       "      <td>1.344396e-01</td>\n",
       "      <td>5.900000</td>\n",
       "      <td>1.385737e-02</td>\n",
       "      <td>1.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6083</th>\n",
       "      <td>A/chicken/Guangdong/F94/2022</td>\n",
       "      <td>H9N2</td>\n",
       "      <td>EPI2167482</td>\n",
       "      <td>EPI2167484</td>\n",
       "      <td>MKTVSLITILLAATVSNADKICIGYQSTNSTETVDTLTENNVPVTH...</td>\n",
       "      <td>MNPNQKITAIGSVSLIIAIICLLMQIAILTTTMTLHFGQKECSNPL...</td>\n",
       "      <td>[0.021926223328056015, 0.021926223328056015, 0...</td>\n",
       "      <td>[0.07341886162130176, 0.07341886162130176, 0.0...</td>\n",
       "      <td>[0.04012229251177334, 0.04012229251177334, 0.0...</td>\n",
       "      <td>6.033752</td>\n",
       "      <td>7.888609e-31</td>\n",
       "      <td>6.111792</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6084</th>\n",
       "      <td>A/chicken/Guangdong/F99/2022</td>\n",
       "      <td>H9N2</td>\n",
       "      <td>EPI2167503</td>\n",
       "      <td>EPI2167506</td>\n",
       "      <td>MEAVSLITILLAATVSNADKICIGYQSTNSTETVDTLTENNVPVTH...</td>\n",
       "      <td>MNPNQKITAIGSVSLIIAIICLLMQIAILTTTMTLHFRQKECSNPS...</td>\n",
       "      <td>[0.01898769074353181, 0.01898769074353181, 0.0...</td>\n",
       "      <td>[0.05202795688766431, 0.05202795688766431, 0.0...</td>\n",
       "      <td>[0.03143072947611582, 0.03143072947611582, 0.0...</td>\n",
       "      <td>6.202124</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>6.258162</td>\n",
       "      <td>7.888609e-31</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6085</th>\n",
       "      <td>A/chicken/Egypt/Menoufia/2021</td>\n",
       "      <td>H9N2</td>\n",
       "      <td>EPI2175832</td>\n",
       "      <td>EPI2175833</td>\n",
       "      <td>MEVIPLMIMLLLVTTNNADKICIGHQSTNSTETVDTLTETGVPVTH...</td>\n",
       "      <td>MNPNQRIIALGSASLTVATICLLIQIAILATTMTLHFKQNEYANTS...</td>\n",
       "      <td>[0.05534613906024509, 0.05534613906024509, 0.0...</td>\n",
       "      <td>[0.4610969543711347, 0.4610969543711347, 0.461...</td>\n",
       "      <td>[0.15974960455938633, 0.15974960455938633, 0.1...</td>\n",
       "      <td>4.344723</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>4.715680</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6086</th>\n",
       "      <td>A/chicken/South_Korea/SL20/2020</td>\n",
       "      <td>H9N2</td>\n",
       "      <td>EPI2176583</td>\n",
       "      <td>EPI2176585</td>\n",
       "      <td>METVSLITILVVATVSSADKICIGYQSTNSTETVDTLTENNVPVTH...</td>\n",
       "      <td>MNPNQKITAIGSVSLIIAIICLLMQIAILTTTMTFHFGQKECSNPS...</td>\n",
       "      <td>[0.02999852644258515, 0.02999852644258515, 0.0...</td>\n",
       "      <td>[0.0512932924280397, 0.0512932924280397, 0.051...</td>\n",
       "      <td>[0.03922656229176604, 0.03922656229176604, 0.0...</td>\n",
       "      <td>6.050818</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>6.126595</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6087</th>\n",
       "      <td>A/Greylag_Goose/South_Korea/SW21/2021</td>\n",
       "      <td>H9N2</td>\n",
       "      <td>EPI2188134</td>\n",
       "      <td>EPI2188136</td>\n",
       "      <td>METIALVAILLMATASNADKICIGYQSTNSTETVDTLIENNVPVTH...</td>\n",
       "      <td>MNPNQKIITIGSVSLTIAIVCFLMQIAILATTITLHFKQNECSIPS...</td>\n",
       "      <td>[0.0464078674105025, 0.0464078674105025, 0.046...</td>\n",
       "      <td>[0.46130905496342184, 0.46130905496342184, 0.4...</td>\n",
       "      <td>[0.14631599180543012, 0.14631599180543012, 0.1...</td>\n",
       "      <td>4.475996</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>4.814980</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>0.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6088 rows × 14 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                         id subtype ha_accession na_accession  \\\n",
       "0                A/swine/Shandong/1207/2016    H1N1      IRAT7.5      IRAT6.9   \n",
       "1                            A/Ohio/13/2017    H3N2      IRAT6.6      IRAT5.8   \n",
       "2                      A/Hong Kong/125/2017    H7N9      IRAT6.5      IRAT7.5   \n",
       "3                        A/Shanghai/02/2013    H7N9      IRAT6.4      IRAT7.2   \n",
       "4                   A/Anhui-Lujiang/39/2018    H9N2      IRAT6.2      IRAT5.9   \n",
       "...                                     ...     ...          ...          ...   \n",
       "6083           A/chicken/Guangdong/F94/2022    H9N2   EPI2167482   EPI2167484   \n",
       "6084           A/chicken/Guangdong/F99/2022    H9N2   EPI2167503   EPI2167506   \n",
       "6085          A/chicken/Egypt/Menoufia/2021    H9N2   EPI2175832   EPI2175833   \n",
       "6086        A/chicken/South_Korea/SL20/2020    H9N2   EPI2176583   EPI2176585   \n",
       "6087  A/Greylag_Goose/South_Korea/SW21/2021    H9N2   EPI2188134   EPI2188136   \n",
       "\n",
       "                                                     ha  \\\n",
       "0     MEARLFVLFCAFTTLKADTICVGYHANNSTDTVDTILEKNVTVTHS...   \n",
       "1     MKTIIALSHILCLVFAQKLPGNDNNMATLCLGHHAVPNGTIVKTIT...   \n",
       "2     MNTQILVFALIAIIPTNADKICLGHHAVSNGTKVNTLTERGVEVVN...   \n",
       "3     MNTQILVFALIAIIPTNADKICLGHHAVSNGTKVNTLTERGVEVVN...   \n",
       "4     METVSLITILLVATASNADKICIGYQSTNSTETVDTLTENNVPVTH...   \n",
       "...                                                 ...   \n",
       "6083  MKTVSLITILLAATVSNADKICIGYQSTNSTETVDTLTENNVPVTH...   \n",
       "6084  MEAVSLITILLAATVSNADKICIGYQSTNSTETVDTLTENNVPVTH...   \n",
       "6085  MEVIPLMIMLLLVTTNNADKICIGHQSTNSTETVDTLTETGVPVTH...   \n",
       "6086  METVSLITILVVATVSSADKICIGYQSTNSTETVDTLTENNVPVTH...   \n",
       "6087  METIALVAILLMATASNADKICIGYQSTNSTETVDTLIENNVPVTH...   \n",
       "\n",
       "                                                     na  \\\n",
       "0     MNPNQKIITIGSICMTIGIASLILQIGNIISIWISHSIQIENQNQS...   \n",
       "1     MNPNQKIITIGSVSLIIATICFLMQIAILVTTITLHFKQHNCDSSP...   \n",
       "2     MNPNQKILCTSATAITIGAIAVLIGIANLGLNIGLHLKPGCNCSHS...   \n",
       "3     MNPNQKILCTSATAIIIGAIAVLIGMANLGLNIGLHLKPGCNCSHS...   \n",
       "4     MNPNQKITAIGSVSLIIAIICLLMQIAILTTTMTLHFGQKECSNPS...   \n",
       "...                                                 ...   \n",
       "6083  MNPNQKITAIGSVSLIIAIICLLMQIAILTTTMTLHFGQKECSNPL...   \n",
       "6084  MNPNQKITAIGSVSLIIAIICLLMQIAILTTTMTLHFRQKECSNPS...   \n",
       "6085  MNPNQRIIALGSASLTVATICLLIQIAILATTMTLHFKQNEYANTS...   \n",
       "6086  MNPNQKITAIGSVSLIIAIICLLMQIAILTTTMTFHFGQKECSNPS...   \n",
       "6087  MNPNQKIITIGSVSLTIAIVCFLMQIAILATTITLHFKQNECSIPS...   \n",
       "\n",
       "                                                ha_risk  \\\n",
       "0     [0.059268942695459764, 0.059207549599012387, 0...   \n",
       "1     [0.007443436845452373, 0.007528108280118501, 0...   \n",
       "2     [0.008053012400112526, 0.011702039255706398, 0...   \n",
       "3     [0.0029756019607934496, 0.003893470493957003, ...   \n",
       "4     [0.013913606506215747, 0.014682700295856159, 0...   \n",
       "...                                                 ...   \n",
       "6083  [0.021926223328056015, 0.021926223328056015, 0...   \n",
       "6084  [0.01898769074353181, 0.01898769074353181, 0.0...   \n",
       "6085  [0.05534613906024509, 0.05534613906024509, 0.0...   \n",
       "6086  [0.02999852644258515, 0.02999852644258515, 0.0...   \n",
       "6087  [0.0464078674105025, 0.0464078674105025, 0.046...   \n",
       "\n",
       "                                                na_risk  \\\n",
       "0     [0.04153677158036683, 0.04121188550974603, 0.0...   \n",
       "1     [0.0684551151913223, 0.06848866622907772, 0.06...   \n",
       "2     [0.001702640937331002, 0.0015986998079791897, ...   \n",
       "3     [0.0014946170772583095, 0.0014623661892494138,...   \n",
       "4     [0.025868069685588593, 0.04350993746295142, 0....   \n",
       "...                                                 ...   \n",
       "6083  [0.07341886162130176, 0.07341886162130176, 0.0...   \n",
       "6084  [0.05202795688766431, 0.05202795688766431, 0.0...   \n",
       "6085  [0.4610969543711347, 0.4610969543711347, 0.461...   \n",
       "6086  [0.0512932924280397, 0.0512932924280397, 0.051...   \n",
       "6087  [0.46130905496342184, 0.46130905496342184, 0.4...   \n",
       "\n",
       "                                    geometric_mean_risk  emergence_risk  \\\n",
       "0     [0.04961693798040306, 0.04939691038300986, 0.0...        7.500000   \n",
       "1     [0.02257302209884124, 0.0227066090672604, 0.02...        6.600000   \n",
       "2     [0.0037028892207661, 0.004325280096255358, 0.0...        6.500000   \n",
       "3     [0.00210888252534967, 0.002386143249933443, 0....        6.400000   \n",
       "4     [0.018971508708604274, 0.025275351068975476, 0...        6.200000   \n",
       "...                                                 ...             ...   \n",
       "6083  [0.04012229251177334, 0.04012229251177334, 0.0...        6.033752   \n",
       "6084  [0.03143072947611582, 0.03143072947611582, 0.0...        6.202124   \n",
       "6085  [0.15974960455938633, 0.15974960455938633, 0.1...        4.344723   \n",
       "6086  [0.03922656229176604, 0.03922656229176604, 0.0...        6.050818   \n",
       "6087  [0.14631599180543012, 0.14631599180543012, 0.1...        4.475996   \n",
       "\n",
       "      emergence_risk_var  impact_risk  impact_risk_var  is_irat  \n",
       "0           1.860546e-03     6.900000     2.573638e-06      1.0  \n",
       "1           7.193062e-02     5.800000     3.940741e-03      1.0  \n",
       "2           5.773995e-02     7.500000     2.592511e-03      1.0  \n",
       "3           2.576900e-02     7.200000     5.179005e-04      1.0  \n",
       "4           1.344396e-01     5.900000     1.385737e-02      1.0  \n",
       "...                  ...          ...              ...      ...  \n",
       "6083        7.888609e-31     6.111792     0.000000e+00      0.0  \n",
       "6084        0.000000e+00     6.258162     7.888609e-31      0.0  \n",
       "6085        0.000000e+00     4.715680     0.000000e+00      0.0  \n",
       "6086        0.000000e+00     6.126595     0.000000e+00      0.0  \n",
       "6087        0.000000e+00     4.814980     0.000000e+00      0.0  \n",
       "\n",
       "[6088 rows x 14 columns]"
      ]
     },
     "execution_count": 124,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "mf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 125,
   "metadata": {},
   "outputs": [],
   "source": [
    "# parameters\n",
    "threshold=6  # draw tree above this emergenec risk threshold\n",
    "CONSTRUCT_PHYLO=False  # use constructed tree (tree construction takes hours)\n",
    "OUTPUT_DIR='./'\n",
    "PHYLO_TREE_DIR='./'\n",
    "PHYLO_DIR='./'\n",
    "#COMBINED_RESULTS='../../../../paper_data/irat_enet/results/animal_predictions/combined_results.csv'\n",
    "COMBINED_RESULTS='combined_with_irat.csv'\n",
    "#CONSTRUCT_PHYLO=False\n",
    "num_collapsed=19  # number of mutations within which leaves are collapsed\n",
    "VERBOSE=False"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# class definition to hold multi sequence data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 126,
   "metadata": {},
   "outputs": [],
   "source": [
    "class SeqInfo(object):\n",
    "    \"\"\"Holds information regarding the sequence.\n",
    "    \n",
    "    \"\"\"\n",
    "    def __init__(self, seq, \n",
    "                 protein,\n",
    "                 accession,\n",
    "                 subtype=None,\n",
    "                 id=None,\n",
    "                 name=None,\n",
    "                 host=None, \n",
    "                 date=None, \n",
    "                 erisk=None,\n",
    "                 irisk=None,\n",
    "                 risk_flag=None,\n",
    "                 country=None):\n",
    "        self.name = name\n",
    "        self.id = id\n",
    "        self.protein=protein\n",
    "        self.subtype=subtype        \n",
    "        self.seq = seq\n",
    "        self.accession = accession \n",
    "        self.host = host\n",
    "        self.date = date\n",
    "        self.erisk = erisk\n",
    "        self.irisk = irisk\n",
    "        self.risk_flag = risk_flag\n",
    "        self.country = country\n",
    "        \n",
    "class MultipleSeqInfo(object):\n",
    "    \"\"\"Holds information regarding multiple sequences.\n",
    "    \n",
    "    Args:\n",
    "        dataframe (pandas.DataFrame): list of records parsed from NCBI\n",
    "        accessionname (str): column name for accession id\n",
    "        proteinname (str): protein name \n",
    "        risk_threshold (float): emergence risk threshold to compute distance matrix\n",
    "    \"\"\"\n",
    "    def __init__(self,\n",
    "                 dataframe,\n",
    "                 accessionname,\n",
    "                 proteinname,\n",
    "                 risk_threshold=6.2):\n",
    "        \n",
    "        self.seq_infos = {}\n",
    "        self.risk_threshold = risk_threshold\n",
    "        for i in np.arange(dataframe.index.size):\n",
    "            record=dataframe.iloc[i,:]\n",
    "            seqinfo = SeqInfo(\n",
    "                name=record.id,\n",
    "                seq=record[proteinname], \n",
    "                protein=proteinname,\n",
    "                accession=record[accessionname],\n",
    "                subtype=record.subtype,\n",
    "                erisk=record.emergence_risk,\n",
    "                irisk=record.impact_risk,\n",
    "                risk_flag = record.emergence_risk > self.risk_threshold,\n",
    "                host=None,\n",
    "                date=None,\n",
    "                country=None)\n",
    "            #print(record.predicted_emergence_score > self.risk_threshold)\n",
    "            self.seq_infos[seqinfo.accession] = seqinfo\n",
    "            \n",
    "    \n",
    "    def compute_L_distance_matrix(self):\n",
    "        highriskseq = pd.DataFrame.from_dict({key:val.seq \n",
    "                                              for (key,val) in self.seq_infos.items() \n",
    "                                              if val.risk_flag},orient='index',columns=['seq'])\n",
    "        num=highriskseq.index.size\n",
    "        d=np.zeros([num,num])\n",
    "        for x in tqdm(np.arange(num*num)):\n",
    "            j=x//num\n",
    "            i=x-num*j\n",
    "            if i > j:\n",
    "                d[i,j] = Levenshtein.distance(highriskseq.seq.values[i],\n",
    "                                                  highriskseq.seq.values[j])\n",
    "        ds=pd.DataFrame(d)        \n",
    "        ds=(ds+ds.transpose())\n",
    "        ds.columns=highriskseq.index.values\n",
    "        self.highriskdistancematrix=ds.copy()\n",
    "        \n",
    "        self.highriskdistancematrix.to_csv('dm'+str(self.risk_threshold)+'.csv',index=None)\n",
    "        return \n",
    "    \n",
    "    \n",
    "    def accessions_to_subtype(self, accessions):\n",
    "        \"\"\"Create a dictionary mapping the accession to the host.\n",
    "        \"\"\"\n",
    "        \n",
    "        subtypes = []\n",
    "        for accession in accessions:\n",
    "            seqinfo = self.seq_infos[accession]\n",
    "            subtypes.append(seqinfo.subtype)\n",
    "            \n",
    "        return subtypes\n",
    "\n",
    "    def accessions_to_host(self, accessions):\n",
    "        \"\"\"Create a dictionary mapping the accession to the host.\n",
    "        \"\"\"\n",
    "        \n",
    "        hosts = []\n",
    "        for accession in accessions:\n",
    "            seqinfo = self.seq_infos[accession]\n",
    "            hosts.append(seqinfo.host)\n",
    "        return hosts\n",
    "           \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 127,
   "metadata": {},
   "outputs": [],
   "source": [
    "df=pd.read_csv(COMBINED_RESULTS,index_col=0).reset_index()\n",
    "#ALLinfoHA=MultipleSeqInfo(df.reset_index(),'ha_accession','ha',risk_threshold=6.054)\n",
    "ALLinfoHA=MultipleSeqInfo(df.reset_index(),'ha_accession','ha',risk_threshold=threshold)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 128,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "<__main__.MultipleSeqInfo at 0x7efbdc810650>"
      ]
     },
     "execution_count": 128,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ALLinfoHA"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 129,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "H1N1    896\n",
       "H3N2    822\n",
       "H9N2     37\n",
       "H7N9      6\n",
       "Name: subtype, dtype: int64"
      ]
     },
     "execution_count": 129,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[df.emergence_risk>threshold].subtype.value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "scrolled": true
   },
   "source": [
    "# generate distance matrix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      " 20%|██████▎                        | 633767/3101121 [00:27<01:39, 24895.94it/s]"
     ]
    }
   ],
   "source": [
    "ALLinfoHA.compute_L_distance_matrix()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# tree construction from distance matrix using biopython"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 91,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio.Phylo import TreeConstruction\n",
    "from Bio import Phylo\n",
    "from Bio.SeqRecord import SeqRecord\n",
    "from Bio.Seq import Seq\n",
    "from Bio.Align import MultipleSeqAlignment\n",
    "from Bio import Entrez\n",
    "from Bio import SeqIO\n",
    "\n",
    "\n",
    "def load_dm(file_, upper_diag=True):\n",
    "    \"\"\"Load the distance matrix. \n",
    "    \n",
    "    Also, do some preprocessing. \n",
    "    \"\"\"\n",
    "    \n",
    "    df = pd.read_csv(file_)\n",
    "    #df.set_index('Unnamed: 0', inplace=True)\n",
    "    #assert np.all(df.columns == df.index)\n",
    "    \n",
    "    # drop duplicate columns after reading csv\n",
    "    #df = df.loc[:, ~df.columns.str.replace(\"(\\.\\d+)$\", \"\").duplicated()]\n",
    "    \n",
    "    if upper_diag:\n",
    "        df = df + df.T\n",
    "    return df\n",
    "\n",
    "def save_tree(tree, file_name, save_type='xml'):\n",
    "    \"\"\"Saved the created phylogenetic tree.\"\"\"\n",
    "    \n",
    "    if save_type == 'pickle':\n",
    "        graph = Phylo.to_networkx(tree)\n",
    "        save_pickled(graph, file_name)\n",
    "    elif save_type == 'xml':\n",
    "        Phylo.write(tree, file_name, 'phyloxml')\n",
    "    else:\n",
    "        raise ValueError('Not a correct save type.')\n",
    "    \n",
    "def pandas_dm_to_biopython_dm(dm):\n",
    "    \"\"\"Convert the pandas distance matrix to the biopython distance matrix.\n",
    "    \n",
    "    Returns:\n",
    "        biopython distance matrix\n",
    "    \"\"\"\n",
    "    \n",
    "    accessions = dm.columns\n",
    "    bio_dm = []\n",
    "    for i, accession in enumerate(accessions):\n",
    "        bio_dm.append(list(dm.iloc[i, :i+1].values))\n",
    "        \n",
    "    bio_dm = TreeConstruction._DistanceMatrix(\n",
    "        list(dm.columns), \n",
    "        bio_dm)\n",
    "    \n",
    "    return bio_dm\n",
    "\n",
    "def distance_matrix_to_phylo_tree(dm, outfile=None):\n",
    "    \"\"\"Create a phylogenetic tree from the distance matrix.\"\"\"\n",
    "    \n",
    "    dm = pandas_dm_to_biopython_dm(dm)\n",
    "    \n",
    "    treeConstructor = TreeConstruction.DistanceTreeConstructor()\n",
    "    tree = treeConstructor.nj(dm)\n",
    "    \n",
    "    if outfile is not None:\n",
    "        save_tree(tree, outfile)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# ete3 function, not all of these are used here"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [],
   "source": [
    "from ete3 import Tree, TreeStyle\n",
    "from ete3 import Phyloxml\n",
    "from ete3 import AttrFace, faces, Tree, NodeStyle, TreeStyle\n",
    "\n",
    "def load_pickled(file_name):\n",
    "    with open(file_name, 'rb') as f:\n",
    "        return pickle.load(f, encoding='latin')\n",
    "\n",
    "\n",
    "def get_farthest_node(tree, sequence):\n",
    "    return (tree&sequence).get_farthest_node()\n",
    "\n",
    "def get_all_accessions_from_tree(tree):\n",
    "    return [leaf_node.name for leaf_node in tree.get_leaves()]\n",
    "\n",
    "def remove_certain_hosts_from_tree(tree, hosts):\n",
    "    \"\"\"Remove leaf nodes if the host of that leaf is in `hosts`\"\"\"\n",
    "    \n",
    "    tree = copy.deepcopy(tree)\n",
    "    \n",
    "    removed_accessions = []\n",
    "    for leaf_node in tree.get_leaves():\n",
    "        if leaf_node.host in hosts:\n",
    "            leaf_node.detach()\n",
    "            \n",
    "    return tree\n",
    "\n",
    "def set_midpoint_outgroup(tree):\n",
    "    tree.set_outgroup(tree.get_midpoint_outgroup())\n",
    "\n",
    "\n",
    "def load_tree(filename, type_='phyloxml'):\n",
    "    \"\"\"Load saved phylogenetic tree.\n",
    "    \"\"\"\n",
    "    \n",
    "    if type_ == 'phyloxml':\n",
    "        project = Phyloxml()\n",
    "        project.build_from_file(filename)\n",
    "\n",
    "        for tree in project.get_phylogeny():\n",
    "            break\n",
    "\n",
    "        t=tree\n",
    "        \n",
    "    elif type_ == 'newick':\n",
    "        t = Tree(filename, format=1)\n",
    "    else:\n",
    "        raise ValueError('Not a correct type.')\n",
    "    \n",
    "    return t"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# construct tree from distance matrix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 93,
   "metadata": {},
   "outputs": [],
   "source": [
    "#CONSTRUCT_PHYLO=True\n",
    "if CONSTRUCT_PHYLO:\n",
    "    ALL_dm_ldistance = load_dm(\n",
    "        OUTPUT_DIR + 'dm'+str(threshold)+'.csv', \n",
    "        upper_diag=False)\n",
    "    \n",
    "    distance_matrix_to_phylo_tree(\n",
    "        ALL_dm_ldistance, PHYLO_TREE_DIR + 'ldistance'+str(threshold)+'.xml')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# convert phyloxml tree to newick tree to manipulate trees"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "Phylo.convert(\n",
    "    PHYLO_DIR + 'ldistance'+str(threshold)+'.xml','phyloxml',\n",
    "    PHYLO_DIR + 'ldistance'+str(threshold)+'.nhx','newick')\n",
    "\n",
    "ltree = load_tree(\n",
    "    PHYLO_DIR + 'ldistance'+str(threshold)+'.nhx',\n",
    "    type_='newick')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# label nodes in tree to add other attributes like subtype risk etc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {},
   "outputs": [],
   "source": [
    "def bandify(val,min=6.05,max=6.71):\n",
    "    maptoten=int(np.ceil(((val-min)/(max-min))*10))\n",
    "    return ' '+u'\\u2580'*maptoten\n",
    "\n",
    "def label_nodes(\n",
    "        tree, \n",
    "        recordinfo):\n",
    "    \"\"\"Label the nodes of the tree.\n",
    "    \n",
    "    We label nodes on whether:\n",
    "        it is covid19\n",
    "    \"\"\"\n",
    "    \n",
    "    tree = copy.deepcopy(tree)\n",
    "    \n",
    "    for node in tree:\n",
    "        name = node.name      \n",
    "        node.subtype = recordinfo.seq_infos[name].subtype\n",
    "        node.erisk =recordinfo.seq_infos[name].erisk\n",
    "        node.id = recordinfo.seq_infos[name].name + bandify(recordinfo.seq_infos[name].erisk,min=threshold)\n",
    "        if VERBOSE:\n",
    "            print(node.name,node.subtype,node.id,node.erisk)\n",
    "    return tree"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# construct labelled tree"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {},
   "outputs": [],
   "source": [
    "labelled_tree=label_nodes(\n",
    "    ltree, ALLinfoHA)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# functions to collapse simialr leaves"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 97,
   "metadata": {},
   "outputs": [],
   "source": [
    "def prune_nodes(t):\n",
    "    # collapsed nodes are labeled, so you locate them and prune them\n",
    "    for n in t.search_nodes(collapsed=True):\n",
    "        for ch in n.get_children():\n",
    "            ch.detach()\n",
    "            \n",
    "            \n",
    "def mean(array):\n",
    "    return sum(array)/float(len(array))\n",
    "\n",
    "def cache_distances(tree):\n",
    "    ''' precalculate distances of all nodes to the root''' \n",
    "    node2rootdist = {tree:0}\n",
    "    for node in tree.iter_descendants('preorder'):\n",
    "        node2rootdist[node] = node.dist + node2rootdist[node.up]\n",
    "    return node2rootdist\n",
    "\n",
    "def closest_node(node, node2tips, root_distance):\n",
    "    \"\"\"Find the closest node.\"\"\"\n",
    "    \n",
    "    tips = []\n",
    "    distances = []\n",
    "    for tip in node2tips[node]:\n",
    "        distances.append(root_distance[tip]-root_distance[node])\n",
    "        tips.append(tip)\n",
    "        #     index = np.argmin([root_distance[tip]-root_distance[node] for tip in node2tips[node]])\n",
    "    index = np.argmin(distances)\n",
    "    return tips[index]\n",
    "\n",
    "def riskiest_node(node, node2tips):\n",
    "    \"\"\"Find the closest node.\"\"\"\n",
    "    \n",
    "    tips = []\n",
    "    risks = []\n",
    "    for tip in node2tips[node]:\n",
    "        risks.append(tip.erisk)\n",
    "        tips.append(tip)\n",
    "        #     index = np.argmin([root_distance[tip]-root_distance[node] for tip in node2tips[node]])\n",
    "    index = np.argmax(risks)\n",
    "    return tips[index]\n",
    "\n",
    "def collapse(tree, min_dist,AllrecordInfo):\n",
    "    # cache the tip content of each node to reduce the number of times the tree is traversed\n",
    "    \n",
    "    tree = copy.deepcopy(tree)\n",
    "    \n",
    "    node2tips = tree.get_cached_content()\n",
    "    root_distance = cache_distances(tree)\n",
    "\n",
    "    for node in tree.get_descendants('preorder'):\n",
    "        if not node.is_leaf():\n",
    "            avg_distance_to_tips = mean([root_distance[tip]-root_distance[node]\n",
    "                                         for tip in node2tips[node]])\n",
    "            if VERBOSE:\n",
    "                print(avg_distance_to_tips)\n",
    "            if avg_distance_to_tips < min_dist:\n",
    "                # do whatever, ete support node annotation, deletion, labeling, etc.\n",
    "            \n",
    "                #closest_name = closest_node(node, node2tips, root_distance).name\n",
    "                closest_name = riskiest_node(node, node2tips).name\n",
    "                node.subtype = AllrecordInfo.seq_infos[closest_name].subtype\n",
    "                node.id = AllrecordInfo.seq_infos[closest_name].name + bandify(AllrecordInfo.seq_infos[closest_name].erisk,min=threshold)\n",
    "                node.name = '%s (%g)' %(closest_name,avg_distance_to_tips)\n",
    "                \n",
    "            \n",
    "                node.add_features(collapsed=True)\n",
    "\n",
    "                # set drawing attribute so they look collapsed when displayed with tree.show()\n",
    "                node.img_style['draw_descendants'] = False\n",
    "\n",
    "    return tree"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 98,
   "metadata": {},
   "outputs": [],
   "source": [
    "# collapse leaved"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {},
   "outputs": [],
   "source": [
    "num_collapsed=1\n",
    "ltree_collapsed = collapse(\n",
    "    labelled_tree, \n",
    "    min_dist=num_collapsed, \n",
    "    AllrecordInfo=ALLinfoHA)\n",
    "\n",
    "prune_nodes(ltree_collapsed)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# code for actual rendering"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 111,
   "metadata": {},
   "outputs": [],
   "source": [
    "# COLBAT='DarkRed'\n",
    "# COLRAT='SteelBlue'\n",
    "COLH3N2='Blue'\n",
    "COLH1N1='DarkRed'\n",
    "COLH7N9='DarkGreen'\n",
    "COLH9N2='#BD890F'\n",
    "COLDEF='BLACK'\n",
    "\n",
    "FS=50\n",
    "PW=10\n",
    "\n",
    "\n",
    "def nodeAttribConstruct(color, node):\n",
    "    N = AttrFace(\n",
    "        \"id\", fsize=FS, \n",
    "        text_prefix=\" \",penwidth=PW,ftype='Arial',\n",
    "        fgcolor=color,fstyle='bold')\n",
    "    faces.add_face_to_node(N, node, 1, position=\"branch-right\")\n",
    "    return N\n",
    "\n",
    "def layout(node):\n",
    "    if node.is_leaf():\n",
    "        if  node.subtype == 'H1N1':\n",
    "            N = nodeAttribConstruct(COLH1N1,node)\n",
    "        elif node.subtype == 'H3N2':\n",
    "            N = nodeAttribConstruct(COLH3N2,node)\n",
    "        elif node.subtype == 'H7N9':\n",
    "            N = nodeAttribConstruct(COLH7N9,node)\n",
    "        elif node.subtype == 'H9N2':\n",
    "            N = nodeAttribConstruct(COLH9N2,node)\n",
    "        else:\n",
    "            N = nodeAttribConstruct(COLDEF,node)\n",
    "            \n",
    "\n",
    "            \n",
    "def render_tree(tree, outfile):# all_seq_data, display_type='nearest_host'):\n",
    "    \"\"\"Render the tree inside the file to a circular \n",
    "    phylogenetic tree.\n",
    "    \n",
    "    NOTE: outfile should be in .pdf for best visuals\n",
    "    Returns:\n",
    "    \"\"\"\n",
    "    #tree = Tree(nwfile,format=1)\n",
    "\n",
    "    ts = TreeStyle()\n",
    "    ns = NodeStyle()\n",
    "    ts.show_leaf_name = False\n",
    "    #ts.rotation = 90\n",
    "    ts.mode = \"r\"\n",
    "    #ts.arc_start = -360 # 0 degrees = 3 o'clock\n",
    "    #ts.arc_span = 360\n",
    "    ts.scale=5\n",
    "    ts.show_scale=False\n",
    "    ts.branch_vertical_margin = .5 # 10 pixels between adjacent branches\n",
    "    # ts.show_branch_length=True\n",
    "    #ts.min_leaf_separation=10\n",
    "    #ts.optimal_scale_level='full'\n",
    "    #ts.branch_vertical_margin=0\n",
    "    \n",
    "    ns.hz_line_width=2\n",
    "    ns.vt_line_width=1\n",
    "    #ts.layout_fn = layout\n",
    "    ns[\"vt_line_width\"] = 16\n",
    "    ns[\"hz_line_width\"] = 16\n",
    "    #     ns['fsize'] = 20\n",
    "    for n in tree.traverse():\n",
    "        n.set_style(ns)\n",
    "        \n",
    "    #all_accessions = all_seq_data['accessions'].values\n",
    "    for n in tree:\n",
    "        ts.layout_fn = layout\n",
    "\n",
    "        \n",
    "    tree.set_style(ns)\n",
    "    tree.set_style(ts)\n",
    "    \n",
    "    ax=tree.render(\n",
    "        outfile, \n",
    "        dpi=300, \n",
    "        w=500,\n",
    "        tree_style=ts)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 116,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "6.3748157030451695"
      ]
     },
     "execution_count": 116,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ALLinfoHA.seq_infos[node.name.split()[0]].erisk"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>subtype</th>\n",
       "      <th>ha_accession</th>\n",
       "      <th>na_accession</th>\n",
       "      <th>impact_risk</th>\n",
       "      <th>emergence_risk</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>5745</th>\n",
       "      <td>A/Chicken/Shanxi/1012/2021</td>\n",
       "      <td>H7N9</td>\n",
       "      <td>EPI1857209</td>\n",
       "      <td>EPI1857211</td>\n",
       "      <td>6.048705</td>\n",
       "      <td>5.960484</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                              id subtype ha_accession na_accession  \\\n",
       "5745  A/Chicken/Shanxi/1012/2021    H7N9   EPI1857209   EPI1857211   \n",
       "\n",
       "      impact_risk  emergence_risk  \n",
       "5745     6.048705        5.960484  "
      ]
     },
     "execution_count": 113,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "r=5\n",
    "medrisknames=[]\n",
    "for node in ltree_collapsed:\n",
    "    if ALLinfoHA.seq_infos[node.name.split()[0]].erisk > r:\n",
    "        medrisknames=np.append(medrisknames,\n",
    "                                ALLinfoHA.seq_infos[node.name.split()[0]].name)\n",
    "medrisknames=df[df.id.isin(medrisknames)][['id','subtype',\n",
    "                                               'ha_accession',\n",
    "                                               'na_accession',\n",
    "                                               'impact_risk',\n",
    "                                               'emergence_risk']].sort_values('emergence_risk',\n",
    "                                                                                         ascending=False)\n",
    "medrisknames.to_csv('allriskystrains_collapsed.csv',index=None)\n",
    "medrisknames[medrisknames.id.str.contains('Sha')]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Render phylogenetic trees"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 102,
   "metadata": {},
   "outputs": [],
   "source": [
    "ax=render_tree(\n",
    "    labelled_tree,\n",
    "    './riskyphylo'+str(threshold)+'_irat.pdf')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 103,
   "metadata": {},
   "outputs": [],
   "source": [
    "ax=render_tree(\n",
    "    ltree_collapsed,\n",
    "    'riskyphylo'+str(threshold)+'_collapsed_'+str(num_collapsed)+'_irat.pdf')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 104,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Tree node 'Inner1771' (0x7efb9fe9335)"
      ]
     },
     "execution_count": 104,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ltree_collapsed"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Find high risk strains which are on distinct branches on phylogenetic tree"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>subtype</th>\n",
       "      <th>HA accession</th>\n",
       "      <th>NA accession</th>\n",
       "      <th>predicted IRAT impact</th>\n",
       "      <th>predicted IRAT emergence</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>strain</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A/swine/Missouri/A02524711/2020</th>\n",
       "      <td>H1N1</td>\n",
       "      <td>EPI1818121</td>\n",
       "      <td>EPI1818122</td>\n",
       "      <td>6.767254</td>\n",
       "      <td>6.782248</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A/swine/Indiana/A02524710/2020</th>\n",
       "      <td>H3N2</td>\n",
       "      <td>EPI1818137</td>\n",
       "      <td>EPI1818138</td>\n",
       "      <td>6.720489</td>\n",
       "      <td>6.729258</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A/swine/North_Carolina/A02479173/2020</th>\n",
       "      <td>H1N1</td>\n",
       "      <td>EPI1780425</td>\n",
       "      <td>EPI1780426</td>\n",
       "      <td>6.713621</td>\n",
       "      <td>6.721477</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A/Camel/Inner_Mongolia/XL/2020</th>\n",
       "      <td>H7N9</td>\n",
       "      <td>EPI2026200</td>\n",
       "      <td>EPI2026202</td>\n",
       "      <td>6.699016</td>\n",
       "      <td>6.704867</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A/swine/Tennessee/A02524414/2022</th>\n",
       "      <td>H1N1</td>\n",
       "      <td>EPI2149257</td>\n",
       "      <td>EPI2149258</td>\n",
       "      <td>6.650119</td>\n",
       "      <td>6.649429</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A/swine/Illinois/A02525253/2021</th>\n",
       "      <td>H3N2</td>\n",
       "      <td>EPI1910375</td>\n",
       "      <td>EPI1910376</td>\n",
       "      <td>6.397379</td>\n",
       "      <td>6.361542</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A/swine/Kansas/A02245381/2020(H1N1)</th>\n",
       "      <td>H1N1</td>\n",
       "      <td>EPI1777723</td>\n",
       "      <td>EPI1777724</td>\n",
       "      <td>6.344661</td>\n",
       "      <td>6.301257</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A/canine/Texas/21-011409-001/2021</th>\n",
       "      <td>H3N2</td>\n",
       "      <td>EPI1896555</td>\n",
       "      <td>EPI1896557</td>\n",
       "      <td>6.344011</td>\n",
       "      <td>6.300499</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A/swine/Iowa/A02246996/2021</th>\n",
       "      <td>H1N1</td>\n",
       "      <td>EPI2146133</td>\n",
       "      <td>EPI2146134</td>\n",
       "      <td>6.326870</td>\n",
       "      <td>6.280888</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A/mink/China/chick_embryo/2020</th>\n",
       "      <td>H9N2</td>\n",
       "      <td>EPI2161544</td>\n",
       "      <td>EPI2161548</td>\n",
       "      <td>6.318778</td>\n",
       "      <td>6.271620</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>68 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                      subtype HA accession NA accession  \\\n",
       "strain                                                                    \n",
       "A/swine/Missouri/A02524711/2020          H1N1   EPI1818121   EPI1818122   \n",
       "A/swine/Indiana/A02524710/2020           H3N2   EPI1818137   EPI1818138   \n",
       "A/swine/North_Carolina/A02479173/2020    H1N1   EPI1780425   EPI1780426   \n",
       "A/Camel/Inner_Mongolia/XL/2020           H7N9   EPI2026200   EPI2026202   \n",
       "A/swine/Tennessee/A02524414/2022         H1N1   EPI2149257   EPI2149258   \n",
       "...                                       ...          ...          ...   \n",
       "A/swine/Illinois/A02525253/2021          H3N2   EPI1910375   EPI1910376   \n",
       "A/swine/Kansas/A02245381/2020(H1N1)      H1N1   EPI1777723   EPI1777724   \n",
       "A/canine/Texas/21-011409-001/2021        H3N2   EPI1896555   EPI1896557   \n",
       "A/swine/Iowa/A02246996/2021              H1N1   EPI2146133   EPI2146134   \n",
       "A/mink/China/chick_embryo/2020           H9N2   EPI2161544   EPI2161548   \n",
       "\n",
       "                                       predicted IRAT impact  \\\n",
       "strain                                                         \n",
       "A/swine/Missouri/A02524711/2020                     6.767254   \n",
       "A/swine/Indiana/A02524710/2020                      6.720489   \n",
       "A/swine/North_Carolina/A02479173/2020               6.713621   \n",
       "A/Camel/Inner_Mongolia/XL/2020                      6.699016   \n",
       "A/swine/Tennessee/A02524414/2022                    6.650119   \n",
       "...                                                      ...   \n",
       "A/swine/Illinois/A02525253/2021                     6.397379   \n",
       "A/swine/Kansas/A02245381/2020(H1N1)                 6.344661   \n",
       "A/canine/Texas/21-011409-001/2021                   6.344011   \n",
       "A/swine/Iowa/A02246996/2021                         6.326870   \n",
       "A/mink/China/chick_embryo/2020                      6.318778   \n",
       "\n",
       "                                       predicted IRAT emergence  \n",
       "strain                                                           \n",
       "A/swine/Missouri/A02524711/2020                        6.782248  \n",
       "A/swine/Indiana/A02524710/2020                         6.729258  \n",
       "A/swine/North_Carolina/A02479173/2020                  6.721477  \n",
       "A/Camel/Inner_Mongolia/XL/2020                         6.704867  \n",
       "A/swine/Tennessee/A02524414/2022                       6.649429  \n",
       "...                                                         ...  \n",
       "A/swine/Illinois/A02525253/2021                        6.361542  \n",
       "A/swine/Kansas/A02245381/2020(H1N1)                    6.301257  \n",
       "A/canine/Texas/21-011409-001/2021                      6.300499  \n",
       "A/swine/Iowa/A02246996/2021                            6.280888  \n",
       "A/mink/China/chick_embryo/2020                         6.271620  \n",
       "\n",
       "[68 rows x 5 columns]"
      ]
     },
     "execution_count": 31,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "r=6.057\n",
    "r=6.25\n",
    "highrisknames=[]\n",
    "for node in ltree_collapsed:\n",
    "    if ALLinfoHA.seq_infos[node.name.split()[0]].erisk > r:\n",
    "        highrisknames=np.append(highrisknames,\n",
    "                                ALLinfoHA.seq_infos[node.name.split()[0]].name)\n",
    "highrisknamesdf=df[df.id.isin(highrisknames)][['id','subtype',\n",
    "                                               'ha_accession',\n",
    "                                               'na_accession',\n",
    "                                               'impact_risk',\n",
    "                                               'emergence_risk']].sort_values('emergence_risk',\n",
    "                                                                                         ascending=False)\n",
    "highrisknamesdf = highrisknamesdf.rename(columns={'id':'strain',\n",
    "                                                  'ha_accession':'HA accession',\n",
    "                                                  'na_accession':'NA accession',\n",
    "                                                  'impact_risk':'predicted IRAT impact',\n",
    "                                                  'emergence_risk':'predicted IRAT emergence'}).set_index('strain')\n",
    "highrisknamesdf#.drop_duplicates()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>subtype</th>\n",
       "      <th>HA accession</th>\n",
       "      <th>NA accession</th>\n",
       "      <th>predicted IRAT impact</th>\n",
       "      <th>predicted IRAT emergence</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>strain</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>\\rowcolor{Red3!20}A/swine/Missouri/A02524711/2020</th>\n",
       "      <td>H1N1</td>\n",
       "      <td>EPI1818121</td>\n",
       "      <td>EPI1818122</td>\n",
       "      <td>6.767254</td>\n",
       "      <td>6.782248</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>\\rowcolor{Blue1!30}A/swine/Indiana/A02524710/2020</th>\n",
       "      <td>H3N2</td>\n",
       "      <td>EPI1818137</td>\n",
       "      <td>EPI1818138</td>\n",
       "      <td>6.720489</td>\n",
       "      <td>6.729258</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>\\rowcolor{Red3!20}A/swine/North_Carolina/A02479173/2020</th>\n",
       "      <td>H1N1</td>\n",
       "      <td>EPI1780425</td>\n",
       "      <td>EPI1780426</td>\n",
       "      <td>6.713621</td>\n",
       "      <td>6.721477</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>\\rowcolor{Green3!50}A/Camel/Inner_Mongolia/XL/2020</th>\n",
       "      <td>H7N9</td>\n",
       "      <td>EPI2026200</td>\n",
       "      <td>EPI2026202</td>\n",
       "      <td>6.699016</td>\n",
       "      <td>6.704867</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>\\rowcolor{Red3!20}A/swine/Tennessee/A02524414/2022</th>\n",
       "      <td>H1N1</td>\n",
       "      <td>EPI2149257</td>\n",
       "      <td>EPI2149258</td>\n",
       "      <td>6.650119</td>\n",
       "      <td>6.649429</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>\\rowcolor{Blue1!30}A/swine/Illinois/A02525253/2021</th>\n",
       "      <td>H3N2</td>\n",
       "      <td>EPI1910375</td>\n",
       "      <td>EPI1910376</td>\n",
       "      <td>6.397379</td>\n",
       "      <td>6.361542</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>\\rowcolor{Red3!20}A/swine/Kansas/A02245381/2020(H1N1)</th>\n",
       "      <td>H1N1</td>\n",
       "      <td>EPI1777723</td>\n",
       "      <td>EPI1777724</td>\n",
       "      <td>6.344661</td>\n",
       "      <td>6.301257</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>\\rowcolor{Blue1!30}A/canine/Texas/21-011409-001/2021</th>\n",
       "      <td>H3N2</td>\n",
       "      <td>EPI1896555</td>\n",
       "      <td>EPI1896557</td>\n",
       "      <td>6.344011</td>\n",
       "      <td>6.300499</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>\\rowcolor{Red3!20}A/swine/Iowa/A02246996/2021</th>\n",
       "      <td>H1N1</td>\n",
       "      <td>EPI2146133</td>\n",
       "      <td>EPI2146134</td>\n",
       "      <td>6.326870</td>\n",
       "      <td>6.280888</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>\\rowcolor{DarkOrange!40}A/mink/China/chick_embryo/2020</th>\n",
       "      <td>H9N2</td>\n",
       "      <td>EPI2161544</td>\n",
       "      <td>EPI2161548</td>\n",
       "      <td>6.318778</td>\n",
       "      <td>6.271620</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>68 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   subtype HA accession  \\\n",
       "strain                                                                    \n",
       "\\rowcolor{Red3!20}A/swine/Missouri/A02524711/2020     H1N1   EPI1818121   \n",
       "\\rowcolor{Blue1!30}A/swine/Indiana/A02524710/2020     H3N2   EPI1818137   \n",
       "\\rowcolor{Red3!20}A/swine/North_Carolina/A02479...    H1N1   EPI1780425   \n",
       "\\rowcolor{Green3!50}A/Camel/Inner_Mongolia/XL/2020    H7N9   EPI2026200   \n",
       "\\rowcolor{Red3!20}A/swine/Tennessee/A02524414/2022    H1N1   EPI2149257   \n",
       "...                                                    ...          ...   \n",
       "\\rowcolor{Blue1!30}A/swine/Illinois/A02525253/2021    H3N2   EPI1910375   \n",
       "\\rowcolor{Red3!20}A/swine/Kansas/A02245381/2020...    H1N1   EPI1777723   \n",
       "\\rowcolor{Blue1!30}A/canine/Texas/21-011409-001...    H3N2   EPI1896555   \n",
       "\\rowcolor{Red3!20}A/swine/Iowa/A02246996/2021         H1N1   EPI2146133   \n",
       "\\rowcolor{DarkOrange!40}A/mink/China/chick_embr...    H9N2   EPI2161544   \n",
       "\n",
       "                                                   NA accession  \\\n",
       "strain                                                            \n",
       "\\rowcolor{Red3!20}A/swine/Missouri/A02524711/2020    EPI1818122   \n",
       "\\rowcolor{Blue1!30}A/swine/Indiana/A02524710/2020    EPI1818138   \n",
       "\\rowcolor{Red3!20}A/swine/North_Carolina/A02479...   EPI1780426   \n",
       "\\rowcolor{Green3!50}A/Camel/Inner_Mongolia/XL/2020   EPI2026202   \n",
       "\\rowcolor{Red3!20}A/swine/Tennessee/A02524414/2022   EPI2149258   \n",
       "...                                                         ...   \n",
       "\\rowcolor{Blue1!30}A/swine/Illinois/A02525253/2021   EPI1910376   \n",
       "\\rowcolor{Red3!20}A/swine/Kansas/A02245381/2020...   EPI1777724   \n",
       "\\rowcolor{Blue1!30}A/canine/Texas/21-011409-001...   EPI1896557   \n",
       "\\rowcolor{Red3!20}A/swine/Iowa/A02246996/2021        EPI2146134   \n",
       "\\rowcolor{DarkOrange!40}A/mink/China/chick_embr...   EPI2161548   \n",
       "\n",
       "                                                    predicted IRAT impact  \\\n",
       "strain                                                                      \n",
       "\\rowcolor{Red3!20}A/swine/Missouri/A02524711/2020                6.767254   \n",
       "\\rowcolor{Blue1!30}A/swine/Indiana/A02524710/2020                6.720489   \n",
       "\\rowcolor{Red3!20}A/swine/North_Carolina/A02479...               6.713621   \n",
       "\\rowcolor{Green3!50}A/Camel/Inner_Mongolia/XL/2020               6.699016   \n",
       "\\rowcolor{Red3!20}A/swine/Tennessee/A02524414/2022               6.650119   \n",
       "...                                                                   ...   \n",
       "\\rowcolor{Blue1!30}A/swine/Illinois/A02525253/2021               6.397379   \n",
       "\\rowcolor{Red3!20}A/swine/Kansas/A02245381/2020...               6.344661   \n",
       "\\rowcolor{Blue1!30}A/canine/Texas/21-011409-001...               6.344011   \n",
       "\\rowcolor{Red3!20}A/swine/Iowa/A02246996/2021                    6.326870   \n",
       "\\rowcolor{DarkOrange!40}A/mink/China/chick_embr...               6.318778   \n",
       "\n",
       "                                                    predicted IRAT emergence  \n",
       "strain                                                                        \n",
       "\\rowcolor{Red3!20}A/swine/Missouri/A02524711/2020                   6.782248  \n",
       "\\rowcolor{Blue1!30}A/swine/Indiana/A02524710/2020                   6.729258  \n",
       "\\rowcolor{Red3!20}A/swine/North_Carolina/A02479...                  6.721477  \n",
       "\\rowcolor{Green3!50}A/Camel/Inner_Mongolia/XL/2020                  6.704867  \n",
       "\\rowcolor{Red3!20}A/swine/Tennessee/A02524414/2022                  6.649429  \n",
       "...                                                                      ...  \n",
       "\\rowcolor{Blue1!30}A/swine/Illinois/A02525253/2021                  6.361542  \n",
       "\\rowcolor{Red3!20}A/swine/Kansas/A02245381/2020...                  6.301257  \n",
       "\\rowcolor{Blue1!30}A/canine/Texas/21-011409-001...                  6.300499  \n",
       "\\rowcolor{Red3!20}A/swine/Iowa/A02246996/2021                       6.280888  \n",
       "\\rowcolor{DarkOrange!40}A/mink/China/chick_embr...                  6.271620  \n",
       "\n",
       "[68 rows x 5 columns]"
      ]
     },
     "execution_count": 32,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "COLDICT={'H1N1':'Red3!20','H3N2':'Blue1!30','H7N9':'Green3!50','H9N2':'DarkOrange!40'}\n",
    "def rowcolor(row):\n",
    "    return '\\\\rowcolor{' + COLDICT[row.subtype]+'}' + row['strain']\n",
    "\n",
    "highrisknamesdf1 = highrisknamesdf.reset_index(drop=False)\n",
    "highrisknamesdf1['strain']=highrisknamesdf1.apply(rowcolor,axis=1)\n",
    "highrisknamesdf1=highrisknamesdf1.set_index('strain')\n",
    "highrisknamesdf1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\\begin{tabular}{L{1.95in}|L{.25in}|L{.60in}|L{.6in}|C{1in}|C{1in}}\\hline\r\n",
      "strain&subtype& HA  accession & NA  accession & predicted  IRAT  impact & predicted  IRAT  emergence \\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Missouri/A02524711/2020 &H1N1&EPI1818121&EPI1818122&6.7673&6.7822\\\\\r\n",
      " \\rowcolor{Blue1!30}A/swine/Indiana/A02524710/2020 &H3N2&EPI1818137&EPI1818138&6.7205&6.7293\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/North\\_Carolina/A02479173/2020 &H1N1&EPI1780425&EPI1780426&6.7136&6.7215\\\\\r\n",
      " \\rowcolor{Green3!50}A/Camel/Inner\\_Mongolia/XL/2020 &H7N9&EPI2026200&EPI2026202&6.6990&6.7049\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Tennessee/A02524414/2022 &H1N1&EPI2149257&EPI2149258&6.6501&6.6494\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Minnesota/A02635976/2021 &H1N1&EPI1912208&EPI1912209&6.5776&6.5670\\\\\r\n",
      " \\rowcolor{Blue1!30}A/swine/Chile/VN1401-5054/2020 &H3N2&EPI1974975&EPI1974978&6.5318&6.5149\\\\\r\n",
      " \\rowcolor{Blue1!30}A/swine/Italy/56910/2020 &H3N2&EPI2142217&EPI2142173&6.5292&6.5119\\\\\r\n",
      " \\rowcolor{Blue1!30}A/swine/Minnesota/A02245643/2020 &H3N2&EPI1769178&EPI1769179&6.5067&6.4863\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Iowa/A02479005/2020 &H1N1&EPI1777621&EPI1777622&6.4872&6.4641\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Indiana/A02636638/2022 &H1N1&EPI2153370&EPI2153371&6.4534&6.4255\\\\\r\n",
      " \\rowcolor{Blue1!30}A/swine/Iowa/A02524874/2020 &H3N2&EPI1907838&EPI1907839&6.4392&6.4093\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Minnesota/A02248037/2021 &H1N1&EPI1912188&EPI1912189&6.4366&6.4063\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Iowa/A02635917/2021 &H1N1&EPI1911753&EPI1911754&6.4356&6.4052\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Illinois/A02635936/2021 &H1N1&EPI1911791&EPI1911792&6.4347&6.4042\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Minnesota/A02711801/2022 &H1N1&EPI2153420&EPI2153421&6.4334&6.4027\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/South\\_Dakota/A02524453/2020 &H1N1&EPI1765555&EPI1765556&6.4321&6.4012\\\\\r\n",
      " \\rowcolor{Blue1!30}A/swine/Illinois/A02479007/2020 &H3N2&EPI1777629&EPI1777630&6.4315&6.4005\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Minnesota/A02248061/2021 &H1N1&EPI1912494&EPI1912495&6.4278&6.3963\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Iowa/A02524875/2020 &H1N1&EPI1907858&EPI1907859&6.4260&6.3943\\\\\r\n",
      " \\rowcolor{Blue1!30}A/swine/Spain/44579-1/2020 &H3N2&EPI1930744&EPI1930748&6.4255&6.3937\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Iowa/A02636439/2022 &H1N1&EPI2147475&EPI2147476&6.4250&6.3931\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Minnesota/A02248060/2021 &H1N1&EPI1912500&EPI1912501&6.4234&6.3913\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Nebraska/A02636117/2021 &H1N1&EPI1932937&EPI1932938&6.4226&6.3903\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Iowa/A02524513/2020 &H1N1&EPI1832647&EPI1832648&6.4223&6.3901\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Nebraska/A02524954/2020 &H1N1&EPI1908393&EPI1908394&6.4222&6.3899\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Iowa/A02524994/2020 &H1N1&EPI1908427&EPI1908428&6.4222&6.3899\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Nebraska/A02479337/2020 &H1N1&EPI1769116&EPI1769117&6.4222&6.3899\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Iowa/A02479383/2020 &H1N1&EPI1771027&EPI1771028&6.4222&6.3899\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Nebraska/A02479212/2020 &H1N1&EPI1775884&EPI1775885&6.4222&6.3899\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Minnesota/A02479051/2020 &H1N1&EPI1778572&EPI1778573&6.4222&6.3899\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Minnesota/A02245424/2020 &H1N1&EPI1780207&EPI1780208&6.4222&6.3899\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Iowa/A02524646/2020 &H1N1&EPI1817164&EPI1817165&6.4222&6.3899\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Iowa/A02524724/2020 &H1N1&EPI1818387&EPI1818388&6.4222&6.3899\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Iowa/A02635719/2021 &H1N1&EPI1910907&EPI1910908&6.4222&6.3899\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Iowa/A02525313/2021 &H1N1&EPI1910761&EPI1910762&6.4222&6.3899\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Missouri/A02525065/2021 &H1N1&EPI1908581&EPI1908582&6.4222&6.3899\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Missouri/A02524951/2020 &H1N1&EPI1908429&EPI1908430&6.4222&6.3899\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Iowa/A02524892/2020 &H1N1&EPI1907881&EPI1907882&6.4222&6.3899\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Nebraska/A02479186/2020 &H1N1&EPI1774141&EPI1774142&6.4220&6.3897\\\\\r\n",
      " \\rowcolor{Blue1!30}A/swine/Minnesota/A02245699/2020 &H3N2&EPI1833007&EPI1833008&6.4220&6.3897\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Iowa/A02479156/2020 &H1N1&EPI1780249&EPI1780250&6.4216&6.3892\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Iowa/A02479229/2020 &H1N1&EPI1775914&EPI1775915&6.4215&6.3891\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Iowa/A02479303/2020 &H1N1&EPI1768639&EPI1768640&6.4207&6.3882\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Minnesota/A02710691/2021 &H1N1&EPI2146090&EPI2146091&6.4200&6.3874\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Iowa/A02635881/2021 &H1N1&EPI1911668&EPI1911669&6.4198&6.3872\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Iowa/A02525354/2021 &H1N1&EPI1910789&EPI1910790&6.4198&6.3872\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Iowa/A02635823/2021 &H1N1&EPI1911263&EPI1911264&6.4176&6.3847\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Iowa/A02524808/2020 &H1N1&EPI1907251&EPI1907252&6.4173&6.3843\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Iowa/A02479141/2020 &H1N1&EPI1780241&EPI1780242&6.4161&6.3829\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Nebraska/A02525367/2021 &H1N1&EPI1910817&EPI1910818&6.4136&6.3801\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Iowa/A02245587/2020 &H1N1&EPI1775817&EPI1775818&6.4119&6.3781\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Iowa/A02750621/2022 &H1N1&EPI2161576&EPI2161577&6.4116&6.3779\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Iowa/A02636496/2022 &H1N1&EPI2148086&EPI2148087&6.4105&6.3765\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Iowa/A02636145/2021 &H1N1&EPI1932055&EPI1932930&6.4090&6.3748\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Iowa/A02525217/2021 &H1N1&EPI1909087&EPI1909088&6.4090&6.3748\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Iowa/A02636114/2021 &H1N1&EPI1931853&EPI1931854&6.4090&6.3748\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Iowa/A02635871/2021 &H1N1&EPI1911656&EPI1911657&6.4089&6.3747\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Iowa/A02479067/2020 &H1N1&EPI1778734&EPI1778735&6.4083&6.3741\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Minnesota/A02246459/2021 &H1N1&EPI1912518&EPI1912519&6.4061&6.3716\\\\\r\n",
      " \\rowcolor{Blue1!30}A/swine/Minnesota/A02711797/2022 &H3N2&EPI2153382&EPI2153383&6.4029&6.3679\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Minnesota/A02525371/2021 &H1N1&EPI1910849&EPI1910850&6.3996&6.3641\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Iowa/A02479343/2020 &H1N1&EPI1769114&EPI1769115&6.3985&6.3628\\\\\r\n",
      " \\rowcolor{Blue1!30}A/swine/Illinois/A02525253/2021 &H3N2&EPI1910375&EPI1910376&6.3974&6.3615\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Kansas/A02245381/2020(H1N1) &H1N1&EPI1777723&EPI1777724&6.3447&6.3013\\\\\r\n",
      " \\rowcolor{Blue1!30}A/canine/Texas/21-011409-001/2021 &H3N2&EPI1896555&EPI1896557&6.3440&6.3005\\\\\r\n",
      " \\rowcolor{Red3!20}A/swine/Iowa/A02246996/2021 &H1N1&EPI2146133&EPI2146134&6.3269&6.2809\\\\\r\n",
      " \\rowcolor{DarkOrange!40}A/mink/China/chick\\_embryo/2020 &H9N2&EPI2161544&EPI2161548&6.3188&6.2716\\\\\r\n",
      "\\hline\\end{tabular}\r\n"
     ]
    }
   ],
   "source": [
    "from zedstat.textable import textable\n",
    "textable(highrisknamesdf1,#pd.concat([highrisknamesdf1.head(30),highrisknamesdf1.tail(10)]),\n",
    "         tabname='highrisk35.tex',\n",
    "         FORMAT='%1.4f',INDEX=True,\n",
    "         TABFORMAT='L{1.95in}|L{.25in}|L{.60in}|L{.6in}|C{1in}|C{1in}',LNTERM='\\\\\\\\\\n')\n",
    "! cat highrisk35.tex"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>H1N1</th>\n",
       "      <th>H3N2</th>\n",
       "      <th>H7N9</th>\n",
       "      <th>H9N2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A/swine/Missouri/A02524711/2020</td>\n",
       "      <td>A/swine/Indiana/A02524710/2020</td>\n",
       "      <td>A/Camel/Inner_Mongolia/XL/2020</td>\n",
       "      <td>A/mink/China/chick_embryo/2020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>6.782248</td>\n",
       "      <td>6.729258</td>\n",
       "      <td>6.704867</td>\n",
       "      <td>6.27162</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                              H1N1                            H3N2  \\\n",
       "0  A/swine/Missouri/A02524711/2020  A/swine/Indiana/A02524710/2020   \n",
       "1                         6.782248                        6.729258   \n",
       "\n",
       "                             H7N9                            H9N2  \n",
       "0  A/Camel/Inner_Mongolia/XL/2020  A/mink/China/chick_embryo/2020  \n",
       "1                        6.704867                         6.27162  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>6.0</th>\n",
       "      <th>6.2</th>\n",
       "      <th>6.3</th>\n",
       "      <th>6.4</th>\n",
       "      <th>6.5</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>H1N1</th>\n",
       "      <td>61</td>\n",
       "      <td>57</td>\n",
       "      <td>54</td>\n",
       "      <td>11</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>H3N2</th>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>11</td>\n",
       "      <td>6</td>\n",
       "      <td>3</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>H7N9</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>H9N2</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      6.0  6.2  6.3  6.4  6.5\n",
       "H1N1   61   57   54   11    4\n",
       "H3N2   11   11   11    6    3\n",
       "H7N9    1    1    1    1    1\n",
       "H9N2    1    1    0    0    0"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "75\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>subtype</th>\n",
       "      <th>ha_accession</th>\n",
       "      <th>na_accession</th>\n",
       "      <th>ha</th>\n",
       "      <th>na</th>\n",
       "      <th>ha_risk</th>\n",
       "      <th>na_risk</th>\n",
       "      <th>geometric_mean_risk</th>\n",
       "      <th>emergence_risk</th>\n",
       "      <th>emergence_risk_var</th>\n",
       "      <th>impact_risk</th>\n",
       "      <th>impact_risk_var</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>218</th>\n",
       "      <td>A/swine/Missouri/A02524711/2020</td>\n",
       "      <td>H1N1</td>\n",
       "      <td>EPI1818121</td>\n",
       "      <td>EPI1818122</td>\n",
       "      <td>MKAILVVMLYTFTTANADTLCIGYHANNSTDTVDTVLEKNVTVTHS...</td>\n",
       "      <td>MNPNQKIITIGSICMTIGTANLILQIGNIISIWVSHSIQIGNQSQI...</td>\n",
       "      <td>[0.007702821053332326, 0.00794167832708008, 0....</td>\n",
       "      <td>[0.0016000691389129699, 0.0019043683598783635,...</td>\n",
       "      <td>[0.0035107045233124007, 0.003888943420830784, ...</td>\n",
       "      <td>6.782248</td>\n",
       "      <td>0.000006</td>\n",
       "      <td>6.767254</td>\n",
       "      <td>0.000005</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1902</th>\n",
       "      <td>A/swine/Indiana/A02524710/2020</td>\n",
       "      <td>H3N2</td>\n",
       "      <td>EPI1818137</td>\n",
       "      <td>EPI1818138</td>\n",
       "      <td>MKTIIALSYILCLVFAQKIPGNDNGTATLCLGHHAVPNGTIVKTIT...</td>\n",
       "      <td>MNPNQKIITIGSVSLTISTICFFMQIAILITTIALHFKQYEFNSPP...</td>\n",
       "      <td>[0.008423100311281287, 0.0044835948774065975, ...</td>\n",
       "      <td>[0.006596476585994605, 0.006619358238135353, 0...</td>\n",
       "      <td>[0.00745404480700585, 0.00544779961894918, 0.0...</td>\n",
       "      <td>6.729258</td>\n",
       "      <td>0.000613</td>\n",
       "      <td>6.720489</td>\n",
       "      <td>0.000476</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5814</th>\n",
       "      <td>A/Camel/Inner_Mongolia/XL/2020</td>\n",
       "      <td>H7N9</td>\n",
       "      <td>EPI2026200</td>\n",
       "      <td>EPI2026202</td>\n",
       "      <td>MNTQILVFALIAIIPTNADKICLGHHAVSNGTKVNTLTEREVEVVN...</td>\n",
       "      <td>MNPNQKILCTSATAITIGAIAVLIGIANLGLNIGLHLKPGCNCSHS...</td>\n",
       "      <td>[0.025061424930009856, 0.01711340116551936, 0....</td>\n",
       "      <td>[0.0011227960579890436, 0.0012230115569459082,...</td>\n",
       "      <td>[0.0053046082908169015, 0.004574919387714034, ...</td>\n",
       "      <td>6.704867</td>\n",
       "      <td>0.004832</td>\n",
       "      <td>6.699016</td>\n",
       "      <td>0.003755</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6049</th>\n",
       "      <td>A/mink/China/chick_embryo/2020</td>\n",
       "      <td>H9N2</td>\n",
       "      <td>EPI2161544</td>\n",
       "      <td>EPI2161548</td>\n",
       "      <td>METVSLITILLAATVSNADKICIGYQSSNSTETVDTLTENNVPVTH...</td>\n",
       "      <td>MNPNQKITAIGSVSLIIAIICLLMQIAILTMTMTLHFGQKECSNPS...</td>\n",
       "      <td>[0.016907786181883842, 0.016907786181883842, 0...</td>\n",
       "      <td>[0.04614861940588559, 0.04614861940588559, 0.0...</td>\n",
       "      <td>[0.027933331156592276, 0.027933331156592276, 0...</td>\n",
       "      <td>6.271620</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>6.318778</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   id subtype ha_accession na_accession  \\\n",
       "218   A/swine/Missouri/A02524711/2020    H1N1   EPI1818121   EPI1818122   \n",
       "1902   A/swine/Indiana/A02524710/2020    H3N2   EPI1818137   EPI1818138   \n",
       "5814   A/Camel/Inner_Mongolia/XL/2020    H7N9   EPI2026200   EPI2026202   \n",
       "6049   A/mink/China/chick_embryo/2020    H9N2   EPI2161544   EPI2161548   \n",
       "\n",
       "                                                     ha  \\\n",
       "218   MKAILVVMLYTFTTANADTLCIGYHANNSTDTVDTVLEKNVTVTHS...   \n",
       "1902  MKTIIALSYILCLVFAQKIPGNDNGTATLCLGHHAVPNGTIVKTIT...   \n",
       "5814  MNTQILVFALIAIIPTNADKICLGHHAVSNGTKVNTLTEREVEVVN...   \n",
       "6049  METVSLITILLAATVSNADKICIGYQSSNSTETVDTLTENNVPVTH...   \n",
       "\n",
       "                                                     na  \\\n",
       "218   MNPNQKIITIGSICMTIGTANLILQIGNIISIWVSHSIQIGNQSQI...   \n",
       "1902  MNPNQKIITIGSVSLTISTICFFMQIAILITTIALHFKQYEFNSPP...   \n",
       "5814  MNPNQKILCTSATAITIGAIAVLIGIANLGLNIGLHLKPGCNCSHS...   \n",
       "6049  MNPNQKITAIGSVSLIIAIICLLMQIAILTMTMTLHFGQKECSNPS...   \n",
       "\n",
       "                                                ha_risk  \\\n",
       "218   [0.007702821053332326, 0.00794167832708008, 0....   \n",
       "1902  [0.008423100311281287, 0.0044835948774065975, ...   \n",
       "5814  [0.025061424930009856, 0.01711340116551936, 0....   \n",
       "6049  [0.016907786181883842, 0.016907786181883842, 0...   \n",
       "\n",
       "                                                na_risk  \\\n",
       "218   [0.0016000691389129699, 0.0019043683598783635,...   \n",
       "1902  [0.006596476585994605, 0.006619358238135353, 0...   \n",
       "5814  [0.0011227960579890436, 0.0012230115569459082,...   \n",
       "6049  [0.04614861940588559, 0.04614861940588559, 0.0...   \n",
       "\n",
       "                                    geometric_mean_risk  emergence_risk  \\\n",
       "218   [0.0035107045233124007, 0.003888943420830784, ...        6.782248   \n",
       "1902  [0.00745404480700585, 0.00544779961894918, 0.0...        6.729258   \n",
       "5814  [0.0053046082908169015, 0.004574919387714034, ...        6.704867   \n",
       "6049  [0.027933331156592276, 0.027933331156592276, 0...        6.271620   \n",
       "\n",
       "      emergence_risk_var  impact_risk  impact_risk_var  \n",
       "218             0.000006     6.767254         0.000005  \n",
       "1902            0.000613     6.720489         0.000476  \n",
       "5814            0.004832     6.699016         0.003755  \n",
       "6049            0.000000     6.318778         0.000000  "
      ]
     },
     "execution_count": 34,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "count=0\n",
    "Subtype={'H1N1':0,'H3N2':0,'H7N9':0,'H9N2':0}\n",
    "MaxriskStrain={'H1N1':None,'H3N2':None,'H7N9':None,'H9N2':None}\n",
    "Subtype_strat={ 6.0:{'H1N1':0,'H3N2':0,'H7N9':0,'H9N2':0},  6.2:{'H1N1':0,'H3N2':0,'H7N9':0,'H9N2':0},              6.3:{'H1N1':0,'H3N2':0,'H7N9':0,'H9N2':0},\n",
    "               6.4:{'H1N1':0,'H3N2':0,'H7N9':0,'H9N2':0},6.5:{'H1N1':0,'H3N2':0,'H7N9':0,'H9N2':0}}\n",
    "for node in ltree_collapsed:\n",
    "    Subtype[node.subtype]=Subtype[node.subtype]+1\n",
    "    if MaxriskStrain[node.subtype] is None:\n",
    "        MaxriskStrain[node.subtype]=(ALLinfoHA.seq_infos[node.name.split()[0]].name,\n",
    "                                     ALLinfoHA.seq_infos[node.name.split()[0]].erisk)\n",
    "    else:\n",
    "        if MaxriskStrain[node.subtype][1]<ALLinfoHA.seq_infos[node.name.split()[0]].erisk:\n",
    "            MaxriskStrain[node.subtype]=(ALLinfoHA.seq_infos[node.name.split()[0]].name,\n",
    "                                         ALLinfoHA.seq_infos[node.name.split()[0]].erisk)\n",
    "    for r in [6.0, 6.2,6.3,6.4,6.5]:\n",
    "        if r<ALLinfoHA.seq_infos[node.name.split()[0]].erisk:\n",
    "            #print(node.subtype,ALLinfoHA.seq_infos[node.name.split()[0]].erisk)\n",
    "            Subtype_strat[r][node.subtype]=Subtype_strat[r][node.subtype]+1\n",
    "        \n",
    "    count=count+1\n",
    "maxriskdf = pd.DataFrame(MaxriskStrain)\n",
    "Subtype_strat_df = pd.DataFrame(Subtype_strat)\n",
    "display(maxriskdf)\n",
    "display(Subtype_strat_df)\n",
    "print(count)\n",
    "df[df.id.isin(maxriskdf.iloc[0,:].values)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 35,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\\begin{tabular}{L{.75in}|L{.25in}|L{.60in}|L{.6in}|L{.6in}|L{.6in}}\\hline\r\n",
      "subtype&6.0&6.2&6.3&6.4&6.5\\\\\r\n",
      "H1N1&61&57&54&11&4\\\\\r\n",
      "H3N2&11&11&11&6&3\\\\\r\n",
      "H7N9&1&1&1&1&1\\\\\r\n",
      "H9N2&1&1&0&0&0\\\\\r\n",
      "\\hline\\end{tabular}\r\n"
     ]
    }
   ],
   "source": [
    "from zedstat.textable import textable\n",
    "Subtype_strat_df.columns=[str(x) for x in Subtype_strat_df.columns]\n",
    "Subtype_strat_df.index.name='subtype'\n",
    "Subtype_strat_df=Subtype_strat_df.reset_index()\n",
    "textable(Subtype_strat_df,#pd.concat([highrisknamesdf1.head(30),highrisknamesdf1.tail(10)]),\n",
    "         tabname='riskycount.tex',\n",
    "         FORMAT='%1.4f',INDEX=False,\n",
    "         TABFORMAT='L{.75in}|L{.25in}|L{.60in}|L{.6in}|L{.6in}|L{.6in}',LNTERM='\\\\\\\\\\n')\n",
    "! cat riskycount.tex"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 36,
   "metadata": {},
   "outputs": [],
   "source": [
    "def write_fasta(seqs, fasta_file, wrap=80):\n",
    "    \"\"\"Write sequences to a fasta file.\n",
    "\n",
    "    Parameters\n",
    "    ----------\n",
    "    seqs : dict[seq_id] -> seq\n",
    "        Sequences indexed by sequence id.\n",
    "    fasta_file : str\n",
    "        Path to write the sequences to.\n",
    "    wrap: int\n",
    "        Number of AA/NT before the line is wrapped.\n",
    "    \"\"\"\n",
    "    with open(fasta_file, 'w') as f:\n",
    "        for gid, gseq in seqs.items():\n",
    "            f.write('>{}\\n'.format(gid))\n",
    "            for i in range(0, len(gseq), wrap):\n",
    "                f.write('{}\\n'.format(gseq[i:i + wrap])) \n",
    "\n",
    "#for i,name in zip(df[df.id.isin(maxriskdf.iloc[0,:].values)].ha.values,\n",
    "#                 df[df.id.isin(maxriskdf.iloc[0,:].values)].ha_accession.values):\n",
    "#    write_fasta({name:i},name+'.fasta')\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [],
   "source": [
    "id_list=['A/swine/North_Carolina/A02479173/2020','A/swine/Tennessee/A02524414/2022']\n",
    "RFILES={name+'('+subtype+')':i for i,name,subtype in zip(df[df.id.isin(id_list)].ha.values,\n",
    "                 df[df.id.isin(id_list)].ha_accession.values,\n",
    "                 df[df.id.isin(id_list)].subtype.values)}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 38,
   "metadata": {},
   "outputs": [],
   "source": [
    "FASTANAME='risky4.fasta'\n",
    "FASTAFILES={name+'('+subtype+')':i for i,name,subtype in zip(df[df.id.isin(maxriskdf.iloc[0,:].values)].ha.values,\n",
    "                 df[df.id.isin(maxriskdf.iloc[0,:].values)].ha_accession.values,\n",
    "                 df[df.id.isin(maxriskdf.iloc[0,:].values)].subtype.values)}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 43,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>sequence</th>\n",
       "      <th>subtype</th>\n",
       "      <th>protein</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>strain</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>north_h1n1_ha</th>\n",
       "      <td>A/Ireland/20935/2022</td>\n",
       "      <td>MKAILVVLLYTFTTANADTLCIGYHANNSTDTVDTVLEKNVTVTHS...</td>\n",
       "      <td>H1N1</td>\n",
       "      <td>HA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>north_h1n1_na</th>\n",
       "      <td>A/Cote_d'Ivoire/3729/2021</td>\n",
       "      <td>MNPNQKIITIGSICMAIGTANLILQIGNIISIWVSHSIQIGNQSQI...</td>\n",
       "      <td>H1N1</td>\n",
       "      <td>NA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>north_h3n2_ha</th>\n",
       "      <td>A/Human/New_York/PV60641/2022</td>\n",
       "      <td>MKTIIALSNILCLVFAQKIPGNDNSTATLCLGHHAVPNGTIVKTIT...</td>\n",
       "      <td>H3N2</td>\n",
       "      <td>HA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>north_h3n2_na</th>\n",
       "      <td>A/Stockholm/10/2022</td>\n",
       "      <td>MNPNQKIITIGSVSLTISTICFFMQIAILITTVTLHFKQYEFNSPP...</td>\n",
       "      <td>H3N2</td>\n",
       "      <td>NA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>south_h1n1_ha</th>\n",
       "      <td>A/Abidjan/457/2021</td>\n",
       "      <td>MKAILVVLLYTFTTANADTLCIGYHANNSTDTVDTVLEKNVTVTHS...</td>\n",
       "      <td>H1N1</td>\n",
       "      <td>HA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>south_h1n1_na</th>\n",
       "      <td>A/Cote_D'Ivoire/1496/2021</td>\n",
       "      <td>MNPNQKIITIGSICMAIGTANLILQIGNIISIWVSHSIQIGNQSQI...</td>\n",
       "      <td>H1N1</td>\n",
       "      <td>NA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>south_h3n2_ha</th>\n",
       "      <td>A/Darwin/9a/2021</td>\n",
       "      <td>MKTIIALSNILCLVFAQKIPGNDNSTATLCLGHHAVPNGTIVKTIT...</td>\n",
       "      <td>H3N2</td>\n",
       "      <td>HA</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>south_h3n2_na</th>\n",
       "      <td>A/India/PUN-NIV301718/2021</td>\n",
       "      <td>MNPNQKIITIGSVSLTISTICFFMQIAILITTVTLHFKQYEFNSPP...</td>\n",
       "      <td>H3N2</td>\n",
       "      <td>NA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        name  \\\n",
       "strain                                         \n",
       "north_h1n1_ha           A/Ireland/20935/2022   \n",
       "north_h1n1_na      A/Cote_d'Ivoire/3729/2021   \n",
       "north_h3n2_ha  A/Human/New_York/PV60641/2022   \n",
       "north_h3n2_na            A/Stockholm/10/2022   \n",
       "south_h1n1_ha             A/Abidjan/457/2021   \n",
       "south_h1n1_na      A/Cote_D'Ivoire/1496/2021   \n",
       "south_h3n2_ha               A/Darwin/9a/2021   \n",
       "south_h3n2_na     A/India/PUN-NIV301718/2021   \n",
       "\n",
       "                                                        sequence subtype  \\\n",
       "strain                                                                     \n",
       "north_h1n1_ha  MKAILVVLLYTFTTANADTLCIGYHANNSTDTVDTVLEKNVTVTHS...    H1N1   \n",
       "north_h1n1_na  MNPNQKIITIGSICMAIGTANLILQIGNIISIWVSHSIQIGNQSQI...    H1N1   \n",
       "north_h3n2_ha  MKTIIALSNILCLVFAQKIPGNDNSTATLCLGHHAVPNGTIVKTIT...    H3N2   \n",
       "north_h3n2_na  MNPNQKIITIGSVSLTISTICFFMQIAILITTVTLHFKQYEFNSPP...    H3N2   \n",
       "south_h1n1_ha  MKAILVVLLYTFTTANADTLCIGYHANNSTDTVDTVLEKNVTVTHS...    H1N1   \n",
       "south_h1n1_na  MNPNQKIITIGSICMAIGTANLILQIGNIISIWVSHSIQIGNQSQI...    H1N1   \n",
       "south_h3n2_ha  MKTIIALSNILCLVFAQKIPGNDNSTATLCLGHHAVPNGTIVKTIT...    H3N2   \n",
       "south_h3n2_na  MNPNQKIITIGSVSLTISTICFFMQIAILITTVTLHFKQYEFNSPP...    H3N2   \n",
       "\n",
       "              protein  \n",
       "strain                 \n",
       "north_h1n1_ha      HA  \n",
       "north_h1n1_na      NA  \n",
       "north_h3n2_ha      HA  \n",
       "north_h3n2_na      NA  \n",
       "south_h1n1_ha      HA  \n",
       "south_h1n1_na      NA  \n",
       "south_h3n2_ha      HA  \n",
       "south_h3n2_na      NA  "
      ]
     },
     "execution_count": 43,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "human_dom_seq='../../../qnet_predictions/results/dominant_sequences_2021_2022.csv'\n",
    "domf=pd.read_csv(human_dom_seq,index_col=0)\n",
    "domf=domf.assign(subtype=[x.split('_')[1].upper() for x in domf.index.values],protein=[x.split('_')[2].upper() for x in domf.index.values])\n",
    "domf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 55,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'DOM_HUMAN(H1N1)': 'MKAILVVLLYTFTTANADTLCIGYHANNSTDTVDTVLEKNVTVTHSVNLLEDKHNGKLCKLRGVAPLHLGKCNIAGWILGNPECESLSTARSWSYIVETSNSDNGTCYPGDFINYEELREQLSSVSSFERFEIFPKTSSWPNHDSDKGVTAACPHAGAKSFYKNLIWLVKKGNSYPKLNQTYINDKGKEVLVLWGIHHPPTIAAQESLYQNADAYVFVGTSRYSKKFKPEIATRPKVRDQEGRMNYYWTLVEPGDKITFEATGNLVVPRYAFTMERDAGSGIIISDTPVHDCNTTCQTPEGAINTSLPFQNVHPITIGKCPKYVKSTKLRLATGLRNVPSIQSRGLFGAIAGFIEGGWTGMVDGWYGYHHQNEQGSGYAADLKSTQNAIDKITNKVNSVIEKMNTQFTAVGKEFNHLEKRIENLNKKVDDGFLDIWTYNAELLVLLENERTLDYHDSNVKNLYEKVRNQLKNNAKEIGNGCFEFYHKCDNTCMESVKNGTYDYPKYSEEAKLNREKIDGVKLESTRIYQILAIYSTVASSLVLVVSLGAISFWMCSNGSLQCRIC',\n",
       " 'EPI1818121(H1N1)': 'MKAILVVMLYTFTTANADTLCIGYHANNSTDTVDTVLEKNVTVTHSVNLLEDKHNGKLCKLRGVAPLHLGKCNIAGWILGNPECESLSTARSWSYIVETSNSDNGTCYPGDFINYEELREQLSSVSSFERFEIFPKTSSWPNHDSDNGVTAACPHAGAKSFYKNLIWLVKKGKSYPKINQTYINDKGKEVLVLWGIHHPPTIADQQSLYQNADAYVFVGTSRYSKKFKPEIATRPKVRDQEGRMNYYWTLVEPGDKITFEATGNLVAPRYAFTMERDAGSGIIISDTPVHDCNTTCQTPEGAINTSLPFQNVHPITIGKCPKYVKSTKLRLATGLRNVPSIQSRGLFGAIAGFIEGGWTGMVDGWYGYHHQNEQGSGYAADLKSTQNAIDKITNKVNSVIEKMNTQFTAVGKEFNHLEKRIENLNKKVDDGFLDIWTYNAELLVLLENERTLDYHDSNVKNLYEKVRNQLKNNAKEIGNGCFEFYHKCDNTCMESVKNGTYDYPKYSEEAKLNREKIDGVKLDSTRIYQILAIYSTVASSLVLVVSLGAI',\n",
       " 'EPI1818137(H3N2)': 'MKTIIALSYILCLVFAQKIPGNDNGTATLCLGHHAVPNGTIVKTITDDRIEVTNATELVQNSSIGEICDSPHQVLDGGNCTLIDALLGDPQCDGFQNKKWDLFVERSRAYSNCYPYDVPDYASLRSLVASSGTLEFKNESFNWTGVTQNGKSSSCIRGSSSSFFSRLNWLTXLNYTYPALNVIMPNKEQFDKLYIWGVHHPGTDNDQIFLYARSSGRIIVSTKRSQQAVIPNIGSRPRIRDIPSRISIYWTIVKPGDILLINSTGNLIAPRGYFKIRSGKSSIMRSDAPIGKCKSECITPNGSIPNDKPFQNVNRITYGACPRYIKQSTLKLATGMRNVPEKQTRGIFGAIAGFIENGWEGMVDGWYGFRHQNSEGRGQAADLKSTQAAIDQINGKLNRLIGKTNEKFHQIEKEFSEVEGRVQDLEKYVEDTKIDLWSYNAELLVALENQHTIDLTDSEMNKLFEKTKKQLRENAEDMGNGCFKIYHKCDNACIGSIRNETYDHNVYRDEALNNRFQIKGVELKSGYKDWILWISFAISCFLLCVALLGF',\n",
       " 'EPI2026200(H7N9)': 'MNTQILVFALIAIIPTNADKICLGHHAVSNGTKVNTLTEREVEVVNATETVERTNIPRICSKGKRTVDLGQCGLLGTITGPPQCDQFLEFSADLIIERREGSDVCYPGKFVNEEALRQILRESGGIDKEAMGFTYNGIRTNGVTSACRRSGSSFYAEMKWLLSNTDNAAFPQMTKSYKNTRKSPAIIVWGIHHSVSTAEQTKLYGSGNKLVTVGSSNYQQSFVPSPGARPQVNGISGRIDFHWLMLNPNDTVTFSFNGAFIAPDRASFLRGKSMGIQSGVQVDANCEGDCYHSGGTIISNLPFQNIDSRAVGKCPRYVKQRSLLLATGMKNVPEIPKGRGLFGAIAGFIENGWEGLIDGWYGFRHQNAQGEGTAADYKSTQSAIDQITGKLNRLIAKTNQQFELIDNEFNEVEKQIGNVINWTRDSITEVWSYNAELLVAMENQHTIDLADSEMDKLYERVKRQLRENAEEDGTGCFEIFHKCDDDCMASIRNNTYDHRKYREEAMQSRIQIDPVKLSSGYKDVILWFSFGASCFILLAIVMGLVFICVK',\n",
       " 'EPI2161544(H9N2)': 'METVSLITILLAATVSNADKICIGYQSSNSTETVDTLTENNVPVTHAKELLHTEHNGMLCATGLGQPLVLDTCTIEGLIYGNPSCDLSLEGREWSYIVERPSAVHGLCYPGNVEDLEELRSLFSSARSYQRIQIFPDTIWNVSYDGTSTACSGSFYKSMRWLTRKNGEYPTQDAQYTNNQGKNILFMWGINHPPTDDTQRGLYTRTDTTTSVATEEINRIFKPLIGPRPLVNGLMGRINYYWSVLKPGQTLRIKSDGNLIAPWYGHILSGESHGRILKTDLKRGNCTVQCQTEKGGLNTTLPFQNVSKYAFGNCSKYIGVKSLKLAVGLRNVPSRSSRGLFGAIAGFIEGGWPGLVAGWYGFQHSNDQGVGMAADRDSTQKAIDKITSKVNNIVDKMNKQYEIIDHEFSEVETRLNMINNKIDDQIQDIWAYNAELLVLLENQKTLDEHDANVNNLYNKVKRALGSNAAEDGKGCFELYHKCDDQCMETIRNGTYNRRKYQEESKLERQRIEGVKLESEGTYKILTIYSTVASSLVIAMGFAAFLFWAMS'}"
      ]
     },
     "execution_count": 55,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "human_dom_seq='../../../../paper_data/enet_predictions/results/dominant_sequences/north_h1n1_ha/north_h1n1_ha_21_22.csv'\n",
    "domf=pd.read_csv(human_dom_seq,index_col=0)\n",
    "domf=domf.assign(subtype='H1N1',protein='HA')\n",
    "domf=domf[domf.protein=='HA'].head(2)\n",
    "DOMFILES={'DOM_HUMAN'+'('+subtype+')':i for i,name,subtype in zip(domf.sequence.values,\n",
    "                 domf.id.values,\n",
    "                 domf.subtype.values)}\n",
    "DOMFILES.update(FASTAFILES)\n",
    "DOMFILES"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "human_dom_seq='../../../../paper_data/enet_predictions/results/dominant_sequences/north_h1n1_ha/north_h1n1_ha_21_22.csv'\n",
    "#human_dom_seq='../../../qnet_predictions/results/dominant_sequences_2021_2022.csv'\n",
    "domf=pd.read_csv(human_dom_seq,index_col=0)\n",
    "domf=domf.assign(subtype=[x.split('_')[1].upper() for x in domf.index.values],protein=[x.split('_')[2].upper() for x in domf.index.values])\n",
    "domf=domf[domf.protein=='HA'].head(2)\n",
    "DOMFILES={'DOM_HUMAN'+'('+subtype+')':i for i,name,subtype in zip(domf.sequence.values,\n",
    "                 domf.name.values,\n",
    "                 domf.subtype.values)}\n",
    "DOMFILES.update(FASTAFILES)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 56,
   "metadata": {},
   "outputs": [],
   "source": [
    "DOMFILES.update(RFILES)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 57,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'DOM_HUMAN(H1N1)': 'MKAILVVLLYTFTTANADTLCIGYHANNSTDTVDTVLEKNVTVTHSVNLLEDKHNGKLCKLRGVAPLHLGKCNIAGWILGNPECESLSTARSWSYIVETSNSDNGTCYPGDFINYEELREQLSSVSSFERFEIFPKTSSWPNHDSDKGVTAACPHAGAKSFYKNLIWLVKKGNSYPKLNQTYINDKGKEVLVLWGIHHPPTIAAQESLYQNADAYVFVGTSRYSKKFKPEIATRPKVRDQEGRMNYYWTLVEPGDKITFEATGNLVVPRYAFTMERDAGSGIIISDTPVHDCNTTCQTPEGAINTSLPFQNVHPITIGKCPKYVKSTKLRLATGLRNVPSIQSRGLFGAIAGFIEGGWTGMVDGWYGYHHQNEQGSGYAADLKSTQNAIDKITNKVNSVIEKMNTQFTAVGKEFNHLEKRIENLNKKVDDGFLDIWTYNAELLVLLENERTLDYHDSNVKNLYEKVRNQLKNNAKEIGNGCFEFYHKCDNTCMESVKNGTYDYPKYSEEAKLNREKIDGVKLESTRIYQILAIYSTVASSLVLVVSLGAISFWMCSNGSLQCRIC',\n",
       " 'EPI1818121(H1N1)': 'MKAILVVMLYTFTTANADTLCIGYHANNSTDTVDTVLEKNVTVTHSVNLLEDKHNGKLCKLRGVAPLHLGKCNIAGWILGNPECESLSTARSWSYIVETSNSDNGTCYPGDFINYEELREQLSSVSSFERFEIFPKTSSWPNHDSDNGVTAACPHAGAKSFYKNLIWLVKKGKSYPKINQTYINDKGKEVLVLWGIHHPPTIADQQSLYQNADAYVFVGTSRYSKKFKPEIATRPKVRDQEGRMNYYWTLVEPGDKITFEATGNLVAPRYAFTMERDAGSGIIISDTPVHDCNTTCQTPEGAINTSLPFQNVHPITIGKCPKYVKSTKLRLATGLRNVPSIQSRGLFGAIAGFIEGGWTGMVDGWYGYHHQNEQGSGYAADLKSTQNAIDKITNKVNSVIEKMNTQFTAVGKEFNHLEKRIENLNKKVDDGFLDIWTYNAELLVLLENERTLDYHDSNVKNLYEKVRNQLKNNAKEIGNGCFEFYHKCDNTCMESVKNGTYDYPKYSEEAKLNREKIDGVKLDSTRIYQILAIYSTVASSLVLVVSLGAI',\n",
       " 'EPI1818137(H3N2)': 'MKTIIALSYILCLVFAQKIPGNDNGTATLCLGHHAVPNGTIVKTITDDRIEVTNATELVQNSSIGEICDSPHQVLDGGNCTLIDALLGDPQCDGFQNKKWDLFVERSRAYSNCYPYDVPDYASLRSLVASSGTLEFKNESFNWTGVTQNGKSSSCIRGSSSSFFSRLNWLTXLNYTYPALNVIMPNKEQFDKLYIWGVHHPGTDNDQIFLYARSSGRIIVSTKRSQQAVIPNIGSRPRIRDIPSRISIYWTIVKPGDILLINSTGNLIAPRGYFKIRSGKSSIMRSDAPIGKCKSECITPNGSIPNDKPFQNVNRITYGACPRYIKQSTLKLATGMRNVPEKQTRGIFGAIAGFIENGWEGMVDGWYGFRHQNSEGRGQAADLKSTQAAIDQINGKLNRLIGKTNEKFHQIEKEFSEVEGRVQDLEKYVEDTKIDLWSYNAELLVALENQHTIDLTDSEMNKLFEKTKKQLRENAEDMGNGCFKIYHKCDNACIGSIRNETYDHNVYRDEALNNRFQIKGVELKSGYKDWILWISFAISCFLLCVALLGF',\n",
       " 'EPI2026200(H7N9)': 'MNTQILVFALIAIIPTNADKICLGHHAVSNGTKVNTLTEREVEVVNATETVERTNIPRICSKGKRTVDLGQCGLLGTITGPPQCDQFLEFSADLIIERREGSDVCYPGKFVNEEALRQILRESGGIDKEAMGFTYNGIRTNGVTSACRRSGSSFYAEMKWLLSNTDNAAFPQMTKSYKNTRKSPAIIVWGIHHSVSTAEQTKLYGSGNKLVTVGSSNYQQSFVPSPGARPQVNGISGRIDFHWLMLNPNDTVTFSFNGAFIAPDRASFLRGKSMGIQSGVQVDANCEGDCYHSGGTIISNLPFQNIDSRAVGKCPRYVKQRSLLLATGMKNVPEIPKGRGLFGAIAGFIENGWEGLIDGWYGFRHQNAQGEGTAADYKSTQSAIDQITGKLNRLIAKTNQQFELIDNEFNEVEKQIGNVINWTRDSITEVWSYNAELLVAMENQHTIDLADSEMDKLYERVKRQLRENAEEDGTGCFEIFHKCDDDCMASIRNNTYDHRKYREEAMQSRIQIDPVKLSSGYKDVILWFSFGASCFILLAIVMGLVFICVK',\n",
       " 'EPI2161544(H9N2)': 'METVSLITILLAATVSNADKICIGYQSSNSTETVDTLTENNVPVTHAKELLHTEHNGMLCATGLGQPLVLDTCTIEGLIYGNPSCDLSLEGREWSYIVERPSAVHGLCYPGNVEDLEELRSLFSSARSYQRIQIFPDTIWNVSYDGTSTACSGSFYKSMRWLTRKNGEYPTQDAQYTNNQGKNILFMWGINHPPTDDTQRGLYTRTDTTTSVATEEINRIFKPLIGPRPLVNGLMGRINYYWSVLKPGQTLRIKSDGNLIAPWYGHILSGESHGRILKTDLKRGNCTVQCQTEKGGLNTTLPFQNVSKYAFGNCSKYIGVKSLKLAVGLRNVPSRSSRGLFGAIAGFIEGGWPGLVAGWYGFQHSNDQGVGMAADRDSTQKAIDKITSKVNNIVDKMNKQYEIIDHEFSEVETRLNMINNKIDDQIQDIWAYNAELLVLLENQKTLDEHDANVNNLYNKVKRALGSNAAEDGKGCFELYHKCDDQCMETIRNGTYNRRKYQEESKLERQRIEGVKLESEGTYKILTIYSTVASSLVIAMGFAAFLFWAMS',\n",
       " 'EPI1780425(H1N1)': 'MKAILVVLLYAFTTANADTLCIGYHANNSTDTVDTVLEKNVTVTHSVNLLEEKHNGKLCKLRGVAPLHLGKCNIAGWILGNPECESLFTASSWSYIVETSNSDNGTCYPGDFINYEELREQLSSVSSFERFEIFPKASSWPNYETSRGVTAACPYAGANSFYRNLIWLVKKGNSYPKLSKSYINNKEKEVLVLWGIHHPSTSTDQQSLYQNADAYVFVGSSKYSKKFKPEIATRPKVRNQAGRMNYYWTLVEPGDTITFEATGNLVVPRYAFAMKRGSGSGIIISDTSVHDCNTNCQTPKGAINTSLPFQNIHPVTIGECPKYVKSKKLRMATGLRNIPSIQSRGLFGAIAGFIEGGWTGMIDGWYGYHHQNEQGSGYAADRKSTQSAIDGITNKVNSIIEKMNSQFTAVGKEFNHLEKRIENLNRKVDDGFLDVWTYNAELLVLLENERTLDYHDSNVKNLYEKVRSQLKNNAKEIGNGCFEFYHKCDDTCMESVKNGTYDYPKYSEEAKLNREEIDGVKLESTEVYQILAIYSTVASSLVLLVSLGAI',\n",
       " 'EPI2149257(H1N1)': 'MKAILLVLLHTLATANADTICIGYHANNSTDTVDTVLEKNVTVTHSVNLLEDKHNGKLCKLGGKSPLYLGKCNIAGWLLGNPECESILTVSSWSYIVETSNSDNGTCYPGSFTNYEELREQLSSVSSFEKFEIFPIESSWPNYKTGETASCPYAGKSSFYRNLMWLVKKADSYPRLNISYVNNKGKEVLVLWGIHHPPTGNDQQWLYQNANASVFVGTSTYSQKFKPEIATRPKVRGQAGRMNYYWTLVEPGDTITFEATGNLVVPRYAFAMNRGSGSGIIVSDAPIHDCNTPCQTPKGALNTSLPFQNVHPVTIGECPKYVKSRRLKMATGLRNIPSIQSRGLFGAIAGFIEGGWTGMVDGWYGYHHQNDQGSGYAADQKSTQRAIDGITNKVNSVIEKMNTQFTAVGKEFNNLERRIENLNRKVDDGFLDVWTYNAELLVLLENERTLDFHDSNVKNLYEKVRNQLRNNAKEIGNGCFEFYHKCDNTCMESVKNGTYNYLKYSEESKLNREEIDGVKLDSTRVYQILAIYSTVASSLVLLVSLGALSF'}"
      ]
     },
     "execution_count": 57,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "DOMFILES"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 58,
   "metadata": {},
   "outputs": [],
   "source": [
    "write_fasta(DOMFILES,FASTANAME)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 49,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import SeqIO\n",
    "from Bio.Blast import NCBIWWW\n",
    "my_query = SeqIO.read(\"EPI1818121.fasta\", format=\"fasta\")\n",
    "result_handle = NCBIWWW.qblast(\"blastp\", \"nr\", my_query.seq)\n",
    "blast_result = open(\"my_blast.xml\", \"w\")\n",
    "blast_result.write(result_handle.read())\n",
    "blast_result.close()\n",
    "result_handle.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 50,
   "metadata": {},
   "outputs": [],
   "source": [
    "def parse_blast(resultfile): #takes in the BLAST result, outputs list that can be made into csv\n",
    "    from Bio.Blast import NCBIXML\n",
    "    result_handle = open(resultfile)\n",
    "    blast_records = NCBIXML.parse(result_handle)\n",
    "    csv_list = []\n",
    "    \n",
    "    header = [  'accession','Query',\n",
    "                'Name', 'Length', 'Score', 'Expect',\n",
    "                'QueryStart', 'QueryEnd',\n",
    "                'SubjectStart', 'SubjectEnd','pct'\n",
    "            ]\n",
    "    \n",
    "    #csv_list.append(header)\n",
    "    count = 0\n",
    "    for blast_record in blast_records:\n",
    "        '''help(blast_record.alignments[0].hsps[0])''' # these give help info for the parts \n",
    "        '''help(blast_record.alignments[0])        '''\n",
    "        count +=1\n",
    "        \n",
    "        query = blast_record.query\n",
    "        for alignment in blast_record.alignments:\n",
    "\n",
    "            name = alignment.title\n",
    "            length = alignment.length\n",
    "    \n",
    "            hsp = alignment.hsps[0] # I don't know if we will ever have more than one, so might as well take the first one.\n",
    "            score = hsp.score\n",
    "            expect = hsp.expect\n",
    "            querystart = hsp.query_start\n",
    "            queryend = hsp.query_end\n",
    "            subjectstart = hsp.sbjct_start\n",
    "            subjectend = hsp.sbjct_end\n",
    "            pct=hsp.positives/hsp.align_length\n",
    "            accession=alignment.accession\n",
    "            row = [accession,query,name,length,score,expect,querystart,queryend,subjectstart,subjectend,pct]\n",
    "            csv_list.append(row)\n",
    "            \n",
    "    result_handle.close()\n",
    "    return pd.DataFrame(csv_list,columns=header)\n",
    " "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 51,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>accession</th>\n",
       "      <th>Query</th>\n",
       "      <th>Name</th>\n",
       "      <th>Length</th>\n",
       "      <th>Score</th>\n",
       "      <th>Expect</th>\n",
       "      <th>QueryStart</th>\n",
       "      <th>QueryEnd</th>\n",
       "      <th>SubjectStart</th>\n",
       "      <th>SubjectEnd</th>\n",
       "      <th>pct</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>QIA58660</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QIA58660.1| hemagglutinin [Influenza A viru...</td>\n",
       "      <td>566</td>\n",
       "      <td>2969.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>WHU33766</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|WHU33766.1| hemagglutinin [Influenza A viru...</td>\n",
       "      <td>566</td>\n",
       "      <td>2967.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>QKG28282</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QKG28282.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2967.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>QIZ32536</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QIZ32536.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2967.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>QJI52882</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QJI52882.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2967.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>QJT24758</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QJT24758.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2967.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>QKV49724</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QKV49724.1| hemagglutinin [Influenza A viru...</td>\n",
       "      <td>566</td>\n",
       "      <td>2967.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>QIQ45861</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QIQ45861.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2967.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>QKM75781</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QKM75781.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2966.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>QJX57625</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QJX57625.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2966.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>WBG95744</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|WBG95744.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2966.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>QIQ45830</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QIQ45830.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2966.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>QOY44156</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QOY44156.1| hemagglutinin [Influenza A viru...</td>\n",
       "      <td>566</td>\n",
       "      <td>2966.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>QIQ45216</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QIQ45216.1| hemagglutinin [Influenza A viru...</td>\n",
       "      <td>566</td>\n",
       "      <td>2965.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>QMX87502</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QMX87502.1| hemagglutinin [Influenza A viru...</td>\n",
       "      <td>566</td>\n",
       "      <td>2965.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>QJD23117</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QJD23117.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2965.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>QIQ45677</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QIQ45677.1| hemagglutinin [Influenza A viru...</td>\n",
       "      <td>566</td>\n",
       "      <td>2965.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>QJT24395</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QJT24395.1| hemagglutinin [Influenza A viru...</td>\n",
       "      <td>566</td>\n",
       "      <td>2965.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>WBG95731</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|WBG95731.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2965.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>QIQ45346</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QIQ45346.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2965.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>QIZ32395</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QIZ32395.1| hemagglutinin [Influenza A viru...</td>\n",
       "      <td>566</td>\n",
       "      <td>2964.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>QIZ32042</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QIZ32042.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2964.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>QIZ33251</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QIZ33251.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2964.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>0.998182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>QIQ44996</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QIQ44996.1| hemagglutinin [Influenza A viru...</td>\n",
       "      <td>566</td>\n",
       "      <td>2964.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>0.998182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>QTJ06578</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QTJ06578.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2964.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>0.998182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>WBG95844</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|WBG95844.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2964.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>QJT24499</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QJT24499.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2964.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>QIV14744</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QIV14744.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2964.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>QIQ45654</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QIQ45654.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2964.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>QIA57747</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QIA57747.1| hemagglutinin [Influenza A viru...</td>\n",
       "      <td>566</td>\n",
       "      <td>2964.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>0.998182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>QIZ31801</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QIZ31801.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2964.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>0.998182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>QKM75836</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QKM75836.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2964.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>QKS74061</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QKS74061.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2964.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>QIH95333</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QIH95333.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2963.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>QIZ32602</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QIZ32602.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2963.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>0.998182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>QJT24351</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QJT24351.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2963.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>0.998182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>QLV95699</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QLV95699.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2963.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>QKM76085</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QKM76085.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2963.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>0.998182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>QJI52636</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QJI52636.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2963.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>QIQ45093</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QIQ45093.1| hemagglutinin [Influenza A viru...</td>\n",
       "      <td>566</td>\n",
       "      <td>2963.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>0.998182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>QJT24428</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QJT24428.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2963.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>0.998182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>QIA58044</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QIA58044.1| hemagglutinin [Influenza A viru...</td>\n",
       "      <td>566</td>\n",
       "      <td>2963.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>0.998182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>QIQ45082</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QIQ45082.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2963.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>0.998182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>QJT24466</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QJT24466.1| hemagglutinin [Influenza A viru...</td>\n",
       "      <td>566</td>\n",
       "      <td>2963.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>QIZ31721</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QIZ31721.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2962.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>QTK18161</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QTK18161.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2962.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>0.998182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>QIV14634</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QIV14634.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2962.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>0.998182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>QJI52715</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QJI52715.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2962.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>0.998182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>QTK18105</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QTK18105.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2962.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>0.998182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>QIZ32778</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QIZ32778.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2962.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>0.998182</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   accession                    Query  \\\n",
       "0   QIA58660  unnamed protein product   \n",
       "1   WHU33766  unnamed protein product   \n",
       "2   QKG28282  unnamed protein product   \n",
       "3   QIZ32536  unnamed protein product   \n",
       "4   QJI52882  unnamed protein product   \n",
       "5   QJT24758  unnamed protein product   \n",
       "6   QKV49724  unnamed protein product   \n",
       "7   QIQ45861  unnamed protein product   \n",
       "8   QKM75781  unnamed protein product   \n",
       "9   QJX57625  unnamed protein product   \n",
       "10  WBG95744  unnamed protein product   \n",
       "11  QIQ45830  unnamed protein product   \n",
       "12  QOY44156  unnamed protein product   \n",
       "13  QIQ45216  unnamed protein product   \n",
       "14  QMX87502  unnamed protein product   \n",
       "15  QJD23117  unnamed protein product   \n",
       "16  QIQ45677  unnamed protein product   \n",
       "17  QJT24395  unnamed protein product   \n",
       "18  WBG95731  unnamed protein product   \n",
       "19  QIQ45346  unnamed protein product   \n",
       "20  QIZ32395  unnamed protein product   \n",
       "21  QIZ32042  unnamed protein product   \n",
       "22  QIZ33251  unnamed protein product   \n",
       "23  QIQ44996  unnamed protein product   \n",
       "24  QTJ06578  unnamed protein product   \n",
       "25  WBG95844  unnamed protein product   \n",
       "26  QJT24499  unnamed protein product   \n",
       "27  QIV14744  unnamed protein product   \n",
       "28  QIQ45654  unnamed protein product   \n",
       "29  QIA57747  unnamed protein product   \n",
       "30  QIZ31801  unnamed protein product   \n",
       "31  QKM75836  unnamed protein product   \n",
       "32  QKS74061  unnamed protein product   \n",
       "33  QIH95333  unnamed protein product   \n",
       "34  QIZ32602  unnamed protein product   \n",
       "35  QJT24351  unnamed protein product   \n",
       "36  QLV95699  unnamed protein product   \n",
       "37  QKM76085  unnamed protein product   \n",
       "38  QJI52636  unnamed protein product   \n",
       "39  QIQ45093  unnamed protein product   \n",
       "40  QJT24428  unnamed protein product   \n",
       "41  QIA58044  unnamed protein product   \n",
       "42  QIQ45082  unnamed protein product   \n",
       "43  QJT24466  unnamed protein product   \n",
       "44  QIZ31721  unnamed protein product   \n",
       "45  QTK18161  unnamed protein product   \n",
       "46  QIV14634  unnamed protein product   \n",
       "47  QJI52715  unnamed protein product   \n",
       "48  QTK18105  unnamed protein product   \n",
       "49  QIZ32778  unnamed protein product   \n",
       "\n",
       "                                                 Name  Length   Score  Expect  \\\n",
       "0   gb|QIA58660.1| hemagglutinin [Influenza A viru...     566  2969.0     0.0   \n",
       "1   gb|WHU33766.1| hemagglutinin [Influenza A viru...     566  2967.0     0.0   \n",
       "2    gb|QKG28282.1| hemagglutinin [Influenza A virus]     566  2967.0     0.0   \n",
       "3    gb|QIZ32536.1| hemagglutinin [Influenza A virus]     566  2967.0     0.0   \n",
       "4    gb|QJI52882.1| hemagglutinin [Influenza A virus]     566  2967.0     0.0   \n",
       "5    gb|QJT24758.1| hemagglutinin [Influenza A virus]     566  2967.0     0.0   \n",
       "6   gb|QKV49724.1| hemagglutinin [Influenza A viru...     566  2967.0     0.0   \n",
       "7    gb|QIQ45861.1| hemagglutinin [Influenza A virus]     566  2967.0     0.0   \n",
       "8    gb|QKM75781.1| hemagglutinin [Influenza A virus]     566  2966.0     0.0   \n",
       "9    gb|QJX57625.1| hemagglutinin [Influenza A virus]     566  2966.0     0.0   \n",
       "10   gb|WBG95744.1| hemagglutinin [Influenza A virus]     566  2966.0     0.0   \n",
       "11   gb|QIQ45830.1| hemagglutinin [Influenza A virus]     566  2966.0     0.0   \n",
       "12  gb|QOY44156.1| hemagglutinin [Influenza A viru...     566  2966.0     0.0   \n",
       "13  gb|QIQ45216.1| hemagglutinin [Influenza A viru...     566  2965.0     0.0   \n",
       "14  gb|QMX87502.1| hemagglutinin [Influenza A viru...     566  2965.0     0.0   \n",
       "15   gb|QJD23117.1| hemagglutinin [Influenza A virus]     566  2965.0     0.0   \n",
       "16  gb|QIQ45677.1| hemagglutinin [Influenza A viru...     566  2965.0     0.0   \n",
       "17  gb|QJT24395.1| hemagglutinin [Influenza A viru...     566  2965.0     0.0   \n",
       "18   gb|WBG95731.1| hemagglutinin [Influenza A virus]     566  2965.0     0.0   \n",
       "19   gb|QIQ45346.1| hemagglutinin [Influenza A virus]     566  2965.0     0.0   \n",
       "20  gb|QIZ32395.1| hemagglutinin [Influenza A viru...     566  2964.0     0.0   \n",
       "21   gb|QIZ32042.1| hemagglutinin [Influenza A virus]     566  2964.0     0.0   \n",
       "22   gb|QIZ33251.1| hemagglutinin [Influenza A virus]     566  2964.0     0.0   \n",
       "23  gb|QIQ44996.1| hemagglutinin [Influenza A viru...     566  2964.0     0.0   \n",
       "24   gb|QTJ06578.1| hemagglutinin [Influenza A virus]     566  2964.0     0.0   \n",
       "25   gb|WBG95844.1| hemagglutinin [Influenza A virus]     566  2964.0     0.0   \n",
       "26   gb|QJT24499.1| hemagglutinin [Influenza A virus]     566  2964.0     0.0   \n",
       "27   gb|QIV14744.1| hemagglutinin [Influenza A virus]     566  2964.0     0.0   \n",
       "28   gb|QIQ45654.1| hemagglutinin [Influenza A virus]     566  2964.0     0.0   \n",
       "29  gb|QIA57747.1| hemagglutinin [Influenza A viru...     566  2964.0     0.0   \n",
       "30   gb|QIZ31801.1| hemagglutinin [Influenza A virus]     566  2964.0     0.0   \n",
       "31   gb|QKM75836.1| hemagglutinin [Influenza A virus]     566  2964.0     0.0   \n",
       "32   gb|QKS74061.1| hemagglutinin [Influenza A virus]     566  2964.0     0.0   \n",
       "33   gb|QIH95333.1| hemagglutinin [Influenza A virus]     566  2963.0     0.0   \n",
       "34   gb|QIZ32602.1| hemagglutinin [Influenza A virus]     566  2963.0     0.0   \n",
       "35   gb|QJT24351.1| hemagglutinin [Influenza A virus]     566  2963.0     0.0   \n",
       "36   gb|QLV95699.1| hemagglutinin [Influenza A virus]     566  2963.0     0.0   \n",
       "37   gb|QKM76085.1| hemagglutinin [Influenza A virus]     566  2963.0     0.0   \n",
       "38   gb|QJI52636.1| hemagglutinin [Influenza A virus]     566  2963.0     0.0   \n",
       "39  gb|QIQ45093.1| hemagglutinin [Influenza A viru...     566  2963.0     0.0   \n",
       "40   gb|QJT24428.1| hemagglutinin [Influenza A virus]     566  2963.0     0.0   \n",
       "41  gb|QIA58044.1| hemagglutinin [Influenza A viru...     566  2963.0     0.0   \n",
       "42   gb|QIQ45082.1| hemagglutinin [Influenza A virus]     566  2963.0     0.0   \n",
       "43  gb|QJT24466.1| hemagglutinin [Influenza A viru...     566  2963.0     0.0   \n",
       "44   gb|QIZ31721.1| hemagglutinin [Influenza A virus]     566  2962.0     0.0   \n",
       "45   gb|QTK18161.1| hemagglutinin [Influenza A virus]     566  2962.0     0.0   \n",
       "46   gb|QIV14634.1| hemagglutinin [Influenza A virus]     566  2962.0     0.0   \n",
       "47   gb|QJI52715.1| hemagglutinin [Influenza A virus]     566  2962.0     0.0   \n",
       "48   gb|QTK18105.1| hemagglutinin [Influenza A virus]     566  2962.0     0.0   \n",
       "49   gb|QIZ32778.1| hemagglutinin [Influenza A virus]     566  2962.0     0.0   \n",
       "\n",
       "    QueryStart  QueryEnd  SubjectStart  SubjectEnd       pct  \n",
       "0            1       550             1         550  1.000000  \n",
       "1            1       550             1         550  1.000000  \n",
       "2            1       550             1         550  1.000000  \n",
       "3            1       550             1         550  1.000000  \n",
       "4            1       550             1         550  1.000000  \n",
       "5            1       550             1         550  1.000000  \n",
       "6            1       550             1         550  1.000000  \n",
       "7            1       550             1         550  1.000000  \n",
       "8            1       550             1         550  1.000000  \n",
       "9            1       550             1         550  1.000000  \n",
       "10           1       550             1         550  1.000000  \n",
       "11           1       550             1         550  1.000000  \n",
       "12           1       550             1         550  1.000000  \n",
       "13           1       550             1         550  1.000000  \n",
       "14           1       550             1         550  1.000000  \n",
       "15           1       550             1         550  1.000000  \n",
       "16           1       550             1         550  1.000000  \n",
       "17           1       550             1         550  1.000000  \n",
       "18           1       550             1         550  1.000000  \n",
       "19           1       550             1         550  1.000000  \n",
       "20           1       550             1         550  1.000000  \n",
       "21           1       550             1         550  1.000000  \n",
       "22           1       550             1         550  0.998182  \n",
       "23           1       550             1         550  0.998182  \n",
       "24           1       550             1         550  0.998182  \n",
       "25           1       550             1         550  1.000000  \n",
       "26           1       550             1         550  1.000000  \n",
       "27           1       550             1         550  1.000000  \n",
       "28           1       550             1         550  1.000000  \n",
       "29           1       550             1         550  0.998182  \n",
       "30           1       550             1         550  0.998182  \n",
       "31           1       550             1         550  1.000000  \n",
       "32           1       550             1         550  1.000000  \n",
       "33           1       550             1         550  1.000000  \n",
       "34           1       550             1         550  0.998182  \n",
       "35           1       550             1         550  0.998182  \n",
       "36           1       550             1         550  1.000000  \n",
       "37           1       550             1         550  0.998182  \n",
       "38           1       550             1         550  1.000000  \n",
       "39           1       550             1         550  0.998182  \n",
       "40           1       550             1         550  0.998182  \n",
       "41           1       550             1         550  0.998182  \n",
       "42           1       550             1         550  0.998182  \n",
       "43           1       550             1         550  1.000000  \n",
       "44           1       550             1         550  1.000000  \n",
       "45           1       550             1         550  0.998182  \n",
       "46           1       550             1         550  0.998182  \n",
       "47           1       550             1         550  0.998182  \n",
       "48           1       550             1         550  0.998182  \n",
       "49           1       550             1         550  0.998182  "
      ]
     },
     "execution_count": 51,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "csv_list=parse_blast('my_blast.xml')\n",
    "csv_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 52,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['1841676337']"
      ]
     },
     "execution_count": 52,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from Bio import Entrez\n",
    "Entrez.email = \"ishanu@uchicago.edu\"     # Always tell NCBI who you are\n",
    "handle = Entrez.esearch(db=\"protein\", term=\"QJT24340\")\n",
    "record = Entrez.read(handle)\n",
    "id=record[\"IdList\"]\n",
    "id"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 53,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'LOCUS       QJD23117                 566 aa            linear   VRL 28-APR-2020\\nDEFINITION  hemagglutinin [Influenza A virus].\\nACCESSION   QJD23117\\nVERSION     QJD23117.1\\nDBSOURCE    accession MT372532.1\\nKEYWORDS    .\\nSOURCE      Influenza A virus\\n  ORGANISM  Influenza A virus\\n            Viruses; Riboviria; Orthornavirae; Negarnaviricota;\\n            Polyploviricotina; Insthoviricetes; Articulavirales;\\n            Orthomyxoviridae; Alphainfluenzavirus; Alphainfluenzavirus\\n            influenzae.\\nREFERENCE   1  (residues 1 to 566)\\n  AUTHORS   Swine Surveillance,A.\\n  TITLE     USDA NAHLN Voluntary Swine Influenza Surveillance\\n  JOURNAL   Unpublished\\nREFERENCE   2  (residues 1 to 566)\\n  AUTHORS   Swine Surveillance,A.\\n  TITLE     Direct Submission\\n  JOURNAL   Submitted (23-APR-2020) DiVL, USDA NAHLN Voluntary Swine Influenza\\n            Surveillance, 1920 Dayton Ave., Ames, IA 50010, USA\\nCOMMENT     Method: conceptual translation.\\nFEATURES             Location/Qualifiers\\n     source          1..566\\n                     /organism=\"Influenza A virus\"\\n                     /strain=\"A/swine/Minnesota/A02245569/2020\"\\n                     /serotype=\"H1N1\"\\n                     /isolation_source=\"lung\"\\n                     /host=\"swine\"\\n                     /db_xref=\"taxon:11320\"\\n                     /segment=\"4\"\\n                     /country=\"USA: Minnesota\"\\n                     /collection_date=\"31-Mar-2020\"\\n     Protein         1..566\\n                     /product=\"hemagglutinin\"\\n                     /function=\"receptor binding and fusion protein\"\\n     sig_peptide     1..17\\n     mat_peptide     18..344\\n                     /product=\"HA1\"\\n     Region          20..520\\n                     /region_name=\"Hemagglutinin\"\\n                     /note=\"Haemagglutinin; pfam00509\"\\n                     /db_xref=\"CDD:395409\"\\n     mat_peptide     345..566\\n                     /product=\"HA2\"\\n     CDS             1..566\\n                     /gene=\"HA\"\\n                     /coded_by=\"MT372532.1:1..1701\"\\nORIGIN      \\n        1 mkailvvmly tfttanadtl cigyhannst dtvdtvlekn vtvthsvnll edkhngklck\\n       61 lrgvaplhlg kcniagwilg npeceslsta rswsyivets nsdngtcypg dfinyeelre\\n      121 qlssvssfer feifpktssw pnhdsdngvt aacphagaks fyknliwlvk kgksypkinq\\n      181 tyindkgkev lvlwgihhpp tiadqqslyq nadayvfvgt sryskkfkpe iatrpkvrdq\\n      241 egrmnyywtl vepgdkitfe atgnlvapry aftmerdags giiisdtpvh dcnttcqtpe\\n      301 gaintslpfq nvhpitigkc pkyvkstklr latglrnvps iqsrglfgai agfieggwtg\\n      361 mvdgwygyhh qneqgsgyaa dlkstqnaid kitnkvnsvi ekmntqftav gkefnhlekr\\n      421 ienlnkkvdd gfldiwtyna ellvllener tldyhdsnvk nlyekvrnql knnakeigng\\n      481 cfefyhkcdn tcmesvkngt ydypkyseea klnrekidgv kldstkiyqi laiystvass\\n      541 lvlvvslgai sfwmcsngsl qcrici\\n//\\n\\n'"
      ]
     },
     "execution_count": 53,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "handle = Entrez.efetch(db=\"protein\", rettype=\"gb\", retmode=\"text\", id=\"1834373412\")\n",
    "#record = SeqIO.read(handle, \"genbank\")\n",
    "handle.read()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "H9N2: minks are probable mixing vessels\n",
    "https://www.ncbi.nlm.nih.gov/nuccore/ON870694.1\n",
    " https://www.tandfonline.com/doi/full/10.1080/22221751.2021.1899058\n",
    " Pandemic influenza, typically caused by the reassortment of human and avian influenza viruses, can result in severe or fatal infections in humans. Timely identification of potential pandemic viruses must be a priority in influenza virus surveillance. However, the range of host species responsible for the generation of novel pandemic influenza viruses remains unclear. In this study, we conducted serological surveys for avian and human influenza virus infections in farmed mink and determined the susceptibility of mink to prevailing avian and human virus subtypes. The results showed that farmed mink were commonly infected with human (H3N2 and H1N1/pdm) and avian (H7N9, H5N6, and H9N2) influenza A viruses. Correlational analysis indicated that transmission of human influenza viruses occurred from humans to mink, and that feed source was a probable route of avian influenza virus transmission to farmed mink. Animal experiments showed that mink were susceptible and permissive to circulating avian and human influenza viruses, and that human influenza viruses (H3N2 and H1N1/pdm), but not avian viruses, were capable of aerosol transmission among mink. These results indicate that farmed mink could be highly permissive “mixing vessels” for the reassortment of circulating human and avian influenza viruses. \n",
    " \n",
    " H7N9\n",
    " https://www.bmj.com/content/347/bmj.f4752?tab=responses\n",
    " HH transmission has been suspected\n",
    " Asian lineage H7N9 virus is rated by the Influenza Risk Assessment Tool as having the greatest potential to cause a pandemic, as well as potentially posing the greatest risk to severely impact public health if it were to achieve sustained human-to-human transmission.\n",
    " \n",
    " H3N2 swine\n",
    " https://www.cdc.gov/flu/swineflu/spotlights/first-human-infection-2022.htm\n",
    " Human infection detected in US this Aug\n",
    " \n",
    " H1N1\n",
    " Submitted (03-NOV-2020) USDA Swine Surveillance, USDA Swine\n",
    "            Surveillance, 1920 Dayton, Ames, IA 50010, USA\n",
    "COMMENT     Method: conceptual translation.\n",
    "\n"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.5"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
