{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import copy\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import Levenshtein\n",
    "from tqdm import tqdm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "/home/ishanu/ZED/Research/emergenet/paper_data_old/irat_qnet/results/animal_predictions\r\n"
     ]
    }
   ],
   "source": [
    "! pwd"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "# parameters\n",
    "threshold=6  # draw tree above this emergenec risk threshold\n",
    "CONSTRUCT_PHYLO=True  # use constructed tree (tree construction takes hours)\n",
    "OUTPUT_DIR='./'\n",
    "PHYLO_TREE_DIR='./'\n",
    "PHYLO_DIR='./'\n",
    "COMBINED_RESULTS='../../../../paper_data/irat_enet/results/animal_predictions/combined_results.csv'\n",
    "#COMBINED_RESULTS='/home/ishanu/Downloads/combined_results_irat.csv'\n",
    "#CONSTRUCT_PHYLO=False\n",
    "num_collapsed=19  # number of mutations within which leaves are collapsed\n",
    "VERBOSE=False"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 26,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>subtype</th>\n",
       "      <th>ha_accession</th>\n",
       "      <th>na_accession</th>\n",
       "      <th>ha</th>\n",
       "      <th>na</th>\n",
       "      <th>ha_risk</th>\n",
       "      <th>na_risk</th>\n",
       "      <th>geometric_mean_risk</th>\n",
       "      <th>emergence_risk</th>\n",
       "      <th>emergence_risk_var</th>\n",
       "      <th>impact_risk</th>\n",
       "      <th>impact_risk_var</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A/swine/Missouri/A02524459/2020</td>\n",
       "      <td>H1N1</td>\n",
       "      <td>EPI1765551</td>\n",
       "      <td>EPI1765552</td>\n",
       "      <td>MKAILVVLLYTFTTANADTLCIGYHANNSTDTVDTVLEKNVTVTHS...</td>\n",
       "      <td>MNANQRIIAIGTVCMIVGIISLLLQIGNIVSLWISHSIQTGWENHT...</td>\n",
       "      <td>[0.014528550015281423, 0.014546310217342569, 0...</td>\n",
       "      <td>[0.029578732921608532, 0.02932902190208756, 0....</td>\n",
       "      <td>[0.020730077198125453, 0.020655000628395058, 0...</td>\n",
       "      <td>6.420763</td>\n",
       "      <td>8.045648e-06</td>\n",
       "      <td>6.449227</td>\n",
       "      <td>6.178046e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A/swine/Ohio/A02524448/2020</td>\n",
       "      <td>H1N1</td>\n",
       "      <td>EPI1765553</td>\n",
       "      <td>EPI1765554</td>\n",
       "      <td>MKAVLVVLLYTFTTANADTLCIGYHANNSTDTVDTVLEKNVTVTHS...</td>\n",
       "      <td>MNTNQRIITIGTVCLIVGIISLLLQIGNIVSLWVSHSIQTRWENHT...</td>\n",
       "      <td>[0.019129440067524317, 0.019127869774293836, 0...</td>\n",
       "      <td>[0.030817728728596058, 0.03059500648117531, 0....</td>\n",
       "      <td>[0.02428015433910791, 0.02419126494657932, 0.0...</td>\n",
       "      <td>6.350109</td>\n",
       "      <td>1.971105e-05</td>\n",
       "      <td>6.387369</td>\n",
       "      <td>1.508262e-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A/swine/South_Dakota/A02524453/2020</td>\n",
       "      <td>H1N1</td>\n",
       "      <td>EPI1765555</td>\n",
       "      <td>EPI1765556</td>\n",
       "      <td>MKAILVVLLYTFTTANADTLCIGYHANNSTDTVDTVLEKNVTVTHS...</td>\n",
       "      <td>MNTNQRIITIGTVCMIVGIISLLLQIGNIVSLWISHSIQIEWENHT...</td>\n",
       "      <td>[0.015332923170890989, 0.015282596391980905, 0...</td>\n",
       "      <td>[0.030647302962040043, 0.030735514122289354, 0...</td>\n",
       "      <td>[0.02167747083291729, 0.02167298911620124, 0.0...</td>\n",
       "      <td>6.401165</td>\n",
       "      <td>1.234722e-05</td>\n",
       "      <td>6.432058</td>\n",
       "      <td>9.471705e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A/swine/Indiana/A02524458/2020</td>\n",
       "      <td>H1N1</td>\n",
       "      <td>EPI1765559</td>\n",
       "      <td>EPI1765560</td>\n",
       "      <td>MKAVLVVLLYTFTTANADTLCIGYHANNSTDTVDTVLEKNVTVTHS...</td>\n",
       "      <td>MNTNQRIITIGTVCLIVGIISLLLQIGNIVSLWVSHSIQTRWENHT...</td>\n",
       "      <td>[0.01911712558486812, 0.019116256253057823, 0....</td>\n",
       "      <td>[0.030817728728596058, 0.03059500648117531, 0....</td>\n",
       "      <td>[0.024272337966190424, 0.024183919946073922, 0...</td>\n",
       "      <td>6.350272</td>\n",
       "      <td>1.976990e-05</td>\n",
       "      <td>6.387512</td>\n",
       "      <td>1.512778e-05</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A/swine/Iowa/A02479424/2020</td>\n",
       "      <td>H1N1</td>\n",
       "      <td>EPI1766606</td>\n",
       "      <td>EPI1766607</td>\n",
       "      <td>MKAVLVVMLYTFTTANADTLCIGYHANNSTDTVDTVLEKNVTVTHS...</td>\n",
       "      <td>MNTNQRIITIGTVCLIVGIISLLLQIGNIVSLWVSHSIQTRWENHT...</td>\n",
       "      <td>[0.016496181614260582, 0.01642582283626936, 0....</td>\n",
       "      <td>[0.03411863661430806, 0.03386191944444978, 0.0...</td>\n",
       "      <td>[0.023723979978506695, 0.02358410247795232, 0....</td>\n",
       "      <td>6.360128</td>\n",
       "      <td>1.145076e-05</td>\n",
       "      <td>6.396134</td>\n",
       "      <td>8.766851e-06</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6061</th>\n",
       "      <td>A/chicken/Guangdong/F94/2022</td>\n",
       "      <td>H9N2</td>\n",
       "      <td>EPI2167482</td>\n",
       "      <td>EPI2167484</td>\n",
       "      <td>MKTVSLITILLAATVSNADKICIGYQSTNSTETVDTLTENNVPVTH...</td>\n",
       "      <td>MNPNQKITAIGSVSLIIAIICLLMQIAILTTTMTLHFGQKECSNPL...</td>\n",
       "      <td>[0.021926223328056015, 0.021926223328056015, 0...</td>\n",
       "      <td>[0.07341886162130176, 0.07341886162130176, 0.0...</td>\n",
       "      <td>[0.04012229251177334, 0.04012229251177334, 0.0...</td>\n",
       "      <td>6.033752</td>\n",
       "      <td>7.888609e-31</td>\n",
       "      <td>6.111792</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6062</th>\n",
       "      <td>A/chicken/Guangdong/F99/2022</td>\n",
       "      <td>H9N2</td>\n",
       "      <td>EPI2167503</td>\n",
       "      <td>EPI2167506</td>\n",
       "      <td>MEAVSLITILLAATVSNADKICIGYQSTNSTETVDTLTENNVPVTH...</td>\n",
       "      <td>MNPNQKITAIGSVSLIIAIICLLMQIAILTTTMTLHFRQKECSNPS...</td>\n",
       "      <td>[0.01898769074353181, 0.01898769074353181, 0.0...</td>\n",
       "      <td>[0.05202795688766431, 0.05202795688766431, 0.0...</td>\n",
       "      <td>[0.03143072947611582, 0.03143072947611582, 0.0...</td>\n",
       "      <td>6.202124</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>6.258162</td>\n",
       "      <td>7.888609e-31</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6063</th>\n",
       "      <td>A/chicken/Egypt/Menoufia/2021</td>\n",
       "      <td>H9N2</td>\n",
       "      <td>EPI2175832</td>\n",
       "      <td>EPI2175833</td>\n",
       "      <td>MEVIPLMIMLLLVTTNNADKICIGHQSTNSTETVDTLTETGVPVTH...</td>\n",
       "      <td>MNPNQRIIALGSASLTVATICLLIQIAILATTMTLHFKQNEYANTS...</td>\n",
       "      <td>[0.05534613906024509, 0.05534613906024509, 0.0...</td>\n",
       "      <td>[0.4610969543711347, 0.4610969543711347, 0.461...</td>\n",
       "      <td>[0.15974960455938633, 0.15974960455938633, 0.1...</td>\n",
       "      <td>4.344723</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>4.715680</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6064</th>\n",
       "      <td>A/chicken/South_Korea/SL20/2020</td>\n",
       "      <td>H9N2</td>\n",
       "      <td>EPI2176583</td>\n",
       "      <td>EPI2176585</td>\n",
       "      <td>METVSLITILVVATVSSADKICIGYQSTNSTETVDTLTENNVPVTH...</td>\n",
       "      <td>MNPNQKITAIGSVSLIIAIICLLMQIAILTTTMTFHFGQKECSNPS...</td>\n",
       "      <td>[0.02999852644258515, 0.02999852644258515, 0.0...</td>\n",
       "      <td>[0.0512932924280397, 0.0512932924280397, 0.051...</td>\n",
       "      <td>[0.03922656229176604, 0.03922656229176604, 0.0...</td>\n",
       "      <td>6.050818</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>6.126595</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6065</th>\n",
       "      <td>A/Greylag_Goose/South_Korea/SW21/2021</td>\n",
       "      <td>H9N2</td>\n",
       "      <td>EPI2188134</td>\n",
       "      <td>EPI2188136</td>\n",
       "      <td>METIALVAILLMATASNADKICIGYQSTNSTETVDTLIENNVPVTH...</td>\n",
       "      <td>MNPNQKIITIGSVSLTIAIVCFLMQIAILATTITLHFKQNECSIPS...</td>\n",
       "      <td>[0.0464078674105025, 0.0464078674105025, 0.046...</td>\n",
       "      <td>[0.46130905496342184, 0.46130905496342184, 0.4...</td>\n",
       "      <td>[0.14631599180543012, 0.14631599180543012, 0.1...</td>\n",
       "      <td>4.475996</td>\n",
       "      <td>0.000000e+00</td>\n",
       "      <td>4.814980</td>\n",
       "      <td>0.000000e+00</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>6066 rows × 13 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                         id subtype ha_accession na_accession   \n",
       "0           A/swine/Missouri/A02524459/2020    H1N1   EPI1765551   EPI1765552  \\\n",
       "1               A/swine/Ohio/A02524448/2020    H1N1   EPI1765553   EPI1765554   \n",
       "2       A/swine/South_Dakota/A02524453/2020    H1N1   EPI1765555   EPI1765556   \n",
       "3            A/swine/Indiana/A02524458/2020    H1N1   EPI1765559   EPI1765560   \n",
       "4               A/swine/Iowa/A02479424/2020    H1N1   EPI1766606   EPI1766607   \n",
       "...                                     ...     ...          ...          ...   \n",
       "6061           A/chicken/Guangdong/F94/2022    H9N2   EPI2167482   EPI2167484   \n",
       "6062           A/chicken/Guangdong/F99/2022    H9N2   EPI2167503   EPI2167506   \n",
       "6063          A/chicken/Egypt/Menoufia/2021    H9N2   EPI2175832   EPI2175833   \n",
       "6064        A/chicken/South_Korea/SL20/2020    H9N2   EPI2176583   EPI2176585   \n",
       "6065  A/Greylag_Goose/South_Korea/SW21/2021    H9N2   EPI2188134   EPI2188136   \n",
       "\n",
       "                                                     ha   \n",
       "0     MKAILVVLLYTFTTANADTLCIGYHANNSTDTVDTVLEKNVTVTHS...  \\\n",
       "1     MKAVLVVLLYTFTTANADTLCIGYHANNSTDTVDTVLEKNVTVTHS...   \n",
       "2     MKAILVVLLYTFTTANADTLCIGYHANNSTDTVDTVLEKNVTVTHS...   \n",
       "3     MKAVLVVLLYTFTTANADTLCIGYHANNSTDTVDTVLEKNVTVTHS...   \n",
       "4     MKAVLVVMLYTFTTANADTLCIGYHANNSTDTVDTVLEKNVTVTHS...   \n",
       "...                                                 ...   \n",
       "6061  MKTVSLITILLAATVSNADKICIGYQSTNSTETVDTLTENNVPVTH...   \n",
       "6062  MEAVSLITILLAATVSNADKICIGYQSTNSTETVDTLTENNVPVTH...   \n",
       "6063  MEVIPLMIMLLLVTTNNADKICIGHQSTNSTETVDTLTETGVPVTH...   \n",
       "6064  METVSLITILVVATVSSADKICIGYQSTNSTETVDTLTENNVPVTH...   \n",
       "6065  METIALVAILLMATASNADKICIGYQSTNSTETVDTLIENNVPVTH...   \n",
       "\n",
       "                                                     na   \n",
       "0     MNANQRIIAIGTVCMIVGIISLLLQIGNIVSLWISHSIQTGWENHT...  \\\n",
       "1     MNTNQRIITIGTVCLIVGIISLLLQIGNIVSLWVSHSIQTRWENHT...   \n",
       "2     MNTNQRIITIGTVCMIVGIISLLLQIGNIVSLWISHSIQIEWENHT...   \n",
       "3     MNTNQRIITIGTVCLIVGIISLLLQIGNIVSLWVSHSIQTRWENHT...   \n",
       "4     MNTNQRIITIGTVCLIVGIISLLLQIGNIVSLWVSHSIQTRWENHT...   \n",
       "...                                                 ...   \n",
       "6061  MNPNQKITAIGSVSLIIAIICLLMQIAILTTTMTLHFGQKECSNPL...   \n",
       "6062  MNPNQKITAIGSVSLIIAIICLLMQIAILTTTMTLHFRQKECSNPS...   \n",
       "6063  MNPNQRIIALGSASLTVATICLLIQIAILATTMTLHFKQNEYANTS...   \n",
       "6064  MNPNQKITAIGSVSLIIAIICLLMQIAILTTTMTFHFGQKECSNPS...   \n",
       "6065  MNPNQKIITIGSVSLTIAIVCFLMQIAILATTITLHFKQNECSIPS...   \n",
       "\n",
       "                                                ha_risk   \n",
       "0     [0.014528550015281423, 0.014546310217342569, 0...  \\\n",
       "1     [0.019129440067524317, 0.019127869774293836, 0...   \n",
       "2     [0.015332923170890989, 0.015282596391980905, 0...   \n",
       "3     [0.01911712558486812, 0.019116256253057823, 0....   \n",
       "4     [0.016496181614260582, 0.01642582283626936, 0....   \n",
       "...                                                 ...   \n",
       "6061  [0.021926223328056015, 0.021926223328056015, 0...   \n",
       "6062  [0.01898769074353181, 0.01898769074353181, 0.0...   \n",
       "6063  [0.05534613906024509, 0.05534613906024509, 0.0...   \n",
       "6064  [0.02999852644258515, 0.02999852644258515, 0.0...   \n",
       "6065  [0.0464078674105025, 0.0464078674105025, 0.046...   \n",
       "\n",
       "                                                na_risk   \n",
       "0     [0.029578732921608532, 0.02932902190208756, 0....  \\\n",
       "1     [0.030817728728596058, 0.03059500648117531, 0....   \n",
       "2     [0.030647302962040043, 0.030735514122289354, 0...   \n",
       "3     [0.030817728728596058, 0.03059500648117531, 0....   \n",
       "4     [0.03411863661430806, 0.03386191944444978, 0.0...   \n",
       "...                                                 ...   \n",
       "6061  [0.07341886162130176, 0.07341886162130176, 0.0...   \n",
       "6062  [0.05202795688766431, 0.05202795688766431, 0.0...   \n",
       "6063  [0.4610969543711347, 0.4610969543711347, 0.461...   \n",
       "6064  [0.0512932924280397, 0.0512932924280397, 0.051...   \n",
       "6065  [0.46130905496342184, 0.46130905496342184, 0.4...   \n",
       "\n",
       "                                    geometric_mean_risk  emergence_risk   \n",
       "0     [0.020730077198125453, 0.020655000628395058, 0...        6.420763  \\\n",
       "1     [0.02428015433910791, 0.02419126494657932, 0.0...        6.350109   \n",
       "2     [0.02167747083291729, 0.02167298911620124, 0.0...        6.401165   \n",
       "3     [0.024272337966190424, 0.024183919946073922, 0...        6.350272   \n",
       "4     [0.023723979978506695, 0.02358410247795232, 0....        6.360128   \n",
       "...                                                 ...             ...   \n",
       "6061  [0.04012229251177334, 0.04012229251177334, 0.0...        6.033752   \n",
       "6062  [0.03143072947611582, 0.03143072947611582, 0.0...        6.202124   \n",
       "6063  [0.15974960455938633, 0.15974960455938633, 0.1...        4.344723   \n",
       "6064  [0.03922656229176604, 0.03922656229176604, 0.0...        6.050818   \n",
       "6065  [0.14631599180543012, 0.14631599180543012, 0.1...        4.475996   \n",
       "\n",
       "      emergence_risk_var  impact_risk  impact_risk_var  \n",
       "0           8.045648e-06     6.449227     6.178046e-06  \n",
       "1           1.971105e-05     6.387369     1.508262e-05  \n",
       "2           1.234722e-05     6.432058     9.471705e-06  \n",
       "3           1.976990e-05     6.387512     1.512778e-05  \n",
       "4           1.145076e-05     6.396134     8.766851e-06  \n",
       "...                  ...          ...              ...  \n",
       "6061        7.888609e-31     6.111792     0.000000e+00  \n",
       "6062        0.000000e+00     6.258162     7.888609e-31  \n",
       "6063        0.000000e+00     4.715680     0.000000e+00  \n",
       "6064        0.000000e+00     6.126595     0.000000e+00  \n",
       "6065        0.000000e+00     4.814980     0.000000e+00  \n",
       "\n",
       "[6066 rows x 13 columns]"
      ]
     },
     "execution_count": 26,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "tmpf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "tmpf=pd.read_csv(COMBINED_RESULTS)\n",
    "IRATSEQ=[]#tmpf[tmpf.is_irat==1].id.values.astype(str)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# class definition to hold multi sequence data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [],
   "source": [
    "class SeqInfo(object):\n",
    "    \"\"\"Holds information regarding the sequence.\n",
    "    \n",
    "    \"\"\"\n",
    "    def __init__(self, seq, \n",
    "                 protein,\n",
    "                 accession,\n",
    "                 subtype=None,\n",
    "                 id=None,\n",
    "                 name=None,\n",
    "                 host=None, \n",
    "                 date=None, \n",
    "                 erisk=None,\n",
    "                 irisk=None,\n",
    "                 risk_flag=None,\n",
    "                 country=None):\n",
    "        self.name = name\n",
    "        self.id = id\n",
    "        self.protein=protein\n",
    "        self.subtype=subtype        \n",
    "        self.seq = seq\n",
    "        self.accession = accession \n",
    "        self.host = host\n",
    "        self.date = date\n",
    "        self.erisk = erisk\n",
    "        self.irisk = irisk\n",
    "        self.risk_flag = risk_flag\n",
    "        self.country = country\n",
    "        \n",
    "class MultipleSeqInfo(object):\n",
    "    \"\"\"Holds information regarding multiple sequences.\n",
    "    \n",
    "    Args:\n",
    "        dataframe (pandas.DataFrame): list of records parsed from NCBI\n",
    "        accessionname (str): column name for accession id\n",
    "        proteinname (str): protein name \n",
    "        risk_threshold (float): emergence risk threshold to compute distance matrix\n",
    "    \"\"\"\n",
    "    def __init__(self,\n",
    "                 dataframe,\n",
    "                 accessionname,\n",
    "                 proteinname,\n",
    "                 risk_threshold=6.2):\n",
    "        \n",
    "        self.seq_infos = {}\n",
    "        self.risk_threshold = risk_threshold\n",
    "        for i in np.arange(dataframe.index.size):\n",
    "            record=dataframe.iloc[i,:]\n",
    "            seqinfo = SeqInfo(\n",
    "                name=record.id,\n",
    "                seq=record[proteinname], \n",
    "                protein=proteinname,\n",
    "                accession=record[accessionname],\n",
    "                subtype=record.subtype,\n",
    "                erisk=record.emergence_risk,\n",
    "                irisk=record.impact_risk,\n",
    "                risk_flag = record.emergence_risk > self.risk_threshold,\n",
    "                host=None,\n",
    "                date=None,\n",
    "                country=None)\n",
    "            #print(record.predicted_emergence_score > self.risk_threshold)\n",
    "            self.seq_infos[seqinfo.accession] = seqinfo\n",
    "            \n",
    "    \n",
    "    def compute_L_distance_matrix(self):\n",
    "        highriskseq = pd.DataFrame.from_dict({key:val.seq \n",
    "                                              for (key,val) in self.seq_infos.items() \n",
    "                                              if val.risk_flag},orient='index',columns=['seq'])\n",
    "        num=highriskseq.index.size\n",
    "        d=np.zeros([num,num])\n",
    "        for x in tqdm(np.arange(num*num)):\n",
    "            j=x//num\n",
    "            i=x-num*j\n",
    "            if i > j:\n",
    "                d[i,j] = Levenshtein.distance(highriskseq.seq.values[i],\n",
    "                                                  highriskseq.seq.values[j])\n",
    "        ds=pd.DataFrame(d)        \n",
    "        ds=(ds+ds.transpose())\n",
    "        ds.columns=highriskseq.index.values\n",
    "        self.highriskdistancematrix=ds.copy()\n",
    "        \n",
    "        self.highriskdistancematrix.to_csv('dm'+str(self.risk_threshold)+'.csv',index=None)\n",
    "        return \n",
    "    \n",
    "    \n",
    "    def accessions_to_subtype(self, accessions):\n",
    "        \"\"\"Create a dictionary mapping the accession to the host.\n",
    "        \"\"\"\n",
    "        \n",
    "        subtypes = []\n",
    "        for accession in accessions:\n",
    "            seqinfo = self.seq_infos[accession]\n",
    "            subtypes.append(seqinfo.subtype)\n",
    "            \n",
    "        return subtypes\n",
    "\n",
    "    def accessions_to_host(self, accessions):\n",
    "        \"\"\"Create a dictionary mapping the accession to the host.\n",
    "        \"\"\"\n",
    "        \n",
    "        hosts = []\n",
    "        for accession in accessions:\n",
    "            seqinfo = self.seq_infos[accession]\n",
    "            hosts.append(seqinfo.host)\n",
    "        return hosts\n",
    "           \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {},
   "outputs": [],
   "source": [
    "df=pd.read_csv(COMBINED_RESULTS,index_col=0).reset_index()\n",
    "ALLinfoHA=MultipleSeqInfo(df.reset_index(),'ha_accession','ha',risk_threshold=threshold)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "subtype\n",
       "H1N1    895\n",
       "H3N2    821\n",
       "H9N2     36\n",
       "H7N9      4\n",
       "Name: count, dtype: int64"
      ]
     },
     "execution_count": 7,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[df.emergence_risk>threshold].subtype.value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "scrolled": true
   },
   "source": [
    "# generate distance matrix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████| 3083536/3083536 [00:57<00:00, 53671.71it/s]\n"
     ]
    }
   ],
   "source": [
    "ALLinfoHA.compute_L_distance_matrix()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# tree construction from distance matrix using biopython"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio.Phylo import TreeConstruction\n",
    "from Bio import Phylo\n",
    "from Bio.SeqRecord import SeqRecord\n",
    "from Bio.Seq import Seq\n",
    "from Bio.Align import MultipleSeqAlignment\n",
    "from Bio import Entrez\n",
    "from Bio import SeqIO\n",
    "\n",
    "\n",
    "def load_dm(file_, upper_diag=True):\n",
    "    \"\"\"Load the distance matrix. \n",
    "    \n",
    "    Also, do some preprocessing. \n",
    "    \"\"\"\n",
    "    \n",
    "    df = pd.read_csv(file_)\n",
    "    #df.set_index('Unnamed: 0', inplace=True)\n",
    "    #assert np.all(df.columns == df.index)\n",
    "    \n",
    "    # drop duplicate columns after reading csv\n",
    "    #df = df.loc[:, ~df.columns.str.replace(\"(\\.\\d+)$\", \"\").duplicated()]\n",
    "    \n",
    "    if upper_diag:\n",
    "        df = df + df.T\n",
    "    return df\n",
    "\n",
    "def save_tree(tree, file_name, save_type='xml'):\n",
    "    \"\"\"Saved the created phylogenetic tree.\"\"\"\n",
    "    \n",
    "    if save_type == 'pickle':\n",
    "        graph = Phylo.to_networkx(tree)\n",
    "        save_pickled(graph, file_name)\n",
    "    elif save_type == 'xml':\n",
    "        Phylo.write(tree, file_name, 'phyloxml')\n",
    "    else:\n",
    "        raise ValueError('Not a correct save type.')\n",
    "    \n",
    "def pandas_dm_to_biopython_dm(dm):\n",
    "    \"\"\"Convert the pandas distance matrix to the biopython distance matrix.\n",
    "    \n",
    "    Returns:\n",
    "        biopython distance matrix\n",
    "    \"\"\"\n",
    "    \n",
    "    accessions = dm.columns\n",
    "    bio_dm = []\n",
    "    for i, accession in enumerate(accessions):\n",
    "        bio_dm.append(list(dm.iloc[i, :i+1].values))\n",
    "        \n",
    "    bio_dm = TreeConstruction._DistanceMatrix(\n",
    "        list(dm.columns), \n",
    "        bio_dm)\n",
    "    \n",
    "    return bio_dm\n",
    "\n",
    "def distance_matrix_to_phylo_tree(dm, outfile=None):\n",
    "    \"\"\"Create a phylogenetic tree from the distance matrix.\"\"\"\n",
    "    \n",
    "    dm = pandas_dm_to_biopython_dm(dm)\n",
    "    \n",
    "    treeConstructor = TreeConstruction.DistanceTreeConstructor()\n",
    "    tree = treeConstructor.nj(dm)\n",
    "    \n",
    "    if outfile is not None:\n",
    "        save_tree(tree, outfile)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# ete3 function, not all of these are used here"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "from ete3 import Tree, TreeStyle\n",
    "from ete3 import Phyloxml\n",
    "from ete3 import AttrFace, faces, Tree, NodeStyle, TreeStyle\n",
    "\n",
    "def load_pickled(file_name):\n",
    "    with open(file_name, 'rb') as f:\n",
    "        return pickle.load(f, encoding='latin')\n",
    "\n",
    "\n",
    "def get_farthest_node(tree, sequence):\n",
    "    return (tree&sequence).get_farthest_node()\n",
    "\n",
    "def get_all_accessions_from_tree(tree):\n",
    "    return [leaf_node.name for leaf_node in tree.get_leaves()]\n",
    "\n",
    "def remove_certain_hosts_from_tree(tree, hosts):\n",
    "    \"\"\"Remove leaf nodes if the host of that leaf is in `hosts`\"\"\"\n",
    "    \n",
    "    tree = copy.deepcopy(tree)\n",
    "    \n",
    "    removed_accessions = []\n",
    "    for leaf_node in tree.get_leaves():\n",
    "        if leaf_node.host in hosts:\n",
    "            leaf_node.detach()\n",
    "            \n",
    "    return tree\n",
    "\n",
    "def set_midpoint_outgroup(tree):\n",
    "    tree.set_outgroup(tree.get_midpoint_outgroup())\n",
    "\n",
    "\n",
    "def load_tree(filename, type_='phyloxml'):\n",
    "    \"\"\"Load saved phylogenetic tree.\n",
    "    \"\"\"\n",
    "    \n",
    "    if type_ == 'phyloxml':\n",
    "        project = Phyloxml()\n",
    "        project.build_from_file(filename)\n",
    "\n",
    "        for tree in project.get_phylogeny():\n",
    "            break\n",
    "\n",
    "        t=tree\n",
    "        \n",
    "    elif type_ == 'newick':\n",
    "        t = Tree(filename, format=1)\n",
    "    else:\n",
    "        raise ValueError('Not a correct type.')\n",
    "    \n",
    "    return t"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# construct tree from distance matrix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "CONSTRUCT_PHYLO=True\n",
    "if CONSTRUCT_PHYLO:\n",
    "    ALL_dm_ldistance = load_dm(\n",
    "        OUTPUT_DIR + 'dm'+str(threshold)+'.csv', \n",
    "        upper_diag=False)\n",
    "    \n",
    "    distance_matrix_to_phylo_tree(\n",
    "        ALL_dm_ldistance, PHYLO_TREE_DIR + 'ldistance'+str(threshold)+'.xml')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# convert phyloxml tree to newick tree to manipulate trees"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "Phylo.convert(\n",
    "    PHYLO_DIR + 'ldistance'+str(threshold)+'.xml','phyloxml',\n",
    "    PHYLO_DIR + 'ldistance'+str(threshold)+'.nhx','newick')\n",
    "\n",
    "ltree = load_tree(\n",
    "    PHYLO_DIR + 'ldistance'+str(threshold)+'.nhx',\n",
    "    type_='newick')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# label nodes in tree to add other attributes like subtype risk etc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "def bandify(val,min=6.05,max=6.71,flag=False):\n",
    "    maptoten=int(np.ceil(((val-min)/(max-min))*10))\n",
    "    L=' '+u'\\u2580'*maptoten\n",
    "    \n",
    "    if flag:\n",
    "        L=L+u'\\u21DD'\n",
    "    return L\n",
    "\n",
    "def bandify(val,min=6,max=8,flag=False):\n",
    "    maptoten=int(np.ceil(((val-min)/(max-min))*10))\n",
    "    L=' '+u'\\u2501'*maptoten\n",
    "    if flag:\n",
    "        L=L+u'\\u2605'\n",
    "    return L\n",
    "\n",
    "def label_nodes(\n",
    "        tree, \n",
    "        recordinfo):\n",
    "    \"\"\"Label the nodes of the tree.\n",
    "    \n",
    "    We label nodes on whether:\n",
    "        it is covid19\n",
    "    \"\"\"\n",
    "    \n",
    "    tree = copy.deepcopy(tree)\n",
    "    \n",
    "    for node in tree:\n",
    "        name = node.name      \n",
    "        node.subtype = recordinfo.seq_infos[name].subtype\n",
    "        node.erisk =recordinfo.seq_infos[name].erisk\n",
    "        node.id = recordinfo.seq_infos[name].name + bandify(recordinfo.seq_infos[name].erisk,min=threshold)\n",
    "        if VERBOSE:\n",
    "            print(node.name,node.subtype,node.id,node.erisk)\n",
    "    return tree"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# construct labelled tree"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "labelled_tree=label_nodes(\n",
    "    ltree, ALLinfoHA)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# functions to collapse similar leaves"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "def prune_nodes(t):\n",
    "    # collapsed nodes are labeled, so you locate them and prune them\n",
    "    for n in t.search_nodes(collapsed=True):\n",
    "        for ch in n.get_children():\n",
    "            ch.detach()\n",
    "            \n",
    "            \n",
    "def mean(array):\n",
    "    return sum(array)/float(len(array))\n",
    "\n",
    "def cache_distances(tree):\n",
    "    ''' precalculate distances of all nodes to the root''' \n",
    "    node2rootdist = {tree:0}\n",
    "    for node in tree.iter_descendants('preorder'):\n",
    "        node2rootdist[node] = node.dist + node2rootdist[node.up]\n",
    "    return node2rootdist\n",
    "\n",
    "def closest_node(node, node2tips, root_distance):\n",
    "    \"\"\"Find the closest node.\"\"\"\n",
    "    \n",
    "    tips = []\n",
    "    distances = []\n",
    "    for tip in node2tips[node]:\n",
    "        distances.append(root_distance[tip]-root_distance[node])\n",
    "        tips.append(tip)\n",
    "        #     index = np.argmin([root_distance[tip]-root_distance[node] for tip in node2tips[node]])\n",
    "    index = np.argmin(distances)\n",
    "    return tips[index]\n",
    "\n",
    "def riskiest_node(node, node2tips):\n",
    "    \"\"\"Find the riskiest node.\"\"\"\n",
    "    \n",
    "    tips = []\n",
    "    risks = []\n",
    "    for tip in node2tips[node]:\n",
    "        risks.append(tip.erisk)\n",
    "        tips.append(tip)\n",
    "        #     index = np.argmin([root_distance[tip]-root_distance[node] for tip in node2tips[node]])\n",
    "    index = np.argmax(risks)\n",
    "    return tips[index]\n",
    "\n",
    "\n",
    "def all_collapsed(node, node2tips,AllrecordInfo):\n",
    "    \"\"\"Find all nodes in collapsed set.\"\"\"\n",
    "    \n",
    "    tips = []\n",
    "    for tip in node2tips[node]:\n",
    "        tips.append(AllrecordInfo.seq_infos[tip.name].name)\n",
    "    return tips\n",
    "\n",
    "def all_collapsed(node, node2tips,AllrecordInfo):\n",
    "    \"\"\"Find all nodes in collapsed set.\"\"\"\n",
    "    \n",
    "    tips = []\n",
    "    for tip in node2tips.get(node, []):  # use get() to avoid KeyError if node has no entries in node2tips\n",
    "        tips.append(AllrecordInfo.seq_infos[tip.name].name)\n",
    "        \n",
    "    # if no tips were found and the node is a leaf, append the node itself\n",
    "    if not tips and node.is_leaf():\n",
    "        tips.append(AllrecordInfo.seq_infos[node.name].name)\n",
    "\n",
    "    return tips\n",
    "\n",
    "\n",
    "def collapse(tree, min_dist,AllrecordInfo):\n",
    "    # cache the tip content of each node to reduce the number of times the tree is traversed\n",
    "    \n",
    "    tree = copy.deepcopy(tree)\n",
    "    \n",
    "    node2tips = tree.get_cached_content()\n",
    "    root_distance = cache_distances(tree)\n",
    "\n",
    "    for node in tree.get_descendants('preorder'):\n",
    "        IRAT=False\n",
    "        if not node.is_leaf():\n",
    "            avg_distance_to_tips = mean([root_distance[tip]-root_distance[node]\n",
    "                                         for tip in node2tips[node]])\n",
    "            if VERBOSE:\n",
    "                print(avg_distance_to_tips)\n",
    "            if avg_distance_to_tips <= min_dist:\n",
    "                # do whatever, ete support node annotation, deletion, labeling, etc.\n",
    "            \n",
    "                #closest_name = closest_node(node, node2tips, root_distance).name\n",
    "                \n",
    "                # find if this name is in IRATSEQ ie any IRAT seq is in teh collapsed nosdees\n",
    "                all_collapsed_nodes = all_collapsed(node, node2tips,AllrecordInfo)\n",
    "                #print(all_collapsed_nodes)\n",
    "                \n",
    "                \n",
    "                \n",
    "                for i in IRATSEQ:\n",
    "                    if i in all_collapsed_nodes:\n",
    "                        IRAT=True\n",
    "                        break\n",
    "                                       \n",
    "                closest_name = riskiest_node(node, node2tips).name\n",
    "                node.subtype = AllrecordInfo.seq_infos[closest_name].subtype\n",
    "                node.id = AllrecordInfo.seq_infos[closest_name].name + bandify(AllrecordInfo.seq_infos[closest_name].erisk,min=threshold,flag=IRAT) \n",
    "                node.name = '%s (%g)' %(closest_name,avg_distance_to_tips)\n",
    "                \n",
    "            \n",
    "                node.add_features(collapsed=True)\n",
    "\n",
    "                # set drawing attribute so they look collapsed when displayed with tree.show()\n",
    "                node.img_style['draw_descendants'] = False\n",
    "        else:\n",
    "            all_collapsed_nodes = all_collapsed(node, node2tips,AllrecordInfo)\n",
    "            for i in IRATSEQ:\n",
    "                if i in all_collapsed_nodes:\n",
    "                    IRAT=True\n",
    "                    break\n",
    "            node.id = AllrecordInfo.seq_infos[node.name].name + bandify(AllrecordInfo.seq_infos[node.name].erisk,min=threshold,flag=IRAT) \n",
    "\n",
    "    return tree"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 16,
   "metadata": {},
   "outputs": [],
   "source": [
    "# collapse leaved"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 17,
   "metadata": {
    "scrolled": false
   },
   "outputs": [],
   "source": [
    "num_collapsed=19\n",
    "ltree_collapsed = collapse(\n",
    "    labelled_tree, \n",
    "    min_dist=num_collapsed, \n",
    "    AllrecordInfo=ALLinfoHA)\n",
    "\n",
    "prune_nodes(ltree_collapsed)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "\n",
    "# code for actual rendering"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 18,
   "metadata": {},
   "outputs": [],
   "source": [
    "# COLBAT='DarkRed'\n",
    "# COLRAT='SteelBlue'\n",
    "COLH3N2='Blue'\n",
    "COLH1N1='DarkRed'\n",
    "COLH7N9='DarkGreen'\n",
    "COLH9N2='#BD890F'\n",
    "COLDEF='BLACK'\n",
    "\n",
    "FS=50\n",
    "PW=10\n",
    "\n",
    "\n",
    "def nodeAttribConstruct(color, node):\n",
    "    N = AttrFace(\n",
    "        \"id\", fsize=FS, \n",
    "        text_prefix=\" \",penwidth=PW,ftype='Arial',\n",
    "        fgcolor=color,fstyle='bold')\n",
    "    faces.add_face_to_node(N, node, 1, position=\"branch-right\")\n",
    "    return N\n",
    "\n",
    "def layout(node):\n",
    "    if node.is_leaf():\n",
    "        if  node.subtype == 'H1N1':\n",
    "            N = nodeAttribConstruct(COLH1N1,node)\n",
    "        elif node.subtype == 'H3N2':\n",
    "            N = nodeAttribConstruct(COLH3N2,node)\n",
    "        elif node.subtype == 'H7N9':\n",
    "            N = nodeAttribConstruct(COLH7N9,node)\n",
    "        elif node.subtype == 'H9N2':\n",
    "            N = nodeAttribConstruct(COLH9N2,node)\n",
    "        else:\n",
    "            N = nodeAttribConstruct(COLDEF,node)\n",
    "            \n",
    "\n",
    "            \n",
    "def render_tree(tree, outfile):# all_seq_data, display_type='nearest_host'):\n",
    "    \"\"\"Render the tree inside the file to a circular \n",
    "    phylogenetic tree.\n",
    "    \n",
    "    NOTE: outfile should be in .pdf for best visuals\n",
    "    Returns:\n",
    "    \"\"\"\n",
    "    #tree = Tree(nwfile,format=1)\n",
    "\n",
    "    ts = TreeStyle()\n",
    "    ns = NodeStyle()\n",
    "    ts.show_leaf_name = False\n",
    "    #ts.rotation = 90\n",
    "    ts.mode = \"r\"\n",
    "    #ts.arc_start = -360 # 0 degrees = 3 o'clock\n",
    "    #ts.arc_span = 360\n",
    "    ts.scale=5\n",
    "    ts.show_scale=False\n",
    "    ts.branch_vertical_margin = .5 # 10 pixels between adjacent branches\n",
    "    # ts.show_branch_length=True\n",
    "    #ts.min_leaf_separation=10\n",
    "    #ts.optimal_scale_level='full'\n",
    "    #ts.branch_vertical_margin=0\n",
    "    \n",
    "    ns.hz_line_width=2\n",
    "    ns.vt_line_width=1\n",
    "    #ts.layout_fn = layout\n",
    "    ns[\"vt_line_width\"] = 16\n",
    "    ns[\"hz_line_width\"] = 16\n",
    "    #     ns['fsize'] = 20\n",
    "    for n in tree.traverse():\n",
    "        n.set_style(ns)\n",
    "        \n",
    "    #all_accessions = all_seq_data['accessions'].values\n",
    "    for n in tree:\n",
    "        ts.layout_fn = layout\n",
    "\n",
    "        \n",
    "    tree.set_style(ns)\n",
    "    tree.set_style(ts)\n",
    "    \n",
    "    ax=tree.render(\n",
    "        outfile, \n",
    "        dpi=300, \n",
    "        w=500,\n",
    "        tree_style=ts)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 19,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>subtype</th>\n",
       "      <th>ha_accession</th>\n",
       "      <th>na_accession</th>\n",
       "      <th>impact_risk</th>\n",
       "      <th>emergence_risk</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "Empty DataFrame\n",
       "Columns: [id, subtype, ha_accession, na_accession, impact_risk, emergence_risk]\n",
       "Index: []"
      ]
     },
     "execution_count": 19,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "r=5\n",
    "medrisknames=[]\n",
    "for node in ltree_collapsed:\n",
    "    if ALLinfoHA.seq_infos[node.name.split()[0]].erisk > r:\n",
    "        medrisknames=np.append(medrisknames,\n",
    "                                ALLinfoHA.seq_infos[node.name.split()[0]].name)\n",
    "medrisknames=df[df.id.isin(medrisknames)][['id','subtype',\n",
    "                                               'ha_accession',\n",
    "                                               'na_accession',\n",
    "                                               'impact_risk',\n",
    "                                               'emergence_risk']].sort_values('emergence_risk',\n",
    "                                                                                         ascending=False)\n",
    "medrisknames.to_csv('allriskystrains_collapsed.csv',index=None)\n",
    "medrisknames[medrisknames.id.str.contains('Ohio')]"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Render phylogenetic trees"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "/usr/share/themes/Flat-Remix-GTK-Red-Darkest/gtk-2.0/main.rc:1227: Unable to locate image file in pixmap_path: \"assets/entry-background.png\"\n",
      "/usr/share/themes/Flat-Remix-GTK-Red-Darkest/gtk-2.0/main.rc:1228: Background image options specified without filename\n",
      "/usr/share/themes/Flat-Remix-GTK-Red-Darkest/gtk-2.0/main.rc:1240: Unable to locate image file in pixmap_path: \"assets/entry-background.png\"\n",
      "/usr/share/themes/Flat-Remix-GTK-Red-Darkest/gtk-2.0/main.rc:1241: Background image options specified without filename\n",
      "Gtk-Message: 12:33:45.865: Failed to load module \"pk-gtk-module\"\n"
     ]
    }
   ],
   "source": [
    "ax=render_tree(\n",
    "    labelled_tree,\n",
    "    './riskyphylo'+str(threshold)+'.pdf')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "ax=render_tree(\n",
    "    ltree_collapsed,\n",
    "    'riskyphylo'+str(threshold)+'_collapsed_'+str(num_collapsed)+'.pdf')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Tree node 'Inner1754' (0x7ff3b7c4c13)"
      ]
     },
     "execution_count": 22,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ltree_collapsed"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Find high risk strains which are on distinct branches on phylogenetic tree"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 23,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>subtype</th>\n",
       "      <th>HA accession</th>\n",
       "      <th>NA accession</th>\n",
       "      <th>predicted IRAT impact</th>\n",
       "      <th>predicted IRAT emergence</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>strain</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A/swine/Missouri/A02524711/2020</th>\n",
       "      <td>H1N1</td>\n",
       "      <td>EPI1818121</td>\n",
       "      <td>EPI1818122</td>\n",
       "      <td>6.767254</td>\n",
       "      <td>6.782248</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A/swine/Indiana/A02524710/2020</th>\n",
       "      <td>H3N2</td>\n",
       "      <td>EPI1818137</td>\n",
       "      <td>EPI1818138</td>\n",
       "      <td>6.720489</td>\n",
       "      <td>6.729258</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A/swine/North_Carolina/A02479173/2020</th>\n",
       "      <td>H1N1</td>\n",
       "      <td>EPI1780425</td>\n",
       "      <td>EPI1780426</td>\n",
       "      <td>6.713621</td>\n",
       "      <td>6.721477</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A/Camel/Inner_Mongolia/XL/2020</th>\n",
       "      <td>H7N9</td>\n",
       "      <td>EPI2026200</td>\n",
       "      <td>EPI2026202</td>\n",
       "      <td>6.699016</td>\n",
       "      <td>6.704867</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A/swine/Tennessee/A02524414/2022</th>\n",
       "      <td>H1N1</td>\n",
       "      <td>EPI2149257</td>\n",
       "      <td>EPI2149258</td>\n",
       "      <td>6.650119</td>\n",
       "      <td>6.649429</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A/canine/Texas/21-011409-001/2021</th>\n",
       "      <td>H3N2</td>\n",
       "      <td>EPI1896555</td>\n",
       "      <td>EPI1896557</td>\n",
       "      <td>6.344011</td>\n",
       "      <td>6.300499</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A/swine/Iowa/A02525161/2021</th>\n",
       "      <td>H3N2</td>\n",
       "      <td>EPI1909023</td>\n",
       "      <td>EPI1909024</td>\n",
       "      <td>6.334205</td>\n",
       "      <td>6.289272</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A/swine/Iowa/A02246996/2021</th>\n",
       "      <td>H1N1</td>\n",
       "      <td>EPI2146133</td>\n",
       "      <td>EPI2146134</td>\n",
       "      <td>6.326870</td>\n",
       "      <td>6.280888</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A/Muscovy_duck/Vietnam/QN6297/2020</th>\n",
       "      <td>H3N2</td>\n",
       "      <td>EPI1974815</td>\n",
       "      <td>EPI1974818</td>\n",
       "      <td>6.323945</td>\n",
       "      <td>6.277527</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A/mink/China/chick_embryo/2020</th>\n",
       "      <td>H9N2</td>\n",
       "      <td>EPI2161544</td>\n",
       "      <td>EPI2161548</td>\n",
       "      <td>6.318778</td>\n",
       "      <td>6.271620</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>74 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                      subtype HA accession NA accession   \n",
       "strain                                                                    \n",
       "A/swine/Missouri/A02524711/2020          H1N1   EPI1818121   EPI1818122  \\\n",
       "A/swine/Indiana/A02524710/2020           H3N2   EPI1818137   EPI1818138   \n",
       "A/swine/North_Carolina/A02479173/2020    H1N1   EPI1780425   EPI1780426   \n",
       "A/Camel/Inner_Mongolia/XL/2020           H7N9   EPI2026200   EPI2026202   \n",
       "A/swine/Tennessee/A02524414/2022         H1N1   EPI2149257   EPI2149258   \n",
       "...                                       ...          ...          ...   \n",
       "A/canine/Texas/21-011409-001/2021        H3N2   EPI1896555   EPI1896557   \n",
       "A/swine/Iowa/A02525161/2021              H3N2   EPI1909023   EPI1909024   \n",
       "A/swine/Iowa/A02246996/2021              H1N1   EPI2146133   EPI2146134   \n",
       "A/Muscovy_duck/Vietnam/QN6297/2020       H3N2   EPI1974815   EPI1974818   \n",
       "A/mink/China/chick_embryo/2020           H9N2   EPI2161544   EPI2161548   \n",
       "\n",
       "                                       predicted IRAT impact   \n",
       "strain                                                         \n",
       "A/swine/Missouri/A02524711/2020                     6.767254  \\\n",
       "A/swine/Indiana/A02524710/2020                      6.720489   \n",
       "A/swine/North_Carolina/A02479173/2020               6.713621   \n",
       "A/Camel/Inner_Mongolia/XL/2020                      6.699016   \n",
       "A/swine/Tennessee/A02524414/2022                    6.650119   \n",
       "...                                                      ...   \n",
       "A/canine/Texas/21-011409-001/2021                   6.344011   \n",
       "A/swine/Iowa/A02525161/2021                         6.334205   \n",
       "A/swine/Iowa/A02246996/2021                         6.326870   \n",
       "A/Muscovy_duck/Vietnam/QN6297/2020                  6.323945   \n",
       "A/mink/China/chick_embryo/2020                      6.318778   \n",
       "\n",
       "                                       predicted IRAT emergence  \n",
       "strain                                                           \n",
       "A/swine/Missouri/A02524711/2020                        6.782248  \n",
       "A/swine/Indiana/A02524710/2020                         6.729258  \n",
       "A/swine/North_Carolina/A02479173/2020                  6.721477  \n",
       "A/Camel/Inner_Mongolia/XL/2020                         6.704867  \n",
       "A/swine/Tennessee/A02524414/2022                       6.649429  \n",
       "...                                                         ...  \n",
       "A/canine/Texas/21-011409-001/2021                      6.300499  \n",
       "A/swine/Iowa/A02525161/2021                            6.289272  \n",
       "A/swine/Iowa/A02246996/2021                            6.280888  \n",
       "A/Muscovy_duck/Vietnam/QN6297/2020                     6.277527  \n",
       "A/mink/China/chick_embryo/2020                         6.271620  \n",
       "\n",
       "[74 rows x 5 columns]"
      ]
     },
     "execution_count": 23,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "r=6.057\n",
    "r=6.25\n",
    "highrisknames=[]\n",
    "for node in ltree_collapsed:\n",
    "    if ALLinfoHA.seq_infos[node.name.split()[0]].erisk > r:\n",
    "        highrisknames=np.append(highrisknames,\n",
    "                                ALLinfoHA.seq_infos[node.name.split()[0]].name)\n",
    "highrisknamesdf=df[df.id.isin(highrisknames)][['id','subtype',\n",
    "                                               'ha_accession',\n",
    "                                               'na_accession',\n",
    "                                               'impact_risk',\n",
    "                                               'emergence_risk']].sort_values('emergence_risk',\n",
    "                                                                                         ascending=False)\n",
    "highrisknamesdf = highrisknamesdf.rename(columns={'id':'strain',\n",
    "                                                  'ha_accession':'HA accession',\n",
    "                                                  'na_accession':'NA accession',\n",
    "                                                  'impact_risk':'predicted IRAT impact',\n",
    "                                                  'emergence_risk':'predicted IRAT emergence'}).set_index('strain')\n",
    "highrisknamesdf#.drop_duplicates()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 24,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>subtype</th>\n",
       "      <th>HA accession</th>\n",
       "      <th>NA accession</th>\n",
       "      <th>predicted IRAT impact</th>\n",
       "      <th>predicted IRAT emergence</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>strain</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>\\rowcolor{Red3!20}A/swine/Missouri/A02524711/2020</th>\n",
       "      <td>H1N1</td>\n",
       "      <td>EPI1818121</td>\n",
       "      <td>EPI1818122</td>\n",
       "      <td>6.767254</td>\n",
       "      <td>6.782248</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>\\rowcolor{Blue1!30}A/swine/Indiana/A02524710/2020</th>\n",
       "      <td>H3N2</td>\n",
       "      <td>EPI1818137</td>\n",
       "      <td>EPI1818138</td>\n",
       "      <td>6.720489</td>\n",
       "      <td>6.729258</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>\\rowcolor{Red3!20}A/swine/North_Carolina/A02479173/2020</th>\n",
       "      <td>H1N1</td>\n",
       "      <td>EPI1780425</td>\n",
       "      <td>EPI1780426</td>\n",
       "      <td>6.713621</td>\n",
       "      <td>6.721477</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>\\rowcolor{Green3!50}A/Camel/Inner_Mongolia/XL/2020</th>\n",
       "      <td>H7N9</td>\n",
       "      <td>EPI2026200</td>\n",
       "      <td>EPI2026202</td>\n",
       "      <td>6.699016</td>\n",
       "      <td>6.704867</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>\\rowcolor{Red3!20}A/swine/Tennessee/A02524414/2022</th>\n",
       "      <td>H1N1</td>\n",
       "      <td>EPI2149257</td>\n",
       "      <td>EPI2149258</td>\n",
       "      <td>6.650119</td>\n",
       "      <td>6.649429</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>\\rowcolor{Blue1!30}A/canine/Texas/21-011409-001/2021</th>\n",
       "      <td>H3N2</td>\n",
       "      <td>EPI1896555</td>\n",
       "      <td>EPI1896557</td>\n",
       "      <td>6.344011</td>\n",
       "      <td>6.300499</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>\\rowcolor{Blue1!30}A/swine/Iowa/A02525161/2021</th>\n",
       "      <td>H3N2</td>\n",
       "      <td>EPI1909023</td>\n",
       "      <td>EPI1909024</td>\n",
       "      <td>6.334205</td>\n",
       "      <td>6.289272</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>\\rowcolor{Red3!20}A/swine/Iowa/A02246996/2021</th>\n",
       "      <td>H1N1</td>\n",
       "      <td>EPI2146133</td>\n",
       "      <td>EPI2146134</td>\n",
       "      <td>6.326870</td>\n",
       "      <td>6.280888</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>\\rowcolor{Blue1!30}A/Muscovy_duck/Vietnam/QN6297/2020</th>\n",
       "      <td>H3N2</td>\n",
       "      <td>EPI1974815</td>\n",
       "      <td>EPI1974818</td>\n",
       "      <td>6.323945</td>\n",
       "      <td>6.277527</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>\\rowcolor{DarkOrange!40}A/mink/China/chick_embryo/2020</th>\n",
       "      <td>H9N2</td>\n",
       "      <td>EPI2161544</td>\n",
       "      <td>EPI2161548</td>\n",
       "      <td>6.318778</td>\n",
       "      <td>6.271620</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>74 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   subtype HA accession   \n",
       "strain                                                                    \n",
       "\\rowcolor{Red3!20}A/swine/Missouri/A02524711/2020     H1N1   EPI1818121  \\\n",
       "\\rowcolor{Blue1!30}A/swine/Indiana/A02524710/2020     H3N2   EPI1818137   \n",
       "\\rowcolor{Red3!20}A/swine/North_Carolina/A02479...    H1N1   EPI1780425   \n",
       "\\rowcolor{Green3!50}A/Camel/Inner_Mongolia/XL/2020    H7N9   EPI2026200   \n",
       "\\rowcolor{Red3!20}A/swine/Tennessee/A02524414/2022    H1N1   EPI2149257   \n",
       "...                                                    ...          ...   \n",
       "\\rowcolor{Blue1!30}A/canine/Texas/21-011409-001...    H3N2   EPI1896555   \n",
       "\\rowcolor{Blue1!30}A/swine/Iowa/A02525161/2021        H3N2   EPI1909023   \n",
       "\\rowcolor{Red3!20}A/swine/Iowa/A02246996/2021         H1N1   EPI2146133   \n",
       "\\rowcolor{Blue1!30}A/Muscovy_duck/Vietnam/QN629...    H3N2   EPI1974815   \n",
       "\\rowcolor{DarkOrange!40}A/mink/China/chick_embr...    H9N2   EPI2161544   \n",
       "\n",
       "                                                   NA accession   \n",
       "strain                                                            \n",
       "\\rowcolor{Red3!20}A/swine/Missouri/A02524711/2020    EPI1818122  \\\n",
       "\\rowcolor{Blue1!30}A/swine/Indiana/A02524710/2020    EPI1818138   \n",
       "\\rowcolor{Red3!20}A/swine/North_Carolina/A02479...   EPI1780426   \n",
       "\\rowcolor{Green3!50}A/Camel/Inner_Mongolia/XL/2020   EPI2026202   \n",
       "\\rowcolor{Red3!20}A/swine/Tennessee/A02524414/2022   EPI2149258   \n",
       "...                                                         ...   \n",
       "\\rowcolor{Blue1!30}A/canine/Texas/21-011409-001...   EPI1896557   \n",
       "\\rowcolor{Blue1!30}A/swine/Iowa/A02525161/2021       EPI1909024   \n",
       "\\rowcolor{Red3!20}A/swine/Iowa/A02246996/2021        EPI2146134   \n",
       "\\rowcolor{Blue1!30}A/Muscovy_duck/Vietnam/QN629...   EPI1974818   \n",
       "\\rowcolor{DarkOrange!40}A/mink/China/chick_embr...   EPI2161548   \n",
       "\n",
       "                                                    predicted IRAT impact   \n",
       "strain                                                                      \n",
       "\\rowcolor{Red3!20}A/swine/Missouri/A02524711/2020                6.767254  \\\n",
       "\\rowcolor{Blue1!30}A/swine/Indiana/A02524710/2020                6.720489   \n",
       "\\rowcolor{Red3!20}A/swine/North_Carolina/A02479...               6.713621   \n",
       "\\rowcolor{Green3!50}A/Camel/Inner_Mongolia/XL/2020               6.699016   \n",
       "\\rowcolor{Red3!20}A/swine/Tennessee/A02524414/2022               6.650119   \n",
       "...                                                                   ...   \n",
       "\\rowcolor{Blue1!30}A/canine/Texas/21-011409-001...               6.344011   \n",
       "\\rowcolor{Blue1!30}A/swine/Iowa/A02525161/2021                   6.334205   \n",
       "\\rowcolor{Red3!20}A/swine/Iowa/A02246996/2021                    6.326870   \n",
       "\\rowcolor{Blue1!30}A/Muscovy_duck/Vietnam/QN629...               6.323945   \n",
       "\\rowcolor{DarkOrange!40}A/mink/China/chick_embr...               6.318778   \n",
       "\n",
       "                                                    predicted IRAT emergence  \n",
       "strain                                                                        \n",
       "\\rowcolor{Red3!20}A/swine/Missouri/A02524711/2020                   6.782248  \n",
       "\\rowcolor{Blue1!30}A/swine/Indiana/A02524710/2020                   6.729258  \n",
       "\\rowcolor{Red3!20}A/swine/North_Carolina/A02479...                  6.721477  \n",
       "\\rowcolor{Green3!50}A/Camel/Inner_Mongolia/XL/2020                  6.704867  \n",
       "\\rowcolor{Red3!20}A/swine/Tennessee/A02524414/2022                  6.649429  \n",
       "...                                                                      ...  \n",
       "\\rowcolor{Blue1!30}A/canine/Texas/21-011409-001...                  6.300499  \n",
       "\\rowcolor{Blue1!30}A/swine/Iowa/A02525161/2021                      6.289272  \n",
       "\\rowcolor{Red3!20}A/swine/Iowa/A02246996/2021                       6.280888  \n",
       "\\rowcolor{Blue1!30}A/Muscovy_duck/Vietnam/QN629...                  6.277527  \n",
       "\\rowcolor{DarkOrange!40}A/mink/China/chick_embr...                  6.271620  \n",
       "\n",
       "[74 rows x 5 columns]"
      ]
     },
     "execution_count": 24,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "COLDICT={'H1N1':'Red3!20','H3N2':'Blue1!30','H7N9':'Green3!50','H9N2':'DarkOrange!40'}\n",
    "def rowcolor(row):\n",
    "    return '\\\\rowcolor{' + COLDICT[row.subtype]+'}' + row['strain']\n",
    "\n",
    "highrisknamesdf1 = highrisknamesdf.reset_index(drop=False)\n",
    "highrisknamesdf1['strain']=highrisknamesdf1.apply(rowcolor,axis=1)\n",
    "highrisknamesdf1=highrisknamesdf1.set_index('strain')\n",
    "highrisknamesdf1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 27,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\\begin{tabular}{L{1.95in}|L{.25in}|L{.60in}|L{.6in}|C{1in}|C{1in}}\\hline\r\n",
      "strain&subtype& HA  accession & NA  accession & predicted  IRAT  impact & predicted  IRAT  emergence \\\\\r\n",
      "\\rowcolor{Red3!20}A/swine/Missouri/A02524711/2020&H1N1&EPI1818121&EPI1818122&6.7673&6.7822\\\\\r\n",
      "\\rowcolor{Blue1!30}A/swine/Indiana/A02524710/2020&H3N2&EPI1818137&EPI1818138&6.7205&6.7293\\\\\r\n",
      "\\rowcolor{Red3!20}A/swine/North\\_Carolina/A02479173/2020&H1N1&EPI1780425&EPI1780426&6.7136&6.7215\\\\\r\n",
      "\\rowcolor{Green3!50}A/Camel/Inner\\_Mongolia/XL/2020&H7N9&EPI2026200&EPI2026202&6.6990&6.7049\\\\\r\n",
      "\\rowcolor{Red3!20}A/swine/Tennessee/A02524414/2022&H1N1&EPI2149257&EPI2149258&6.6501&6.6494\\\\\r\n",
      "\\rowcolor{Blue1!30}A/swine/Nebraska/A02245526/2020&H3N2&EPI1780195&EPI1780196&6.6058&6.5991\\\\\r\n",
      "\\rowcolor{Red3!20}A/swine/Minnesota/A02635976/2021&H1N1&EPI1912208&EPI1912209&6.5776&6.5670\\\\\r\n",
      "\\rowcolor{Blue1!30}A/swine/Chile/VN1401-5054/2020&H3N2&EPI1974975&EPI1974978&6.5318&6.5149\\\\\r\n",
      "\\rowcolor{Blue1!30}A/swine/Italy/56910/2020&H3N2&EPI2142217&EPI2142173&6.5292&6.5119\\\\\r\n",
      "\\rowcolor{Blue1!30}A/swine/Minnesota/A02245643/2020&H3N2&EPI1769178&EPI1769179&6.5067&6.4863\\\\\r\n",
      "\\rowcolor{Red3!20}A/swine/Iowa/A02479005/2020&H1N1&EPI1777621&EPI1777622&6.4872&6.4641\\\\\r\n",
      "\\rowcolor{Blue1!30}A/swine/Iowa/A02524878/2020&H3N2&EPI1907866&EPI1907867&6.4566&6.4291\\\\\r\n",
      "\\rowcolor{Red3!20}A/swine/Indiana/A02636638/2022&H1N1&EPI2153370&EPI2153371&6.4534&6.4255\\\\\r\n",
      "\\rowcolor{Blue1!30}A/swine/Iowa/A02524874/2020&H3N2&EPI1907838&EPI1907839&6.4392&6.4093\\\\\r\n",
      "\\rowcolor{Red3!20}A/swine/Minnesota/A02248037/2021&H1N1&EPI1912188&EPI1912189&6.4366&6.4063\\\\\r\n",
      "\\rowcolor{Red3!20}A/swine/Iowa/A02635917/2021&H1N1&EPI1911753&EPI1911754&6.4356&6.4052\\\\\r\n",
      "\\rowcolor{Red3!20}A/swine/Illinois/A02635936/2021&H1N1&EPI1911791&EPI1911792&6.4347&6.4042\\\\\r\n",
      "\\rowcolor{Red3!20}A/swine/Minnesota/A02711801/2022&H1N1&EPI2153420&EPI2153421&6.4334&6.4027\\\\\r\n",
      "\\rowcolor{Red3!20}A/swine/South\\_Dakota/A02524453/2020&H1N1&EPI1765555&EPI1765556&6.4321&6.4012\\\\\r\n",
      "\\rowcolor{Blue1!30}A/swine/Illinois/A02479007/2020&H3N2&EPI1777629&EPI1777630&6.4315&6.4005\\\\\r\n",
      "\\rowcolor{Red3!20}A/swine/Minnesota/A02248061/2021&H1N1&EPI1912494&EPI1912495&6.4278&6.3963\\\\\r\n",
      "\\rowcolor{Red3!20}A/swine/Iowa/A02524875/2020&H1N1&EPI1907858&EPI1907859&6.4260&6.3943\\\\\r\n",
      "\\rowcolor{Blue1!30}A/swine/Spain/44579-1/2020&H3N2&EPI1930744&EPI1930748&6.4255&6.3937\\\\\r\n",
      "\\rowcolor{Red3!20}A/swine/Iowa/A02636439/2022&H1N1&EPI2147475&EPI2147476&6.4250&6.3931\\\\\r\n",
      "\\rowcolor{Red3!20}A/swine/Minnesota/A02248060/2021&H1N1&EPI1912500&EPI1912501&6.4234&6.3913\\\\\r\n",
      "\\rowcolor{Red3!20}A/swine/Nebraska/A02636117/2021&H1N1&EPI1932937&EPI1932938&6.4226&6.3903\\\\\r\n",
      "\\rowcolor{Red3!20}A/swine/Iowa/A02524513/2020&H1N1&EPI1832647&EPI1832648&6.4223&6.3901\\\\\r\n",
      "\\rowcolor{Red3!20}A/swine/Iowa/A02524646/2020&H1N1&EPI1817164&EPI1817165&6.4222&6.3899\\\\\r\n",
      "\\rowcolor{Red3!20}A/swine/Nebraska/A02479337/2020&H1N1&EPI1769116&EPI1769117&6.4222&6.3899\\\\\r\n",
      "\\rowcolor{Red3!20}A/swine/Iowa/A02479383/2020&H1N1&EPI1771027&EPI1771028&6.4222&6.3899\\\\\r\n",
      "\\rowcolor{Red3!20}A/swine/Nebraska/A02479212/2020&H1N1&EPI1775884&EPI1775885&6.4222&6.3899\\\\\r\n",
      "\\rowcolor{Red3!20}A/swine/Minnesota/A02479051/2020&H1N1&EPI1778572&EPI1778573&6.4222&6.3899\\\\\r\n",
      "\\rowcolor{Red3!20}A/swine/Minnesota/A02245424/2020&H1N1&EPI1780207&EPI1780208&6.4222&6.3899\\\\\r\n",
      "\\rowcolor{Red3!20}A/swine/Iowa/A02524724/2020&H1N1&EPI1818387&EPI1818388&6.4222&6.3899\\\\\r\n",
      "\\rowcolor{Red3!20}A/swine/Missouri/A02525065/2021&H1N1&EPI1908581&EPI1908582&6.4222&6.3899\\\\\r\n",
      "\\rowcolor{Red3!20}A/swine/Iowa/A02525313/2021&H1N1&EPI1910761&EPI1910762&6.4222&6.3899\\\\\r\n",
      "\\rowcolor{Red3!20}A/swine/Iowa/A02524994/2020&H1N1&EPI1908427&EPI1908428&6.4222&6.3899\\\\\r\n",
      "\\rowcolor{Red3!20}A/swine/Iowa/A02525345/2021&H1N1&EPI1910807&EPI1910808&6.4222&6.3899\\\\\r\n",
      "\\rowcolor{Red3!20}A/swine/Iowa/A02524892/2020&H1N1&EPI1907881&EPI1907882&6.4222&6.3899\\\\\r\n",
      "\\rowcolor{Red3!20}A/swine/Nebraska/A02524935/2020&H1N1&EPI1908118&EPI1908119&6.4222&6.3899\\\\\r\n",
      "\\rowcolor{Red3!20}A/swine/Missouri/A02524951/2020&H1N1&EPI1908429&EPI1908430&6.4222&6.3899\\\\\r\n",
      "\\rowcolor{Red3!20}A/swine/Nebraska/A02479186/2020&H1N1&EPI1774141&EPI1774142&6.4220&6.3897\\\\\r\n",
      "\\rowcolor{Blue1!30}A/swine/Minnesota/A02245699/2020&H3N2&EPI1833007&EPI1833008&6.4220&6.3897\\\\\r\n",
      "\\rowcolor{Red3!20}A/swine/Iowa/A02479156/2020&H1N1&EPI1780249&EPI1780250&6.4216&6.3892\\\\\r\n",
      "\\rowcolor{Red3!20}A/swine/Iowa/A02479229/2020&H1N1&EPI1775914&EPI1775915&6.4215&6.3891\\\\\r\n",
      "\\rowcolor{Red3!20}A/swine/Iowa/A02479303/2020&H1N1&EPI1768639&EPI1768640&6.4207&6.3882\\\\\r\n",
      "\\rowcolor{Red3!20}A/swine/Minnesota/A02710691/2021&H1N1&EPI2146090&EPI2146091&6.4200&6.3874\\\\\r\n",
      "\\rowcolor{Red3!20}A/swine/Iowa/A02635881/2021&H1N1&EPI1911668&EPI1911669&6.4198&6.3872\\\\\r\n",
      "\\rowcolor{Red3!20}A/swine/Iowa/A02525354/2021&H1N1&EPI1910789&EPI1910790&6.4198&6.3872\\\\\r\n",
      "\\rowcolor{Red3!20}A/swine/Iowa/A02635823/2021&H1N1&EPI1911263&EPI1911264&6.4176&6.3847\\\\\r\n",
      "\\rowcolor{Red3!20}A/swine/Iowa/A02524968/2020&H1N1&EPI1908419&EPI1908420&6.4173&6.3843\\\\\r\n",
      "\\rowcolor{Red3!20}A/swine/Iowa/A02479141/2020&H1N1&EPI1780241&EPI1780242&6.4161&6.3829\\\\\r\n",
      "\\rowcolor{Red3!20}A/swine/Nebraska/A02525367/2021&H1N1&EPI1910817&EPI1910818&6.4136&6.3801\\\\\r\n",
      "\\rowcolor{Red3!20}A/swine/Iowa/A02245587/2020&H1N1&EPI1775817&EPI1775818&6.4119&6.3781\\\\\r\n",
      "\\rowcolor{Red3!20}A/swine/Iowa/A02750621/2022&H1N1&EPI2161576&EPI2161577&6.4116&6.3779\\\\\r\n",
      "\\rowcolor{Red3!20}A/swine/Iowa/A02636496/2022&H1N1&EPI2148086&EPI2148087&6.4105&6.3765\\\\\r\n",
      "\\rowcolor{Blue1!30}A/swine/South\\_Dakota/A02524914/2020&H3N2&EPI1908070&EPI1908071&6.4101&6.3761\\\\\r\n",
      "\\rowcolor{Red3!20}A/swine/Iowa/A02525217/2021&H1N1&EPI1909087&EPI1909088&6.4090&6.3748\\\\\r\n",
      "\\rowcolor{Red3!20}A/swine/Iowa/A02636145/2021&H1N1&EPI1932055&EPI1932930&6.4090&6.3748\\\\\r\n",
      "\\rowcolor{Red3!20}A/swine/Iowa/A02636114/2021&H1N1&EPI1931853&EPI1931854&6.4090&6.3748\\\\\r\n",
      "\\rowcolor{Red3!20}A/swine/Iowa/A02635871/2021&H1N1&EPI1911656&EPI1911657&6.4089&6.3747\\\\\r\n",
      "\\rowcolor{Red3!20}A/swine/Iowa/A02479067/2020&H1N1&EPI1778734&EPI1778735&6.4083&6.3741\\\\\r\n",
      "\\rowcolor{Red3!20}A/swine/Minnesota/A02246459/2021&H1N1&EPI1912518&EPI1912519&6.4061&6.3716\\\\\r\n",
      "\\rowcolor{Blue1!30}A/swine/Minnesota/A02711797/2022&H3N2&EPI2153382&EPI2153383&6.4029&6.3679\\\\\r\n",
      "\\rowcolor{Red3!20}A/swine/Minnesota/A02525348/2021&H1N1&EPI1910795&EPI1910796&6.3996&6.3641\\\\\r\n",
      "\\rowcolor{Red3!20}A/swine/Iowa/A02479343/2020&H1N1&EPI1769114&EPI1769115&6.3985&6.3628\\\\\r\n",
      "\\rowcolor{Blue1!30}A/swine/Illinois/A02525253/2021&H3N2&EPI1910375&EPI1910376&6.3974&6.3615\\\\\r\n",
      "\\rowcolor{Blue1!30}A/swine/Colorado/A02710706/2022&H3N2&EPI2176699&EPI2176700&6.3629&6.3221\\\\\r\n",
      "\\rowcolor{Red3!20}A/swine/Kansas/A02245381/2020(H1N1)&H1N1&EPI1777723&EPI1777724&6.3447&6.3013\\\\\r\n",
      "\\rowcolor{Blue1!30}A/canine/Texas/21-011409-001/2021&H3N2&EPI1896555&EPI1896557&6.3440&6.3005\\\\\r\n",
      "\\rowcolor{Blue1!30}A/swine/Iowa/A02525161/2021&H3N2&EPI1909023&EPI1909024&6.3342&6.2893\\\\\r\n",
      "\\rowcolor{Red3!20}A/swine/Iowa/A02246996/2021&H1N1&EPI2146133&EPI2146134&6.3269&6.2809\\\\\r\n",
      "\\rowcolor{Blue1!30}A/Muscovy\\_duck/Vietnam/QN6297/2020&H3N2&EPI1974815&EPI1974818&6.3239&6.2775\\\\\r\n",
      "\\rowcolor{DarkOrange!40}A/mink/China/chick\\_embryo/2020&H9N2&EPI2161544&EPI2161548&6.3188&6.2716\\\\\r\n",
      "\\hline\\end{tabular}\r\n"
     ]
    }
   ],
   "source": [
    "from zedstat.textable import textable\n",
    "textable(highrisknamesdf1,#pd.concat([highrisknamesdf1.head(30),highrisknamesdf1.tail(10)]),\n",
    "         tabname='../../../../tex/overleaf/Figures/tabdata/highrisk35_upd.tex',\n",
    "         FORMAT='%1.4f',INDEX=True,\n",
    "         TABFORMAT='L{1.95in}|L{.25in}|L{.60in}|L{.6in}|C{1in}|C{1in}',LNTERM='\\\\\\\\\\n')\n",
    "! cat ../../../../tex/overleaf/Figures/tabdata/highrisk35_upd.tex"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 28,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>H1N1</th>\n",
       "      <th>H3N2</th>\n",
       "      <th>H7N9</th>\n",
       "      <th>H9N2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A/swine/Missouri/A02524711/2020</td>\n",
       "      <td>A/swine/Indiana/A02524710/2020</td>\n",
       "      <td>A/Camel/Inner_Mongolia/XL/2020</td>\n",
       "      <td>A/mink/China/chick_embryo/2020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>6.782248</td>\n",
       "      <td>6.729258</td>\n",
       "      <td>6.704867</td>\n",
       "      <td>6.27162</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                              H1N1                            H3N2   \n",
       "0  A/swine/Missouri/A02524711/2020  A/swine/Indiana/A02524710/2020  \\\n",
       "1                         6.782248                        6.729258   \n",
       "\n",
       "                             H7N9                            H9N2  \n",
       "0  A/Camel/Inner_Mongolia/XL/2020  A/mink/China/chick_embryo/2020  \n",
       "1                        6.704867                         6.27162  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>6.0</th>\n",
       "      <th>6.2</th>\n",
       "      <th>6.3</th>\n",
       "      <th>6.4</th>\n",
       "      <th>6.5</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>H1N1</th>\n",
       "      <td>64</td>\n",
       "      <td>57</td>\n",
       "      <td>54</td>\n",
       "      <td>11</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>H3N2</th>\n",
       "      <td>17</td>\n",
       "      <td>17</td>\n",
       "      <td>15</td>\n",
       "      <td>8</td>\n",
       "      <td>4</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>H7N9</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>H9N2</th>\n",
       "      <td>2</td>\n",
       "      <td>2</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      6.0  6.2  6.3  6.4  6.5\n",
       "H1N1   64   57   54   11    4\n",
       "H3N2   17   17   15    8    4\n",
       "H7N9    1    1    1    1    1\n",
       "H9N2    2    2    0    0    0"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "84\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>subtype</th>\n",
       "      <th>ha_accession</th>\n",
       "      <th>na_accession</th>\n",
       "      <th>ha</th>\n",
       "      <th>na</th>\n",
       "      <th>ha_risk</th>\n",
       "      <th>na_risk</th>\n",
       "      <th>geometric_mean_risk</th>\n",
       "      <th>emergence_risk</th>\n",
       "      <th>emergence_risk_var</th>\n",
       "      <th>impact_risk</th>\n",
       "      <th>impact_risk_var</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>218</th>\n",
       "      <td>A/swine/Missouri/A02524711/2020</td>\n",
       "      <td>H1N1</td>\n",
       "      <td>EPI1818121</td>\n",
       "      <td>EPI1818122</td>\n",
       "      <td>MKAILVVMLYTFTTANADTLCIGYHANNSTDTVDTVLEKNVTVTHS...</td>\n",
       "      <td>MNPNQKIITIGSICMTIGTANLILQIGNIISIWVSHSIQIGNQSQI...</td>\n",
       "      <td>[0.007702821053332326, 0.00794167832708008, 0....</td>\n",
       "      <td>[0.0016000691389129699, 0.0019043683598783635,...</td>\n",
       "      <td>[0.0035107045233124007, 0.003888943420830784, ...</td>\n",
       "      <td>6.782248</td>\n",
       "      <td>0.000006</td>\n",
       "      <td>6.767254</td>\n",
       "      <td>0.000005</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1902</th>\n",
       "      <td>A/swine/Indiana/A02524710/2020</td>\n",
       "      <td>H3N2</td>\n",
       "      <td>EPI1818137</td>\n",
       "      <td>EPI1818138</td>\n",
       "      <td>MKTIIALSYILCLVFAQKIPGNDNGTATLCLGHHAVPNGTIVKTIT...</td>\n",
       "      <td>MNPNQKIITIGSVSLTISTICFFMQIAILITTIALHFKQYEFNSPP...</td>\n",
       "      <td>[0.008423100311281287, 0.0044835948774065975, ...</td>\n",
       "      <td>[0.006596476585994605, 0.006619358238135353, 0...</td>\n",
       "      <td>[0.00745404480700585, 0.00544779961894918, 0.0...</td>\n",
       "      <td>6.729258</td>\n",
       "      <td>0.000613</td>\n",
       "      <td>6.720489</td>\n",
       "      <td>0.000476</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5814</th>\n",
       "      <td>A/Camel/Inner_Mongolia/XL/2020</td>\n",
       "      <td>H7N9</td>\n",
       "      <td>EPI2026200</td>\n",
       "      <td>EPI2026202</td>\n",
       "      <td>MNTQILVFALIAIIPTNADKICLGHHAVSNGTKVNTLTEREVEVVN...</td>\n",
       "      <td>MNPNQKILCTSATAITIGAIAVLIGIANLGLNIGLHLKPGCNCSHS...</td>\n",
       "      <td>[0.025061424930009856, 0.01711340116551936, 0....</td>\n",
       "      <td>[0.0011227960579890436, 0.0012230115569459082,...</td>\n",
       "      <td>[0.0053046082908169015, 0.004574919387714034, ...</td>\n",
       "      <td>6.704867</td>\n",
       "      <td>0.004832</td>\n",
       "      <td>6.699016</td>\n",
       "      <td>0.003755</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6049</th>\n",
       "      <td>A/mink/China/chick_embryo/2020</td>\n",
       "      <td>H9N2</td>\n",
       "      <td>EPI2161544</td>\n",
       "      <td>EPI2161548</td>\n",
       "      <td>METVSLITILLAATVSNADKICIGYQSSNSTETVDTLTENNVPVTH...</td>\n",
       "      <td>MNPNQKITAIGSVSLIIAIICLLMQIAILTMTMTLHFGQKECSNPS...</td>\n",
       "      <td>[0.016907786181883842, 0.016907786181883842, 0...</td>\n",
       "      <td>[0.04614861940588559, 0.04614861940588559, 0.0...</td>\n",
       "      <td>[0.027933331156592276, 0.027933331156592276, 0...</td>\n",
       "      <td>6.271620</td>\n",
       "      <td>0.000000</td>\n",
       "      <td>6.318778</td>\n",
       "      <td>0.000000</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   id subtype ha_accession na_accession   \n",
       "218   A/swine/Missouri/A02524711/2020    H1N1   EPI1818121   EPI1818122  \\\n",
       "1902   A/swine/Indiana/A02524710/2020    H3N2   EPI1818137   EPI1818138   \n",
       "5814   A/Camel/Inner_Mongolia/XL/2020    H7N9   EPI2026200   EPI2026202   \n",
       "6049   A/mink/China/chick_embryo/2020    H9N2   EPI2161544   EPI2161548   \n",
       "\n",
       "                                                     ha   \n",
       "218   MKAILVVMLYTFTTANADTLCIGYHANNSTDTVDTVLEKNVTVTHS...  \\\n",
       "1902  MKTIIALSYILCLVFAQKIPGNDNGTATLCLGHHAVPNGTIVKTIT...   \n",
       "5814  MNTQILVFALIAIIPTNADKICLGHHAVSNGTKVNTLTEREVEVVN...   \n",
       "6049  METVSLITILLAATVSNADKICIGYQSSNSTETVDTLTENNVPVTH...   \n",
       "\n",
       "                                                     na   \n",
       "218   MNPNQKIITIGSICMTIGTANLILQIGNIISIWVSHSIQIGNQSQI...  \\\n",
       "1902  MNPNQKIITIGSVSLTISTICFFMQIAILITTIALHFKQYEFNSPP...   \n",
       "5814  MNPNQKILCTSATAITIGAIAVLIGIANLGLNIGLHLKPGCNCSHS...   \n",
       "6049  MNPNQKITAIGSVSLIIAIICLLMQIAILTMTMTLHFGQKECSNPS...   \n",
       "\n",
       "                                                ha_risk   \n",
       "218   [0.007702821053332326, 0.00794167832708008, 0....  \\\n",
       "1902  [0.008423100311281287, 0.0044835948774065975, ...   \n",
       "5814  [0.025061424930009856, 0.01711340116551936, 0....   \n",
       "6049  [0.016907786181883842, 0.016907786181883842, 0...   \n",
       "\n",
       "                                                na_risk   \n",
       "218   [0.0016000691389129699, 0.0019043683598783635,...  \\\n",
       "1902  [0.006596476585994605, 0.006619358238135353, 0...   \n",
       "5814  [0.0011227960579890436, 0.0012230115569459082,...   \n",
       "6049  [0.04614861940588559, 0.04614861940588559, 0.0...   \n",
       "\n",
       "                                    geometric_mean_risk  emergence_risk   \n",
       "218   [0.0035107045233124007, 0.003888943420830784, ...        6.782248  \\\n",
       "1902  [0.00745404480700585, 0.00544779961894918, 0.0...        6.729258   \n",
       "5814  [0.0053046082908169015, 0.004574919387714034, ...        6.704867   \n",
       "6049  [0.027933331156592276, 0.027933331156592276, 0...        6.271620   \n",
       "\n",
       "      emergence_risk_var  impact_risk  impact_risk_var  \n",
       "218             0.000006     6.767254         0.000005  \n",
       "1902            0.000613     6.720489         0.000476  \n",
       "5814            0.004832     6.699016         0.003755  \n",
       "6049            0.000000     6.318778         0.000000  "
      ]
     },
     "execution_count": 28,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "count=0\n",
    "Subtype={'H1N1':0,'H3N2':0,'H7N9':0,'H9N2':0}\n",
    "MaxriskStrain={'H1N1':None,'H3N2':None,'H7N9':None,'H9N2':None}\n",
    "Subtype_strat={ 6.0:{'H1N1':0,'H3N2':0,'H7N9':0,'H9N2':0},  6.2:{'H1N1':0,'H3N2':0,'H7N9':0,'H9N2':0},              6.3:{'H1N1':0,'H3N2':0,'H7N9':0,'H9N2':0},\n",
    "               6.4:{'H1N1':0,'H3N2':0,'H7N9':0,'H9N2':0},6.5:{'H1N1':0,'H3N2':0,'H7N9':0,'H9N2':0}}\n",
    "for node in ltree_collapsed:\n",
    "    Subtype[node.subtype]=Subtype[node.subtype]+1\n",
    "    if MaxriskStrain[node.subtype] is None:\n",
    "        MaxriskStrain[node.subtype]=(ALLinfoHA.seq_infos[node.name.split()[0]].name,\n",
    "                                     ALLinfoHA.seq_infos[node.name.split()[0]].erisk)\n",
    "    else:\n",
    "        if MaxriskStrain[node.subtype][1]<ALLinfoHA.seq_infos[node.name.split()[0]].erisk:\n",
    "            MaxriskStrain[node.subtype]=(ALLinfoHA.seq_infos[node.name.split()[0]].name,\n",
    "                                         ALLinfoHA.seq_infos[node.name.split()[0]].erisk)\n",
    "    for r in [6.0, 6.2,6.3,6.4,6.5]:\n",
    "        if r<ALLinfoHA.seq_infos[node.name.split()[0]].erisk:\n",
    "            #print(node.subtype,ALLinfoHA.seq_infos[node.name.split()[0]].erisk)\n",
    "            Subtype_strat[r][node.subtype]=Subtype_strat[r][node.subtype]+1\n",
    "        \n",
    "    count=count+1\n",
    "maxriskdf = pd.DataFrame(MaxriskStrain)\n",
    "Subtype_strat_df = pd.DataFrame(Subtype_strat)\n",
    "display(maxriskdf)\n",
    "display(Subtype_strat_df)\n",
    "print(count)\n",
    "df[df.id.isin(maxriskdf.iloc[0,:].values)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 29,
   "metadata": {},
   "outputs": [],
   "source": [
    "dcf=100*(Subtype_strat_df/Subtype_strat_df.sum(axis=0))"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 30,
   "metadata": {},
   "outputs": [],
   "source": [
    "df_=Subtype_strat_df\n",
    "dcfs=' ('+dcf.round(2).astype(str)+'\\%)'\n",
    "df__=df_.astype(str)+dcfs"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 31,
   "metadata": {},
   "outputs": [],
   "source": [
    "df__.columns=df__.columns.astype(str)\n",
    "df__.index.name='subtype'"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 32,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\\begin{tabular}{L{.4in}|L{.6in}|L{.6in}|L{.6in}|L{.6in}|L{.6in}}\\hline\r\n",
      "subtype&6.0&6.2&6.3&6.4&6.5\\\\\r\n",
      "H1N1& 64  (76.19\\%) & 57  (74.03\\%) & 54  (77.14\\%) & 11  (55.0\\%) & 4  (44.44\\%) \\\\\r\n",
      "H3N2& 17  (20.24\\%) & 17  (22.08\\%) & 15  (21.43\\%) & 8  (40.0\\%) & 4  (44.44\\%) \\\\\r\n",
      "H7N9& 1  (1.19\\%) & 1  (1.3\\%) & 1  (1.43\\%) & 1  (5.0\\%) & 1  (11.11\\%) \\\\\r\n",
      "H9N2& 2  (2.38\\%) & 2  (2.6\\%) & 0  (0.0\\%) & 0  (0.0\\%) & 0  (0.0\\%) \\\\\r\n",
      "\\hline\\end{tabular}\r\n"
     ]
    }
   ],
   "source": [
    "from zedstat.textable import textable\n",
    "#Subtype_strat_df.columns=[str(x) for x in Subtype_strat_df.columns]\n",
    "#Subtype_strat_df.index.name='subtype'\n",
    "#Subtype_strat_df=Subtype_strat_df.reset_index()\n",
    "textable(df__,#pd.concat([highrisknamesdf1.head(30),highrisknamesdf1.tail(10)]),\n",
    "         tabname='../../../../tex/overleaf/Figures/tabdata/riskycount_upd.tex',\n",
    "         FORMAT='%s',INDEX=True,\n",
    "         TABFORMAT='L{.4in}|L{.6in}|L{.6in}|L{.6in}|L{.6in}|L{.6in}',LNTERM='\\\\\\\\\\n')\n",
    "! cat ../../../../tex/overleaf/Figures/tabdata/riskycount_upd.tex"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [],
   "source": [
    "def write_fasta(seqs, fasta_file, wrap=80):\n",
    "    \"\"\"Write sequences to a fasta file.\n",
    "\n",
    "    Parameters\n",
    "    ----------\n",
    "    seqs : dict[seq_id] -> seq\n",
    "        Sequences indexed by sequence id.\n",
    "    fasta_file : str\n",
    "        Path to write the sequences to.\n",
    "    wrap: int\n",
    "        Number of AA/NT before the line is wrapped.\n",
    "    \"\"\"\n",
    "    with open(fasta_file, 'w') as f:\n",
    "        for gid, gseq in seqs.items():\n",
    "            f.write('>{}\\n'.format(gid))\n",
    "            for i in range(0, len(gseq), wrap):\n",
    "                f.write('{}\\n'.format(gseq[i:i + wrap])) \n",
    "\n",
    "#for i,name in zip(df[df.id.isin(maxriskdf.iloc[0,:].values)].ha.values,\n",
    "#                 df[df.id.isin(maxriskdf.iloc[0,:].values)].ha_accession.values):\n",
    "#    write_fasta({name:i},name+'.fasta')\n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 45,
   "metadata": {},
   "outputs": [],
   "source": [
    "id_list=['A/swine/North_Carolina/A02479173/2020','A/swine/Tennessee/A02524414/2022']\n",
    "RFILES={name+'('+subtype+')':i for i,name,subtype in zip(df[df.id.isin(id_list)].ha.values,\n",
    "                 df[df.id.isin(id_list)].ha_accession.values,\n",
    "                 df[df.id.isin(id_list)].subtype.values)}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 34,
   "metadata": {},
   "outputs": [],
   "source": [
    "FASTANAME='../../../../tex/overleaf/Figures/plotdata/risky4.fasta'\n",
    "FASTAFILES={name+'('+subtype+')':i for i,name,subtype in zip(df[df.id.isin(maxriskdf.iloc[0,:].values)].ha.values,\n",
    "                 df[df.id.isin(maxriskdf.iloc[0,:].values)].ha_accession.values,\n",
    "                 df[df.id.isin(maxriskdf.iloc[0,:].values)].subtype.values)}"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 42,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>date</th>\n",
       "      <th>sequence</th>\n",
       "      <th>acc_na</th>\n",
       "      <th>sequence_na</th>\n",
       "      <th>cluster_size</th>\n",
       "      <th>subtype</th>\n",
       "      <th>protein</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>acc</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>EPI1877316</th>\n",
       "      <td>A/Togo/0172/2021</td>\n",
       "      <td>2021-02-18</td>\n",
       "      <td>MKAILVVLLYTFTTANADTLCIGYHANNSTDTVDTVLEKNVTVTHS...</td>\n",
       "      <td>EPI1877315</td>\n",
       "      <td>MNPNQKIITIGSICMAIGTANLILQIGNIISIWVSHSIQIGNQSQI...</td>\n",
       "      <td>531</td>\n",
       "      <td>H1N1</td>\n",
       "      <td>HA</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                        name        date   \n",
       "acc                                        \n",
       "EPI1877316  A/Togo/0172/2021  2021-02-18  \\\n",
       "\n",
       "                                                     sequence      acc_na   \n",
       "acc                                                                         \n",
       "EPI1877316  MKAILVVLLYTFTTANADTLCIGYHANNSTDTVDTVLEKNVTVTHS...  EPI1877315  \\\n",
       "\n",
       "                                                  sequence_na  cluster_size   \n",
       "acc                                                                           \n",
       "EPI1877316  MNPNQKIITIGSICMAIGTANLILQIGNIISIWVSHSIQIGNQSQI...           531  \\\n",
       "\n",
       "           subtype protein  \n",
       "acc                         \n",
       "EPI1877316    H1N1      HA  "
      ]
     },
     "execution_count": 42,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "human_dom_seq='../../../qnet_predictions/results/dominant_sequences_2021_2022.csv'\n",
    "human_dom_seq='../../../../paper_data_v2/enet_predictions/dominant_sequences/results/north_h1n1_ha/north_h1n1_ha_21_22/dom_seqs.csv'\n",
    "domf=pd.read_csv(human_dom_seq,index_col=0)\n",
    "#domf=domf.assign(subtype=[x.split('_')[1].upper() for x in domf.index.values],protein=[x.split('_')[2].upper() for x in domf.index.values])\n",
    "domf=domf.assign(subtype='H1N1',protein='HA').head(1)\n",
    "domf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 37,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>name</th>\n",
       "      <th>sequence</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>strain</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>north_h1n1_ha</th>\n",
       "      <td>A/Ireland/20935/2022</td>\n",
       "      <td>MKAILVVLLYTFTTANADTLCIGYHANNSTDTVDTVLEKNVTVTHS...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>north_h1n1_na</th>\n",
       "      <td>A/Cote_d'Ivoire/3729/2021</td>\n",
       "      <td>MNPNQKIITIGSICMAIGTANLILQIGNIISIWVSHSIQIGNQSQI...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>north_h3n2_ha</th>\n",
       "      <td>A/Human/New_York/PV60641/2022</td>\n",
       "      <td>MKTIIALSNILCLVFAQKIPGNDNSTATLCLGHHAVPNGTIVKTIT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>north_h3n2_na</th>\n",
       "      <td>A/Stockholm/10/2022</td>\n",
       "      <td>MNPNQKIITIGSVSLTISTICFFMQIAILITTVTLHFKQYEFNSPP...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>south_h1n1_ha</th>\n",
       "      <td>A/Abidjan/457/2021</td>\n",
       "      <td>MKAILVVLLYTFTTANADTLCIGYHANNSTDTVDTVLEKNVTVTHS...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>south_h1n1_na</th>\n",
       "      <td>A/Cote_D'Ivoire/1496/2021</td>\n",
       "      <td>MNPNQKIITIGSICMAIGTANLILQIGNIISIWVSHSIQIGNQSQI...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>south_h3n2_ha</th>\n",
       "      <td>A/Darwin/9a/2021</td>\n",
       "      <td>MKTIIALSNILCLVFAQKIPGNDNSTATLCLGHHAVPNGTIVKTIT...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>south_h3n2_na</th>\n",
       "      <td>A/India/PUN-NIV301718/2021</td>\n",
       "      <td>MNPNQKIITIGSVSLTISTICFFMQIAILITTVTLHFKQYEFNSPP...</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                        name   \n",
       "strain                                         \n",
       "north_h1n1_ha           A/Ireland/20935/2022  \\\n",
       "north_h1n1_na      A/Cote_d'Ivoire/3729/2021   \n",
       "north_h3n2_ha  A/Human/New_York/PV60641/2022   \n",
       "north_h3n2_na            A/Stockholm/10/2022   \n",
       "south_h1n1_ha             A/Abidjan/457/2021   \n",
       "south_h1n1_na      A/Cote_D'Ivoire/1496/2021   \n",
       "south_h3n2_ha               A/Darwin/9a/2021   \n",
       "south_h3n2_na     A/India/PUN-NIV301718/2021   \n",
       "\n",
       "                                                        sequence  \n",
       "strain                                                            \n",
       "north_h1n1_ha  MKAILVVLLYTFTTANADTLCIGYHANNSTDTVDTVLEKNVTVTHS...  \n",
       "north_h1n1_na  MNPNQKIITIGSICMAIGTANLILQIGNIISIWVSHSIQIGNQSQI...  \n",
       "north_h3n2_ha  MKTIIALSNILCLVFAQKIPGNDNSTATLCLGHHAVPNGTIVKTIT...  \n",
       "north_h3n2_na  MNPNQKIITIGSVSLTISTICFFMQIAILITTVTLHFKQYEFNSPP...  \n",
       "south_h1n1_ha  MKAILVVLLYTFTTANADTLCIGYHANNSTDTVDTVLEKNVTVTHS...  \n",
       "south_h1n1_na  MNPNQKIITIGSICMAIGTANLILQIGNIISIWVSHSIQIGNQSQI...  \n",
       "south_h3n2_ha  MKTIIALSNILCLVFAQKIPGNDNSTATLCLGHHAVPNGTIVKTIT...  \n",
       "south_h3n2_na  MNPNQKIITIGSVSLTISTICFFMQIAILITTVTLHFKQYEFNSPP...  "
      ]
     },
     "execution_count": 37,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "domf"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 107,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'DOM_HUMAN(H1N1)': 'MKAILVVLLYTFTTANADTLCIGYHANNSTDTVDTVLEKNVTVTHSVNLLEDKHNGKLCKLRGVAPLHLGKCNIAGWILGNPECESLSTARSWSYIVETSNSDNGTCYPGDFINYEELREQLSSVSSFERFEIFPKTSSWPNHDSDKGVTAACPHAGAKSFYKNLIWLVKKGNSYPKLNQTYINDKGKEVLVLWGIHHPPTIAAQESLYQNADAYVFVGTSRYSKKFKPEIATRPKVRDQEGRMNYYWTLVEPGDKITFEATGNLVVPRYAFTMERDAGSGIIISDTPVHDCNTTCQTPEGAINTSLPFQNVHPITIGKCPKYVKSTKLRLATGLRNVPSIQSRGLFGAIAGFIEGGWTGMVDGWYGYHHQNEQGSGYAADLKSTQNAIDKITNKVNSVIEKMNTQFTAVGKEFNHLEKRIENLNKKVDDGFLDIWTYNAELLVLLENERTLDYHDSNVKNLYEKVRNQLKNNAKEIGNGCFEFYHKCDNTCMESVKNGTYDYPKYSEEAKLNREKIDGVKLESTRIYQILAIYSTVASSLVLVVSLGAISFWMCSNGSLQCRIC',\n",
       " 'EPI1818121(H1N1)': 'MKAILVVMLYTFTTANADTLCIGYHANNSTDTVDTVLEKNVTVTHSVNLLEDKHNGKLCKLRGVAPLHLGKCNIAGWILGNPECESLSTARSWSYIVETSNSDNGTCYPGDFINYEELREQLSSVSSFERFEIFPKTSSWPNHDSDNGVTAACPHAGAKSFYKNLIWLVKKGKSYPKINQTYINDKGKEVLVLWGIHHPPTIADQQSLYQNADAYVFVGTSRYSKKFKPEIATRPKVRDQEGRMNYYWTLVEPGDKITFEATGNLVAPRYAFTMERDAGSGIIISDTPVHDCNTTCQTPEGAINTSLPFQNVHPITIGKCPKYVKSTKLRLATGLRNVPSIQSRGLFGAIAGFIEGGWTGMVDGWYGYHHQNEQGSGYAADLKSTQNAIDKITNKVNSVIEKMNTQFTAVGKEFNHLEKRIENLNKKVDDGFLDIWTYNAELLVLLENERTLDYHDSNVKNLYEKVRNQLKNNAKEIGNGCFEFYHKCDNTCMESVKNGTYDYPKYSEEAKLNREKIDGVKLDSTRIYQILAIYSTVASSLVLVVSLGAI',\n",
       " 'EPI1818137(H3N2)': 'MKTIIALSYILCLVFAQKIPGNDNGTATLCLGHHAVPNGTIVKTITDDRIEVTNATELVQNSSIGEICDSPHQVLDGGNCTLIDALLGDPQCDGFQNKKWDLFVERSRAYSNCYPYDVPDYASLRSLVASSGTLEFKNESFNWTGVTQNGKSSSCIRGSSSSFFSRLNWLTXLNYTYPALNVIMPNKEQFDKLYIWGVHHPGTDNDQIFLYARSSGRIIVSTKRSQQAVIPNIGSRPRIRDIPSRISIYWTIVKPGDILLINSTGNLIAPRGYFKIRSGKSSIMRSDAPIGKCKSECITPNGSIPNDKPFQNVNRITYGACPRYIKQSTLKLATGMRNVPEKQTRGIFGAIAGFIENGWEGMVDGWYGFRHQNSEGRGQAADLKSTQAAIDQINGKLNRLIGKTNEKFHQIEKEFSEVEGRVQDLEKYVEDTKIDLWSYNAELLVALENQHTIDLTDSEMNKLFEKTKKQLRENAEDMGNGCFKIYHKCDNACIGSIRNETYDHNVYRDEALNNRFQIKGVELKSGYKDWILWISFAISCFLLCVALLGF',\n",
       " 'EPI2026200(H7N9)': 'MNTQILVFALIAIIPTNADKICLGHHAVSNGTKVNTLTEREVEVVNATETVERTNIPRICSKGKRTVDLGQCGLLGTITGPPQCDQFLEFSADLIIERREGSDVCYPGKFVNEEALRQILRESGGIDKEAMGFTYNGIRTNGVTSACRRSGSSFYAEMKWLLSNTDNAAFPQMTKSYKNTRKSPAIIVWGIHHSVSTAEQTKLYGSGNKLVTVGSSNYQQSFVPSPGARPQVNGISGRIDFHWLMLNPNDTVTFSFNGAFIAPDRASFLRGKSMGIQSGVQVDANCEGDCYHSGGTIISNLPFQNIDSRAVGKCPRYVKQRSLLLATGMKNVPEIPKGRGLFGAIAGFIENGWEGLIDGWYGFRHQNAQGEGTAADYKSTQSAIDQITGKLNRLIAKTNQQFELIDNEFNEVEKQIGNVINWTRDSITEVWSYNAELLVAMENQHTIDLADSEMDKLYERVKRQLRENAEEDGTGCFEIFHKCDDDCMASIRNNTYDHRKYREEAMQSRIQIDPVKLSSGYKDVILWFSFGASCFILLAIVMGLVFICVK',\n",
       " 'EPI2161544(H9N2)': 'METVSLITILLAATVSNADKICIGYQSSNSTETVDTLTENNVPVTHAKELLHTEHNGMLCATGLGQPLVLDTCTIEGLIYGNPSCDLSLEGREWSYIVERPSAVHGLCYPGNVEDLEELRSLFSSARSYQRIQIFPDTIWNVSYDGTSTACSGSFYKSMRWLTRKNGEYPTQDAQYTNNQGKNILFMWGINHPPTDDTQRGLYTRTDTTTSVATEEINRIFKPLIGPRPLVNGLMGRINYYWSVLKPGQTLRIKSDGNLIAPWYGHILSGESHGRILKTDLKRGNCTVQCQTEKGGLNTTLPFQNVSKYAFGNCSKYIGVKSLKLAVGLRNVPSRSSRGLFGAIAGFIEGGWPGLVAGWYGFQHSNDQGVGMAADRDSTQKAIDKITSKVNNIVDKMNKQYEIIDHEFSEVETRLNMINNKIDDQIQDIWAYNAELLVLLENQKTLDEHDANVNNLYNKVKRALGSNAAEDGKGCFELYHKCDDQCMETIRNGTYNRRKYQEESKLERQRIEGVKLESEGTYKILTIYSTVASSLVIAMGFAAFLFWAMS'}"
      ]
     },
     "execution_count": 107,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "human_dom_seq='../../../../paper_data/enet_predictions/results/dominant_sequences/north_h1n1_ha/north_h1n1_ha_21_22.csv'\n",
    "domf=pd.read_csv(human_dom_seq,index_col=0)\n",
    "domf=domf.assign(subtype='H1N1',protein='HA')\n",
    "domf=domf[domf.protein=='HA'].head(2)\n",
    "DOMFILES={'DOM_HUMAN'+'('+subtype+')':i for i,name,subtype in zip(domf.sequence.values,\n",
    "                 domf.id.values,\n",
    "                 domf.subtype.values)}\n",
    "DOMFILES.update(FASTAFILES)\n",
    "DOMFILES"
   ]
  },
  {
   "cell_type": "raw",
   "metadata": {},
   "source": [
    "human_dom_seq='../../../../paper_data/enet_predictions/results/dominant_sequences/north_h1n1_ha/north_h1n1_ha_21_22.csv'\n",
    "#human_dom_seq='../../../qnet_predictions/results/dominant_sequences_2021_2022.csv'\n",
    "domf=pd.read_csv(human_dom_seq,index_col=0)\n",
    "domf=domf.assign(subtype=[x.split('_')[1].upper() for x in domf.index.values],protein=[x.split('_')[2].upper() for x in domf.index.values])\n",
    "domf=domf[domf.protein=='HA'].head(2)\n",
    "DOMFILES={'DOM_HUMAN'+'('+subtype+')':i for i,name,subtype in zip(domf.sequence.values,\n",
    "                 domf.name.values,\n",
    "                 domf.subtype.values)}\n",
    "DOMFILES.update(FASTAFILES)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 108,
   "metadata": {},
   "outputs": [],
   "source": [
    "DOMFILES.update(RFILES)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 109,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "{'DOM_HUMAN(H1N1)': 'MKAILVVLLYTFTTANADTLCIGYHANNSTDTVDTVLEKNVTVTHSVNLLEDKHNGKLCKLRGVAPLHLGKCNIAGWILGNPECESLSTARSWSYIVETSNSDNGTCYPGDFINYEELREQLSSVSSFERFEIFPKTSSWPNHDSDKGVTAACPHAGAKSFYKNLIWLVKKGNSYPKLNQTYINDKGKEVLVLWGIHHPPTIAAQESLYQNADAYVFVGTSRYSKKFKPEIATRPKVRDQEGRMNYYWTLVEPGDKITFEATGNLVVPRYAFTMERDAGSGIIISDTPVHDCNTTCQTPEGAINTSLPFQNVHPITIGKCPKYVKSTKLRLATGLRNVPSIQSRGLFGAIAGFIEGGWTGMVDGWYGYHHQNEQGSGYAADLKSTQNAIDKITNKVNSVIEKMNTQFTAVGKEFNHLEKRIENLNKKVDDGFLDIWTYNAELLVLLENERTLDYHDSNVKNLYEKVRNQLKNNAKEIGNGCFEFYHKCDNTCMESVKNGTYDYPKYSEEAKLNREKIDGVKLESTRIYQILAIYSTVASSLVLVVSLGAISFWMCSNGSLQCRIC',\n",
       " 'EPI1818121(H1N1)': 'MKAILVVMLYTFTTANADTLCIGYHANNSTDTVDTVLEKNVTVTHSVNLLEDKHNGKLCKLRGVAPLHLGKCNIAGWILGNPECESLSTARSWSYIVETSNSDNGTCYPGDFINYEELREQLSSVSSFERFEIFPKTSSWPNHDSDNGVTAACPHAGAKSFYKNLIWLVKKGKSYPKINQTYINDKGKEVLVLWGIHHPPTIADQQSLYQNADAYVFVGTSRYSKKFKPEIATRPKVRDQEGRMNYYWTLVEPGDKITFEATGNLVAPRYAFTMERDAGSGIIISDTPVHDCNTTCQTPEGAINTSLPFQNVHPITIGKCPKYVKSTKLRLATGLRNVPSIQSRGLFGAIAGFIEGGWTGMVDGWYGYHHQNEQGSGYAADLKSTQNAIDKITNKVNSVIEKMNTQFTAVGKEFNHLEKRIENLNKKVDDGFLDIWTYNAELLVLLENERTLDYHDSNVKNLYEKVRNQLKNNAKEIGNGCFEFYHKCDNTCMESVKNGTYDYPKYSEEAKLNREKIDGVKLDSTRIYQILAIYSTVASSLVLVVSLGAI',\n",
       " 'EPI1818137(H3N2)': 'MKTIIALSYILCLVFAQKIPGNDNGTATLCLGHHAVPNGTIVKTITDDRIEVTNATELVQNSSIGEICDSPHQVLDGGNCTLIDALLGDPQCDGFQNKKWDLFVERSRAYSNCYPYDVPDYASLRSLVASSGTLEFKNESFNWTGVTQNGKSSSCIRGSSSSFFSRLNWLTXLNYTYPALNVIMPNKEQFDKLYIWGVHHPGTDNDQIFLYARSSGRIIVSTKRSQQAVIPNIGSRPRIRDIPSRISIYWTIVKPGDILLINSTGNLIAPRGYFKIRSGKSSIMRSDAPIGKCKSECITPNGSIPNDKPFQNVNRITYGACPRYIKQSTLKLATGMRNVPEKQTRGIFGAIAGFIENGWEGMVDGWYGFRHQNSEGRGQAADLKSTQAAIDQINGKLNRLIGKTNEKFHQIEKEFSEVEGRVQDLEKYVEDTKIDLWSYNAELLVALENQHTIDLTDSEMNKLFEKTKKQLRENAEDMGNGCFKIYHKCDNACIGSIRNETYDHNVYRDEALNNRFQIKGVELKSGYKDWILWISFAISCFLLCVALLGF',\n",
       " 'EPI2026200(H7N9)': 'MNTQILVFALIAIIPTNADKICLGHHAVSNGTKVNTLTEREVEVVNATETVERTNIPRICSKGKRTVDLGQCGLLGTITGPPQCDQFLEFSADLIIERREGSDVCYPGKFVNEEALRQILRESGGIDKEAMGFTYNGIRTNGVTSACRRSGSSFYAEMKWLLSNTDNAAFPQMTKSYKNTRKSPAIIVWGIHHSVSTAEQTKLYGSGNKLVTVGSSNYQQSFVPSPGARPQVNGISGRIDFHWLMLNPNDTVTFSFNGAFIAPDRASFLRGKSMGIQSGVQVDANCEGDCYHSGGTIISNLPFQNIDSRAVGKCPRYVKQRSLLLATGMKNVPEIPKGRGLFGAIAGFIENGWEGLIDGWYGFRHQNAQGEGTAADYKSTQSAIDQITGKLNRLIAKTNQQFELIDNEFNEVEKQIGNVINWTRDSITEVWSYNAELLVAMENQHTIDLADSEMDKLYERVKRQLRENAEEDGTGCFEIFHKCDDDCMASIRNNTYDHRKYREEAMQSRIQIDPVKLSSGYKDVILWFSFGASCFILLAIVMGLVFICVK',\n",
       " 'EPI2161544(H9N2)': 'METVSLITILLAATVSNADKICIGYQSSNSTETVDTLTENNVPVTHAKELLHTEHNGMLCATGLGQPLVLDTCTIEGLIYGNPSCDLSLEGREWSYIVERPSAVHGLCYPGNVEDLEELRSLFSSARSYQRIQIFPDTIWNVSYDGTSTACSGSFYKSMRWLTRKNGEYPTQDAQYTNNQGKNILFMWGINHPPTDDTQRGLYTRTDTTTSVATEEINRIFKPLIGPRPLVNGLMGRINYYWSVLKPGQTLRIKSDGNLIAPWYGHILSGESHGRILKTDLKRGNCTVQCQTEKGGLNTTLPFQNVSKYAFGNCSKYIGVKSLKLAVGLRNVPSRSSRGLFGAIAGFIEGGWPGLVAGWYGFQHSNDQGVGMAADRDSTQKAIDKITSKVNNIVDKMNKQYEIIDHEFSEVETRLNMINNKIDDQIQDIWAYNAELLVLLENQKTLDEHDANVNNLYNKVKRALGSNAAEDGKGCFELYHKCDDQCMETIRNGTYNRRKYQEESKLERQRIEGVKLESEGTYKILTIYSTVASSLVIAMGFAAFLFWAMS',\n",
       " 'EPI1780425(H1N1)': 'MKAILVVLLYAFTTANADTLCIGYHANNSTDTVDTVLEKNVTVTHSVNLLEEKHNGKLCKLRGVAPLHLGKCNIAGWILGNPECESLFTASSWSYIVETSNSDNGTCYPGDFINYEELREQLSSVSSFERFEIFPKASSWPNYETSRGVTAACPYAGANSFYRNLIWLVKKGNSYPKLSKSYINNKEKEVLVLWGIHHPSTSTDQQSLYQNADAYVFVGSSKYSKKFKPEIATRPKVRNQAGRMNYYWTLVEPGDTITFEATGNLVVPRYAFAMKRGSGSGIIISDTSVHDCNTNCQTPKGAINTSLPFQNIHPVTIGECPKYVKSKKLRMATGLRNIPSIQSRGLFGAIAGFIEGGWTGMIDGWYGYHHQNEQGSGYAADRKSTQSAIDGITNKVNSIIEKMNSQFTAVGKEFNHLEKRIENLNRKVDDGFLDVWTYNAELLVLLENERTLDYHDSNVKNLYEKVRSQLKNNAKEIGNGCFEFYHKCDDTCMESVKNGTYDYPKYSEEAKLNREEIDGVKLESTEVYQILAIYSTVASSLVLLVSLGAI',\n",
       " 'EPI2149257(H1N1)': 'MKAILLVLLHTLATANADTICIGYHANNSTDTVDTVLEKNVTVTHSVNLLEDKHNGKLCKLGGKSPLYLGKCNIAGWLLGNPECESILTVSSWSYIVETSNSDNGTCYPGSFTNYEELREQLSSVSSFEKFEIFPIESSWPNYKTGETASCPYAGKSSFYRNLMWLVKKADSYPRLNISYVNNKGKEVLVLWGIHHPPTGNDQQWLYQNANASVFVGTSTYSQKFKPEIATRPKVRGQAGRMNYYWTLVEPGDTITFEATGNLVVPRYAFAMNRGSGSGIIVSDAPIHDCNTPCQTPKGALNTSLPFQNVHPVTIGECPKYVKSRRLKMATGLRNIPSIQSRGLFGAIAGFIEGGWTGMVDGWYGYHHQNDQGSGYAADQKSTQRAIDGITNKVNSVIEKMNTQFTAVGKEFNNLERRIENLNRKVDDGFLDVWTYNAELLVLLENERTLDFHDSNVKNLYEKVRNQLRNNAKEIGNGCFEFYHKCDNTCMESVKNGTYNYLKYSEESKLNREEIDGVKLDSTRVYQILAIYSTVASSLVLLVSLGALSF'}"
      ]
     },
     "execution_count": 109,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "DOMFILES"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 110,
   "metadata": {},
   "outputs": [],
   "source": [
    "write_fasta(DOMFILES,FASTANAME)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import SeqIO\n",
    "from Bio.Blast import NCBIWWW\n",
    "my_query = SeqIO.read(\"EPI1818121.fasta\", format=\"fasta\")\n",
    "result_handle = NCBIWWW.qblast(\"blastp\", \"nr\", my_query.seq)\n",
    "blast_result = open(\"my_blast.xml\", \"w\")\n",
    "blast_result.write(result_handle.read())\n",
    "blast_result.close()\n",
    "result_handle.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "def parse_blast(resultfile): #takes in the BLAST result, outputs list that can be made into csv\n",
    "    from Bio.Blast import NCBIXML\n",
    "    result_handle = open(resultfile)\n",
    "    blast_records = NCBIXML.parse(result_handle)\n",
    "    csv_list = []\n",
    "    \n",
    "    header = [  'accession','Query',\n",
    "                'Name', 'Length', 'Score', 'Expect',\n",
    "                'QueryStart', 'QueryEnd',\n",
    "                'SubjectStart', 'SubjectEnd','pct'\n",
    "            ]\n",
    "    \n",
    "    #csv_list.append(header)\n",
    "    count = 0\n",
    "    for blast_record in blast_records:\n",
    "        '''help(blast_record.alignments[0].hsps[0])''' # these give help info for the parts \n",
    "        '''help(blast_record.alignments[0])        '''\n",
    "        count +=1\n",
    "        \n",
    "        query = blast_record.query\n",
    "        for alignment in blast_record.alignments:\n",
    "\n",
    "            name = alignment.title\n",
    "            length = alignment.length\n",
    "    \n",
    "            hsp = alignment.hsps[0] # I don't know if we will ever have more than one, so might as well take the first one.\n",
    "            score = hsp.score\n",
    "            expect = hsp.expect\n",
    "            querystart = hsp.query_start\n",
    "            queryend = hsp.query_end\n",
    "            subjectstart = hsp.sbjct_start\n",
    "            subjectend = hsp.sbjct_end\n",
    "            pct=hsp.positives/hsp.align_length\n",
    "            accession=alignment.accession\n",
    "            row = [accession,query,name,length,score,expect,querystart,queryend,subjectstart,subjectend,pct]\n",
    "            csv_list.append(row)\n",
    "            \n",
    "    result_handle.close()\n",
    "    return pd.DataFrame(csv_list,columns=header)\n",
    " "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "csv_list=parse_blast('my_blast.xml')\n",
    "csv_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import Entrez\n",
    "Entrez.email = \"ishanu@uchicago.edu\"     # Always tell NCBI who you are\n",
    "handle = Entrez.esearch(db=\"protein\", term=\"QJT24340\")\n",
    "record = Entrez.read(handle)\n",
    "id=record[\"IdList\"]\n",
    "id"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "handle = Entrez.efetch(db=\"protein\", rettype=\"gb\", retmode=\"text\", id=\"1834373412\")\n",
    "#record = SeqIO.read(handle, \"genbank\")\n",
    "handle.read()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "H9N2: minks are probable mixing vessels\n",
    "https://www.ncbi.nlm.nih.gov/nuccore/ON870694.1\n",
    " https://www.tandfonline.com/doi/full/10.1080/22221751.2021.1899058\n",
    " Pandemic influenza, typically caused by the reassortment of human and avian influenza viruses, can result in severe or fatal infections in humans. Timely identification of potential pandemic viruses must be a priority in influenza virus surveillance. However, the range of host species responsible for the generation of novel pandemic influenza viruses remains unclear. In this study, we conducted serological surveys for avian and human influenza virus infections in farmed mink and determined the susceptibility of mink to prevailing avian and human virus subtypes. The results showed that farmed mink were commonly infected with human (H3N2 and H1N1/pdm) and avian (H7N9, H5N6, and H9N2) influenza A viruses. Correlational analysis indicated that transmission of human influenza viruses occurred from humans to mink, and that feed source was a probable route of avian influenza virus transmission to farmed mink. Animal experiments showed that mink were susceptible and permissive to circulating avian and human influenza viruses, and that human influenza viruses (H3N2 and H1N1/pdm), but not avian viruses, were capable of aerosol transmission among mink. These results indicate that farmed mink could be highly permissive “mixing vessels” for the reassortment of circulating human and avian influenza viruses. \n",
    " \n",
    " H7N9\n",
    " https://www.bmj.com/content/347/bmj.f4752?tab=responses\n",
    " HH transmission has been suspected\n",
    " Asian lineage H7N9 virus is rated by the Influenza Risk Assessment Tool as having the greatest potential to cause a pandemic, as well as potentially posing the greatest risk to severely impact public health if it were to achieve sustained human-to-human transmission.\n",
    " \n",
    " H3N2 swine\n",
    " https://www.cdc.gov/flu/swineflu/spotlights/first-human-infection-2022.htm\n",
    " Human infection detected in US this Aug\n",
    " \n",
    " H1N1\n",
    " Submitted (03-NOV-2020) USDA Swine Surveillance, USDA Swine\n",
    "            Surveillance, 1920 Dayton, Ames, IA 50010, USA\n",
    "COMMENT     Method: conceptual translation.\n",
    "\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.8"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
