{
 "cells": [
  {
   "cell_type": "markdown",
   "id": "e71c30fc",
   "metadata": {},
   "source": [
    "# IRAT vs. Qnet Comparison - All Sequences\n",
    "**H1N1, H1N2, and H3N2 only**\n",
    "- Compares risk assesment from IRAT and risk assessment using Qnet q-distance\n",
    "- Use both NA and HA segments\n",
    "- For each strain previously analyzed by IRAT\n",
    "    - Collect all strains of that variety\n",
    "    - Note: had difficulty finding 'A/duck/New York/1996', only NA available\n",
    "    - Construct a Qnet using these strains\n",
    "        - For H1N2, the HA is similar to that of H1N1 and the NA to that of H3N2, so use H1N1 HA for HA Qnet and H3N2 NA for NA Qnet\n",
    "    - Compute the average q-distance among the strain in question and the circulating human strains for both NA and HA\n",
    "    - Average the NA and HA averages"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "b1b87e7e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# basic imports\n",
    "import os \n",
    "import numpy as np\n",
    "import pandas as pd\n",
    "import math\n",
    "import warnings\n",
    "warnings.filterwarnings('ignore')\n",
    "import tqdm\n",
    "from tqdm.notebook import trange, tqdm\n",
    "\n",
    "# visualization\n",
    "import seaborn as sns \n",
    "import matplotlib.pyplot as plt\n",
    "%matplotlib inline\n",
    "\n",
    "# other\n",
    "from Bio import SeqIO\n",
    "from collections import Counter\n",
    "import Levenshtein as lev\n",
    "\n",
    "# qnet\n",
    "from quasinet.qnet import Qnet, qdistance, qdistance_matrix, membership_degree, save_qnet, load_qnet\n",
    "from quasinet.qseqtools import list_trained_qnets, load_trained_qnet"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "318664a6",
   "metadata": {},
   "source": [
    "## Data Sources\n",
    "- IRAT (CDC): https://www.cdc.gov/flu/pandemic-resources/monitoring/irat-virus-summaries.htm#H1N2variant\n",
    "- NCBI: https://www.ncbi.nlm.nih.gov/labs/virus/vssi/#/virus?SeqType_s=Protein\n",
    "- GISAID: https://platform.epicov.org/epi3/cfrontend#586f5f"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "id": "4b043e8f",
   "metadata": {},
   "outputs": [],
   "source": [
    "NCBI_PATH = 'raw_data/ncbi/'\n",
    "GISAID_PATH = 'raw_data/gisaid/'\n",
    "IRAT_PATH = 'results/'\n",
    "\n",
    "FILES = ['h1n1_ha', 'h1n1_na', 'h3n2_ha', 'h3n2_na']\n",
    "\n",
    "NA_TRUNC = 469\n",
    "HA_TRUNC = 565\n",
    "\n",
    "df = pd.read_csv(IRAT_PATH + 'irat_data_all_sequences.csv')"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "a3426ff0",
   "metadata": {},
   "source": [
    "## Creating New Qnet\n",
    "- Truncate NA at 469 amino acids, HA at 565 amino acids (this is possible because H1N1, H1N2, and H3N2 Irat strains are at least this long, as opposed to others which are shorter)\n",
    "- Save Qnet as \"h1n1_na.joblib\" or \"h1n1_ha.joblib\" or \"h3n2_na.joblib\" or \"h3n2_ha.joblib\""
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "id": "fa90d586",
   "metadata": {},
   "outputs": [],
   "source": [
    "# input: fasta file name, length to truncate each sequence\n",
    "# output: dataframe of sequences\n",
    "def parse_fasta(file_name, trunc):\n",
    "    acc = []\n",
    "    seq = []\n",
    "    for record in SeqIO.parse(file_name, 'fasta'):\n",
    "        if len(record.seq) < trunc:\n",
    "            continue\n",
    "        acc.append(record.id.split('|')[0])\n",
    "        seq.append(np.array(record.seq[:trunc].upper()))\n",
    "    df = pd.DataFrame({'name':acc, 'sequence':seq})\n",
    "    return df\n",
    "\n",
    "\n",
    "# input: dataframe of sequences, number of samples, list of IRAT strains\n",
    "# output: array of nucleotide lists\n",
    "def sequence_array(seq_df, sample_size, IRAT_strains):\n",
    "    seqs = seq_df['sequence'].sample(sample_size, random_state = 42).values\n",
    "    seq_lst = []\n",
    "    for seq in seqs:\n",
    "        seq_lst.append(seq)\n",
    "    for seq in IRAT_strains:\n",
    "        seq_lst.append(np.array(list(seq)))\n",
    "    return np.array(seq_lst)\n",
    "\n",
    "\n",
    "\n",
    "# input: name to call qnet, array of nucleotide lists, number of nucleotides\n",
    "# output: save qnet as joblib\n",
    "def train_save_qnet(name, seq_arr, num_nuc):\n",
    "    myqnet = Qnet(feature_names=['x'+str(i) for i in np.arange(num_nuc)],n_jobs=1)\n",
    "    myqnet.fit(seq_arr)\n",
    "    save_qnet(myqnet, 'qnet_models/' + name + '.joblib')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "id": "0f14bb45",
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Influenza Virus</th>\n",
       "      <th>Virus Type</th>\n",
       "      <th>Dates of Risk Assessment</th>\n",
       "      <th>Potential Emergence Estimate</th>\n",
       "      <th>Potential Impact Estimate</th>\n",
       "      <th>Summary Risk Score Category</th>\n",
       "      <th>HA Sequence</th>\n",
       "      <th>NA Sequence</th>\n",
       "      <th>HA Qnet Sample</th>\n",
       "      <th>NA Qnet Sample</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A/swine/Shandong/1207/2016</td>\n",
       "      <td>H1N1</td>\n",
       "      <td>Jul 2020</td>\n",
       "      <td>7.5</td>\n",
       "      <td>6.9</td>\n",
       "      <td>Moderate</td>\n",
       "      <td>MEARLFVLFCAFTTLKADTICVGYHANNSTDTVDTILEKNVTVTHS...</td>\n",
       "      <td>MNPNQKIITIGSICMTIGIASLILQIGNIISIWISHSIQIENQNQS...</td>\n",
       "      <td>17402.0</td>\n",
       "      <td>14687.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A/duck/New York/1996</td>\n",
       "      <td>H1N1</td>\n",
       "      <td>Nov 2011</td>\n",
       "      <td>2.3</td>\n",
       "      <td>2.4</td>\n",
       "      <td>Low</td>\n",
       "      <td>-1</td>\n",
       "      <td>MNPNQKIITIGSICMAIGIISLVLQIGNIISIWVSHSIQTGSQSHP...</td>\n",
       "      <td>17402.0</td>\n",
       "      <td>14687.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A/California/62/2018</td>\n",
       "      <td>H1N2</td>\n",
       "      <td>Jul 2019</td>\n",
       "      <td>5.8</td>\n",
       "      <td>5.7</td>\n",
       "      <td>Moderate</td>\n",
       "      <td>MKVKLMVLLCTFTATYADTICVGYHANNSTDTVDTVLEKNVTVTHS...</td>\n",
       "      <td>MNPNQKIITIGSISLTLAAMCFLMQTAILVTNVTLHFNQCECHYPP...</td>\n",
       "      <td>17402.0</td>\n",
       "      <td>14057.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A/Ohio/13/2017</td>\n",
       "      <td>H3N2</td>\n",
       "      <td>Jul 2019</td>\n",
       "      <td>6.6</td>\n",
       "      <td>5.8</td>\n",
       "      <td>Moderate</td>\n",
       "      <td>MKTIIALSHILCLVFAQKLPGNDNNMATLCLGHHAVPNGTIVKTIT...</td>\n",
       "      <td>MNPNQKIITIGSVSLIIATICFLMQIAILVTTITLHFKQHNCDSSP...</td>\n",
       "      <td>17423.0</td>\n",
       "      <td>14057.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A/Indiana/08/2011</td>\n",
       "      <td>H3N2</td>\n",
       "      <td>Dec 2012</td>\n",
       "      <td>6.0</td>\n",
       "      <td>4.5</td>\n",
       "      <td>Moderate</td>\n",
       "      <td>MKTIIAFSCILCLIFAQKLPGSDNSMATLCLGHHAVPNGTLVKTIT...</td>\n",
       "      <td>MNPNQKIITIGSVSLIIATICFLMQIAILVTTVTLHFKQHDYNSPP...</td>\n",
       "      <td>17423.0</td>\n",
       "      <td>14057.0</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>A/canine/Illinois/12191/2015</td>\n",
       "      <td>H3N2</td>\n",
       "      <td>Jun 2016</td>\n",
       "      <td>3.7</td>\n",
       "      <td>3.7</td>\n",
       "      <td>Low</td>\n",
       "      <td>MKTVIALSYIFCLAFGQNLLGNENNAATLCLGHHAVPNGTMVKTIT...</td>\n",
       "      <td>MNPNQKIIAIGSVSLTIATVCFLLQIAILATTVTLYFKQNECNIPS...</td>\n",
       "      <td>17423.0</td>\n",
       "      <td>14057.0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                Influenza Virus Virus Type Dates of Risk Assessment  \\\n",
       "0    A/swine/Shandong/1207/2016       H1N1                 Jul 2020   \n",
       "1          A/duck/New York/1996       H1N1                 Nov 2011   \n",
       "2          A/California/62/2018       H1N2                 Jul 2019   \n",
       "3                A/Ohio/13/2017       H3N2                 Jul 2019   \n",
       "4             A/Indiana/08/2011       H3N2                 Dec 2012   \n",
       "5  A/canine/Illinois/12191/2015       H3N2                 Jun 2016   \n",
       "\n",
       "   Potential Emergence Estimate  Potential Impact Estimate  \\\n",
       "0                           7.5                        6.9   \n",
       "1                           2.3                        2.4   \n",
       "2                           5.8                        5.7   \n",
       "3                           6.6                        5.8   \n",
       "4                           6.0                        4.5   \n",
       "5                           3.7                        3.7   \n",
       "\n",
       "  Summary Risk Score Category  \\\n",
       "0                    Moderate   \n",
       "1                         Low   \n",
       "2                    Moderate   \n",
       "3                    Moderate   \n",
       "4                    Moderate   \n",
       "5                         Low   \n",
       "\n",
       "                                         HA Sequence  \\\n",
       "0  MEARLFVLFCAFTTLKADTICVGYHANNSTDTVDTILEKNVTVTHS...   \n",
       "1                                                 -1   \n",
       "2  MKVKLMVLLCTFTATYADTICVGYHANNSTDTVDTVLEKNVTVTHS...   \n",
       "3  MKTIIALSHILCLVFAQKLPGNDNNMATLCLGHHAVPNGTIVKTIT...   \n",
       "4  MKTIIAFSCILCLIFAQKLPGSDNSMATLCLGHHAVPNGTLVKTIT...   \n",
       "5  MKTVIALSYIFCLAFGQNLLGNENNAATLCLGHHAVPNGTMVKTIT...   \n",
       "\n",
       "                                         NA Sequence  HA Qnet Sample  \\\n",
       "0  MNPNQKIITIGSICMTIGIASLILQIGNIISIWISHSIQIENQNQS...         17402.0   \n",
       "1  MNPNQKIITIGSICMAIGIISLVLQIGNIISIWVSHSIQTGSQSHP...         17402.0   \n",
       "2  MNPNQKIITIGSISLTLAAMCFLMQTAILVTNVTLHFNQCECHYPP...         17402.0   \n",
       "3  MNPNQKIITIGSVSLIIATICFLMQIAILVTTITLHFKQHNCDSSP...         17423.0   \n",
       "4  MNPNQKIITIGSVSLIIATICFLMQIAILVTTVTLHFKQHDYNSPP...         17423.0   \n",
       "5  MNPNQKIIAIGSVSLTIATVCFLLQIAILATTVTLYFKQNECNIPS...         17423.0   \n",
       "\n",
       "   NA Qnet Sample  \n",
       "0         14687.0  \n",
       "1         14687.0  \n",
       "2         14057.0  \n",
       "3         14057.0  \n",
       "4         14057.0  \n",
       "5         14057.0  "
      ]
     },
     "execution_count": 4,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "id": "a683460a",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "application/vnd.jupyter.widget-view+json": {
       "model_id": "12edbd8e17c94fe0b6f278b7ee740d8e",
       "version_major": 2,
       "version_minor": 0
      },
      "text/plain": [
       "  0%|          | 0/2 [00:00<?, ?it/s]"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    }
   ],
   "source": [
    "h1n1_ha_irat = [df['HA Sequence'][0][:HA_TRUNC], \n",
    "                df['HA Sequence'][2][:HA_TRUNC]]\n",
    "h3n2_ha_irat = [df['HA Sequence'][3][:HA_TRUNC], \n",
    "                df['HA Sequence'][4][:HA_TRUNC], \n",
    "                df['HA Sequence'][5][:HA_TRUNC]]\n",
    "h1n1_na_irat = [df['NA Sequence'][0][:NA_TRUNC], \n",
    "                df['NA Sequence'][1][:NA_TRUNC]]\n",
    "h3n2_na_irat = [df['NA Sequence'][2][:NA_TRUNC], \n",
    "                df['NA Sequence'][3][:NA_TRUNC], \n",
    "                df['NA Sequence'][4][:NA_TRUNC], \n",
    "                df['NA Sequence'][5][:NA_TRUNC]]\n",
    "irat_seq_dict = {'h1n1_ha':h1n1_ha_irat, 'h3n2_ha':h3n2_ha_irat, \n",
    "                 'h1n1_na':h1n1_na_irat, 'h3n2_na':h3n2_na_irat}\n",
    "\n",
    "FILES = ['h1n1_na', 'h3n2_na']\n",
    "\n",
    "# create qnets for each dataset\n",
    "for FILE in tqdm(FILES):\n",
    "    TRUNC = HA_TRUNC\n",
    "    if 'na' in FILE:\n",
    "        TRUNC = NA_TRUNC\n",
    "    seq_df = parse_fasta(NCBI_PATH + FILE + \".fasta\", TRUNC) \n",
    "    seq_arr = sequence_array(seq_df, min(1000, len(seq_df)), irat_seq_dict[FILE])\n",
    "    train_save_qnet(FILE, seq_arr, TRUNC)"
   ]
  },
  {
   "cell_type": "markdown",
   "id": "f6af439a",
   "metadata": {},
   "source": [
    "## Average Qdistance\n",
    "- Compute average qdistance between IRAT strain and the rest of the strains"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "2d637d3f",
   "metadata": {},
   "outputs": [],
   "source": [
    "avg_qdists_ha = []\n",
    "avg_qdists_na = []\n",
    "avg_qdists_both = []\n",
    "\n",
    "for i in trange(len(df)):\n",
    "    VIRUS_TYPE_HA = df['Virus Type'].iloc[i].lower() + '_ha'\n",
    "    VIRUS_TYPE_NA = df['Virus Type'].iloc[i].lower() + '_na'\n",
    "    if df['Virus Type'].iloc[i] == 'H1N2':\n",
    "        VIRUS_TYPE_HA = 'h1n1_ha'\n",
    "        VIRUS_TYPE_NA = 'h3n2_na'\n",
    "        \n",
    "    # load qnets\n",
    "    ha_qnet = load_qnet('qnet_models/' + VIRUS_TYPE_HA + '.joblib')\n",
    "    na_qnet = load_qnet('qnet_models/' + VIRUS_TYPE_NA + '.joblib')\n",
    "\n",
    "    # skip if HA doesn't exist\n",
    "    ha_exists = True\n",
    "    if df['HA Sequence'].iloc[i] == '-1':\n",
    "        ha_exists = False\n",
    "        \n",
    "    if ha_exists:\n",
    "        # access irat sequences and all sequences\n",
    "        ha_irat_seq = np.array(list(df['HA Sequence'].iloc[i][:HA_TRUNC]))\n",
    "        na_irat_seq = np.array(list(df['NA Sequence'].iloc[i][:NA_TRUNC]))\n",
    "        ha_df = parse_fasta(NCBI_PATH + VIRUS_TYPE_HA + \".fasta\", HA_TRUNC)\n",
    "        na_df = parse_fasta(NCBI_PATH + VIRUS_TYPE_NA + \".fasta\", NA_TRUNC)\n",
    "        ha_arr = sequence_array(ha_df, min(10000, len(ha_df)), [])\n",
    "        na_arr = sequence_array(na_df, min(10000, len(na_df)), [])\n",
    "\n",
    "        # compute qdistance sum\n",
    "        ha_qdist_sum = 0\n",
    "        na_qdist_sum = 0\n",
    "        ha_nan = 0\n",
    "        na_nan = 0\n",
    "        for j in range(len(ha_arr)):\n",
    "            if not math.isnan(qdistance(ha_irat_seq, ha_arr[j], ha_qnet, ha_qnet)):\n",
    "                ha_qdist_sum += qdistance(ha_irat_seq, ha_arr[j], ha_qnet, ha_qnet)\n",
    "            else:\n",
    "                ha_nan += 1\n",
    "        for j in range(len(na_arr)):\n",
    "            if not math.isnan(qdistance(na_irat_seq, na_arr[j], na_qnet, na_qnet)):\n",
    "                na_qdist_sum += qdistance(na_irat_seq, na_arr[j], na_qnet, na_qnet)\n",
    "            else:\n",
    "                na_nan += 1\n",
    "        print('HA NaN: ', ha_nan)\n",
    "        print('NA NaN: ', na_nan)\n",
    "\n",
    "        # compute qdistance averages\n",
    "        avg_qdists_ha.append(ha_qdist_sum/len(ha_arr))\n",
    "        avg_qdists_na.append(na_qdist_sum/len(na_arr))\n",
    "        avg_qdists_both.append((ha_qdist_sum + na_qdist_sum)/(len(ha_arr) + len(na_arr)))\n",
    "    \n",
    "    else:\n",
    "        # access irat sequences and all sequences\n",
    "        na_irat_seq = np.array(list(df['NA Sequence'].iloc[i][:NA_TRUNC]))\n",
    "        na_df = parse_fasta(NCBI_PATH + VIRUS_TYPE_NA + \".fasta\", NA_TRUNC)\n",
    "        na_arr = sequence_array(na_df, min(10000, len(na_df)), [])\n",
    "\n",
    "        # compute qdistance sum\n",
    "        na_qdist_sum = 0\n",
    "        for j in range(len(na_arr)):\n",
    "            if not math.isnan(qdistance(na_irat_seq, na_arr[j], na_qnet, na_qnet)):\n",
    "                na_qdist_sum += qdistance(na_irat_seq, na_arr[j], na_qnet, na_qnet)\n",
    "            else:\n",
    "                na_nan += 1\n",
    "        print('NA NaN: ', na_nan)\n",
    "\n",
    "        # compute qdistance averages\n",
    "        avg_qdists_ha.append(-1)\n",
    "        avg_qdists_na.append(na_qdist_sum/len(na_arr))\n",
    "        avg_qdists_both.append(-1)\n",
    "    \n",
    "\n",
    "df['HA Average Qdistance'] = avg_qdists_ha\n",
    "df['NA Average Qdistance'] = avg_qdists_na\n",
    "df['Both Average Qdistance'] = avg_qdists_both"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "id": "a0ee2260",
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>Influenza Virus</th>\n",
       "      <th>Virus Type</th>\n",
       "      <th>Dates of Risk Assessment</th>\n",
       "      <th>Potential Emergence Estimate</th>\n",
       "      <th>Potential Impact Estimate</th>\n",
       "      <th>Summary Risk Score Category</th>\n",
       "      <th>HA Sequence</th>\n",
       "      <th>NA Sequence</th>\n",
       "      <th>HA Qnet Sample</th>\n",
       "      <th>NA Qnet Sample</th>\n",
       "      <th>HA Average Qdistance</th>\n",
       "      <th>NA Average Qdistance</th>\n",
       "      <th>Both Average Qdistance</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>A/duck/New York/1996</td>\n",
       "      <td>H1N1</td>\n",
       "      <td>Nov 2011</td>\n",
       "      <td>2.3</td>\n",
       "      <td>2.4</td>\n",
       "      <td>Low</td>\n",
       "      <td>-1</td>\n",
       "      <td>MNPNQKIITIGSICMAIGIISLVLQIGNIISIWVSHSIQTGSQSHP...</td>\n",
       "      <td>17402.0</td>\n",
       "      <td>14687.0</td>\n",
       "      <td>-1.000000</td>\n",
       "      <td>0.037525</td>\n",
       "      <td>-1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>A/canine/Illinois/12191/2015</td>\n",
       "      <td>H3N2</td>\n",
       "      <td>Jun 2016</td>\n",
       "      <td>3.7</td>\n",
       "      <td>3.7</td>\n",
       "      <td>Low</td>\n",
       "      <td>MKTVIALSYIFCLAFGQNLLGNENNAATLCLGHHAVPNGTMVKTIT...</td>\n",
       "      <td>MNPNQKIIAIGSVSLTIATVCFLLQIAILATTVTLYFKQNECNIPS...</td>\n",
       "      <td>17423.0</td>\n",
       "      <td>14057.0</td>\n",
       "      <td>0.017681</td>\n",
       "      <td>0.041427</td>\n",
       "      <td>0.029554</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>A/California/62/2018</td>\n",
       "      <td>H1N2</td>\n",
       "      <td>Jul 2019</td>\n",
       "      <td>5.8</td>\n",
       "      <td>5.7</td>\n",
       "      <td>Moderate</td>\n",
       "      <td>MKVKLMVLLCTFTATYADTICVGYHANNSTDTVDTVLEKNVTVTHS...</td>\n",
       "      <td>MNPNQKIITIGSISLTLAAMCFLMQTAILVTNVTLHFNQCECHYPP...</td>\n",
       "      <td>17402.0</td>\n",
       "      <td>14057.0</td>\n",
       "      <td>0.379929</td>\n",
       "      <td>0.029437</td>\n",
       "      <td>0.204683</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>A/Indiana/08/2011</td>\n",
       "      <td>H3N2</td>\n",
       "      <td>Dec 2012</td>\n",
       "      <td>6.0</td>\n",
       "      <td>4.5</td>\n",
       "      <td>Moderate</td>\n",
       "      <td>MKTIIAFSCILCLIFAQKLPGSDNSMATLCLGHHAVPNGTLVKTIT...</td>\n",
       "      <td>MNPNQKIITIGSVSLIIATICFLMQIAILVTTVTLHFKQHDYNSPP...</td>\n",
       "      <td>17423.0</td>\n",
       "      <td>14057.0</td>\n",
       "      <td>0.008264</td>\n",
       "      <td>0.021427</td>\n",
       "      <td>0.014846</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>A/Ohio/13/2017</td>\n",
       "      <td>H3N2</td>\n",
       "      <td>Jul 2019</td>\n",
       "      <td>6.6</td>\n",
       "      <td>5.8</td>\n",
       "      <td>Moderate</td>\n",
       "      <td>MKTIIALSHILCLVFAQKLPGNDNNMATLCLGHHAVPNGTIVKTIT...</td>\n",
       "      <td>MNPNQKIITIGSVSLIIATICFLMQIAILVTTITLHFKQHNCDSSP...</td>\n",
       "      <td>17423.0</td>\n",
       "      <td>14057.0</td>\n",
       "      <td>0.006036</td>\n",
       "      <td>0.022038</td>\n",
       "      <td>0.014037</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A/swine/Shandong/1207/2016</td>\n",
       "      <td>H1N1</td>\n",
       "      <td>Jul 2020</td>\n",
       "      <td>7.5</td>\n",
       "      <td>6.9</td>\n",
       "      <td>Moderate</td>\n",
       "      <td>MEARLFVLFCAFTTLKADTICVGYHANNSTDTVDTILEKNVTVTHS...</td>\n",
       "      <td>MNPNQKIITIGSICMTIGIASLILQIGNIISIWISHSIQIENQNQS...</td>\n",
       "      <td>17402.0</td>\n",
       "      <td>14687.0</td>\n",
       "      <td>0.088669</td>\n",
       "      <td>0.043072</td>\n",
       "      <td>0.065871</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                Influenza Virus Virus Type Dates of Risk Assessment  \\\n",
       "1          A/duck/New York/1996       H1N1                 Nov 2011   \n",
       "5  A/canine/Illinois/12191/2015       H3N2                 Jun 2016   \n",
       "2          A/California/62/2018       H1N2                 Jul 2019   \n",
       "4             A/Indiana/08/2011       H3N2                 Dec 2012   \n",
       "3                A/Ohio/13/2017       H3N2                 Jul 2019   \n",
       "0    A/swine/Shandong/1207/2016       H1N1                 Jul 2020   \n",
       "\n",
       "   Potential Emergence Estimate  Potential Impact Estimate  \\\n",
       "1                           2.3                        2.4   \n",
       "5                           3.7                        3.7   \n",
       "2                           5.8                        5.7   \n",
       "4                           6.0                        4.5   \n",
       "3                           6.6                        5.8   \n",
       "0                           7.5                        6.9   \n",
       "\n",
       "  Summary Risk Score Category  \\\n",
       "1                         Low   \n",
       "5                         Low   \n",
       "2                    Moderate   \n",
       "4                    Moderate   \n",
       "3                    Moderate   \n",
       "0                    Moderate   \n",
       "\n",
       "                                         HA Sequence  \\\n",
       "1                                                 -1   \n",
       "5  MKTVIALSYIFCLAFGQNLLGNENNAATLCLGHHAVPNGTMVKTIT...   \n",
       "2  MKVKLMVLLCTFTATYADTICVGYHANNSTDTVDTVLEKNVTVTHS...   \n",
       "4  MKTIIAFSCILCLIFAQKLPGSDNSMATLCLGHHAVPNGTLVKTIT...   \n",
       "3  MKTIIALSHILCLVFAQKLPGNDNNMATLCLGHHAVPNGTIVKTIT...   \n",
       "0  MEARLFVLFCAFTTLKADTICVGYHANNSTDTVDTILEKNVTVTHS...   \n",
       "\n",
       "                                         NA Sequence  HA Qnet Sample  \\\n",
       "1  MNPNQKIITIGSICMAIGIISLVLQIGNIISIWVSHSIQTGSQSHP...         17402.0   \n",
       "5  MNPNQKIIAIGSVSLTIATVCFLLQIAILATTVTLYFKQNECNIPS...         17423.0   \n",
       "2  MNPNQKIITIGSISLTLAAMCFLMQTAILVTNVTLHFNQCECHYPP...         17402.0   \n",
       "4  MNPNQKIITIGSVSLIIATICFLMQIAILVTTVTLHFKQHDYNSPP...         17423.0   \n",
       "3  MNPNQKIITIGSVSLIIATICFLMQIAILVTTITLHFKQHNCDSSP...         17423.0   \n",
       "0  MNPNQKIITIGSICMTIGIASLILQIGNIISIWISHSIQIENQNQS...         17402.0   \n",
       "\n",
       "   NA Qnet Sample  HA Average Qdistance  NA Average Qdistance  \\\n",
       "1         14687.0             -1.000000              0.037525   \n",
       "5         14057.0              0.017681              0.041427   \n",
       "2         14057.0              0.379929              0.029437   \n",
       "4         14057.0              0.008264              0.021427   \n",
       "3         14057.0              0.006036              0.022038   \n",
       "0         14687.0              0.088669              0.043072   \n",
       "\n",
       "   Both Average Qdistance  \n",
       "1               -1.000000  \n",
       "5                0.029554  \n",
       "2                0.204683  \n",
       "4                0.014846  \n",
       "3                0.014037  \n",
       "0                0.065871  "
      ]
     },
     "execution_count": 12,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df.sort_values(by = 'Potential Emergence Estimate')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "id": "ff5aeb7e",
   "metadata": {},
   "outputs": [],
   "source": [
    "# save dataframe as csv\n",
    "os.makedirs('results', exist_ok=True)\n",
    "df.to_csv('results/irat_average_qdistances_all_sequences.csv', index=False)  \n",
    "df.drop(columns=['HA Sequence', 'NA Sequence']).to_latex('results/irat_average_qdistances_all_sequences.tex', index=False)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "id": "847cc034",
   "metadata": {},
   "outputs": [],
   "source": []
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.7.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 5
}
