{
 "cells": [
  {
   "cell_type": "code",
   "execution_count": 1,
   "metadata": {},
   "outputs": [],
   "source": [
    "import copy\n",
    "import pandas as pd\n",
    "import numpy as np\n",
    "import Levenshtein\n",
    "from tqdm import tqdm"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 2,
   "metadata": {},
   "outputs": [],
   "source": [
    "# parameters\n",
    "threshold=6  # draw tree above this emergenec risk threshold\n",
    "CONSTRUCT_PHYLO=True  # use constructed tree (tree construction takes hours)\n",
    "OUTPUT_DIR='./'\n",
    "PHYLO_TREE_DIR='./'\n",
    "PHYLO_DIR='./'\n",
    "COMBINED_RESULTS='./combined_results.csv'\n",
    "CONSTRUCT_PHYLO=False\n",
    "num_collapsed=18  # number of mutations within which leaves are collapsed\n",
    "VERBOSE=False"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# class definition to hold multi sequence data"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 3,
   "metadata": {},
   "outputs": [],
   "source": [
    "class SeqInfo(object):\n",
    "    \"\"\"Holds information regarding the sequence.\n",
    "    \n",
    "    \"\"\"\n",
    "    def __init__(self, seq, \n",
    "                 protein,\n",
    "                 accession,\n",
    "                 subtype=None,\n",
    "                 id=None,\n",
    "                 name=None,\n",
    "                 host=None, \n",
    "                 date=None, \n",
    "                 erisk=None,\n",
    "                 irisk=None,\n",
    "                 risk_flag=None,\n",
    "                 country=None):\n",
    "        self.name = name\n",
    "        self.id = id\n",
    "        self.protein=protein\n",
    "        self.subtype=subtype        \n",
    "        self.seq = seq\n",
    "        self.accession = accession \n",
    "        self.host = host\n",
    "        self.date = date\n",
    "        self.erisk = erisk\n",
    "        self.irisk = irisk\n",
    "        self.risk_flag = risk_flag\n",
    "        self.country = country\n",
    "        \n",
    "class MultipleSeqInfo(object):\n",
    "    \"\"\"Holds information regarding multiple sequences.\n",
    "    \n",
    "    Args:\n",
    "        dataframe (pandas.DataFrame): list of records parsed from NCBI\n",
    "        accessionname (str): column name for accession id\n",
    "        proteinname (str): protein name \n",
    "        risk_threshold (float): emergence risk threshold to compute distance matrix\n",
    "    \"\"\"\n",
    "    def __init__(self,\n",
    "                 dataframe,\n",
    "                 accessionname,\n",
    "                 proteinname,\n",
    "                 risk_threshold=6.2):\n",
    "        \n",
    "        self.seq_infos = {}\n",
    "        self.risk_threshold = risk_threshold\n",
    "        for i in np.arange(dataframe.index.size):\n",
    "            record=dataframe.iloc[i,:]\n",
    "            seqinfo = SeqInfo(\n",
    "                name=record.id,\n",
    "                seq=record[proteinname], \n",
    "                protein=proteinname,\n",
    "                accession=record[accessionname],\n",
    "                subtype=record.subtype,\n",
    "                erisk=record.predicted_emergence_score,\n",
    "                irisk=record.predicted_impact_score,\n",
    "                risk_flag = record.predicted_emergence_score > self.risk_threshold,\n",
    "                host=None,\n",
    "                date=None,\n",
    "                country=None)\n",
    "            #print(record.predicted_emergence_score > self.risk_threshold)\n",
    "            self.seq_infos[seqinfo.accession] = seqinfo\n",
    "            \n",
    "    \n",
    "    def compute_L_distance_matrix(self):\n",
    "        highriskseq = pd.DataFrame.from_dict({key:val.seq \n",
    "                                              for (key,val) in self.seq_infos.items() \n",
    "                                              if val.risk_flag},orient='index',columns=['seq'])\n",
    "        num=highriskseq.index.size\n",
    "        d=np.zeros([num,num])\n",
    "        for x in tqdm(np.arange(num*num)):\n",
    "            j=x//num\n",
    "            i=x-num*j\n",
    "            if i > j:\n",
    "                d[i,j] = Levenshtein.distance(highriskseq.seq.values[i],\n",
    "                                                  highriskseq.seq.values[j])\n",
    "        ds=pd.DataFrame(d)        \n",
    "        ds=(ds+ds.transpose())\n",
    "        ds.columns=highriskseq.index.values\n",
    "        self.highriskdistancematrix=ds.copy()\n",
    "        \n",
    "        self.highriskdistancematrix.to_csv('dm'+str(self.risk_threshold)+'.csv',index=None)\n",
    "        return \n",
    "    \n",
    "    \n",
    "    def accessions_to_subtype(self, accessions):\n",
    "        \"\"\"Create a dictionary mapping the accession to the host.\n",
    "        \"\"\"\n",
    "        \n",
    "        subtypes = []\n",
    "        for accession in accessions:\n",
    "            seqinfo = self.seq_infos[accession]\n",
    "            subtypes.append(seqinfo.subtype)\n",
    "            \n",
    "        return subtypes\n",
    "\n",
    "    def accessions_to_host(self, accessions):\n",
    "        \"\"\"Create a dictionary mapping the accession to the host.\n",
    "        \"\"\"\n",
    "        \n",
    "        hosts = []\n",
    "        for accession in accessions:\n",
    "            seqinfo = self.seq_infos[accession]\n",
    "            hosts.append(seqinfo.host)\n",
    "        return hosts\n",
    "           \n",
    "    "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 4,
   "metadata": {},
   "outputs": [],
   "source": [
    "df=pd.read_csv(COMBINED_RESULTS,index_col=0).reset_index()\n",
    "#ALLinfoHA=MultipleSeqInfo(df.reset_index(),'ha_accession','ha',risk_threshold=6.054)\n",
    "ALLinfoHA=MultipleSeqInfo(df.reset_index(),'ha_accession','ha',risk_threshold=threshold)"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 5,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "H1N1    890\n",
       "H3N2    812\n",
       "H9N2     29\n",
       "H7N9      1\n",
       "Name: subtype, dtype: int64"
      ]
     },
     "execution_count": 5,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "df[df.predicted_emergence_score>threshold].subtype.value_counts()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {
    "scrolled": true
   },
   "source": [
    "# generate distance matrix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 6,
   "metadata": {
    "scrolled": true
   },
   "outputs": [
    {
     "name": "stderr",
     "output_type": "stream",
     "text": [
      "100%|██████████████████████████████| 2999824/2999824 [01:14<00:00, 40519.69it/s]\n"
     ]
    }
   ],
   "source": [
    "ALLinfoHA.compute_L_distance_matrix()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# tree construction from distance matrix using biopython"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 7,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio.Phylo import TreeConstruction\n",
    "from Bio import Phylo\n",
    "from Bio.SeqRecord import SeqRecord\n",
    "from Bio.Seq import Seq\n",
    "from Bio.Align import MultipleSeqAlignment\n",
    "from Bio import Entrez\n",
    "from Bio import SeqIO\n",
    "\n",
    "\n",
    "def load_dm(file_, upper_diag=True):\n",
    "    \"\"\"Load the distance matrix. \n",
    "    \n",
    "    Also, do some preprocessing. \n",
    "    \"\"\"\n",
    "    \n",
    "    df = pd.read_csv(file_)\n",
    "    #df.set_index('Unnamed: 0', inplace=True)\n",
    "    #assert np.all(df.columns == df.index)\n",
    "    \n",
    "    # drop duplicate columns after reading csv\n",
    "    #df = df.loc[:, ~df.columns.str.replace(\"(\\.\\d+)$\", \"\").duplicated()]\n",
    "    \n",
    "    if upper_diag:\n",
    "        df = df + df.T\n",
    "    return df\n",
    "\n",
    "def save_tree(tree, file_name, save_type='xml'):\n",
    "    \"\"\"Saved the created phylogenetic tree.\"\"\"\n",
    "    \n",
    "    if save_type == 'pickle':\n",
    "        graph = Phylo.to_networkx(tree)\n",
    "        save_pickled(graph, file_name)\n",
    "    elif save_type == 'xml':\n",
    "        Phylo.write(tree, file_name, 'phyloxml')\n",
    "    else:\n",
    "        raise ValueError('Not a correct save type.')\n",
    "    \n",
    "def pandas_dm_to_biopython_dm(dm):\n",
    "    \"\"\"Convert the pandas distance matrix to the biopython distance matrix.\n",
    "    \n",
    "    Returns:\n",
    "        biopython distance matrix\n",
    "    \"\"\"\n",
    "    \n",
    "    accessions = dm.columns\n",
    "    bio_dm = []\n",
    "    for i, accession in enumerate(accessions):\n",
    "        bio_dm.append(list(dm.iloc[i, :i+1].values))\n",
    "        \n",
    "    bio_dm = TreeConstruction._DistanceMatrix(\n",
    "        list(dm.columns), \n",
    "        bio_dm)\n",
    "    \n",
    "    return bio_dm\n",
    "\n",
    "def distance_matrix_to_phylo_tree(dm, outfile=None):\n",
    "    \"\"\"Create a phylogenetic tree from the distance matrix.\"\"\"\n",
    "    \n",
    "    dm = pandas_dm_to_biopython_dm(dm)\n",
    "    \n",
    "    treeConstructor = TreeConstruction.DistanceTreeConstructor()\n",
    "    tree = treeConstructor.nj(dm)\n",
    "    \n",
    "    if outfile is not None:\n",
    "        save_tree(tree, outfile)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# ete3 function, not all of these are used here"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 8,
   "metadata": {},
   "outputs": [],
   "source": [
    "from ete3 import Tree, TreeStyle\n",
    "from ete3 import Phyloxml\n",
    "from ete3 import AttrFace, faces, Tree, NodeStyle, TreeStyle\n",
    "\n",
    "def load_pickled(file_name):\n",
    "    with open(file_name, 'rb') as f:\n",
    "        return pickle.load(f, encoding='latin')\n",
    "\n",
    "\n",
    "def get_farthest_node(tree, sequence):\n",
    "    return (tree&sequence).get_farthest_node()\n",
    "\n",
    "def get_all_accessions_from_tree(tree):\n",
    "    return [leaf_node.name for leaf_node in tree.get_leaves()]\n",
    "\n",
    "def remove_certain_hosts_from_tree(tree, hosts):\n",
    "    \"\"\"Remove leaf nodes if the host of that leaf is in `hosts`\"\"\"\n",
    "    \n",
    "    tree = copy.deepcopy(tree)\n",
    "    \n",
    "    removed_accessions = []\n",
    "    for leaf_node in tree.get_leaves():\n",
    "        if leaf_node.host in hosts:\n",
    "            leaf_node.detach()\n",
    "            \n",
    "    return tree\n",
    "\n",
    "def set_midpoint_outgroup(tree):\n",
    "    tree.set_outgroup(tree.get_midpoint_outgroup())\n",
    "\n",
    "\n",
    "def load_tree(filename, type_='phyloxml'):\n",
    "    \"\"\"Load saved phylogenetic tree.\n",
    "    \"\"\"\n",
    "    \n",
    "    if type_ == 'phyloxml':\n",
    "        project = Phyloxml()\n",
    "        project.build_from_file(filename)\n",
    "\n",
    "        for tree in project.get_phylogeny():\n",
    "            break\n",
    "\n",
    "        t=tree\n",
    "        \n",
    "    elif type_ == 'newick':\n",
    "        t = Tree(filename, format=1)\n",
    "    else:\n",
    "        raise ValueError('Not a correct type.')\n",
    "    \n",
    "    return t"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# construct tree from distance matrix"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 9,
   "metadata": {},
   "outputs": [],
   "source": [
    "if CONSTRUCT_PHYLO:\n",
    "    ALL_dm_ldistance = load_dm(\n",
    "        OUTPUT_DIR + 'dm'+str(threshold)+'.csv', \n",
    "        upper_diag=False)\n",
    "    \n",
    "    distance_matrix_to_phylo_tree(\n",
    "        ALL_dm_ldistance, PHYLO_TREE_DIR + 'ldistance'+str(threshold)+'.xml')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# convert phyloxml tree to newick tree to manipulate trees"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 10,
   "metadata": {},
   "outputs": [],
   "source": [
    "\n",
    "Phylo.convert(\n",
    "    PHYLO_DIR + 'ldistance'+str(threshold)+'.xml','phyloxml',\n",
    "    PHYLO_DIR + 'ldistance'+str(threshold)+'.nhx','newick')\n",
    "\n",
    "ltree = load_tree(\n",
    "    PHYLO_DIR + 'ldistance'+str(threshold)+'.nhx',\n",
    "    type_='newick')"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# label nodes in tree to add other attributes like subtype risk etc"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 11,
   "metadata": {},
   "outputs": [],
   "source": [
    "def bandify(val,min=6.27,max=6.295):\n",
    "    maptoten=int(np.ceil(((val-min)/(max-min))*10))\n",
    "    return ' '+u'\\u2580'*maptoten\n",
    "\n",
    "def label_nodes(\n",
    "        tree, \n",
    "        recordinfo):\n",
    "    \"\"\"Label the nodes of the tree.\n",
    "    \n",
    "    We label nodes on whether:\n",
    "        it is covid19\n",
    "    \"\"\"\n",
    "    \n",
    "    tree = copy.deepcopy(tree)\n",
    "    \n",
    "    for node in tree:\n",
    "        name = node.name      \n",
    "        node.subtype = recordinfo.seq_infos[name].subtype\n",
    "        node.erisk =recordinfo.seq_infos[name].erisk\n",
    "        node.id = recordinfo.seq_infos[name].name + bandify(recordinfo.seq_infos[name].erisk,min=threshold)\n",
    "        if VERBOSE:\n",
    "            print(node.name,node.subtype,node.id,node.erisk)\n",
    "    return tree"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# construct labelled tree"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 12,
   "metadata": {},
   "outputs": [],
   "source": [
    "labelled_tree=label_nodes(\n",
    "    ltree, ALLinfoHA)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# functions to collapse simialr leaves"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 13,
   "metadata": {},
   "outputs": [],
   "source": [
    "def prune_nodes(t):\n",
    "    # collapsed nodes are labeled, so you locate them and prune them\n",
    "    for n in t.search_nodes(collapsed=True):\n",
    "        for ch in n.get_children():\n",
    "            ch.detach()\n",
    "            \n",
    "            \n",
    "def mean(array):\n",
    "    return sum(array)/float(len(array))\n",
    "\n",
    "def cache_distances(tree):\n",
    "    ''' precalculate distances of all nodes to the root''' \n",
    "    node2rootdist = {tree:0}\n",
    "    for node in tree.iter_descendants('preorder'):\n",
    "        node2rootdist[node] = node.dist + node2rootdist[node.up]\n",
    "    return node2rootdist\n",
    "\n",
    "def closest_node(node, node2tips, root_distance):\n",
    "    \"\"\"Find the closest node.\"\"\"\n",
    "    \n",
    "    tips = []\n",
    "    distances = []\n",
    "    for tip in node2tips[node]:\n",
    "        distances.append(root_distance[tip]-root_distance[node])\n",
    "        tips.append(tip)\n",
    "        #     index = np.argmin([root_distance[tip]-root_distance[node] for tip in node2tips[node]])\n",
    "    index = np.argmin(distances)\n",
    "    return tips[index]\n",
    "\n",
    "def riskiest_node(node, node2tips):\n",
    "    \"\"\"Find the closest node.\"\"\"\n",
    "    \n",
    "    tips = []\n",
    "    risks = []\n",
    "    for tip in node2tips[node]:\n",
    "        risks.append(tip.erisk)\n",
    "        tips.append(tip)\n",
    "        #     index = np.argmin([root_distance[tip]-root_distance[node] for tip in node2tips[node]])\n",
    "    index = np.argmax(risks)\n",
    "    return tips[index]\n",
    "\n",
    "def collapse(tree, min_dist,AllrecordInfo):\n",
    "    # cache the tip content of each node to reduce the number of times the tree is traversed\n",
    "    \n",
    "    tree = copy.deepcopy(tree)\n",
    "    \n",
    "    node2tips = tree.get_cached_content()\n",
    "    root_distance = cache_distances(tree)\n",
    "\n",
    "    for node in tree.get_descendants('preorder'):\n",
    "        if not node.is_leaf():\n",
    "            avg_distance_to_tips = mean([root_distance[tip]-root_distance[node]\n",
    "                                         for tip in node2tips[node]])\n",
    "            if VERBOSE:\n",
    "                print(avg_distance_to_tips)\n",
    "            if avg_distance_to_tips < min_dist:\n",
    "                # do whatever, ete support node annotation, deletion, labeling, etc.\n",
    "            \n",
    "                #closest_name = closest_node(node, node2tips, root_distance).name\n",
    "                closest_name = riskiest_node(node, node2tips).name\n",
    "                node.subtype = AllrecordInfo.seq_infos[closest_name].subtype\n",
    "                node.id = AllrecordInfo.seq_infos[closest_name].name + bandify(AllrecordInfo.seq_infos[closest_name].erisk,min=threshold)\n",
    "                node.name = '%s (%g)' %(closest_name,avg_distance_to_tips)\n",
    "                \n",
    "            \n",
    "                node.add_features(collapsed=True)\n",
    "\n",
    "                # set drawing attribute so they look collapsed when displayed with tree.show()\n",
    "                node.img_style['draw_descendants'] = False\n",
    "\n",
    "    return tree"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 14,
   "metadata": {},
   "outputs": [],
   "source": [
    "# collapse leaved"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 15,
   "metadata": {},
   "outputs": [],
   "source": [
    "ltree_collapsed = collapse(\n",
    "    labelled_tree, \n",
    "    min_dist=num_collapsed, \n",
    "    AllrecordInfo=ALLinfoHA)\n",
    "\n",
    "prune_nodes(ltree_collapsed)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# code for actual rendering"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 20,
   "metadata": {},
   "outputs": [],
   "source": [
    "# COLBAT='DarkRed'\n",
    "# COLRAT='SteelBlue'\n",
    "COLH3N2='Blue'\n",
    "COLH1N1='DarkRed'\n",
    "COLH7N9='DarkGreen'\n",
    "COLH9N2='#BD890F'\n",
    "COLDEF='BLACK'\n",
    "\n",
    "FS=50\n",
    "PW=10\n",
    "\n",
    "\n",
    "def nodeAttribConstruct(color, node):\n",
    "    N = AttrFace(\n",
    "        \"id\", fsize=FS, \n",
    "        text_prefix=\" \",penwidth=PW,ftype='Arial',\n",
    "        fgcolor=color,fstyle='bold')\n",
    "    faces.add_face_to_node(N, node, 1, position=\"branch-right\")\n",
    "    return N\n",
    "\n",
    "def layout(node):\n",
    "    if node.is_leaf():\n",
    "        if  node.subtype == 'H1N1':\n",
    "            N = nodeAttribConstruct(COLH1N1,node)\n",
    "        elif node.subtype == 'H3N2':\n",
    "            N = nodeAttribConstruct(COLH3N2,node)\n",
    "        elif node.subtype == 'H7N9':\n",
    "            N = nodeAttribConstruct(COLH7N9,node)\n",
    "        elif node.subtype == 'H9N2':\n",
    "            N = nodeAttribConstruct(COLH9N2,node)\n",
    "        else:\n",
    "            N = nodeAttribConstruct(COLDEF,node)\n",
    "            \n",
    "\n",
    "            \n",
    "def render_tree(tree, outfile):# all_seq_data, display_type='nearest_host'):\n",
    "    \"\"\"Render the tree inside the file to a circular \n",
    "    phylogenetic tree.\n",
    "    \n",
    "    NOTE: outfile should be in .pdf for best visuals\n",
    "    Returns:\n",
    "    \"\"\"\n",
    "    #tree = Tree(nwfile,format=1)\n",
    "\n",
    "    ts = TreeStyle()\n",
    "    ns = NodeStyle()\n",
    "    ts.show_leaf_name = False\n",
    "    #ts.rotation = 90\n",
    "    ts.mode = \"r\"\n",
    "    #ts.arc_start = -360 # 0 degrees = 3 o'clock\n",
    "    #ts.arc_span = 360\n",
    "    ts.scale=5\n",
    "    ts.show_scale=True\n",
    "    ts.branch_vertical_margin = .5 # 10 pixels between adjacent branches\n",
    "    # ts.show_branch_length=True\n",
    "    #ts.min_leaf_separation=10\n",
    "    #ts.optimal_scale_level='full'\n",
    "    #ts.branch_vertical_margin=0\n",
    "    \n",
    "    ns.hz_line_width=2\n",
    "    ns.vt_line_width=1\n",
    "    #ts.layout_fn = layout\n",
    "    ns[\"vt_line_width\"] = 16\n",
    "    ns[\"hz_line_width\"] = 16\n",
    "    #     ns['fsize'] = 20\n",
    "    for n in tree.traverse():\n",
    "        n.set_style(ns)\n",
    "        \n",
    "    #all_accessions = all_seq_data['accessions'].values\n",
    "    for n in tree:\n",
    "        ts.layout_fn = layout\n",
    "\n",
    "        \n",
    "    tree.set_style(ns)\n",
    "    tree.set_style(ts)\n",
    "    \n",
    "    ax=tree.render(\n",
    "        outfile, \n",
    "        dpi=300, \n",
    "        h=500,\n",
    "        tree_style=ts)"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Render phylogenetic trees"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 21,
   "metadata": {},
   "outputs": [],
   "source": [
    "ax=render_tree(\n",
    "    labelled_tree,\n",
    "    './riskyphylo'+str(threshold)+'.pdf')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 22,
   "metadata": {},
   "outputs": [],
   "source": [
    "ax=render_tree(\n",
    "    ltree_collapsed,\n",
    "    './riskyphylo'+str(threshold)+'_collapsed_'+str(num_collapsed)+'.pdf')"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 113,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "Tree node 'Inner1730' (0x7fe11d5c9ba)"
      ]
     },
     "execution_count": 113,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "ltree_collapsed"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "# Find high risk strains which are on distinct branches on phylogenetic tree"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 200,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>subtype</th>\n",
       "      <th>HA accession</th>\n",
       "      <th>NA accession</th>\n",
       "      <th>predicted IRAT impact</th>\n",
       "      <th>predicted IRAT emergence</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>strain</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>A/swine/Missouri/A02524711/2020</th>\n",
       "      <td>H1N1</td>\n",
       "      <td>EPI1818121</td>\n",
       "      <td>EPI1818122</td>\n",
       "      <td>6.064677</td>\n",
       "      <td>6.294432</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A/swine/Indiana/A02524710/2020</th>\n",
       "      <td>H3N2</td>\n",
       "      <td>EPI1818137</td>\n",
       "      <td>EPI1818138</td>\n",
       "      <td>6.046863</td>\n",
       "      <td>6.288258</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A/Camel/Inner_Mongolia/XL/2020</th>\n",
       "      <td>H7N9</td>\n",
       "      <td>EPI2026200</td>\n",
       "      <td>EPI2026202</td>\n",
       "      <td>6.067189</td>\n",
       "      <td>6.273290</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A/swine/North_Carolina/A02479173/2020</th>\n",
       "      <td>H1N1</td>\n",
       "      <td>EPI1780425</td>\n",
       "      <td>EPI1780426</td>\n",
       "      <td>6.064405</td>\n",
       "      <td>6.229882</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A/swine/Kansas/A02479028/2020</th>\n",
       "      <td>H3N2</td>\n",
       "      <td>EPI1777753</td>\n",
       "      <td>EPI1777754</td>\n",
       "      <td>6.000545</td>\n",
       "      <td>6.228595</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A/swine/Iowa/A02525161/2021</th>\n",
       "      <td>H3N2</td>\n",
       "      <td>EPI1909023</td>\n",
       "      <td>EPI1909024</td>\n",
       "      <td>5.890407</td>\n",
       "      <td>6.078649</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A/mink/China/chick_embryo/2020</th>\n",
       "      <td>H9N2</td>\n",
       "      <td>EPI2161544</td>\n",
       "      <td>EPI2161548</td>\n",
       "      <td>5.897720</td>\n",
       "      <td>6.076332</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A/swine/Italy/127069/2020</th>\n",
       "      <td>H1N1</td>\n",
       "      <td>EPI2142221</td>\n",
       "      <td>EPI2142147</td>\n",
       "      <td>5.996242</td>\n",
       "      <td>6.060233</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A/chicken/China/2096/2021</th>\n",
       "      <td>H9N2</td>\n",
       "      <td>EPI2116845</td>\n",
       "      <td>EPI2116843</td>\n",
       "      <td>5.892288</td>\n",
       "      <td>6.057454</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>A/chicken/China/2042/2020</th>\n",
       "      <td>H9N2</td>\n",
       "      <td>EPI2116789</td>\n",
       "      <td>EPI2116787</td>\n",
       "      <td>5.896311</td>\n",
       "      <td>6.057069</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>78 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                      subtype HA accession NA accession  \\\n",
       "strain                                                                    \n",
       "A/swine/Missouri/A02524711/2020          H1N1   EPI1818121   EPI1818122   \n",
       "A/swine/Indiana/A02524710/2020           H3N2   EPI1818137   EPI1818138   \n",
       "A/Camel/Inner_Mongolia/XL/2020           H7N9   EPI2026200   EPI2026202   \n",
       "A/swine/North_Carolina/A02479173/2020    H1N1   EPI1780425   EPI1780426   \n",
       "A/swine/Kansas/A02479028/2020            H3N2   EPI1777753   EPI1777754   \n",
       "...                                       ...          ...          ...   \n",
       "A/swine/Iowa/A02525161/2021              H3N2   EPI1909023   EPI1909024   \n",
       "A/mink/China/chick_embryo/2020           H9N2   EPI2161544   EPI2161548   \n",
       "A/swine/Italy/127069/2020                H1N1   EPI2142221   EPI2142147   \n",
       "A/chicken/China/2096/2021                H9N2   EPI2116845   EPI2116843   \n",
       "A/chicken/China/2042/2020                H9N2   EPI2116789   EPI2116787   \n",
       "\n",
       "                                       predicted IRAT impact  \\\n",
       "strain                                                         \n",
       "A/swine/Missouri/A02524711/2020                     6.064677   \n",
       "A/swine/Indiana/A02524710/2020                      6.046863   \n",
       "A/Camel/Inner_Mongolia/XL/2020                      6.067189   \n",
       "A/swine/North_Carolina/A02479173/2020               6.064405   \n",
       "A/swine/Kansas/A02479028/2020                       6.000545   \n",
       "...                                                      ...   \n",
       "A/swine/Iowa/A02525161/2021                         5.890407   \n",
       "A/mink/China/chick_embryo/2020                      5.897720   \n",
       "A/swine/Italy/127069/2020                           5.996242   \n",
       "A/chicken/China/2096/2021                           5.892288   \n",
       "A/chicken/China/2042/2020                           5.896311   \n",
       "\n",
       "                                       predicted IRAT emergence  \n",
       "strain                                                           \n",
       "A/swine/Missouri/A02524711/2020                        6.294432  \n",
       "A/swine/Indiana/A02524710/2020                         6.288258  \n",
       "A/Camel/Inner_Mongolia/XL/2020                         6.273290  \n",
       "A/swine/North_Carolina/A02479173/2020                  6.229882  \n",
       "A/swine/Kansas/A02479028/2020                          6.228595  \n",
       "...                                                         ...  \n",
       "A/swine/Iowa/A02525161/2021                            6.078649  \n",
       "A/mink/China/chick_embryo/2020                         6.076332  \n",
       "A/swine/Italy/127069/2020                              6.060233  \n",
       "A/chicken/China/2096/2021                              6.057454  \n",
       "A/chicken/China/2042/2020                              6.057069  \n",
       "\n",
       "[78 rows x 5 columns]"
      ]
     },
     "execution_count": 200,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "r=6.057\n",
    "highrisknames=[]\n",
    "for node in ltree_collapsed:\n",
    "    if ALLinfoHA.seq_infos[node.name.split()[0]].erisk > r:\n",
    "        highrisknames=np.append(highrisknames,\n",
    "                                ALLinfoHA.seq_infos[node.name.split()[0]].name)\n",
    "highrisknamesdf=df[df.id.isin(highrisknames)][['id','subtype',\n",
    "                                               'ha_accession',\n",
    "                                               'na_accession',\n",
    "                                               'predicted_impact_score',\n",
    "                                               'predicted_emergence_score']].sort_values('predicted_emergence_score',\n",
    "                                                                                         ascending=False)\n",
    "highrisknamesdf = highrisknamesdf.rename(columns={'id':'strain',\n",
    "                                                  'ha_accession':'HA accession',\n",
    "                                                  'na_accession':'NA accession',\n",
    "                                                  'predicted_impact_score':'predicted IRAT impact',\n",
    "                                                  'predicted_emergence_score':'predicted IRAT emergence'}).set_index('strain')\n",
    "highrisknamesdf#.drop_duplicates()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 204,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>subtype</th>\n",
       "      <th>HA accession</th>\n",
       "      <th>NA accession</th>\n",
       "      <th>predicted IRAT impact</th>\n",
       "      <th>predicted IRAT emergence</th>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>strain</th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "      <th></th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>\\rowcolor{Red3!20}A/swine/Missouri/A02524711/2020</th>\n",
       "      <td>H1N1</td>\n",
       "      <td>EPI1818121</td>\n",
       "      <td>EPI1818122</td>\n",
       "      <td>6.064677</td>\n",
       "      <td>6.294432</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>\\rowcolor{Blue1!30}A/swine/Indiana/A02524710/2020</th>\n",
       "      <td>H3N2</td>\n",
       "      <td>EPI1818137</td>\n",
       "      <td>EPI1818138</td>\n",
       "      <td>6.046863</td>\n",
       "      <td>6.288258</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>\\rowcolor{Green3!50}A/Camel/Inner_Mongolia/XL/2020</th>\n",
       "      <td>H7N9</td>\n",
       "      <td>EPI2026200</td>\n",
       "      <td>EPI2026202</td>\n",
       "      <td>6.067189</td>\n",
       "      <td>6.273290</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>\\rowcolor{Red3!20}A/swine/North_Carolina/A02479173/2020</th>\n",
       "      <td>H1N1</td>\n",
       "      <td>EPI1780425</td>\n",
       "      <td>EPI1780426</td>\n",
       "      <td>6.064405</td>\n",
       "      <td>6.229882</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>\\rowcolor{Blue1!30}A/swine/Kansas/A02479028/2020</th>\n",
       "      <td>H3N2</td>\n",
       "      <td>EPI1777753</td>\n",
       "      <td>EPI1777754</td>\n",
       "      <td>6.000545</td>\n",
       "      <td>6.228595</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>...</th>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "      <td>...</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>\\rowcolor{Blue1!30}A/swine/Iowa/A02525161/2021</th>\n",
       "      <td>H3N2</td>\n",
       "      <td>EPI1909023</td>\n",
       "      <td>EPI1909024</td>\n",
       "      <td>5.890407</td>\n",
       "      <td>6.078649</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>\\rowcolor{DarkOrange!40}A/mink/China/chick_embryo/2020</th>\n",
       "      <td>H9N2</td>\n",
       "      <td>EPI2161544</td>\n",
       "      <td>EPI2161548</td>\n",
       "      <td>5.897720</td>\n",
       "      <td>6.076332</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>\\rowcolor{Red3!20}A/swine/Italy/127069/2020</th>\n",
       "      <td>H1N1</td>\n",
       "      <td>EPI2142221</td>\n",
       "      <td>EPI2142147</td>\n",
       "      <td>5.996242</td>\n",
       "      <td>6.060233</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>\\rowcolor{DarkOrange!40}A/chicken/China/2096/2021</th>\n",
       "      <td>H9N2</td>\n",
       "      <td>EPI2116845</td>\n",
       "      <td>EPI2116843</td>\n",
       "      <td>5.892288</td>\n",
       "      <td>6.057454</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>\\rowcolor{DarkOrange!40}A/chicken/China/2042/2020</th>\n",
       "      <td>H9N2</td>\n",
       "      <td>EPI2116789</td>\n",
       "      <td>EPI2116787</td>\n",
       "      <td>5.896311</td>\n",
       "      <td>6.057069</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "<p>78 rows × 5 columns</p>\n",
       "</div>"
      ],
      "text/plain": [
       "                                                   subtype HA accession  \\\n",
       "strain                                                                    \n",
       "\\rowcolor{Red3!20}A/swine/Missouri/A02524711/2020     H1N1   EPI1818121   \n",
       "\\rowcolor{Blue1!30}A/swine/Indiana/A02524710/2020     H3N2   EPI1818137   \n",
       "\\rowcolor{Green3!50}A/Camel/Inner_Mongolia/XL/2020    H7N9   EPI2026200   \n",
       "\\rowcolor{Red3!20}A/swine/North_Carolina/A02479...    H1N1   EPI1780425   \n",
       "\\rowcolor{Blue1!30}A/swine/Kansas/A02479028/2020      H3N2   EPI1777753   \n",
       "...                                                    ...          ...   \n",
       "\\rowcolor{Blue1!30}A/swine/Iowa/A02525161/2021        H3N2   EPI1909023   \n",
       "\\rowcolor{DarkOrange!40}A/mink/China/chick_embr...    H9N2   EPI2161544   \n",
       "\\rowcolor{Red3!20}A/swine/Italy/127069/2020           H1N1   EPI2142221   \n",
       "\\rowcolor{DarkOrange!40}A/chicken/China/2096/2021     H9N2   EPI2116845   \n",
       "\\rowcolor{DarkOrange!40}A/chicken/China/2042/2020     H9N2   EPI2116789   \n",
       "\n",
       "                                                   NA accession  \\\n",
       "strain                                                            \n",
       "\\rowcolor{Red3!20}A/swine/Missouri/A02524711/2020    EPI1818122   \n",
       "\\rowcolor{Blue1!30}A/swine/Indiana/A02524710/2020    EPI1818138   \n",
       "\\rowcolor{Green3!50}A/Camel/Inner_Mongolia/XL/2020   EPI2026202   \n",
       "\\rowcolor{Red3!20}A/swine/North_Carolina/A02479...   EPI1780426   \n",
       "\\rowcolor{Blue1!30}A/swine/Kansas/A02479028/2020     EPI1777754   \n",
       "...                                                         ...   \n",
       "\\rowcolor{Blue1!30}A/swine/Iowa/A02525161/2021       EPI1909024   \n",
       "\\rowcolor{DarkOrange!40}A/mink/China/chick_embr...   EPI2161548   \n",
       "\\rowcolor{Red3!20}A/swine/Italy/127069/2020          EPI2142147   \n",
       "\\rowcolor{DarkOrange!40}A/chicken/China/2096/2021    EPI2116843   \n",
       "\\rowcolor{DarkOrange!40}A/chicken/China/2042/2020    EPI2116787   \n",
       "\n",
       "                                                    predicted IRAT impact  \\\n",
       "strain                                                                      \n",
       "\\rowcolor{Red3!20}A/swine/Missouri/A02524711/2020                6.064677   \n",
       "\\rowcolor{Blue1!30}A/swine/Indiana/A02524710/2020                6.046863   \n",
       "\\rowcolor{Green3!50}A/Camel/Inner_Mongolia/XL/2020               6.067189   \n",
       "\\rowcolor{Red3!20}A/swine/North_Carolina/A02479...               6.064405   \n",
       "\\rowcolor{Blue1!30}A/swine/Kansas/A02479028/2020                 6.000545   \n",
       "...                                                                   ...   \n",
       "\\rowcolor{Blue1!30}A/swine/Iowa/A02525161/2021                   5.890407   \n",
       "\\rowcolor{DarkOrange!40}A/mink/China/chick_embr...               5.897720   \n",
       "\\rowcolor{Red3!20}A/swine/Italy/127069/2020                      5.996242   \n",
       "\\rowcolor{DarkOrange!40}A/chicken/China/2096/2021                5.892288   \n",
       "\\rowcolor{DarkOrange!40}A/chicken/China/2042/2020                5.896311   \n",
       "\n",
       "                                                    predicted IRAT emergence  \n",
       "strain                                                                        \n",
       "\\rowcolor{Red3!20}A/swine/Missouri/A02524711/2020                   6.294432  \n",
       "\\rowcolor{Blue1!30}A/swine/Indiana/A02524710/2020                   6.288258  \n",
       "\\rowcolor{Green3!50}A/Camel/Inner_Mongolia/XL/2020                  6.273290  \n",
       "\\rowcolor{Red3!20}A/swine/North_Carolina/A02479...                  6.229882  \n",
       "\\rowcolor{Blue1!30}A/swine/Kansas/A02479028/2020                    6.228595  \n",
       "...                                                                      ...  \n",
       "\\rowcolor{Blue1!30}A/swine/Iowa/A02525161/2021                      6.078649  \n",
       "\\rowcolor{DarkOrange!40}A/mink/China/chick_embr...                  6.076332  \n",
       "\\rowcolor{Red3!20}A/swine/Italy/127069/2020                         6.060233  \n",
       "\\rowcolor{DarkOrange!40}A/chicken/China/2096/2021                   6.057454  \n",
       "\\rowcolor{DarkOrange!40}A/chicken/China/2042/2020                   6.057069  \n",
       "\n",
       "[78 rows x 5 columns]"
      ]
     },
     "execution_count": 204,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "COLDICT={'H1N1':'Red3!20','H3N2':'Blue1!30','H7N9':'Green3!50','H9N2':'DarkOrange!40'}\n",
    "def rowcolor(row):\n",
    "    return '\\\\rowcolor{' + COLDICT[row.subtype]+'}' + row['strain']\n",
    "\n",
    "highrisknamesdf1 = highrisknamesdf.reset_index(drop=False)\n",
    "highrisknamesdf1['strain']=highrisknamesdf1.apply(rowcolor,axis=1)\n",
    "highrisknamesdf1=highrisknamesdf1.set_index('strain')\n",
    "highrisknamesdf1"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 208,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "\\begin{tabular}{L{1.2in}|L{.3in}|L{.3in}|L{.3in}|L{.3in}|L{.3in}}\\hline\r\n",
      "name&subtype& HA  accession & NA  accession & predicted  IRAT  impact & predicted  IRAT  emergence \\\\\r\n",
      "A/swine/Missouri/A02524711/2020&H1N1&EPI1818121&EPI1818122&6.1&6.3\\\\\r\n",
      "A/swine/Indiana/A02524710/2020&H3N2&EPI1818137&EPI1818138&6.0&6.3\\\\\r\n",
      " A/Camel/Inner\\_Mongolia/XL/2020 &H7N9&EPI2026200&EPI2026202&6.1&6.3\\\\\r\n",
      " A/swine/North\\_Carolina/A02479173/2020 &H1N1&EPI1780425&EPI1780426&6.1&6.2\\\\\r\n",
      "A/swine/Kansas/A02479028/2020&H3N2&EPI1777753&EPI1777754&6.0&6.2\\\\\r\n",
      "A/swine/Minnesota/A02635976/2021&H1N1&EPI1912208&EPI1912209&6.0&6.2\\\\\r\n",
      "A/swine/Italy/56910/2020&H3N2&EPI2142217&EPI2142173&6.0&6.2\\\\\r\n",
      "A/swine/Chile/VN1401-5054/2020&H3N2&EPI1974975&EPI1974978&6.0&6.2\\\\\r\n",
      "A/swine/Minnesota/A02245643/2020&H3N2&EPI1769178&EPI1769179&6.0&6.2\\\\\r\n",
      "A/swine/Tennessee/A02524414/2022&H1N1&EPI2149257&EPI2149258&6.1&6.2\\\\\r\n",
      "A/swine/Iowa/A02479005/2020&H1N1&EPI1777621&EPI1777622&6.0&6.2\\\\\r\n",
      "A/swine/Iowa/A02524878/2020&H3N2&EPI1907866&EPI1907867&6.0&6.2\\\\\r\n",
      "A/swine/Indiana/A02636638/2022&H1N1&EPI2153370&EPI2153371&6.0&6.2\\\\\r\n",
      "A/swine/Illinois/A02479007/2020&H3N2&EPI1777629&EPI1777630&6.0&6.2\\\\\r\n",
      "A/swine/Iowa/A02524874/2020&H3N2&EPI1907838&EPI1907839&6.0&6.2\\\\\r\n",
      "A/swine/Minnesota/A02245699/2020&H3N2&EPI1833007&EPI1833008&6.0&6.2\\\\\r\n",
      "A/swine/Minnesota/A02248037/2021&H1N1&EPI1912188&EPI1912189&6.0&6.2\\\\\r\n",
      "A/swine/Minnesota/A02711801/2022&H1N1&EPI2153420&EPI2153421&6.0&6.2\\\\\r\n",
      "A/swine/Iowa/A02635917/2021&H1N1&EPI1911753&EPI1911754&6.0&6.2\\\\\r\n",
      "A/swine/Illinois/A02635936/2021&H1N1&EPI1911791&EPI1911792&6.0&6.2\\\\\r\n",
      "A/swine/Minnesota/A02711797/2022&H3N2&EPI2153382&EPI2153383&6.0&6.1\\\\\r\n",
      "A/swine/Spain/44579-1/2020&H3N2&EPI1930744&EPI1930748&6.0&6.1\\\\\r\n",
      "A/swine/Minnesota/A02248061/2021&H1N1&EPI1912494&EPI1912495&6.0&6.1\\\\\r\n",
      " A/swine/South\\_Dakota/A02524914/2020 &H3N2&EPI1908070&EPI1908071&6.0&6.1\\\\\r\n",
      " A/swine/South\\_Dakota/A02524453/2020 &H1N1&EPI1765555&EPI1765556&6.0&6.1\\\\\r\n",
      "A/swine/Minnesota/A02248060/2021&H1N1&EPI1912500&EPI1912501&6.0&6.1\\\\\r\n",
      "A/swine/Iowa/A02636439/2022&H1N1&EPI2147475&EPI2147476&6.0&6.1\\\\\r\n",
      "A/swine/Iowa/A02524875/2020&H1N1&EPI1907858&EPI1907859&6.0&6.1\\\\\r\n",
      "A/swine/Nebraska/A02636117/2021&H1N1&EPI1932937&EPI1932938&6.0&6.1\\\\\r\n",
      "A/swine/Iowa/A02524513/2020&H1N1&EPI1832647&EPI1832648&6.0&6.1\\\\\r\n",
      "A/swine/Nebraska/A02524954/2020&H1N1&EPI1908393&EPI1908394&6.0&6.1\\\\\r\n",
      "A/swine/Iowa/A02524994/2020&H1N1&EPI1908427&EPI1908428&6.0&6.1\\\\\r\n",
      "A/swine/Minnesota/A02245424/2020&H1N1&EPI1780207&EPI1780208&6.0&6.1\\\\\r\n",
      "A/swine/Minnesota/A02479051/2020&H1N1&EPI1778572&EPI1778573&6.0&6.1\\\\\r\n",
      "A/swine/Iowa/A02524724/2020&H1N1&EPI1818387&EPI1818388&6.0&6.1\\\\\r\n",
      "A/swine/Iowa/A02524892/2020&H1N1&EPI1907881&EPI1907882&6.0&6.1\\\\\r\n",
      "A/swine/Iowa/A02524646/2020&H1N1&EPI1817164&EPI1817165&6.0&6.1\\\\\r\n",
      "A/swine/Nebraska/A02479212/2020&H1N1&EPI1775884&EPI1775885&6.0&6.1\\\\\r\n",
      "A/swine/Iowa/A02525345/2021&H1N1&EPI1910807&EPI1910808&6.0&6.1\\\\\r\n",
      "A/swine/Iowa/A02479383/2020&H1N1&EPI1771027&EPI1771028&6.0&6.1\\\\\r\n",
      "A/swine/Nebraska/A02479337/2020&H1N1&EPI1769116&EPI1769117&6.0&6.1\\\\\r\n",
      "A/swine/Iowa/A02525313/2021&H1N1&EPI1910761&EPI1910762&6.0&6.1\\\\\r\n",
      "A/swine/Missouri/A02525065/2021&H1N1&EPI1908581&EPI1908582&6.0&6.1\\\\\r\n",
      "A/swine/Missouri/A02524951/2020&H1N1&EPI1908429&EPI1908430&6.0&6.1\\\\\r\n",
      "A/swine/Nebraska/A02479186/2020&H1N1&EPI1774141&EPI1774142&6.0&6.1\\\\\r\n",
      "A/swine/Iowa/A02479156/2020&H1N1&EPI1780249&EPI1780250&6.0&6.1\\\\\r\n",
      "A/swine/Iowa/A02479229/2020&H1N1&EPI1775914&EPI1775915&6.0&6.1\\\\\r\n",
      "A/swine/Iowa/A02524739/2020&H1N1&EPI1818383&EPI1818384&6.0&6.1\\\\\r\n",
      "A/swine/Minnesota/A02245881/2021&H1N1&EPI1908939&EPI1908940&6.0&6.1\\\\\r\n",
      "A/swine/Iowa/A02479303/2020&H1N1&EPI1768639&EPI1768640&6.0&6.1\\\\\r\n",
      "A/swine/Iowa/A02635823/2021&H1N1&EPI1911263&EPI1911264&6.0&6.1\\\\\r\n",
      "A/swine/Iowa/A02525354/2021&H1N1&EPI1910789&EPI1910790&6.0&6.1\\\\\r\n",
      "A/swine/Iowa/A02635881/2021&H1N1&EPI1911668&EPI1911669&6.0&6.1\\\\\r\n",
      "A/swine/Iowa/A02635955/2021&H1N1&EPI1912240&EPI1912241&6.0&6.1\\\\\r\n",
      "A/swine/Iowa/A02245587/2020&H1N1&EPI1775817&EPI1775818&6.0&6.1\\\\\r\n",
      "A/swine/Illinois/A02525253/2021&H3N2&EPI1910375&EPI1910376&6.0&6.1\\\\\r\n",
      "A/swine/Iowa/A02479141/2020&H1N1&EPI1780241&EPI1780242&6.0&6.1\\\\\r\n",
      "A/swine/Iowa/A02635871/2021&H1N1&EPI1911656&EPI1911657&6.0&6.1\\\\\r\n",
      "A/swine/Iowa/A02479067/2020&H1N1&EPI1778734&EPI1778735&6.0&6.1\\\\\r\n",
      "A/swine/Iowa/A02750621/2022&H1N1&EPI2161576&EPI2161577&6.0&6.1\\\\\r\n",
      "A/swine/Minnesota/A02525348/2021&H1N1&EPI1910795&EPI1910796&6.0&6.1\\\\\r\n",
      "A/swine/Iowa/A02525217/2021&H1N1&EPI1909087&EPI1909088&5.9&6.1\\\\\r\n",
      "A/swine/Iowa/A02636145/2021&H1N1&EPI1932055&EPI1932930&5.9&6.1\\\\\r\n",
      "A/swine/Iowa/A02636114/2021&H1N1&EPI1931853&EPI1931854&5.9&6.1\\\\\r\n",
      "A/swine/Minnesota/A02246459/2021&H1N1&EPI1912518&EPI1912519&5.9&6.1\\\\\r\n",
      "A/swine/Iowa/A02479306/2020&H1N1&EPI1768625&EPI1768626&5.9&6.1\\\\\r\n",
      "A/swine/Iowa/A02479343/2020&H1N1&EPI1769114&EPI1769115&5.9&6.1\\\\\r\n",
      "A/swine/Colorado/A02710706/2022&H3N2&EPI2176699&EPI2176700&5.9&6.1\\\\\r\n",
      "A/swine/Kansas/A02245381/2020(H1N1)&H1N1&EPI1777723&EPI1777724&6.0&6.1\\\\\r\n",
      "A/swine/Iowa/A02246996/2021&H1N1&EPI2146133&EPI2146134&5.9&6.1\\\\\r\n",
      "\\hline\\end{tabular}\r\n"
     ]
    }
   ],
   "source": [
    "from zedstat.textable import textable\n",
    "         tabname='../../../tex/Figures/tabdata/highrisk.tex',\n",
    "         FORMAT='%1.4f',INDEX=True,\n",
    "         TABFORMAT='L{1.95in}|L{.25in}|L{.60in}|L{.6in}|C{1in}|C{1in}',LNTERM='\\\\\\\\\\n')\n",
    "! cat highrisk.tex"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 166,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>H1N1</th>\n",
       "      <th>H3N2</th>\n",
       "      <th>H7N9</th>\n",
       "      <th>H9N2</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>A/swine/Missouri/A02524711/2020</td>\n",
       "      <td>A/swine/Indiana/A02524710/2020</td>\n",
       "      <td>A/Camel/Inner_Mongolia/XL/2020</td>\n",
       "      <td>A/mink/China/chick_embryo/2020</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>6.294432</td>\n",
       "      <td>6.288258</td>\n",
       "      <td>6.27329</td>\n",
       "      <td>6.076332</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                              H1N1                            H3N2  \\\n",
       "0  A/swine/Missouri/A02524711/2020  A/swine/Indiana/A02524710/2020   \n",
       "1                         6.294432                        6.288258   \n",
       "\n",
       "                             H7N9                            H9N2  \n",
       "0  A/Camel/Inner_Mongolia/XL/2020  A/mink/China/chick_embryo/2020  \n",
       "1                         6.27329                        6.076332  "
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>6.00</th>\n",
       "      <th>6.10</th>\n",
       "      <th>6.20</th>\n",
       "      <th>6.25</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>H1N1</th>\n",
       "      <td>63</td>\n",
       "      <td>55</td>\n",
       "      <td>3</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>H3N2</th>\n",
       "      <td>17</td>\n",
       "      <td>14</td>\n",
       "      <td>4</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>H7N9</th>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "      <td>1</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>H9N2</th>\n",
       "      <td>3</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "      <td>0</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "      6.00  6.10  6.20  6.25\n",
       "H1N1    63    55     3     1\n",
       "H3N2    17    14     4     1\n",
       "H7N9     1     1     1     1\n",
       "H9N2     3     0     0     0"
      ]
     },
     "metadata": {},
     "output_type": "display_data"
    },
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "84\n"
     ]
    },
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>id</th>\n",
       "      <th>subtype</th>\n",
       "      <th>ha_accession</th>\n",
       "      <th>na_accession</th>\n",
       "      <th>ha</th>\n",
       "      <th>na</th>\n",
       "      <th>HA_Avg_Qdist</th>\n",
       "      <th>ha_variance</th>\n",
       "      <th>NA_Avg_Qdist</th>\n",
       "      <th>na_variance</th>\n",
       "      <th>Geometric_Mean</th>\n",
       "      <th>predicted_impact_score</th>\n",
       "      <th>predicted_emergence_score</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>218</th>\n",
       "      <td>A/swine/Missouri/A02524711/2020</td>\n",
       "      <td>H1N1</td>\n",
       "      <td>EPI1818121</td>\n",
       "      <td>EPI1818122</td>\n",
       "      <td>MKAILVVMLYTFTTANADTLCIGYHANNSTDTVDTVLEKNVTVTHS...</td>\n",
       "      <td>MNPNQKIITIGSICMTIGTANLILQIGNIISIWVSHSIQIGNQSQI...</td>\n",
       "      <td>0.007931</td>\n",
       "      <td>9.678707e-06</td>\n",
       "      <td>0.001876</td>\n",
       "      <td>2.438430e-06</td>\n",
       "      <td>0.003857</td>\n",
       "      <td>6.064677</td>\n",
       "      <td>6.294432</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1902</th>\n",
       "      <td>A/swine/Indiana/A02524710/2020</td>\n",
       "      <td>H3N2</td>\n",
       "      <td>EPI1818137</td>\n",
       "      <td>EPI1818138</td>\n",
       "      <td>MKTIIALSYILCLVFAQKIPGNDNGTATLCLGHHAVPNGTIVKTIT...</td>\n",
       "      <td>MNPNQKIITIGSVSLTISTICFFMQIAILITTIALHFKQYEFNSPP...</td>\n",
       "      <td>0.004542</td>\n",
       "      <td>4.497950e-07</td>\n",
       "      <td>0.006619</td>\n",
       "      <td>1.048722e-06</td>\n",
       "      <td>0.005483</td>\n",
       "      <td>6.046863</td>\n",
       "      <td>6.288258</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5814</th>\n",
       "      <td>A/Camel/Inner_Mongolia/XL/2020</td>\n",
       "      <td>H7N9</td>\n",
       "      <td>EPI2026200</td>\n",
       "      <td>EPI2026202</td>\n",
       "      <td>MNTQILVFALIAIIPTNADKICLGHHAVSNGTKVNTLTEREVEVVN...</td>\n",
       "      <td>MNPNQKILCTSATAITIGAIAVLIGIANLGLNIGLHLKPGCNCSHS...</td>\n",
       "      <td>0.014006</td>\n",
       "      <td>2.854999e-03</td>\n",
       "      <td>0.001182</td>\n",
       "      <td>2.218687e-07</td>\n",
       "      <td>0.004069</td>\n",
       "      <td>6.067189</td>\n",
       "      <td>6.273290</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6049</th>\n",
       "      <td>A/mink/China/chick_embryo/2020</td>\n",
       "      <td>H9N2</td>\n",
       "      <td>EPI2161544</td>\n",
       "      <td>EPI2161548</td>\n",
       "      <td>METVSLITILLAATVSNADKICIGYQSSNSTETVDTLTENNVPVTH...</td>\n",
       "      <td>MNPNQKITAIGSVSLIIAIICLLMQIAILTMTMTLHFGQKECSNPS...</td>\n",
       "      <td>0.016908</td>\n",
       "      <td>1.293100e-04</td>\n",
       "      <td>0.046149</td>\n",
       "      <td>1.553612e-02</td>\n",
       "      <td>0.027933</td>\n",
       "      <td>5.897720</td>\n",
       "      <td>6.076332</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "                                   id subtype ha_accession na_accession  \\\n",
       "218   A/swine/Missouri/A02524711/2020    H1N1   EPI1818121   EPI1818122   \n",
       "1902   A/swine/Indiana/A02524710/2020    H3N2   EPI1818137   EPI1818138   \n",
       "5814   A/Camel/Inner_Mongolia/XL/2020    H7N9   EPI2026200   EPI2026202   \n",
       "6049   A/mink/China/chick_embryo/2020    H9N2   EPI2161544   EPI2161548   \n",
       "\n",
       "                                                     ha  \\\n",
       "218   MKAILVVMLYTFTTANADTLCIGYHANNSTDTVDTVLEKNVTVTHS...   \n",
       "1902  MKTIIALSYILCLVFAQKIPGNDNGTATLCLGHHAVPNGTIVKTIT...   \n",
       "5814  MNTQILVFALIAIIPTNADKICLGHHAVSNGTKVNTLTEREVEVVN...   \n",
       "6049  METVSLITILLAATVSNADKICIGYQSSNSTETVDTLTENNVPVTH...   \n",
       "\n",
       "                                                     na  HA_Avg_Qdist  \\\n",
       "218   MNPNQKIITIGSICMTIGTANLILQIGNIISIWVSHSIQIGNQSQI...      0.007931   \n",
       "1902  MNPNQKIITIGSVSLTISTICFFMQIAILITTIALHFKQYEFNSPP...      0.004542   \n",
       "5814  MNPNQKILCTSATAITIGAIAVLIGIANLGLNIGLHLKPGCNCSHS...      0.014006   \n",
       "6049  MNPNQKITAIGSVSLIIAIICLLMQIAILTMTMTLHFGQKECSNPS...      0.016908   \n",
       "\n",
       "       ha_variance  NA_Avg_Qdist   na_variance  Geometric_Mean  \\\n",
       "218   9.678707e-06      0.001876  2.438430e-06        0.003857   \n",
       "1902  4.497950e-07      0.006619  1.048722e-06        0.005483   \n",
       "5814  2.854999e-03      0.001182  2.218687e-07        0.004069   \n",
       "6049  1.293100e-04      0.046149  1.553612e-02        0.027933   \n",
       "\n",
       "      predicted_impact_score  predicted_emergence_score  \n",
       "218                 6.064677                   6.294432  \n",
       "1902                6.046863                   6.288258  \n",
       "5814                6.067189                   6.273290  \n",
       "6049                5.897720                   6.076332  "
      ]
     },
     "execution_count": 166,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "count=0\n",
    "Subtype={'H1N1':0,'H3N2':0,'H7N9':0,'H9N2':0}\n",
    "MaxriskStrain={'H1N1':None,'H3N2':None,'H7N9':None,'H9N2':None}\n",
    "Subtype_strat={6:{'H1N1':0,'H3N2':0,'H7N9':0,'H9N2':0},\n",
    "               6.1:{'H1N1':0,'H3N2':0,'H7N9':0,'H9N2':0},\n",
    "               6.2:{'H1N1':0,'H3N2':0,'H7N9':0,'H9N2':0},\n",
    "               6.25:{'H1N1':0,'H3N2':0,'H7N9':0,'H9N2':0}}\n",
    "for node in ltree_collapsed:\n",
    "    Subtype[node.subtype]=Subtype[node.subtype]+1\n",
    "    if MaxriskStrain[node.subtype] is None:\n",
    "        MaxriskStrain[node.subtype]=(ALLinfoHA.seq_infos[node.name.split()[0]].name,\n",
    "                                     ALLinfoHA.seq_infos[node.name.split()[0]].erisk)\n",
    "    else:\n",
    "        if MaxriskStrain[node.subtype][1]<ALLinfoHA.seq_infos[node.name.split()[0]].erisk:\n",
    "            MaxriskStrain[node.subtype]=(ALLinfoHA.seq_infos[node.name.split()[0]].name,\n",
    "                                         ALLinfoHA.seq_infos[node.name.split()[0]].erisk)\n",
    "    for r in [6.25,6.2,6.1,6]:\n",
    "        if r<ALLinfoHA.seq_infos[node.name.split()[0]].erisk:\n",
    "            #print(node.subtype,ALLinfoHA.seq_infos[node.name.split()[0]].erisk)\n",
    "            Subtype_strat[r][node.subtype]=Subtype_strat[r][node.subtype]+1\n",
    "        \n",
    "    count=count+1\n",
    "maxriskdf = pd.DataFrame(MaxriskStrain)\n",
    "Subtype_strat_df = pd.DataFrame(Subtype_strat)\n",
    "display(maxriskdf)\n",
    "display(Subtype_strat_df)\n",
    "print(count)\n",
    "df[df.id.isin(maxriskdf.iloc[0,:].values)]"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 33,
   "metadata": {},
   "outputs": [
    {
     "name": "stdout",
     "output_type": "stream",
     "text": [
      "-rw-rw-r--. 1 ishanu ishanu 164K Nov 14 12:19 ha.fasta\r\n",
      "-rw-rw-r--. 1 ishanu ishanu  569 Nov 15 12:05 EPI2161544.fasta\r\n",
      "-rw-rw-r--. 1 ishanu ishanu  569 Nov 15 12:05 EPI2026200.fasta\r\n",
      "-rw-rw-r--. 1 ishanu ishanu  569 Nov 15 12:05 EPI1818137.fasta\r\n",
      "-rw-rw-r--. 1 ishanu ishanu  569 Nov 15 12:05 EPI1818121.fasta\r\n"
     ]
    }
   ],
   "source": [
    "def write_fasta(seqs, fasta_file, wrap=80):\n",
    "    \"\"\"Write sequences to a fasta file.\n",
    "\n",
    "    Parameters\n",
    "    ----------\n",
    "    seqs : dict[seq_id] -> seq\n",
    "        Sequences indexed by sequence id.\n",
    "    fasta_file : str\n",
    "        Path to write the sequences to.\n",
    "    wrap: int\n",
    "        Number of AA/NT before the line is wrapped.\n",
    "    \"\"\"\n",
    "    with open(fasta_file, 'w') as f:\n",
    "        for gid, gseq in seqs.items():\n",
    "            f.write('>{}\\n'.format(gid))\n",
    "            for i in range(0, len(gseq), wrap):\n",
    "                f.write('{}\\n'.format(gseq[i:i + wrap])) \n",
    "\n",
    "for i,name in zip(df[df.id.isin(maxriskdf.iloc[0,:].values)].ha.values,\n",
    "                 df[df.id.isin(maxriskdf.iloc[0,:].values)].ha_accession.values):\n",
    "    write_fasta({name:i},name+'.fasta')\n",
    "    \n",
    "! ls -lhtr *fasta"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": null,
   "metadata": {},
   "outputs": [],
   "source": [
    "from Bio import SeqIO\n",
    "from Bio.Blast import NCBIWWW\n",
    "my_query = SeqIO.read(\"EPI1818121.fasta\", format=\"fasta\")\n",
    "result_handle = NCBIWWW.qblast(\"blastp\", \"nr\", my_query.seq)\n",
    "blast_result = open(\"my_blast.xml\", \"w\")\n",
    "blast_result.write(result_handle.read())\n",
    "blast_result.close()\n",
    "result_handle.close()"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 92,
   "metadata": {},
   "outputs": [],
   "source": [
    "def parse_blast(resultfile): #takes in the BLAST result, outputs list that can be made into csv\n",
    "    from Bio.Blast import NCBIXML\n",
    "    result_handle = open(resultfile)\n",
    "    blast_records = NCBIXML.parse(result_handle)\n",
    "    csv_list = []\n",
    "    \n",
    "    header = [  'accession','Query',\n",
    "                'Name', 'Length', 'Score', 'Expect',\n",
    "                'QueryStart', 'QueryEnd',\n",
    "                'SubjectStart', 'SubjectEnd','pct'\n",
    "            ]\n",
    "    \n",
    "    #csv_list.append(header)\n",
    "    count = 0\n",
    "    for blast_record in blast_records:\n",
    "        '''help(blast_record.alignments[0].hsps[0])''' # these give help info for the parts \n",
    "        '''help(blast_record.alignments[0])        '''\n",
    "        count +=1\n",
    "        \n",
    "        query = blast_record.query\n",
    "        for alignment in blast_record.alignments:\n",
    "\n",
    "            name = alignment.title\n",
    "            length = alignment.length\n",
    "    \n",
    "            hsp = alignment.hsps[0] # I don't know if we will ever have more than one, so might as well take the first one.\n",
    "            score = hsp.score\n",
    "            expect = hsp.expect\n",
    "            querystart = hsp.query_start\n",
    "            queryend = hsp.query_end\n",
    "            subjectstart = hsp.sbjct_start\n",
    "            subjectend = hsp.sbjct_end\n",
    "            pct=hsp.positives/hsp.align_length\n",
    "            accession=alignment.accession\n",
    "            row = [accession,query,name,length,score,expect,querystart,queryend,subjectstart,subjectend,pct]\n",
    "            csv_list.append(row)\n",
    "            \n",
    "    result_handle.close()\n",
    "    return pd.DataFrame(csv_list,columns=header)\n",
    " "
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 94,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/html": [
       "<div>\n",
       "<style scoped>\n",
       "    .dataframe tbody tr th:only-of-type {\n",
       "        vertical-align: middle;\n",
       "    }\n",
       "\n",
       "    .dataframe tbody tr th {\n",
       "        vertical-align: top;\n",
       "    }\n",
       "\n",
       "    .dataframe thead th {\n",
       "        text-align: right;\n",
       "    }\n",
       "</style>\n",
       "<table border=\"1\" class=\"dataframe\">\n",
       "  <thead>\n",
       "    <tr style=\"text-align: right;\">\n",
       "      <th></th>\n",
       "      <th>accession</th>\n",
       "      <th>Query</th>\n",
       "      <th>Name</th>\n",
       "      <th>Length</th>\n",
       "      <th>Score</th>\n",
       "      <th>Expect</th>\n",
       "      <th>QueryStart</th>\n",
       "      <th>QueryEnd</th>\n",
       "      <th>SubjectStart</th>\n",
       "      <th>SubjectEnd</th>\n",
       "      <th>pct</th>\n",
       "    </tr>\n",
       "  </thead>\n",
       "  <tbody>\n",
       "    <tr>\n",
       "      <th>0</th>\n",
       "      <td>QIA58660</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QIA58660.1| hemagglutinin [Influenza A viru...</td>\n",
       "      <td>566</td>\n",
       "      <td>2969.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>1</th>\n",
       "      <td>QKG28282</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QKG28282.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2967.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>2</th>\n",
       "      <td>QIZ32536</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QIZ32536.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2967.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>3</th>\n",
       "      <td>QJI52882</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QJI52882.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2967.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>4</th>\n",
       "      <td>QJT24758</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QJT24758.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2967.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>5</th>\n",
       "      <td>QKV49724</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QKV49724.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2967.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>6</th>\n",
       "      <td>QIQ45861</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QIQ45861.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2967.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>7</th>\n",
       "      <td>QKM75781</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QKM75781.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2966.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>8</th>\n",
       "      <td>QJX57625</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QJX57625.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2966.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>9</th>\n",
       "      <td>QIQ45830</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QIQ45830.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2966.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>10</th>\n",
       "      <td>QOY44156</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QOY44156.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2966.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>11</th>\n",
       "      <td>QIQ45216</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QIQ45216.1| hemagglutinin [Influenza A viru...</td>\n",
       "      <td>566</td>\n",
       "      <td>2965.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>12</th>\n",
       "      <td>QMX87502</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QMX87502.1| hemagglutinin [Influenza A viru...</td>\n",
       "      <td>566</td>\n",
       "      <td>2965.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>13</th>\n",
       "      <td>QJD23117</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QJD23117.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2965.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>14</th>\n",
       "      <td>QIQ45677</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QIQ45677.1| hemagglutinin [Influenza A viru...</td>\n",
       "      <td>566</td>\n",
       "      <td>2965.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>15</th>\n",
       "      <td>QJT24395</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QJT24395.1| hemagglutinin [Influenza A viru...</td>\n",
       "      <td>566</td>\n",
       "      <td>2965.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>16</th>\n",
       "      <td>QIQ45346</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QIQ45346.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2965.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>17</th>\n",
       "      <td>QIZ32395</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QIZ32395.1| hemagglutinin [Influenza A viru...</td>\n",
       "      <td>566</td>\n",
       "      <td>2964.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>18</th>\n",
       "      <td>QIZ32042</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QIZ32042.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2964.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>19</th>\n",
       "      <td>QIZ33251</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QIZ33251.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2964.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>0.998182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>20</th>\n",
       "      <td>QIQ44996</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QIQ44996.1| hemagglutinin [Influenza A viru...</td>\n",
       "      <td>566</td>\n",
       "      <td>2964.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>0.998182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>21</th>\n",
       "      <td>QTJ06578</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QTJ06578.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2964.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>0.998182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>22</th>\n",
       "      <td>QJT24499</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QJT24499.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2964.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>23</th>\n",
       "      <td>QIV14744</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QIV14744.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2964.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>24</th>\n",
       "      <td>QIQ45654</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QIQ45654.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2964.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>25</th>\n",
       "      <td>QIA57747</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QIA57747.1| hemagglutinin [Influenza A viru...</td>\n",
       "      <td>566</td>\n",
       "      <td>2964.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>0.998182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>26</th>\n",
       "      <td>QIZ31801</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QIZ31801.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2964.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>0.998182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>27</th>\n",
       "      <td>QKM75836</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QKM75836.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2964.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>28</th>\n",
       "      <td>QKS74061</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QKS74061.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2964.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>29</th>\n",
       "      <td>QIH95333</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QIH95333.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2963.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>30</th>\n",
       "      <td>QIZ32602</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QIZ32602.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2963.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>0.998182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>31</th>\n",
       "      <td>QJT24351</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QJT24351.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2963.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>0.998182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>32</th>\n",
       "      <td>QLV95699</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QLV95699.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2963.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>33</th>\n",
       "      <td>QKM76085</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QKM76085.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2963.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>0.998182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>34</th>\n",
       "      <td>QJI52636</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QJI52636.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2963.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>35</th>\n",
       "      <td>QIQ45093</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QIQ45093.1| hemagglutinin [Influenza A viru...</td>\n",
       "      <td>566</td>\n",
       "      <td>2963.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>0.998182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>36</th>\n",
       "      <td>QJT24428</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QJT24428.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2963.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>0.998182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>37</th>\n",
       "      <td>QIA58044</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QIA58044.1| hemagglutinin [Influenza A viru...</td>\n",
       "      <td>566</td>\n",
       "      <td>2963.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>0.998182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>38</th>\n",
       "      <td>QIQ45082</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QIQ45082.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2963.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>0.998182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>39</th>\n",
       "      <td>QJT24466</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QJT24466.1| hemagglutinin [Influenza A viru...</td>\n",
       "      <td>566</td>\n",
       "      <td>2963.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>40</th>\n",
       "      <td>QIZ31721</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QIZ31721.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2962.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1.000000</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>41</th>\n",
       "      <td>QTK18161</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QTK18161.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2962.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>0.998182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>42</th>\n",
       "      <td>QIV14634</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QIV14634.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2962.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>0.998182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>43</th>\n",
       "      <td>QJI52715</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QJI52715.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2962.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>0.998182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>44</th>\n",
       "      <td>QTK18105</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QTK18105.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2962.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>0.998182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>45</th>\n",
       "      <td>QIZ32778</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QIZ32778.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2962.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>0.998182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>46</th>\n",
       "      <td>QIZ33133</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QIZ33133.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2962.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>0.998182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>47</th>\n",
       "      <td>QIZ32822</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QIZ32822.1| hemagglutinin [Influenza A viru...</td>\n",
       "      <td>566</td>\n",
       "      <td>2962.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>0.998182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>48</th>\n",
       "      <td>QJT24659</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QJT24659.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2962.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>0.998182</td>\n",
       "    </tr>\n",
       "    <tr>\n",
       "      <th>49</th>\n",
       "      <td>QJT24340</td>\n",
       "      <td>unnamed protein product</td>\n",
       "      <td>gb|QJT24340.1| hemagglutinin [Influenza A virus]</td>\n",
       "      <td>566</td>\n",
       "      <td>2962.0</td>\n",
       "      <td>0.0</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>1</td>\n",
       "      <td>550</td>\n",
       "      <td>0.998182</td>\n",
       "    </tr>\n",
       "  </tbody>\n",
       "</table>\n",
       "</div>"
      ],
      "text/plain": [
       "   accession                    Query  \\\n",
       "0   QIA58660  unnamed protein product   \n",
       "1   QKG28282  unnamed protein product   \n",
       "2   QIZ32536  unnamed protein product   \n",
       "3   QJI52882  unnamed protein product   \n",
       "4   QJT24758  unnamed protein product   \n",
       "5   QKV49724  unnamed protein product   \n",
       "6   QIQ45861  unnamed protein product   \n",
       "7   QKM75781  unnamed protein product   \n",
       "8   QJX57625  unnamed protein product   \n",
       "9   QIQ45830  unnamed protein product   \n",
       "10  QOY44156  unnamed protein product   \n",
       "11  QIQ45216  unnamed protein product   \n",
       "12  QMX87502  unnamed protein product   \n",
       "13  QJD23117  unnamed protein product   \n",
       "14  QIQ45677  unnamed protein product   \n",
       "15  QJT24395  unnamed protein product   \n",
       "16  QIQ45346  unnamed protein product   \n",
       "17  QIZ32395  unnamed protein product   \n",
       "18  QIZ32042  unnamed protein product   \n",
       "19  QIZ33251  unnamed protein product   \n",
       "20  QIQ44996  unnamed protein product   \n",
       "21  QTJ06578  unnamed protein product   \n",
       "22  QJT24499  unnamed protein product   \n",
       "23  QIV14744  unnamed protein product   \n",
       "24  QIQ45654  unnamed protein product   \n",
       "25  QIA57747  unnamed protein product   \n",
       "26  QIZ31801  unnamed protein product   \n",
       "27  QKM75836  unnamed protein product   \n",
       "28  QKS74061  unnamed protein product   \n",
       "29  QIH95333  unnamed protein product   \n",
       "30  QIZ32602  unnamed protein product   \n",
       "31  QJT24351  unnamed protein product   \n",
       "32  QLV95699  unnamed protein product   \n",
       "33  QKM76085  unnamed protein product   \n",
       "34  QJI52636  unnamed protein product   \n",
       "35  QIQ45093  unnamed protein product   \n",
       "36  QJT24428  unnamed protein product   \n",
       "37  QIA58044  unnamed protein product   \n",
       "38  QIQ45082  unnamed protein product   \n",
       "39  QJT24466  unnamed protein product   \n",
       "40  QIZ31721  unnamed protein product   \n",
       "41  QTK18161  unnamed protein product   \n",
       "42  QIV14634  unnamed protein product   \n",
       "43  QJI52715  unnamed protein product   \n",
       "44  QTK18105  unnamed protein product   \n",
       "45  QIZ32778  unnamed protein product   \n",
       "46  QIZ33133  unnamed protein product   \n",
       "47  QIZ32822  unnamed protein product   \n",
       "48  QJT24659  unnamed protein product   \n",
       "49  QJT24340  unnamed protein product   \n",
       "\n",
       "                                                 Name  Length   Score  Expect  \\\n",
       "0   gb|QIA58660.1| hemagglutinin [Influenza A viru...     566  2969.0     0.0   \n",
       "1    gb|QKG28282.1| hemagglutinin [Influenza A virus]     566  2967.0     0.0   \n",
       "2    gb|QIZ32536.1| hemagglutinin [Influenza A virus]     566  2967.0     0.0   \n",
       "3    gb|QJI52882.1| hemagglutinin [Influenza A virus]     566  2967.0     0.0   \n",
       "4    gb|QJT24758.1| hemagglutinin [Influenza A virus]     566  2967.0     0.0   \n",
       "5    gb|QKV49724.1| hemagglutinin [Influenza A virus]     566  2967.0     0.0   \n",
       "6    gb|QIQ45861.1| hemagglutinin [Influenza A virus]     566  2967.0     0.0   \n",
       "7    gb|QKM75781.1| hemagglutinin [Influenza A virus]     566  2966.0     0.0   \n",
       "8    gb|QJX57625.1| hemagglutinin [Influenza A virus]     566  2966.0     0.0   \n",
       "9    gb|QIQ45830.1| hemagglutinin [Influenza A virus]     566  2966.0     0.0   \n",
       "10   gb|QOY44156.1| hemagglutinin [Influenza A virus]     566  2966.0     0.0   \n",
       "11  gb|QIQ45216.1| hemagglutinin [Influenza A viru...     566  2965.0     0.0   \n",
       "12  gb|QMX87502.1| hemagglutinin [Influenza A viru...     566  2965.0     0.0   \n",
       "13   gb|QJD23117.1| hemagglutinin [Influenza A virus]     566  2965.0     0.0   \n",
       "14  gb|QIQ45677.1| hemagglutinin [Influenza A viru...     566  2965.0     0.0   \n",
       "15  gb|QJT24395.1| hemagglutinin [Influenza A viru...     566  2965.0     0.0   \n",
       "16   gb|QIQ45346.1| hemagglutinin [Influenza A virus]     566  2965.0     0.0   \n",
       "17  gb|QIZ32395.1| hemagglutinin [Influenza A viru...     566  2964.0     0.0   \n",
       "18   gb|QIZ32042.1| hemagglutinin [Influenza A virus]     566  2964.0     0.0   \n",
       "19   gb|QIZ33251.1| hemagglutinin [Influenza A virus]     566  2964.0     0.0   \n",
       "20  gb|QIQ44996.1| hemagglutinin [Influenza A viru...     566  2964.0     0.0   \n",
       "21   gb|QTJ06578.1| hemagglutinin [Influenza A virus]     566  2964.0     0.0   \n",
       "22   gb|QJT24499.1| hemagglutinin [Influenza A virus]     566  2964.0     0.0   \n",
       "23   gb|QIV14744.1| hemagglutinin [Influenza A virus]     566  2964.0     0.0   \n",
       "24   gb|QIQ45654.1| hemagglutinin [Influenza A virus]     566  2964.0     0.0   \n",
       "25  gb|QIA57747.1| hemagglutinin [Influenza A viru...     566  2964.0     0.0   \n",
       "26   gb|QIZ31801.1| hemagglutinin [Influenza A virus]     566  2964.0     0.0   \n",
       "27   gb|QKM75836.1| hemagglutinin [Influenza A virus]     566  2964.0     0.0   \n",
       "28   gb|QKS74061.1| hemagglutinin [Influenza A virus]     566  2964.0     0.0   \n",
       "29   gb|QIH95333.1| hemagglutinin [Influenza A virus]     566  2963.0     0.0   \n",
       "30   gb|QIZ32602.1| hemagglutinin [Influenza A virus]     566  2963.0     0.0   \n",
       "31   gb|QJT24351.1| hemagglutinin [Influenza A virus]     566  2963.0     0.0   \n",
       "32   gb|QLV95699.1| hemagglutinin [Influenza A virus]     566  2963.0     0.0   \n",
       "33   gb|QKM76085.1| hemagglutinin [Influenza A virus]     566  2963.0     0.0   \n",
       "34   gb|QJI52636.1| hemagglutinin [Influenza A virus]     566  2963.0     0.0   \n",
       "35  gb|QIQ45093.1| hemagglutinin [Influenza A viru...     566  2963.0     0.0   \n",
       "36   gb|QJT24428.1| hemagglutinin [Influenza A virus]     566  2963.0     0.0   \n",
       "37  gb|QIA58044.1| hemagglutinin [Influenza A viru...     566  2963.0     0.0   \n",
       "38   gb|QIQ45082.1| hemagglutinin [Influenza A virus]     566  2963.0     0.0   \n",
       "39  gb|QJT24466.1| hemagglutinin [Influenza A viru...     566  2963.0     0.0   \n",
       "40   gb|QIZ31721.1| hemagglutinin [Influenza A virus]     566  2962.0     0.0   \n",
       "41   gb|QTK18161.1| hemagglutinin [Influenza A virus]     566  2962.0     0.0   \n",
       "42   gb|QIV14634.1| hemagglutinin [Influenza A virus]     566  2962.0     0.0   \n",
       "43   gb|QJI52715.1| hemagglutinin [Influenza A virus]     566  2962.0     0.0   \n",
       "44   gb|QTK18105.1| hemagglutinin [Influenza A virus]     566  2962.0     0.0   \n",
       "45   gb|QIZ32778.1| hemagglutinin [Influenza A virus]     566  2962.0     0.0   \n",
       "46   gb|QIZ33133.1| hemagglutinin [Influenza A virus]     566  2962.0     0.0   \n",
       "47  gb|QIZ32822.1| hemagglutinin [Influenza A viru...     566  2962.0     0.0   \n",
       "48   gb|QJT24659.1| hemagglutinin [Influenza A virus]     566  2962.0     0.0   \n",
       "49   gb|QJT24340.1| hemagglutinin [Influenza A virus]     566  2962.0     0.0   \n",
       "\n",
       "    QueryStart  QueryEnd  SubjectStart  SubjectEnd       pct  \n",
       "0            1       550             1         550  1.000000  \n",
       "1            1       550             1         550  1.000000  \n",
       "2            1       550             1         550  1.000000  \n",
       "3            1       550             1         550  1.000000  \n",
       "4            1       550             1         550  1.000000  \n",
       "5            1       550             1         550  1.000000  \n",
       "6            1       550             1         550  1.000000  \n",
       "7            1       550             1         550  1.000000  \n",
       "8            1       550             1         550  1.000000  \n",
       "9            1       550             1         550  1.000000  \n",
       "10           1       550             1         550  1.000000  \n",
       "11           1       550             1         550  1.000000  \n",
       "12           1       550             1         550  1.000000  \n",
       "13           1       550             1         550  1.000000  \n",
       "14           1       550             1         550  1.000000  \n",
       "15           1       550             1         550  1.000000  \n",
       "16           1       550             1         550  1.000000  \n",
       "17           1       550             1         550  1.000000  \n",
       "18           1       550             1         550  1.000000  \n",
       "19           1       550             1         550  0.998182  \n",
       "20           1       550             1         550  0.998182  \n",
       "21           1       550             1         550  0.998182  \n",
       "22           1       550             1         550  1.000000  \n",
       "23           1       550             1         550  1.000000  \n",
       "24           1       550             1         550  1.000000  \n",
       "25           1       550             1         550  0.998182  \n",
       "26           1       550             1         550  0.998182  \n",
       "27           1       550             1         550  1.000000  \n",
       "28           1       550             1         550  1.000000  \n",
       "29           1       550             1         550  1.000000  \n",
       "30           1       550             1         550  0.998182  \n",
       "31           1       550             1         550  0.998182  \n",
       "32           1       550             1         550  1.000000  \n",
       "33           1       550             1         550  0.998182  \n",
       "34           1       550             1         550  1.000000  \n",
       "35           1       550             1         550  0.998182  \n",
       "36           1       550             1         550  0.998182  \n",
       "37           1       550             1         550  0.998182  \n",
       "38           1       550             1         550  0.998182  \n",
       "39           1       550             1         550  1.000000  \n",
       "40           1       550             1         550  1.000000  \n",
       "41           1       550             1         550  0.998182  \n",
       "42           1       550             1         550  0.998182  \n",
       "43           1       550             1         550  0.998182  \n",
       "44           1       550             1         550  0.998182  \n",
       "45           1       550             1         550  0.998182  \n",
       "46           1       550             1         550  0.998182  \n",
       "47           1       550             1         550  0.998182  \n",
       "48           1       550             1         550  0.998182  \n",
       "49           1       550             1         550  0.998182  "
      ]
     },
     "execution_count": 94,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "csv_list=parse_blast('my_blast.xml')\n",
    "csv_list"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 95,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "['1841676337']"
      ]
     },
     "execution_count": 95,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "from Bio import Entrez\n",
    "Entrez.email = \"ishanu@uchicago.edu\"     # Always tell NCBI who you are\n",
    "handle = Entrez.esearch(db=\"protein\", term=\"QJT24340\")\n",
    "record = Entrez.read(handle)\n",
    "id=record[\"IdList\"]\n",
    "id"
   ]
  },
  {
   "cell_type": "code",
   "execution_count": 96,
   "metadata": {},
   "outputs": [
    {
     "data": {
      "text/plain": [
       "'LOCUS       QJD23117                 566 aa            linear   VRL 28-APR-2020\\nDEFINITION  hemagglutinin [Influenza A virus].\\nACCESSION   QJD23117\\nVERSION     QJD23117.1\\nDBSOURCE    accession MT372532.1\\nKEYWORDS    .\\nSOURCE      Influenza A virus\\n  ORGANISM  Influenza A virus\\n            Viruses; Riboviria; Orthornavirae; Negarnaviricota;\\n            Polyploviricotina; Insthoviricetes; Articulavirales;\\n            Orthomyxoviridae; Alphainfluenzavirus.\\nREFERENCE   1  (residues 1 to 566)\\n  AUTHORS   Swine Surveillance,A.\\n  TITLE     USDA NAHLN Voluntary Swine Influenza Surveillance\\n  JOURNAL   Unpublished\\nREFERENCE   2  (residues 1 to 566)\\n  AUTHORS   Swine Surveillance,A.\\n  TITLE     Direct Submission\\n  JOURNAL   Submitted (23-APR-2020) DiVL, USDA NAHLN Voluntary Swine Influenza\\n            Surveillance, 1920 Dayton Ave., Ames, IA 50010, USA\\nCOMMENT     Method: conceptual translation.\\nFEATURES             Location/Qualifiers\\n     source          1..566\\n                     /organism=\"Influenza A virus\"\\n                     /strain=\"A/swine/Minnesota/A02245569/2020\"\\n                     /serotype=\"H1N1\"\\n                     /isolation_source=\"lung\"\\n                     /host=\"swine\"\\n                     /db_xref=\"taxon:11320\"\\n                     /segment=\"4\"\\n                     /country=\"USA: Minnesota\"\\n                     /collection_date=\"31-Mar-2020\"\\n     Protein         1..566\\n                     /product=\"hemagglutinin\"\\n                     /function=\"receptor binding and fusion protein\"\\n     sig_peptide     1..17\\n     mat_peptide     18..344\\n                     /product=\"HA1\"\\n     Region          20..520\\n                     /region_name=\"Hemagglutinin\"\\n                     /note=\"Haemagglutinin; pfam00509\"\\n                     /db_xref=\"CDD:395409\"\\n     mat_peptide     345..566\\n                     /product=\"HA2\"\\n     CDS             1..566\\n                     /gene=\"HA\"\\n                     /coded_by=\"MT372532.1:1..1701\"\\nORIGIN      \\n        1 mkailvvmly tfttanadtl cigyhannst dtvdtvlekn vtvthsvnll edkhngklck\\n       61 lrgvaplhlg kcniagwilg npeceslsta rswsyivets nsdngtcypg dfinyeelre\\n      121 qlssvssfer feifpktssw pnhdsdngvt aacphagaks fyknliwlvk kgksypkinq\\n      181 tyindkgkev lvlwgihhpp tiadqqslyq nadayvfvgt sryskkfkpe iatrpkvrdq\\n      241 egrmnyywtl vepgdkitfe atgnlvapry aftmerdags giiisdtpvh dcnttcqtpe\\n      301 gaintslpfq nvhpitigkc pkyvkstklr latglrnvps iqsrglfgai agfieggwtg\\n      361 mvdgwygyhh qneqgsgyaa dlkstqnaid kitnkvnsvi ekmntqftav gkefnhlekr\\n      421 ienlnkkvdd gfldiwtyna ellvllener tldyhdsnvk nlyekvrnql knnakeigng\\n      481 cfefyhkcdn tcmesvkngt ydypkyseea klnrekidgv kldstkiyqi laiystvass\\n      541 lvlvvslgai sfwmcsngsl qcrici\\n//\\n\\n'"
      ]
     },
     "execution_count": 96,
     "metadata": {},
     "output_type": "execute_result"
    }
   ],
   "source": [
    "handle = Entrez.efetch(db=\"protein\", rettype=\"gb\", retmode=\"text\", id=\"1834373412\")\n",
    "#record = SeqIO.read(handle, \"genbank\")\n",
    "handle.read()"
   ]
  },
  {
   "cell_type": "markdown",
   "metadata": {},
   "source": [
    "H9N2: minks are probable mixing vessels\n",
    "https://www.ncbi.nlm.nih.gov/nuccore/ON870694.1\n",
    " https://www.tandfonline.com/doi/full/10.1080/22221751.2021.1899058\n",
    " Pandemic influenza, typically caused by the reassortment of human and avian influenza viruses, can result in severe or fatal infections in humans. Timely identification of potential pandemic viruses must be a priority in influenza virus surveillance. However, the range of host species responsible for the generation of novel pandemic influenza viruses remains unclear. In this study, we conducted serological surveys for avian and human influenza virus infections in farmed mink and determined the susceptibility of mink to prevailing avian and human virus subtypes. The results showed that farmed mink were commonly infected with human (H3N2 and H1N1/pdm) and avian (H7N9, H5N6, and H9N2) influenza A viruses. Correlational analysis indicated that transmission of human influenza viruses occurred from humans to mink, and that feed source was a probable route of avian influenza virus transmission to farmed mink. Animal experiments showed that mink were susceptible and permissive to circulating avian and human influenza viruses, and that human influenza viruses (H3N2 and H1N1/pdm), but not avian viruses, were capable of aerosol transmission among mink. These results indicate that farmed mink could be highly permissive “mixing vessels” for the reassortment of circulating human and avian influenza viruses. \n",
    " \n",
    " H7N9\n",
    " https://www.bmj.com/content/347/bmj.f4752?tab=responses\n",
    " HH transmission has been suspected\n",
    " Asian lineage H7N9 virus is rated by the Influenza Risk Assessment Tool as having the greatest potential to cause a pandemic, as well as potentially posing the greatest risk to severely impact public health if it were to achieve sustained human-to-human transmission.\n",
    " \n",
    " H3N2 swine\n",
    " https://www.cdc.gov/flu/swineflu/spotlights/first-human-infection-2022.htm\n",
    " Human infection detected in US this Aug\n",
    " \n",
    " H1N1\n",
    " Submitted (03-NOV-2020) USDA Swine Surveillance, USDA Swine\n",
    "            Surveillance, 1920 Dayton, Ames, IA 50010, USA\n",
    "COMMENT     Method: conceptual translation.\n",
    "\n"
   ]
  }
 ],
 "metadata": {
  "kernelspec": {
   "display_name": "Python 3 (ipykernel)",
   "language": "python",
   "name": "python3"
  },
  "language_info": {
   "codemirror_mode": {
    "name": "ipython",
    "version": 3
   },
   "file_extension": ".py",
   "mimetype": "text/x-python",
   "name": "python",
   "nbconvert_exporter": "python",
   "pygments_lexer": "ipython3",
   "version": "3.10.6"
  }
 },
 "nbformat": 4,
 "nbformat_minor": 4
}
